# Statistical Genetics

## Phasing

WhatsHap phases genotypes in a pedigree using the Mendelian transmission rules and physical phasing based on reads that span two or more heterozygous variants.

```{bash eval=FALSE}
whatshap phase --ped pedigree.ped \
-o phased_whatshap.vcf joint/output_vqsr_variants.vcf.gz \
aligned_HG002.bam aligned_HG003.bam aligned_HG004.bam \
--tag=PS --no-reference --indels
```

## Kinship

We seek to confirm pedigrees and identify hidden relatedness between samples. We first select SNPs in approximate linkage equilibrium with Linkdatagen based on the genetic linkage map of HapMap SNPs. Linkdatagen also eliminates Mendelian inconsitencies in the data.

```{bash eval=FALSE}
perl vcf2linkdatagen.pl \
-variantCaller mpileup \
-annotfile annotHapMap2U.txt \
-pop CEU -mindepth 10 -missingness 0 \
-idlist MyVCFlist.txt > MySNPs.brlmm

perl linkdatagen.pl \
-data m -pedfile MyPed.ped \
-whichSamplesFile MyWS.ws \
-callFile MySNPs.brlmm \
-annotFile annotHapMap2U.txt \
-pop CEU -binsize 0.3 \
-MendelErrors removeSNPs -prog pl \
-outputDir MyPed_HapMap2_pl > MyPed_HapMap2_pl.out
```

We construct an allele frequency report for the chosen SNPs.

```{r eval=FALSE}
map <- read_delim(file = "plink.map", delim = " ", col_select = 1:4, col_names = c("CHR", "SNP", "cM", "bp"))
ord <- read_tsv(file = "orderedSNPs.txt")
freq_pl <- map %>% left_join(ord, by = "SNP") %>% 
  dplyr::select(CHR, SNP, `Allele freq`) %>% 
  mutate(A1="1") %>% mutate(A1="1") %>% mutate(A2="2") %>% mutate(NCHROBS="300") %>% 
  dplyr::select(CHR, SNP, A1, A2, `Allele freq`, NCHROBS) %>% 
  rename(`Allele freq` = "MAF")
write_delim(freq_pl, "freq_pl.frq", delim = " ")
```

Alternatively, with a sufficient number of samples, we can estimate the linkage disequilibrium between alleles and remove correlated SNPs.

```{bash eval=FALSE}
plink --file data --indep 50 5 2
```

We then calculate the empirical kinship coefficient by estimating identity by descent using identity by state and allele frequencies. We run this calculation in PLINK.

```{bash eval=FALSE}
plink --file plink --genome --read-freq freq_pl.frq
```

We also run this calculation using MendelKinship from the OpenMendel project.

```{bash eval=FALSE}
using MendelKinship, CSV
Kinship("control_compare.txt")
```

## Runs of Homozygosity

HomozygosityMapper <https://www.homozygositymapper.org/>, PLINK

## Linkage Analysis

Linkdatagen, PBAP, MORGAN, MERLIN, Allegro, OpenMendel, Pedminer

```{bash eval=FALSE}
perl linkdatagen.pl \
-data m -pedfile MyPed.ped \
-whichSamplesFile MyWS.ws \
-callFile MySNPs.brlmm \
-annotFile annotHapMap2U.txt \
-pop CEU -binsize 0.3 \
-MendelErrors removeSNPs -prog me \
-outputDir MyPed_HapMap2 > MyPed_HapMap2_me.out
```

Two-point linkage
Multipoint linkage

## Population Substructure

Given a large population sample, we cluster individuals by degree of identity by state using PLINK in order to identify subgroups with shared ancestry.

```{bash eval=FALSE}
plink --file mydata --cluster
# dimensional reduction to 4D then plot two coordinates
plink --file mydata --read-genome plink.genome --cluster --mds-plot 4
```
