# Variant Annotation

Annotate the variants with VEP for molecular consequence, allele frequency and a set of pathogenicity prediction tools.

```{bash eval=FALSE}
apptainer exec -B vep_data:/opt/vep/.vep vep.sif /opt/vep/src/ensembl-vep/vep \
--dir_cache /opt/vep/.vep/ \
--dir_plugins /opt/vep/.vep/Plugins/ \
-i /opt/vep/.vep/output_vqsr_snps_hard_indels.vcf.gz \
-o /opt/vep/.vep/ensembl_annotated.txt \
--individual EC27_proband \
--everything \
--check_existing \
--xref_refseq \
--fasta /opt/vep/.vep/Homo_sapiens.GRCh38.fa \
--hgvs \
--hgvsg \
--plugin CADD,/opt/vep/.vep/Plugins/data/grch38/whole_genome_SNVs_grch38.tsv.gz,/opt/vep/.vep/Plugins/data/grch38/gnomad.genomes.r3.0.indel_grch38.tsv.gz \
--plugin PrimateAI,/opt/vep/.vep/Plugins/data/grch38/PrimateAI_scores_v0.2_GRCh38_sorted.tsv.bgz \
--plugin SpliceAI,snv=/opt/vep/.vep/Plugins/data/grch38/spliceai_scores.raw.snv.hg38.vcf.gz,indel=/opt/vep/.vep/Plugins/data/grch38/spliceai_scores.raw.indel.hg38.vcf.gz \
--plugin EVE,file=/opt/vep/.vep/Plugins/data/grch38/eve_merged.vcf.gz \
--plugin Mastermind,/opt/vep/.vep/Plugins/data/grch38/mastermind_cited_variants_reference-2023.04.02-grch38.vcf.gz \
--plugin dbNSFP,/opt/vep/.vep/Plugins/data/grch38/dbNSFP4.3a_grch38.gz,MutationTaster_score,MutationTaster_pred \
--plugin REVEL,/opt/vep/.vep/Plugins/data/grch38/new_tabbed_revel_grch38.tsv.gz \
--custom /opt/vep/.vep/Plugins/data/grch38/clinvar.vcf.gz,ClinVar,vcf,exact,0,CLNSIG,CLNREVSTAT,CLNDN \
--cache \
--merged \
--offline \
--use_given_ref \
--force_overwrite \
--assembly GRCh38 \
--fork 2 \
--buffer_size 1000 \
--tab
```

## Filtration

Perform a first round of filtering with VEP to reduce the size of the VCF to be loaded into R. In particular, select variants with high or moderate impact and an allele frequency smaller than 0.05.

```{bash eval=FALSE}
apptainer exec -B vep_data:/opt/vep/.vep vep.sif /opt/vep/src/ensembl-vep/filter_vep \
-i /opt/vep/.vep/ensembl_annotated.txt \
-out /opt/vep/.vep/ensembl_ann_vep_filt.txt \
--filter "(IMPACT is HIGH or IMPACT is MODERATE) and (gnomADe_AF <= 0.05 or not gnomADe_AF)" \
--force_overwrite \
--format tab 
```

Load the WhatsHap phased variants into R and select autosomal variants.

```{r eval=FALSE}
library(tidyverse)
library(VariantAnnotation)

mother = "EC25"
father = "EC26"
proband = "EC27"

wh <- readVcf("phased_whatshap_refined.vcf", "hg38")
wh <- expand(wh)
pos <- rowRanges(wh)
pos2 <- as.data.frame(pos) %>% as_tibble 

# Genotype
wh_geno <- geno(wh)$GT %>% as_tibble() %>% dplyr::rename("mother"= sym(mother), "father"= sym(father), "proband"= sym(proband))
# Phasing block
wh_ps <- geno(wh)$PS %>% as_tibble() %>% dplyr::rename("PS_mother" = sym(mother), "PS_father"= sym(father), "PS_proband"= sym(proband))
# Genotype quality
wh_gq <- geno(wh)$GQ %>% as_tibble() %>% dplyr::rename("GQ_mother" = sym(mother), "GQ_father"= sym(father), "GQ_proband"= sym(proband))
# Join genotype, phasing block and genotype quality
wh_geno0 <- bind_cols(pos2,wh_geno,wh_gq,wh_ps)

# Identify alternate alleles with number
wh_geno01 <- wh_geno0 %>% 
  group_by(seqnames,start,end,width,REF) %>% 
  mutate(counter = row_number()) %>%
  ungroup()

# Convert variant formatting to VEP format. Add 1 to indel start site.
wh_geno01 <- wh_geno01 %>% 
  mutate(Location = ifelse(width==1, ifelse(nchar(ALT)==1, paste(seqnames,start,sep=":"), paste0(seqnames,":",start,"-",start+1)), ifelse(start+1==end, paste0(seqnames,":",start+1), paste0(seqnames,":",start+1,"-",end)))) %>% 
  mutate(Allele = ifelse(substr(REF,1,1)==substr(ALT,1,1), substr(ALT,2,nchar(ALT)), ALT))
wh_geno01$Allele[wh_geno01$Allele==""] <- "-"

# Overlapping deletions
wh_geno_overlap <- wh_geno01 %>% dplyr::filter(ALT=="")
wh_geno1 <- wh_geno01 %>% dplyr::filter(ALT!="")
wh_geno1 <- wh_geno1[c(21,22,11:19,20)]

# Select X chromosome
wh_geno_X <- wh_geno1 %>% dplyr::filter(str_detect(Location,"chrX"))

# Select autosomal variants
wh_geno_auto <- wh_geno1 %>%
  dplyr::filter(!str_detect(Location,"chrX")) %>% 
  dplyr::filter(!str_detect(Location,"chrY")) %>%
  dplyr::filter(!str_detect(Location,"M"))
wh_geno_auto
```

Manually remove the hashtag before the line of column titles in ensembl_ann_vep_filt.txt. Load the filtered variants into R. Select variants with an allele frequency \<0.01 that fall in RefSeq transcripts.

```{r eval=FALSE}
result <- read_table("ensembl_ann_vep_filt_refined_08_07_23.txt", comment = "#", na="-", col_types = list(Uploaded_variation= col_factor(), Gene=col_factor(), Feature=col_factor(), MANE_PLUS_CLINICAL = col_character(), MOTIF_NAME = col_character(), HIGH_INF_POS = col_character(), TRANSCRIPTION_FACTORS = col_character(), PUBMED=col_character(), MOTIF_POS = col_character()))
result$Allele[is.na(result$Allele)] <- "-"
result1 <- result %>% as_tibble %>% left_join(wh_geno1, by = c("Location","Allele"))
refseq0 <- result1 %>% 
  dplyr::filter(SOURCE == "RefSeq") %>% 
  dplyr::filter(gnomADe_AF<=0.01|is.na(gnomADe_AF)) %>% 
  dplyr::filter(!(is.na(gnomADe_AF) & gnomADg_AF > 0.01)) %>%
  dplyr::filter(!(str_detect(Feature,"XM"))) %>%
  select_if(~any(!is.na(.)))
```

Add gene-level information.

```{r eval=FALSE}
library(ontologyIndex)

# OMIM
omim <- read_tsv("~/Documents/kidney_panel/omim/genemap2_edit.txt", name_repair="universal", col_types=list(Entrez.Gene.ID = col_factor())) %>% dplyr::rename(OMIM_Phenotypes = Phenotypes) %>% dplyr::select(Entrez.Gene.ID,OMIM_Phenotypes)

# Human Phenotype Ontology
hpo_data <- read_tsv("~/Documents/kidney_panel/genes_to_phenotype.txt", col_types=list(entrez_gene_id = col_factor()))
hpo <- get_ontology(file = '~/Documents/kidney_panel/hp_obo.txt', propagate_relationships = "is_a", extract_tags = "minimal")
hpo_kidney <- get_descendants(hpo, roots="HP:0000077") #Abnormality of the kidney

gen_hpo_kidney <- hpo_data %>% 
  dplyr::filter(hpo_term_id %in% hpo_kidney) %>% 
  dplyr::arrange(frequency_hpo) %>% 
  dplyr::mutate(frequency = case_when(frequency_hpo == "HP:0040284" ~ " (very rare)", frequency_hpo == "HP:0040283" ~ " (occasional)", frequency_hpo == "HP:0040282" ~ " (frequent)", frequency_hpo == "HP:0040281" ~ " (very frequent)", frequency_hpo == "HP:0040280" ~ " (obligate)", TRUE ~ "")) %>% 
  unite("renal_hpo_term", c(hpo_term_name, frequency), remove = FALSE, sep = "") %>% 
  dplyr::group_by(entrez_gene_id) %>% 
  dplyr::summarise(renal_hpo_term = toString(unique(renal_hpo_term)))

gen_hpo_extra <- hpo_data %>% 
  dplyr::filter(frequency_hpo %in% c("HP:0040281", "HP:0040280")) %>%
  dplyr::arrange(frequency_hpo) %>%
  dplyr::filter(!(hpo_term_id %in% hpo_kidney)) %>% 
  dplyr::group_by(entrez_gene_id) %>% 
  dplyr::summarise(extrarenal_hpo_term = toString(unique(hpo_term_name)))

# ClinGen 
clingen <- read_csv("~/Documents/kidney_panel/Clingen-Curation.csv", skip = 3) %>% 
  dplyr::mutate(assertion=gsub("\\s*\\([^\\)]+\\)","", gene_disease_validity_assertion_classifications)) %>%
  dplyr::mutate(assertion=paste("(",assertion,")",sep="")) %>%
  dplyr::mutate(assertion=if_else(assertion=="(NA)",NA,assertion)) %>%
  unite("disease", c(disease_label, assertion), remove = FALSE, na.rm=TRUE, sep = " ") %>%
  dplyr::group_by(hgnc_id) %>% 
  dplyr::summarise(clingen_disease_label = paste(disease,collapse="; "))

# GenCC including Orphanet data
gencc <- read_tsv("~/Documents/kidney_panel/gencc-submissions.tsv")

gencc_def <- gencc %>% 
  dplyr::filter(submitter_title != "Orphanet") %>% 
  dplyr::filter(classification_title == "Definitive") %>% 
  dplyr::group_by(gene_symbol, gene_curie, disease_title) %>% 
  dplyr::summarise(submitter_title = toString(unique(submitter_title)), moi_title = toString(unique(moi_title)), disease_curie = toString(unique(disease_curie)), disease_original_curie = toString(unique(disease_original_curie)), classification_title = toString(classification_title), submitted_as_pmids = toString(unique(submitted_as_pmids))) %>%
  dplyr::mutate(submitter = paste("(", submitter_title, ")", sep = "")) %>% 
  tidyr::unite("gencc_disease_definitive", c(disease_title, submitter), remove = FALSE, sep = " ") %>% 
  dplyr::select(gene_curie, gencc_disease_definitive) %>% 
  dplyr::group_by(gene_curie) %>% 
  dplyr::summarise(gencc_disease_definitive = paste(unique(gencc_disease_definitive), collapse="; "))

gencc_strong <- gencc %>% 
  dplyr::filter(submitter_title != "Orphanet") %>% 
  dplyr::filter(classification_title == "Strong") %>% 
  dplyr::group_by(gene_symbol, gene_curie, disease_title) %>% 
  dplyr::summarise(submitter_title = toString(unique(submitter_title)), moi_title = toString(unique(moi_title)), disease_curie = toString(unique(disease_curie)), disease_original_curie = toString(unique(disease_original_curie)), classification_title = toString(classification_title), submitted_as_pmids = toString(unique(submitted_as_pmids))) %>%
  dplyr::mutate(submitter = paste("(", submitter_title, ")", sep = "")) %>% 
  tidyr::unite("gencc_disease_strong", c(disease_title, submitter), remove = FALSE, sep = " ") %>% 
  dplyr::select(gene_curie, gencc_disease_strong) %>% 
  dplyr::group_by(gene_curie) %>% 
  dplyr::summarise(gencc_disease_strong = paste(unique(gencc_disease_strong), collapse="; "))

orphanet <- gencc %>% 
  dplyr::filter(submitter_title == "Orphanet") %>% 
  dplyr::group_by(gene_curie) %>% 
  dplyr::summarise(orphanet_disease = toString(disease_title), orphanet_moi = toString(unique(moi_title)))

# Join
refseq01 <- refseq0 %>% 
  left_join(omim, by=join_by(Gene==Entrez.Gene.ID)) %>%
  left_join(gen_hpo_kidney, by=join_by(Gene==entrez_gene_id)) %>%
  left_join(gen_hpo_extra, by=join_by(Gene==entrez_gene_id)) %>%
  left_join(clingen, by=join_by(HGNC_ID==hgnc_id)) %>%
  left_join(gencc_def, by=join_by(HGNC_ID==gene_curie)) %>%
  left_join(gencc_strong, by=join_by(HGNC_ID==gene_curie)) %>%
  left_join(orphanet, by=join_by(HGNC_ID==gene_curie))

refseq_X <- refseq01 %>% dplyr::filter(str_detect(Uploaded_variation,"X"))
refseq <- refseq01 %>%
  dplyr::filter(!str_detect(Uploaded_variation,"X")) %>% 
  dplyr::filter(!str_detect(Uploaded_variation,"Y")) %>% 
  dplyr::filter(!str_detect(Uploaded_variation,"M"))
refseq
```

## *De Novo* Variants

Select *de novo* variants in the proband.

```{r eval=FALSE}
geno_novo_all <- wh_geno_auto %>% 
  dplyr::filter(! ( ( ( str_detect(mother, substr(proband,1,1)) | str_detect(mother, "\\.") ) & 
                 (str_detect(father, substr(proband,3,3)) | str_detect(father, "\\.") ) ) | 
               ( ( str_detect(father, substr(proband,1,1)) | str_detect(father, "\\.") ) & 
                   (str_detect(mother, substr(proband,3,3)) | str_detect(mother, "\\.") ) ) ) ) %>%
  dplyr::filter(!(str_detect(mother,as.character(counter)) | str_detect(father,as.character(counter))) |
                  (str_detect(mother, substr(proband,1,1)) & str_detect(mother, substr(proband,3,3))) |
                  (str_detect(father, substr(proband,1,1)) & str_detect(father, substr(proband,3,3)))
                  )
geno_novo_all
```

Discard *de novo* mutations back to the reference allele (e.g. 1/1, 1/1 --\> 0/1) since they are unlikely to be pathogenic.

```{r eval=FALSE}
geno_novo_not_ref <- wh_geno_auto %>% 
  dplyr::filter(! ( ( ( str_detect(mother, substr(proband,1,1)) | str_detect(mother, "\\.") | substr(proband,1,1)==0 ) & 
                 (str_detect(father, substr(proband,3,3)) | str_detect(father, "\\.") | substr(proband,3,3)==0 ) ) | 
               ( ( str_detect(father, substr(proband,1,1)) | str_detect(father, "\\.") | substr(proband,1,1)==0 ) & 
                   (str_detect(mother, substr(proband,3,3)) | str_detect(mother, "\\.") | substr(proband,3,3)==0 ) ) ) ) %>%
  dplyr::filter(!(str_detect(mother,as.character(counter)) | str_detect(father,as.character(counter))) |
                  (str_detect(mother, substr(proband,1,1)) & str_detect(mother, substr(proband,3,3))) |
                  (str_detect(father, substr(proband,1,1)) & str_detect(father, substr(proband,3,3)))
                  )
geno_novo_not_ref[c(1:5)]
```

Increase the cutoff for genotype quality to ensure that the the *de novo* variants are not a result of incorrect variant calling. Set GQ>=30, i.e, the probability that the call is incorrect is =<0.1%.

```{r eval=FALSE}
geno_novo_not_ref_high_qual <- geno_novo_not_ref %>% dplyr::filter(GQ_mother>=30) %>% dplyr::filter(GQ_father>=30) %>% dplyr::filter(GQ_proband>=30)
geno_novo_not_ref_high_qual
```

Select *de novo* variants found in the filtered VEP annotation. For each variant, select a single gene transcript that contains the variant.

```{r eval=FALSE}
prob_novo <- refseq %>% inner_join(geno_novo_not_ref)
prob_novo <- prob_novo %>% group_by(Uploaded_variation) %>% arrange(desc(CANONICAL), as.character(Feature),.by_group = T) %>% dplyr::filter(Feature == Feature[1])
prob_novo_high_qual <- refseq %>% inner_join(geno_novo_not_ref_high_qual)
prob_novo_high_qual <- prob_novo_high_qual %>% group_by(Uploaded_variation) %>% arrange(desc(CANONICAL), as.character(Feature),.by_group = T) %>% dplyr::filter(Feature == Feature[1])
prob_novo_high_qual
```

## Homozygous Variants

Select the homozygous variants in the proband.

```{r eval=FALSE}
geno_hom <- wh_geno_auto %>% dplyr::filter(substr(proband,1,1)==substr(proband,3,3) & substr(proband,1,1) != 0 & substr(proband,1,1) != ".")
geno_hom[c(1:5)]
```

Discard the homozygous variants for which one of the parents is also homozygous (e.g. 1/1, 0/1 -\> 1/1) since these variants are unlikely to be pathogenic.

```{r eval=FALSE}
geno_hom2 <- geno_hom %>%
  dplyr::filter(! ( (substr(mother,1,1)==substr(mother,3,3) & substr(mother,1,1)==substr(proband,1,1)) |
                      (substr(father,1,1)==substr(father,3,3) & substr(father,1,1)==substr(proband,1,1)) )
                  )
geno_hom2[c(1:5)]
```

Select the homozygous variants found in the filtered VEP annotation. For each variant, select a single gene transcript that contains the variant.

```{r eval=FALSE}
prob_hom <- refseq %>% inner_join(geno_hom2)
prob_hom <- prob_hom %>% group_by(Uploaded_variation) %>% arrange(desc(CANONICAL), as.character(Feature),.by_group = T) %>% dplyr::filter(Feature == Feature[1])
prob_hom
```

## Compound Heterozygous Variants

Select the unphased compound heterozygous variants.

```{r eval=FALSE}
prob_2plus_het <- refseq %>% dplyr::filter(substr(proband,1,1)!=substr(proband,3,3)) 
prob_2plus_het <- prob_2plus_het %>% group_by(Feature) %>% dplyr::filter(length(unique(PS_proband))>=2 | length(is.na(PS_proband)[is.na(PS_proband)==TRUE])>=2 ) %>% ungroup()
prob_2plus_het2 <- prob_2plus_het %>% group_by(Gene) %>% arrange(desc(CANONICAL), as.character(Feature),.by_group = T) %>% dplyr::filter(Feature == Feature[1])
prob_2plus_het2
```

Select the phased compound heterozygous variants.

```{r eval=FALSE}
prob_comp_het <- refseq %>% group_by(Feature, PS_proband) %>% dplyr::filter(any(substr(proband,1,1)>=1 & substr(proband,2,2)=="|") & any(substr(proband,3,3)>=1 & substr(proband,2,2)=="|")) 
prob_comp_het2 <- prob_comp_het %>% group_by(Gene) %>% arrange(desc(CANONICAL), as.character(Feature),.by_group = T) %>% dplyr::filter(Feature == Feature[1])
prob_comp_het2
```

## X Chromosome

Remove X chromosome calls that are "heterozygous" for male individuals.

```{r eval=FALSE}
wh_geno_X_filt <- wh_geno_X %>% 
  dplyr::filter(( substr(proband,1,1)==substr(proband,3,3) & substr(father,1,1)==substr(father,3,3) ))
prob_X <- inner_join(refseq_X, wh_geno_X) %>% group_by(Gene) %>% arrange(desc(CANONICAL), as.character(Feature),.by_group = T) %>% dplyr::filter(Feature == Feature[1])
prob_X
```


## Disease Genes

Select one entry for each variant in refseq table, prioritizing canonical and curated transcripts.

```{r eval=FALSE}
refseq2 <- refseq01 %>% group_by(Uploaded_variation) %>% arrange(desc(CANONICAL), as.character(Feature),.by_group = T) %>% dplyr::filter(Feature == Feature[1])
refseq2
```

Select variants that fall in green genes from the UK PanelApp renal broad superpanel.

```{r eval=FALSE}
# UK Panel App
panel <- read_tsv("renal_superpanel_broad.tsv")
panel_green <- panel %>% dplyr::filter(GEL_Status==3)
refseq_green <- refseq2 %>% dplyr::filter(HGNC_ID %in% panel_green$HGNC)

#Australia Panel App
panel_au <- read_tsv("Kidneyome_SuperPanel.tsv") %>% 
  dplyr::filter(!duplicated(HGNC)) %>%
  dplyr::mutate(Status=case_when(GEL_Status==4|GEL_Status==3 ~ "Green", 
                                 GEL_Status==2 ~ "Amber", 
                                 GEL_Status==1|GEL_Status==0 ~ "Red")) %>%
  dplyr::select(HGNC,Status)
refseq_au <- refseq2 %>% inner_join(panel_au, join_by(HGNC_ID==HGNC))
refseq_au
```

Combine annotations.

```{r eval=FALSE}
refseq3 <- refseq01 %>% group_by(Gene) %>% arrange(desc(CANONICAL), as.character(Feature),.by_group = T) %>% ungroup()
refseq4 <- refseq3 %>% 
  left_join(add_column(geno_novo_not_ref,de_novo="de_novo")) %>%
  left_join(add_column(geno_novo_not_ref_high_qual,de_novo_gq="de_novo_gq>=30")) %>%
  left_join(add_column(geno_hom2, homozygote="homozygote")) %>%
  left_join(add_column(prob_2plus_het, unphased_compound_heterozygote="unphased_compound_heterozygote")) %>%
  left_join(add_column(prob_comp_het, phased_compound_heterozygote="phased_compound_heterozygote")) %>%
  mutate(zygosity=coalesce(de_novo_gq,de_novo,homozygote,phased_compound_heterozygote,unphased_compound_heterozygote)) %>%
  dplyr::select(-c(de_novo_gq,de_novo,homozygote,phased_compound_heterozygote,unphased_compound_heterozygote)) %>%
  dplyr::relocate(zygosity,.after=counter)
refseq_sum <- refseq4 %>% group_by(Feature) %>% summarize(z=paste(c(Uploaded_variation,Location,Allele,Consequence,Amino_acids,Codons), collapse = " ")) %>% 
  distinct(z,.keep_all = T)
refseq5 <- refseq4 %>% dplyr::filter(Feature %in% refseq_sum$Feature)
```
