--- 
title: "Whole Exome Sequencing Pipeline"
author: "Marc Henein and Zachary Sentell"
date: "`r Sys.Date()`"
output:
  html_document:
    df_print: paged
  pdf_document: default
documentclass: book
bibliography:
- book.bib
- packages.bib
biblio-style: apalike
description: This is a pipeline for whole exome sequencing analysis.
link-citations: yes
site: bookdown::bookdown_site
always_allow_html: yes
---

# Overview

This is a whole exome sequencing pipeline to identify the genetic causes of Mendelian disorders in singletons, trios and larger pedigrees. It was assembled in the Kitzler lab at the McGill University Health Centre and run on the Compute Canada servers. It includes the following steps:

1. Pre-processing of raw reads.
2. Germline variant calling for SNVs, indels and structural variants.
4. Pedigree-based, read-based and statistical phasing.
3. Estimation of kinship coefficients to verify pedigree relationships.
4. Variant annotation and trio analysis to identify disease-causing variants.

We follow [GATK best practices](https://gatk.broadinstitute.org/hc/en-us/sections/360007226651-Best-Practices-Workflows) for pre-processing and SNV/indel calling. We use [Manta](https://github.com/Illumina/manta) to call structural variants, [WhatsHap](https://whatshap.readthedocs.io/en/latest/) to phase variants, [Linkdatagen](https://bioinf.wehi.edu.au/software/linkdatagen/) to select SNPs in approximate linkage equilibrium and [OpenMendel](https://openmendel.github.io/) and [PLINK](https://zzz.bwh.harvard.edu/plink/) to estimate kinship coefficients. We use [Ensembl VEP](https://useast.ensembl.org/info/docs/tools/vep/script/index.html) to annotate variants. We provide *R* scripts to identify *de novo* variants as well as compound heterozygous variants *in trans* within a gene and to fetch data from the [gnomAD API](https://gnomad.broadinstitute.org/api/), in particular homozygote counts and phasing estimates based on the EM algorithm. We output the results to Excel in a readable format.

We illustrate the expected output and benchmark the pipeline using Illumina paired-end WES data from the [GIAB](https://www.nist.gov/programs-projects/genome-bottle) Ashkenazi Jewish reference trio ([SRR2962669](https://www.ncbi.nlm.nih.gov/sra/?term=SRR2962669), [SRR2962692](https://www.ncbi.nlm.nih.gov/sra/?term=SRR2962692), [SRR2962694](https://www.ncbi.nlm.nih.gov/sra/?term=SRR2962694)). Bash scripts submitted to Slurm as well as associated data and results can be found on the [GitHub repository](https://github.com/kitzlerlab/WES-pipeline).

```{r include=FALSE}
# automatically create a bib database for R packages
knitr::write_bib(c(
  .packages(), 'bookdown', 'knitr', 'rmarkdown'
), 'packages.bib')
```
