<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>3 Variant Calling | Whole Exome Sequencing Platform</title>
  <meta name="description" content="This is a pipeline for whole exome sequencing analysis." />
  <meta name="generator" content="bookdown 0.35 and GitBook 2.6.7" />

  <meta property="og:title" content="3 Variant Calling | Whole Exome Sequencing Platform" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a pipeline for whole exome sequencing analysis." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="3 Variant Calling | Whole Exome Sequencing Platform" />
  
  <meta name="twitter:description" content="This is a pipeline for whole exome sequencing analysis." />
  

<meta name="author" content="Marc Henein and Zachary Sentell" />


<meta name="date" content="2023-09-05" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="align.html"/>
<link rel="next" href="statistical-genetics.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Whole Exome Sequencing Platform</a></li>
<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Overview</a></li>
<li class="chapter" data-level="2" data-path="align.html"><a href="align.html"><i class="fa fa-check"></i><b>2</b> Pre-processing</a>
<ul>
<li class="chapter" data-level="2.1" data-path="align.html"><a href="align.html#sarek-pipeline"><i class="fa fa-check"></i><b>2.1</b> Sarek Pipeline</a></li>
<li class="chapter" data-level="2.2" data-path="align.html"><a href="align.html#quality-control-of-raw-reads"><i class="fa fa-check"></i><b>2.2</b> Quality Control of Raw Reads</a></li>
<li class="chapter" data-level="2.3" data-path="align.html"><a href="align.html#alignment"><i class="fa fa-check"></i><b>2.3</b> Alignment</a></li>
<li class="chapter" data-level="2.4" data-path="align.html"><a href="align.html#mark-duplicates"><i class="fa fa-check"></i><b>2.4</b> Mark Duplicates</a></li>
<li class="chapter" data-level="2.5" data-path="align.html"><a href="align.html#base-quality-score-recalibration"><i class="fa fa-check"></i><b>2.5</b> Base Quality Score Recalibration</a></li>
<li class="chapter" data-level="2.6" data-path="align.html"><a href="align.html#quality-control-of-alignment"><i class="fa fa-check"></i><b>2.6</b> Quality Control of Alignment</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="calling.html"><a href="calling.html"><i class="fa fa-check"></i><b>3</b> Variant Calling</a>
<ul>
<li class="chapter" data-level="3.1" data-path="calling.html"><a href="calling.html#snvindel-calling"><i class="fa fa-check"></i><b>3.1</b> SNV/Indel Calling</a></li>
<li class="chapter" data-level="3.2" data-path="calling.html"><a href="calling.html#variant-filtration"><i class="fa fa-check"></i><b>3.2</b> Variant Filtration</a></li>
<li class="chapter" data-level="3.3" data-path="calling.html"><a href="calling.html#genotype-posteriors"><i class="fa fa-check"></i><b>3.3</b> Genotype Posteriors</a></li>
<li class="chapter" data-level="3.4" data-path="calling.html"><a href="calling.html#benchmarking"><i class="fa fa-check"></i><b>3.4</b> Benchmarking</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="statistical-genetics.html"><a href="statistical-genetics.html"><i class="fa fa-check"></i><b>4</b> Statistical Genetics</a>
<ul>
<li class="chapter" data-level="4.1" data-path="statistical-genetics.html"><a href="statistical-genetics.html#phasing"><i class="fa fa-check"></i><b>4.1</b> Phasing</a></li>
<li class="chapter" data-level="4.2" data-path="statistical-genetics.html"><a href="statistical-genetics.html#kinship"><i class="fa fa-check"></i><b>4.2</b> Kinship</a></li>
<li class="chapter" data-level="4.3" data-path="statistical-genetics.html"><a href="statistical-genetics.html#runs-of-homozygosity"><i class="fa fa-check"></i><b>4.3</b> Runs of Homozygosity</a></li>
<li class="chapter" data-level="4.4" data-path="statistical-genetics.html"><a href="statistical-genetics.html#linkage-analysis"><i class="fa fa-check"></i><b>4.4</b> Linkage Analysis</a></li>
<li class="chapter" data-level="4.5" data-path="statistical-genetics.html"><a href="statistical-genetics.html#population-substructure"><i class="fa fa-check"></i><b>4.5</b> Population Substructure</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="variant-annotation.html"><a href="variant-annotation.html"><i class="fa fa-check"></i><b>5</b> Variant Annotation</a>
<ul>
<li class="chapter" data-level="5.1" data-path="variant-annotation.html"><a href="variant-annotation.html#filtration"><i class="fa fa-check"></i><b>5.1</b> Filtration</a></li>
<li class="chapter" data-level="5.2" data-path="variant-annotation.html"><a href="variant-annotation.html#de-novo-variants"><i class="fa fa-check"></i><b>5.2</b> <em>De Novo</em> Variants</a></li>
<li class="chapter" data-level="5.3" data-path="variant-annotation.html"><a href="variant-annotation.html#homozygous-variants"><i class="fa fa-check"></i><b>5.3</b> Homozygous Variants</a></li>
<li class="chapter" data-level="5.4" data-path="variant-annotation.html"><a href="variant-annotation.html#compound-heterozygous-variants"><i class="fa fa-check"></i><b>5.4</b> Compound Heterozygous Variants</a></li>
<li class="chapter" data-level="5.5" data-path="variant-annotation.html"><a href="variant-annotation.html#x-chromosome"><i class="fa fa-check"></i><b>5.5</b> X Chromosome</a></li>
<li class="chapter" data-level="5.6" data-path="variant-annotation.html"><a href="variant-annotation.html#disease-genes"><i class="fa fa-check"></i><b>5.6</b> Disease Genes</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="summary.html"><a href="summary.html"><i class="fa fa-check"></i><b>6</b> Summary</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Whole Exome Sequencing Platform</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="calling" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number">3</span> Variant Calling<a href="calling.html#calling" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="snvindel-calling" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> SNV/Indel Calling<a href="calling.html#snvindel-calling" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We call variants with GATK4 HaplotypeCaller.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="calling.html#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="va">exome</span><span class="op">=</span>sureselectV4_padded_hg38.bed</span>
<span id="cb1-2"><a href="calling.html#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="va">names</span><span class="op">=</span><span class="st">&#39;father mother proband&#39;</span> </span>
<span id="cb1-3"><a href="calling.html#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> name <span class="kw">in</span> <span class="va">$names</span></span>
<span id="cb1-4"><a href="calling.html#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb1-5"><a href="calling.html#cb1-5" aria-hidden="true" tabindex="-1"></a>    <span class="va">input</span><span class="op">=</span><span class="st">&quot;BAM/</span><span class="va">${name}</span><span class="st">.recal.bam&quot;</span></span>
<span id="cb1-6"><a href="calling.html#cb1-6" aria-hidden="true" tabindex="-1"></a>    <span class="va">output</span><span class="op">=</span><span class="st">&quot;</span><span class="va">${name}</span><span class="st">_raw_variants.g.vcf.gz&quot;</span></span>
<span id="cb1-7"><a href="calling.html#cb1-7" aria-hidden="true" tabindex="-1"></a>    <span class="ex">java</span> <span class="at">-Xmx6g</span> <span class="at">-jar</span> <span class="va">$GATK_JAR</span> HaplotypeCaller <span class="dt">\</span></span>
<span id="cb1-8"><a href="calling.html#cb1-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">-R</span> <span class="va">$reference</span> <span class="dt">\</span></span>
<span id="cb1-9"><a href="calling.html#cb1-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">-I</span> <span class="va">$input</span> <span class="dt">\</span></span>
<span id="cb1-10"><a href="calling.html#cb1-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">-O</span> <span class="va">$output</span> <span class="dt">\</span></span>
<span id="cb1-11"><a href="calling.html#cb1-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">-L</span> <span class="va">$exome</span> <span class="dt">\</span></span>
<span id="cb1-12"><a href="calling.html#cb1-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">-ip</span> 0 <span class="dt">\</span></span>
<span id="cb1-13"><a href="calling.html#cb1-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">-ERC</span> GVCF</span>
<span id="cb1-14"><a href="calling.html#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span></code></pre></div>
<p>Then, combine gVCFs and compile genotypes at sites with a variant in at least one sample.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="calling.html#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine gVCFs</span></span>
<span id="cb2-2"><a href="calling.html#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">find</span> . <span class="at">-maxdepth</span> 1 <span class="at">-type</span> f <span class="at">-name</span> <span class="st">&quot;*_raw_variants.g.vcf.gz&quot;</span> <span class="op">&gt;</span> input.list</span>
<span id="cb2-3"><a href="calling.html#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="ex">java</span> <span class="at">-Xmx6g</span> <span class="at">-jar</span> <span class="va">$GATK_JAR</span> CombineGVCFs <span class="dt">\</span></span>
<span id="cb2-4"><a href="calling.html#cb2-4" aria-hidden="true" tabindex="-1"></a>-R <span class="va">$reference</span> <span class="dt">\</span></span>
<span id="cb2-5"><a href="calling.html#cb2-5" aria-hidden="true" tabindex="-1"></a>-V input.list <span class="dt">\</span></span>
<span id="cb2-6"><a href="calling.html#cb2-6" aria-hidden="true" tabindex="-1"></a>-O joint/joint_raw_variants.g.vcf.gz</span>
<span id="cb2-7"><a href="calling.html#cb2-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-8"><a href="calling.html#cb2-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Genotype gVCF</span></span>
<span id="cb2-9"><a href="calling.html#cb2-9" aria-hidden="true" tabindex="-1"></a><span class="ex">java</span> <span class="at">-Xmx6g</span> <span class="at">-jar</span> <span class="va">$GATK_JAR</span> GenotypeGVCFs <span class="dt">\</span></span>
<span id="cb2-10"><a href="calling.html#cb2-10" aria-hidden="true" tabindex="-1"></a>-R <span class="va">$reference</span> <span class="dt">\</span></span>
<span id="cb2-11"><a href="calling.html#cb2-11" aria-hidden="true" tabindex="-1"></a>-V joint/joint_raw_variants.g.vcf.gz <span class="dt">\</span></span>
<span id="cb2-12"><a href="calling.html#cb2-12" aria-hidden="true" tabindex="-1"></a>-O joint/joint_raw_variants_genotype.vcf.gz</span></code></pre></div>
</div>
<div id="variant-filtration" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> Variant Filtration<a href="calling.html#variant-filtration" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>For SNVs, we use VQSR to filter variants. VQSR creates a training set of high-confidence variants and learns which region of the parameter space is associated with high quality calls.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb3-1"><a href="calling.html#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Sties only</span></span>
<span id="cb3-2"><a href="calling.html#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="ex">java</span> <span class="at">-Xmx2g</span> <span class="at">-jar</span> <span class="va">$GATK_JAR</span> MakeSitesOnlyVcf <span class="dt">\</span></span>
<span id="cb3-3"><a href="calling.html#cb3-3" aria-hidden="true" tabindex="-1"></a>-I joint/joint_raw_variants_genotype.vcf.gz <span class="dt">\</span></span>
<span id="cb3-4"><a href="calling.html#cb3-4" aria-hidden="true" tabindex="-1"></a>-O joint/joint_raw_variants_genotype_sitesonly.vcf.gz</span>
<span id="cb3-5"><a href="calling.html#cb3-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-6"><a href="calling.html#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="co"># VQSR for SNPs</span></span>
<span id="cb3-7"><a href="calling.html#cb3-7" aria-hidden="true" tabindex="-1"></a><span class="ex">java</span> <span class="at">-Xmx6g</span> <span class="at">-jar</span> <span class="va">$GATK_JAR</span> VariantRecalibrator <span class="dt">\</span></span>
<span id="cb3-8"><a href="calling.html#cb3-8" aria-hidden="true" tabindex="-1"></a>-R <span class="va">$reference</span> <span class="dt">\</span></span>
<span id="cb3-9"><a href="calling.html#cb3-9" aria-hidden="true" tabindex="-1"></a>-V joint/joint_raw_variants_genotype_sitesonly.vcf.gz <span class="dt">\</span></span>
<span id="cb3-10"><a href="calling.html#cb3-10" aria-hidden="true" tabindex="-1"></a>--trust-all-polymorphic <span class="dt">\</span></span>
<span id="cb3-11"><a href="calling.html#cb3-11" aria-hidden="true" tabindex="-1"></a>-tranche 100.0 <span class="at">-tranche</span> 99.95 <span class="at">-tranche</span> 99.9 <span class="at">-tranche</span> 99.8 <span class="at">-tranche</span> 99.6 <span class="at">-tranche</span> 99.5 <span class="at">-tranche</span> 99.4 <span class="at">-tranche</span> 99.3 <span class="at">-tranche</span> 99.0 <span class="at">-tranche</span> 98.0 <span class="at">-tranche</span> 97.0 <span class="at">-tranche</span> 90.0 <span class="dt">\</span></span>
<span id="cb3-12"><a href="calling.html#cb3-12" aria-hidden="true" tabindex="-1"></a>-an QD <span class="at">-an</span> MQRankSum <span class="at">-an</span> ReadPosRankSum <span class="at">-an</span> FS <span class="at">-an</span> MQ <span class="at">-an</span> SOR <span class="at">-an</span> DP <span class="dt">\</span></span>
<span id="cb3-13"><a href="calling.html#cb3-13" aria-hidden="true" tabindex="-1"></a>-resource:hapmap,known=false,training=true,truth=true,prior=15 VQSR/hg38/hapmap_3.3.hg38.vcf.gz <span class="dt">\</span></span>
<span id="cb3-14"><a href="calling.html#cb3-14" aria-hidden="true" tabindex="-1"></a>-resource:omni,known=false,training=true,truth=true,prior=12 VQSR/hg38/1000G_omni2.5.hg38.vcf.gz <span class="dt">\</span></span>
<span id="cb3-15"><a href="calling.html#cb3-15" aria-hidden="true" tabindex="-1"></a>-resource:1000G,known=false,training=true,truth=false,prior=10 VQSR/hg38/1000G_phase1.snps.high_confidence.hg38.vcf.gz <span class="dt">\</span></span>
<span id="cb3-16"><a href="calling.html#cb3-16" aria-hidden="true" tabindex="-1"></a>-resource:dbsnp,known=true,training=false,truth=false,prior=7 VQSR/hg38/Homo_sapiens_assembly38.dbsnp138.vcf <span class="dt">\</span></span>
<span id="cb3-17"><a href="calling.html#cb3-17" aria-hidden="true" tabindex="-1"></a>--max-gaussians 4 <span class="dt">\</span></span>
<span id="cb3-18"><a href="calling.html#cb3-18" aria-hidden="true" tabindex="-1"></a>-mode SNP <span class="dt">\</span></span>
<span id="cb3-19"><a href="calling.html#cb3-19" aria-hidden="true" tabindex="-1"></a>-O joint/cohort_snps.recal <span class="dt">\</span></span>
<span id="cb3-20"><a href="calling.html#cb3-20" aria-hidden="true" tabindex="-1"></a>--tranches-file joint/cohort_snps.tranches <span class="dt">\</span></span>
<span id="cb3-21"><a href="calling.html#cb3-21" aria-hidden="true" tabindex="-1"></a>--rscript-file joint/output_snp.plots.R</span>
<span id="cb3-22"><a href="calling.html#cb3-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-23"><a href="calling.html#cb3-23" aria-hidden="true" tabindex="-1"></a><span class="ex">java</span> <span class="at">-Xmx6g</span> <span class="at">-jar</span> <span class="va">$GATK_JAR</span> ApplyVQSR <span class="dt">\</span></span>
<span id="cb3-24"><a href="calling.html#cb3-24" aria-hidden="true" tabindex="-1"></a>-R <span class="va">$reference</span> <span class="dt">\</span></span>
<span id="cb3-25"><a href="calling.html#cb3-25" aria-hidden="true" tabindex="-1"></a>-V joint/joint_raw_variants_genotype.vcf.gz <span class="dt">\</span></span>
<span id="cb3-26"><a href="calling.html#cb3-26" aria-hidden="true" tabindex="-1"></a>-O joint/snp.recalibrated.vcf.gz <span class="dt">\</span></span>
<span id="cb3-27"><a href="calling.html#cb3-27" aria-hidden="true" tabindex="-1"></a>--recal-file joint/cohort_snps.recal <span class="dt">\</span></span>
<span id="cb3-28"><a href="calling.html#cb3-28" aria-hidden="true" tabindex="-1"></a>--tranches-file joint/cohort_snps.tranches <span class="dt">\</span></span>
<span id="cb3-29"><a href="calling.html#cb3-29" aria-hidden="true" tabindex="-1"></a>--truth-sensitivity-filter-level 99.7 <span class="dt">\</span></span>
<span id="cb3-30"><a href="calling.html#cb3-30" aria-hidden="true" tabindex="-1"></a>--create-output-variant-index true <span class="dt">\</span></span>
<span id="cb3-31"><a href="calling.html#cb3-31" aria-hidden="true" tabindex="-1"></a>-mode SNP</span></code></pre></div>
<p>We manually filter indels by setting hard thresholds due to the small number of samples. For large numbers of samples, we can use VQSR to filter indels.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="calling.html#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Select indels</span></span>
<span id="cb4-2"><a href="calling.html#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="ex">java</span> <span class="at">-Xmx2g</span> <span class="at">-jar</span> <span class="va">$GATK_JAR</span> SelectVariants <span class="dt">\</span></span>
<span id="cb4-3"><a href="calling.html#cb4-3" aria-hidden="true" tabindex="-1"></a>-R <span class="va">$reference</span> <span class="dt">\</span></span>
<span id="cb4-4"><a href="calling.html#cb4-4" aria-hidden="true" tabindex="-1"></a>-V joint/joint_raw_variants_genotype.vcf.gz <span class="dt">\</span></span>
<span id="cb4-5"><a href="calling.html#cb4-5" aria-hidden="true" tabindex="-1"></a>--select-type-to-include INDEL <span class="dt">\</span></span>
<span id="cb4-6"><a href="calling.html#cb4-6" aria-hidden="true" tabindex="-1"></a>-O joint/raw_indels.vcf.gz</span>
<span id="cb4-7"><a href="calling.html#cb4-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-8"><a href="calling.html#cb4-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Indels hard threshold</span></span>
<span id="cb4-9"><a href="calling.html#cb4-9" aria-hidden="true" tabindex="-1"></a><span class="ex">java</span> <span class="at">-Xmx2g</span> <span class="at">-jar</span> <span class="va">$GATK_JAR</span> VariantFiltration <span class="dt">\</span></span>
<span id="cb4-10"><a href="calling.html#cb4-10" aria-hidden="true" tabindex="-1"></a>-R <span class="va">$reference</span> <span class="dt">\</span></span>
<span id="cb4-11"><a href="calling.html#cb4-11" aria-hidden="true" tabindex="-1"></a>-V joint/raw_indels.vcf.gz <span class="dt">\</span></span>
<span id="cb4-12"><a href="calling.html#cb4-12" aria-hidden="true" tabindex="-1"></a>-O joint/filtered_indels.vcf.gz <span class="dt">\</span></span>
<span id="cb4-13"><a href="calling.html#cb4-13" aria-hidden="true" tabindex="-1"></a>-filter <span class="st">&quot;QD &lt; 2.0&quot;</span> <span class="at">--filter-name</span> <span class="st">&quot;QD2&quot;</span> <span class="dt">\</span></span>
<span id="cb4-14"><a href="calling.html#cb4-14" aria-hidden="true" tabindex="-1"></a>-filter <span class="st">&quot;QUAL &lt; 30.0&quot;</span> <span class="at">--filter-name</span> <span class="st">&quot;QUAL30&quot;</span> <span class="dt">\</span></span>
<span id="cb4-15"><a href="calling.html#cb4-15" aria-hidden="true" tabindex="-1"></a>-filter <span class="st">&quot;FS &gt; 200.0&quot;</span> <span class="at">--filter-name</span> <span class="st">&quot;FS200&quot;</span> <span class="dt">\</span></span>
<span id="cb4-16"><a href="calling.html#cb4-16" aria-hidden="true" tabindex="-1"></a>-filter <span class="st">&quot;ReadPosRankSum &lt; -20.0&quot;</span> <span class="at">--filter-name</span> <span class="st">&quot;ReadPosRankSum-20&quot;</span></span></code></pre></div>
<p>We merge SNPs and indels into a single VCF.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb5-1"><a href="calling.html#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="ex">java</span> <span class="at">-Xmx2g</span> <span class="at">-jar</span> <span class="va">$GATK_JAR</span> SelectVariants <span class="dt">\</span></span>
<span id="cb5-2"><a href="calling.html#cb5-2" aria-hidden="true" tabindex="-1"></a>-R <span class="va">$reference</span> <span class="dt">\</span></span>
<span id="cb5-3"><a href="calling.html#cb5-3" aria-hidden="true" tabindex="-1"></a>-V joint/snp.recalibrated.vcf.gz <span class="dt">\</span></span>
<span id="cb5-4"><a href="calling.html#cb5-4" aria-hidden="true" tabindex="-1"></a>--select-type-to-include SNP <span class="dt">\</span></span>
<span id="cb5-5"><a href="calling.html#cb5-5" aria-hidden="true" tabindex="-1"></a>-O joint/snp.recalibrated_minus_indel.vcf.gz</span>
<span id="cb5-6"><a href="calling.html#cb5-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-7"><a href="calling.html#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="ex">java</span> <span class="at">-Xmx2g</span> <span class="at">-jar</span> <span class="va">$GATK_JAR</span> MergeVcfs <span class="dt">\</span></span>
<span id="cb5-8"><a href="calling.html#cb5-8" aria-hidden="true" tabindex="-1"></a>-I joint/snp.recalibrated_minus_indel.vcf.gz <span class="dt">\</span></span>
<span id="cb5-9"><a href="calling.html#cb5-9" aria-hidden="true" tabindex="-1"></a>-I joint/filtered_indels.vcf.gz <span class="dt">\</span></span>
<span id="cb5-10"><a href="calling.html#cb5-10" aria-hidden="true" tabindex="-1"></a>-O joint/output_vqsr_snps_hard_indels.vcf.gz</span></code></pre></div>
</div>
<div id="genotype-posteriors" class="section level2 hasAnchor" number="3.3">
<h2><span class="header-section-number">3.3</span> Genotype Posteriors<a href="calling.html#genotype-posteriors" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We recalculate the likelihood of the genotypes given pedigree relationships and population allele frequencies.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb6-1"><a href="calling.html#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="ex">java</span> <span class="at">-Xmx2g</span> <span class="at">-jar</span> <span class="va">$GATK_JAR</span> CalculateGenotypePosteriors <span class="dt">\</span></span>
<span id="cb6-2"><a href="calling.html#cb6-2" aria-hidden="true" tabindex="-1"></a>-V joint/output_vqsr_snps_hard_indels.vcf.gz <span class="dt">\</span></span>
<span id="cb6-3"><a href="calling.html#cb6-3" aria-hidden="true" tabindex="-1"></a>-ped pedigree.ped <span class="dt">\</span></span>
<span id="cb6-4"><a href="calling.html#cb6-4" aria-hidden="true" tabindex="-1"></a>--supporting-callsets af-only-gnomad.hg38.vcf.gz <span class="dt">\</span></span>
<span id="cb6-5"><a href="calling.html#cb6-5" aria-hidden="true" tabindex="-1"></a>-O trio_refined.vcf.gzb</span></code></pre></div>
</div>
<div id="benchmarking" class="section level2 hasAnchor" number="3.4">
<h2><span class="header-section-number">3.4</span> Benchmarking<a href="calling.html#benchmarking" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We benchmarked the pipeline against the GIAB high-confidence variants calls using the Illumina hap.py VCF comparison tool. We obtain the following results for the proband HG002:</p>
<table>
<thead>
<tr class="header">
<th align="left">Type</th>
<th align="left">Truth TP</th>
<th align="left">Truth FN</th>
<th align="left">Query FP</th>
<th align="left">Recall</th>
<th align="left">Precision</th>
<th align="left">F Score</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">SNV</td>
<td align="left">37653</td>
<td align="left">753</td>
<td align="left">369</td>
<td align="left">98.0%</td>
<td align="left">99.0%</td>
<td align="left">0.985</td>
</tr>
<tr class="even">
<td align="left">INDEL</td>
<td align="left">2732</td>
<td align="left">176</td>
<td align="left">382</td>
<td align="left">93.9%</td>
<td align="left">87.9%</td>
<td align="left">0.908</td>
</tr>
</tbody>
</table>
<p>We note that the F-scores are comparable to those obtained by <span class="citation">Kumaran, Subramanian, and Devarajan (<a href="references.html#ref-kumaran2019performance" role="doc-biblioref">2019</a>)</span>: SNV F-score&gt;0.96 and INDEL F-score&gt;0.9.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="align.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="statistical-genetics.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
