<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>4 Statistical Genetics | Whole Exome Sequencing Platform</title>
  <meta name="description" content="This is a pipeline for whole exome sequencing analysis." />
  <meta name="generator" content="bookdown 0.35 and GitBook 2.6.7" />

  <meta property="og:title" content="4 Statistical Genetics | Whole Exome Sequencing Platform" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a pipeline for whole exome sequencing analysis." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="4 Statistical Genetics | Whole Exome Sequencing Platform" />
  
  <meta name="twitter:description" content="This is a pipeline for whole exome sequencing analysis." />
  

<meta name="author" content="Marc Henein and Zachary Sentell" />


<meta name="date" content="2023-09-05" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="calling.html"/>
<link rel="next" href="variant-annotation.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="libs/pagedtable-1.1/js/pagedtable.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Whole Exome Sequencing Platform</a></li>
<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Overview</a></li>
<li class="chapter" data-level="2" data-path="align.html"><a href="align.html"><i class="fa fa-check"></i><b>2</b> Pre-processing</a>
<ul>
<li class="chapter" data-level="2.1" data-path="align.html"><a href="align.html#sarek-pipeline"><i class="fa fa-check"></i><b>2.1</b> Sarek Pipeline</a></li>
<li class="chapter" data-level="2.2" data-path="align.html"><a href="align.html#quality-control-of-raw-reads"><i class="fa fa-check"></i><b>2.2</b> Quality Control of Raw Reads</a></li>
<li class="chapter" data-level="2.3" data-path="align.html"><a href="align.html#alignment"><i class="fa fa-check"></i><b>2.3</b> Alignment</a></li>
<li class="chapter" data-level="2.4" data-path="align.html"><a href="align.html#mark-duplicates"><i class="fa fa-check"></i><b>2.4</b> Mark Duplicates</a></li>
<li class="chapter" data-level="2.5" data-path="align.html"><a href="align.html#base-quality-score-recalibration"><i class="fa fa-check"></i><b>2.5</b> Base Quality Score Recalibration</a></li>
<li class="chapter" data-level="2.6" data-path="align.html"><a href="align.html#quality-control-of-alignment"><i class="fa fa-check"></i><b>2.6</b> Quality Control of Alignment</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="calling.html"><a href="calling.html"><i class="fa fa-check"></i><b>3</b> Variant Calling</a>
<ul>
<li class="chapter" data-level="3.1" data-path="calling.html"><a href="calling.html#snvindel-calling"><i class="fa fa-check"></i><b>3.1</b> SNV/Indel Calling</a></li>
<li class="chapter" data-level="3.2" data-path="calling.html"><a href="calling.html#variant-filtration"><i class="fa fa-check"></i><b>3.2</b> Variant Filtration</a></li>
<li class="chapter" data-level="3.3" data-path="calling.html"><a href="calling.html#genotype-posteriors"><i class="fa fa-check"></i><b>3.3</b> Genotype Posteriors</a></li>
<li class="chapter" data-level="3.4" data-path="calling.html"><a href="calling.html#benchmarking"><i class="fa fa-check"></i><b>3.4</b> Benchmarking</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="statistical-genetics.html"><a href="statistical-genetics.html"><i class="fa fa-check"></i><b>4</b> Statistical Genetics</a>
<ul>
<li class="chapter" data-level="4.1" data-path="statistical-genetics.html"><a href="statistical-genetics.html#phasing"><i class="fa fa-check"></i><b>4.1</b> Phasing</a></li>
<li class="chapter" data-level="4.2" data-path="statistical-genetics.html"><a href="statistical-genetics.html#kinship"><i class="fa fa-check"></i><b>4.2</b> Kinship</a></li>
<li class="chapter" data-level="4.3" data-path="statistical-genetics.html"><a href="statistical-genetics.html#runs-of-homozygosity"><i class="fa fa-check"></i><b>4.3</b> Runs of Homozygosity</a></li>
<li class="chapter" data-level="4.4" data-path="statistical-genetics.html"><a href="statistical-genetics.html#linkage-analysis"><i class="fa fa-check"></i><b>4.4</b> Linkage Analysis</a></li>
<li class="chapter" data-level="4.5" data-path="statistical-genetics.html"><a href="statistical-genetics.html#population-substructure"><i class="fa fa-check"></i><b>4.5</b> Population Substructure</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="variant-annotation.html"><a href="variant-annotation.html"><i class="fa fa-check"></i><b>5</b> Variant Annotation</a>
<ul>
<li class="chapter" data-level="5.1" data-path="variant-annotation.html"><a href="variant-annotation.html#filtration"><i class="fa fa-check"></i><b>5.1</b> Filtration</a></li>
<li class="chapter" data-level="5.2" data-path="variant-annotation.html"><a href="variant-annotation.html#de-novo-variants"><i class="fa fa-check"></i><b>5.2</b> <em>De Novo</em> Variants</a></li>
<li class="chapter" data-level="5.3" data-path="variant-annotation.html"><a href="variant-annotation.html#homozygous-variants"><i class="fa fa-check"></i><b>5.3</b> Homozygous Variants</a></li>
<li class="chapter" data-level="5.4" data-path="variant-annotation.html"><a href="variant-annotation.html#compound-heterozygous-variants"><i class="fa fa-check"></i><b>5.4</b> Compound Heterozygous Variants</a></li>
<li class="chapter" data-level="5.5" data-path="variant-annotation.html"><a href="variant-annotation.html#x-chromosome"><i class="fa fa-check"></i><b>5.5</b> X Chromosome</a></li>
<li class="chapter" data-level="5.6" data-path="variant-annotation.html"><a href="variant-annotation.html#disease-genes"><i class="fa fa-check"></i><b>5.6</b> Disease Genes</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="summary.html"><a href="summary.html"><i class="fa fa-check"></i><b>6</b> Summary</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Whole Exome Sequencing Platform</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="statistical-genetics" class="section level1 hasAnchor" number="4">
<h1><span class="header-section-number">4</span> Statistical Genetics<a href="statistical-genetics.html#statistical-genetics" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="phasing" class="section level2 hasAnchor" number="4.1">
<h2><span class="header-section-number">4.1</span> Phasing<a href="statistical-genetics.html#phasing" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We use WhatsHap to phase genotypes in a pedigree using the Mendelian transmission rules and physical phasing based on reads that span two or more heterozygous variants.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb15-1"><a href="statistical-genetics.html#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="ex">whatshap</span> phase <span class="at">--ped</span> pedigree.ped <span class="dt">\</span></span>
<span id="cb15-2"><a href="statistical-genetics.html#cb15-2" aria-hidden="true" tabindex="-1"></a>-o phased_whatshap.vcf joint/output_vqsr_variants.vcf.gz <span class="dt">\</span></span>
<span id="cb15-3"><a href="statistical-genetics.html#cb15-3" aria-hidden="true" tabindex="-1"></a>aligned_HG002.bam aligned_HG003.bam aligned_HG004.bam <span class="dt">\</span></span>
<span id="cb15-4"><a href="statistical-genetics.html#cb15-4" aria-hidden="true" tabindex="-1"></a>--tag=PS <span class="at">--no-reference</span> <span class="at">--indels</span></span></code></pre></div>
</div>
<div id="kinship" class="section level2 hasAnchor" number="4.2">
<h2><span class="header-section-number">4.2</span> Kinship<a href="statistical-genetics.html#kinship" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We seek to verify pedigree relationships and identify hidden relatedness between samples. We first select SNPs in approximate linkage equilibrium with <a href="https://bioinf.wehi.edu.au/software/linkdatagen/">Linkdatagen</a> based on the genetic linkage map of HapMap SNPs. Linkdatagen also eliminates Mendelian inconsitencies in the data.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb16-1"><a href="statistical-genetics.html#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Split variant calling file into seperate files for each individual</span></span>
<span id="cb16-2"><a href="statistical-genetics.html#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="ex">bcftools</span> +split AJ_output_vqsr_snps_hard_indels.vcf.gz <span class="at">-Ov</span> <span class="at">-o</span> split</span>
<span id="cb16-3"><a href="statistical-genetics.html#cb16-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-4"><a href="statistical-genetics.html#cb16-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Write path to each individual&#39;s vcf in a text file named MyVCFlist.txt</span></span>
<span id="cb16-5"><a href="statistical-genetics.html#cb16-5" aria-hidden="true" tabindex="-1"></a><span class="fu">perl</span> vcf2linkdatagen.pl <span class="dt">\</span></span>
<span id="cb16-6"><a href="statistical-genetics.html#cb16-6" aria-hidden="true" tabindex="-1"></a>-variantCaller unifiedgenotyper <span class="dt">\</span></span>
<span id="cb16-7"><a href="statistical-genetics.html#cb16-7" aria-hidden="true" tabindex="-1"></a>-annotfile annotHapMap2U_hg38.txt <span class="dt">\</span></span>
<span id="cb16-8"><a href="statistical-genetics.html#cb16-8" aria-hidden="true" tabindex="-1"></a>-pop CEU <span class="at">-mindepth</span> 10 <span class="at">-missingness</span> 0 <span class="dt">\</span></span>
<span id="cb16-9"><a href="statistical-genetics.html#cb16-9" aria-hidden="true" tabindex="-1"></a>-idlist MyVCFlist.txt <span class="op">&gt;</span> HG002_trio.brlmm</span>
<span id="cb16-10"><a href="statistical-genetics.html#cb16-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-11"><a href="statistical-genetics.html#cb16-11" aria-hidden="true" tabindex="-1"></a><span class="fu">perl</span> linkdatagen.pl <span class="dt">\</span></span>
<span id="cb16-12"><a href="statistical-genetics.html#cb16-12" aria-hidden="true" tabindex="-1"></a>-data m <span class="at">-pedfile</span> MyPed.ped <span class="dt">\</span></span>
<span id="cb16-13"><a href="statistical-genetics.html#cb16-13" aria-hidden="true" tabindex="-1"></a>-whichSamplesFile MyWS.ws <span class="dt">\</span></span>
<span id="cb16-14"><a href="statistical-genetics.html#cb16-14" aria-hidden="true" tabindex="-1"></a>-callFile MySNPs.brlmm <span class="dt">\</span></span>
<span id="cb16-15"><a href="statistical-genetics.html#cb16-15" aria-hidden="true" tabindex="-1"></a>-annotFile annotHapMap2U.txt <span class="dt">\</span></span>
<span id="cb16-16"><a href="statistical-genetics.html#cb16-16" aria-hidden="true" tabindex="-1"></a>-pop CEU <span class="at">-binsize</span> 0.3 <span class="dt">\</span></span>
<span id="cb16-17"><a href="statistical-genetics.html#cb16-17" aria-hidden="true" tabindex="-1"></a>-MendelErrors removeSNPs <span class="at">-prog</span> pl <span class="dt">\</span></span>
<span id="cb16-18"><a href="statistical-genetics.html#cb16-18" aria-hidden="true" tabindex="-1"></a>-outputDir MyPed_HapMap2_pl <span class="op">&gt;</span> MyPed_HapMap2_pl.out</span></code></pre></div>
<p>We construct an allele frequency report for the chosen SNPs.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="statistical-genetics.html#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb17-2"><a href="statistical-genetics.html#cb17-2" aria-hidden="true" tabindex="-1"></a>map <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="at">file =</span> <span class="st">&quot;data/plink.map&quot;</span>, <span class="at">delim =</span> <span class="st">&quot; &quot;</span>, <span class="at">col_select =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>, <span class="at">col_names =</span> <span class="fu">c</span>(<span class="st">&quot;CHR&quot;</span>, <span class="st">&quot;SNP&quot;</span>, <span class="st">&quot;cM&quot;</span>, <span class="st">&quot;bp&quot;</span>))</span>
<span id="cb17-3"><a href="statistical-genetics.html#cb17-3" aria-hidden="true" tabindex="-1"></a>ord <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="at">file =</span> <span class="st">&quot;data/orderedSNPs.txt&quot;</span>)</span>
<span id="cb17-4"><a href="statistical-genetics.html#cb17-4" aria-hidden="true" tabindex="-1"></a>freq_pl <span class="ot">&lt;-</span> map <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(ord, <span class="at">by =</span> <span class="st">&quot;SNP&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-5"><a href="statistical-genetics.html#cb17-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(CHR, SNP, <span class="st">`</span><span class="at">Allele freq</span><span class="st">`</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-6"><a href="statistical-genetics.html#cb17-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">A1=</span><span class="st">&quot;1&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">A2=</span><span class="st">&quot;2&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">NCHROBS=</span><span class="st">&quot;300&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-7"><a href="statistical-genetics.html#cb17-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(CHR, SNP, A1, A2, <span class="st">`</span><span class="at">Allele freq</span><span class="st">`</span>, NCHROBS) <span class="sc">%&gt;%</span> </span>
<span id="cb17-8"><a href="statistical-genetics.html#cb17-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">MAF =</span> <span class="st">`</span><span class="at">Allele freq</span><span class="st">`</span>)</span>
<span id="cb17-9"><a href="statistical-genetics.html#cb17-9" aria-hidden="true" tabindex="-1"></a><span class="fu">write_delim</span>(freq_pl, <span class="st">&quot;result/freq_pl.frq&quot;</span>, <span class="at">delim =</span> <span class="st">&quot; &quot;</span>)</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["CHR"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["SNP"],"name":[2],"type":["chr"],"align":["left"]},{"label":["A1"],"name":[3],"type":["chr"],"align":["left"]},{"label":["A2"],"name":[4],"type":["chr"],"align":["left"]},{"label":["MAF"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["NCHROBS"],"name":[6],"type":["chr"],"align":["left"]}],"data":[{"1":"1","2":"rs2905036","3":"1","4":"2","5":"0.000","6":"300"},{"1":"1","2":"rs4074137","3":"1","4":"2","5":"0.623","6":"300"},{"1":"1","2":"rs7515488","3":"1","4":"2","5":"0.853","6":"300"},{"1":"1","2":"rs7519837","3":"1","4":"2","5":"0.695","6":"300"},{"1":"1","2":"rs2803337","3":"1","4":"2","5":"0.491","6":"300"},{"1":"1","2":"rs2503701","3":"1","4":"2","5":"0.394","6":"300"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>Alternatively, with a sufficient number of samples, we can estimate the linkage disequilibrium between alleles and remove correlated SNPs using PLINK.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb18-1"><a href="statistical-genetics.html#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="ex">plink</span> <span class="at">--file</span> data <span class="at">--indep</span> 50 5 2</span></code></pre></div>
<p>We then calculate the empirical kinship coefficient by estimating identity by descent based on identity by state and population allele frequencies <span class="citation">(<a href="references.html#ref-lange2002mathematical" role="doc-biblioref">Lange 2002</a>)</span>. We run this calculation in PLINK.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb19-1"><a href="statistical-genetics.html#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="ex">module</span> load plink/1.9b_6.21-x86_64</span>
<span id="cb19-2"><a href="statistical-genetics.html#cb19-2" aria-hidden="true" tabindex="-1"></a><span class="ex">plink</span> <span class="at">--file</span> plink <span class="at">--genome</span> full <span class="at">--read-freq</span> freq_pl.frq</span></code></pre></div>
<p>We also run this calculation using MendelKinship from the OpenMendel project.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb20-1"><a href="statistical-genetics.html#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="ex">using</span> MendelKinship, CSV</span>
<span id="cb20-2"><a href="statistical-genetics.html#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="ex">Kinship</span><span class="er">(</span><span class="st">&quot;control_compare.txt&quot;</span><span class="kw">)</span></span></code></pre></div>
<p>We obtain the following results for the AJ reference trio using PLINK. PI_HAT is the estimated kinship coefficient between individuals IID1 and IID2. We note that the estimated kinship coefficient between individuals HG003 and HG004, which equals the the inbreeding coefficient of their son HG002, is 0.</p>
<table style="width:100%;">
<colgroup>
<col width="7%" />
<col width="7%" />
<col width="7%" />
<col width="7%" />
<col width="7%" />
<col width="7%" />
<col width="7%" />
<col width="7%" />
<col width="7%" />
<col width="7%" />
<col width="7%" />
<col width="7%" />
<col width="7%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">IID1</th>
<th align="left">IID2</th>
<th align="left">RT</th>
<th align="left">EZ</th>
<th align="left">Z0</th>
<th align="left">Z1</th>
<th align="left">Z2</th>
<th align="left">PI_HAT</th>
<th align="left">DST</th>
<th align="left">IBS0</th>
<th align="left">IBS1</th>
<th align="left">IBS2</th>
<th align="left"></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">HG002</td>
<td align="left">HG003</td>
<td align="left">PO</td>
<td align="left">0.5</td>
<td align="left">0</td>
<td align="left">1</td>
<td align="left">0</td>
<td align="left">0.5</td>
<td align="left">0.752334</td>
<td align="left">0</td>
<td align="left">2918</td>
<td align="left">2973</td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left">HG002</td>
<td align="left">HG004</td>
<td align="left">PO</td>
<td align="left">0.5</td>
<td align="left">0</td>
<td align="left">1</td>
<td align="left">0</td>
<td align="left">0.5</td>
<td align="left">0.756153</td>
<td align="left">0</td>
<td align="left">2873</td>
<td align="left">3018</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">HG003</td>
<td align="left">HG004</td>
<td align="left">OT</td>
<td align="left">0</td>
<td align="left">1</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">0.616449</td>
<td align="left">638</td>
<td align="left">3243</td>
<td align="left">2010</td>
<td align="left"></td>
</tr>
</tbody>
</table>
</div>
<div id="runs-of-homozygosity" class="section level2 hasAnchor" number="4.3">
<h2><span class="header-section-number">4.3</span> Runs of Homozygosity<a href="statistical-genetics.html#runs-of-homozygosity" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>If we suspect some degree of consanguinity, we may look for a homozygous pathogenic variant in runs of homozygosity using <a href="https://teufelsberg.charite.de/AutozygosityMapper/">AutozygosityMapper</a>.</p>
</div>
<div id="linkage-analysis" class="section level2 hasAnchor" number="4.4">
<h2><span class="header-section-number">4.4</span> Linkage Analysis<a href="statistical-genetics.html#linkage-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Using the same set of SNPs in linkage equillibrium, we can run linkage analysis on appropriately chosen pedigrees using OpenMendel for two-point linkage or multipoint linkage analysis.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb21-1"><a href="statistical-genetics.html#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Two-point linkage analysis</span></span>
<span id="cb21-2"><a href="statistical-genetics.html#cb21-2" aria-hidden="true" tabindex="-1"></a><span class="ex">using</span> MendelTwoPointLinkage, CSV</span>
<span id="cb21-3"><a href="statistical-genetics.html#cb21-3" aria-hidden="true" tabindex="-1"></a><span class="ex">TwoPointLinkage</span><span class="er">(</span><span class="st">&quot;Control_file.txt&quot;</span><span class="kw">)</span></span>
<span id="cb21-4"><a href="statistical-genetics.html#cb21-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-5"><a href="statistical-genetics.html#cb21-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Multipoint linkage analysis</span></span>
<span id="cb21-6"><a href="statistical-genetics.html#cb21-6" aria-hidden="true" tabindex="-1"></a><span class="ex">using</span> MendelLocationScores, CSV</span>
<span id="cb21-7"><a href="statistical-genetics.html#cb21-7" aria-hidden="true" tabindex="-1"></a><span class="ex">LocationScores</span><span class="er">(</span><span class="st">&quot;Control_file.txt&quot;</span><span class="kw">)</span></span></code></pre></div>
</div>
<div id="population-substructure" class="section level2 hasAnchor" number="4.5">
<h2><span class="header-section-number">4.5</span> Population Substructure<a href="statistical-genetics.html#population-substructure" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Given a large number of samples from a population, we cluster individuals by degree of identity by state using PLINK in order to identify subgroups with shared ancestry.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb22-1"><a href="statistical-genetics.html#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="ex">plink</span> <span class="at">--file</span> mydata <span class="at">--cluster</span></span>
<span id="cb22-2"><a href="statistical-genetics.html#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="co"># dimensional reduction to 4D then plot two coordinates</span></span>
<span id="cb22-3"><a href="statistical-genetics.html#cb22-3" aria-hidden="true" tabindex="-1"></a><span class="ex">plink</span> <span class="at">--file</span> mydata <span class="at">--read-genome</span> plink.genome <span class="at">--cluster</span> <span class="at">--mds-plot</span> 4</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="calling.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="variant-annotation.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
