<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>5 Variant Annotation | Whole Exome Sequencing Platform</title>
  <meta name="description" content="This is a pipeline for whole exome sequencing analysis." />
  <meta name="generator" content="bookdown 0.35 and GitBook 2.6.7" />

  <meta property="og:title" content="5 Variant Annotation | Whole Exome Sequencing Platform" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a pipeline for whole exome sequencing analysis." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="5 Variant Annotation | Whole Exome Sequencing Platform" />
  
  <meta name="twitter:description" content="This is a pipeline for whole exome sequencing analysis." />
  

<meta name="author" content="Marc Henein and Zachary Sentell" />


<meta name="date" content="2023-09-05" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="statistical-genetics.html"/>
<link rel="next" href="summary.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="libs/pagedtable-1.1/js/pagedtable.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Whole Exome Sequencing Platform</a></li>
<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Overview</a></li>
<li class="chapter" data-level="2" data-path="align.html"><a href="align.html"><i class="fa fa-check"></i><b>2</b> Pre-processing</a>
<ul>
<li class="chapter" data-level="2.1" data-path="align.html"><a href="align.html#sarek-pipeline"><i class="fa fa-check"></i><b>2.1</b> Sarek Pipeline</a></li>
<li class="chapter" data-level="2.2" data-path="align.html"><a href="align.html#quality-control-of-raw-reads"><i class="fa fa-check"></i><b>2.2</b> Quality Control of Raw Reads</a></li>
<li class="chapter" data-level="2.3" data-path="align.html"><a href="align.html#alignment"><i class="fa fa-check"></i><b>2.3</b> Alignment</a></li>
<li class="chapter" data-level="2.4" data-path="align.html"><a href="align.html#mark-duplicates"><i class="fa fa-check"></i><b>2.4</b> Mark Duplicates</a></li>
<li class="chapter" data-level="2.5" data-path="align.html"><a href="align.html#base-quality-score-recalibration"><i class="fa fa-check"></i><b>2.5</b> Base Quality Score Recalibration</a></li>
<li class="chapter" data-level="2.6" data-path="align.html"><a href="align.html#quality-control-of-alignment"><i class="fa fa-check"></i><b>2.6</b> Quality Control of Alignment</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="calling.html"><a href="calling.html"><i class="fa fa-check"></i><b>3</b> Variant Calling</a>
<ul>
<li class="chapter" data-level="3.1" data-path="calling.html"><a href="calling.html#snvindel-calling"><i class="fa fa-check"></i><b>3.1</b> SNV/Indel Calling</a></li>
<li class="chapter" data-level="3.2" data-path="calling.html"><a href="calling.html#variant-filtration"><i class="fa fa-check"></i><b>3.2</b> Variant Filtration</a></li>
<li class="chapter" data-level="3.3" data-path="calling.html"><a href="calling.html#genotype-posteriors"><i class="fa fa-check"></i><b>3.3</b> Genotype Posteriors</a></li>
<li class="chapter" data-level="3.4" data-path="calling.html"><a href="calling.html#benchmarking"><i class="fa fa-check"></i><b>3.4</b> Benchmarking</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="statistical-genetics.html"><a href="statistical-genetics.html"><i class="fa fa-check"></i><b>4</b> Statistical Genetics</a>
<ul>
<li class="chapter" data-level="4.1" data-path="statistical-genetics.html"><a href="statistical-genetics.html#phasing"><i class="fa fa-check"></i><b>4.1</b> Phasing</a></li>
<li class="chapter" data-level="4.2" data-path="statistical-genetics.html"><a href="statistical-genetics.html#kinship"><i class="fa fa-check"></i><b>4.2</b> Kinship</a></li>
<li class="chapter" data-level="4.3" data-path="statistical-genetics.html"><a href="statistical-genetics.html#runs-of-homozygosity"><i class="fa fa-check"></i><b>4.3</b> Runs of Homozygosity</a></li>
<li class="chapter" data-level="4.4" data-path="statistical-genetics.html"><a href="statistical-genetics.html#linkage-analysis"><i class="fa fa-check"></i><b>4.4</b> Linkage Analysis</a></li>
<li class="chapter" data-level="4.5" data-path="statistical-genetics.html"><a href="statistical-genetics.html#population-substructure"><i class="fa fa-check"></i><b>4.5</b> Population Substructure</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="variant-annotation.html"><a href="variant-annotation.html"><i class="fa fa-check"></i><b>5</b> Variant Annotation</a>
<ul>
<li class="chapter" data-level="5.1" data-path="variant-annotation.html"><a href="variant-annotation.html#filtration"><i class="fa fa-check"></i><b>5.1</b> Filtration</a></li>
<li class="chapter" data-level="5.2" data-path="variant-annotation.html"><a href="variant-annotation.html#de-novo-variants"><i class="fa fa-check"></i><b>5.2</b> <em>De Novo</em> Variants</a></li>
<li class="chapter" data-level="5.3" data-path="variant-annotation.html"><a href="variant-annotation.html#homozygous-variants"><i class="fa fa-check"></i><b>5.3</b> Homozygous Variants</a></li>
<li class="chapter" data-level="5.4" data-path="variant-annotation.html"><a href="variant-annotation.html#compound-heterozygous-variants"><i class="fa fa-check"></i><b>5.4</b> Compound Heterozygous Variants</a></li>
<li class="chapter" data-level="5.5" data-path="variant-annotation.html"><a href="variant-annotation.html#x-chromosome"><i class="fa fa-check"></i><b>5.5</b> X Chromosome</a></li>
<li class="chapter" data-level="5.6" data-path="variant-annotation.html"><a href="variant-annotation.html#disease-genes"><i class="fa fa-check"></i><b>5.6</b> Disease Genes</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="summary.html"><a href="summary.html"><i class="fa fa-check"></i><b>6</b> Summary</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Whole Exome Sequencing Platform</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="variant-annotation" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">5</span> Variant Annotation<a href="variant-annotation.html#variant-annotation" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>We annotate the variants with VEP for molecular consequence, allele frequency, clinical variant databases, i.e.Â Clinvar, Mastermind, and a set of pathogenicity prediction tools, i.e., CADD, PrimateAI, SpliceAI, EVE, MutationTaster, REVEL.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb23-1"><a href="variant-annotation.html#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="ex">apptainer</span> exec <span class="at">-B</span> vep_data:/opt/vep/.vep vep.sif /opt/vep/src/ensembl-vep/vep <span class="dt">\</span></span>
<span id="cb23-2"><a href="variant-annotation.html#cb23-2" aria-hidden="true" tabindex="-1"></a>--dir_cache /opt/vep/.vep/ <span class="dt">\</span></span>
<span id="cb23-3"><a href="variant-annotation.html#cb23-3" aria-hidden="true" tabindex="-1"></a>--dir_plugins /opt/vep/.vep/Plugins/ <span class="dt">\</span></span>
<span id="cb23-4"><a href="variant-annotation.html#cb23-4" aria-hidden="true" tabindex="-1"></a>-i /opt/vep/.vep/output_vqsr_snps_hard_indels.vcf.gz <span class="dt">\</span></span>
<span id="cb23-5"><a href="variant-annotation.html#cb23-5" aria-hidden="true" tabindex="-1"></a>-o /opt/vep/.vep/ensembl_annotated.txt <span class="dt">\</span></span>
<span id="cb23-6"><a href="variant-annotation.html#cb23-6" aria-hidden="true" tabindex="-1"></a>--individual EC27_proband <span class="dt">\</span></span>
<span id="cb23-7"><a href="variant-annotation.html#cb23-7" aria-hidden="true" tabindex="-1"></a>--everything <span class="dt">\</span></span>
<span id="cb23-8"><a href="variant-annotation.html#cb23-8" aria-hidden="true" tabindex="-1"></a>--check_existing <span class="dt">\</span></span>
<span id="cb23-9"><a href="variant-annotation.html#cb23-9" aria-hidden="true" tabindex="-1"></a>--xref_refseq <span class="dt">\</span></span>
<span id="cb23-10"><a href="variant-annotation.html#cb23-10" aria-hidden="true" tabindex="-1"></a>--fasta /opt/vep/.vep/Homo_sapiens.GRCh38.fa <span class="dt">\</span></span>
<span id="cb23-11"><a href="variant-annotation.html#cb23-11" aria-hidden="true" tabindex="-1"></a>--hgvs <span class="dt">\</span></span>
<span id="cb23-12"><a href="variant-annotation.html#cb23-12" aria-hidden="true" tabindex="-1"></a>--hgvsg <span class="dt">\</span></span>
<span id="cb23-13"><a href="variant-annotation.html#cb23-13" aria-hidden="true" tabindex="-1"></a>--plugin CADD,/opt/vep/.vep/Plugins/data/grch38/whole_genome_SNVs_grch38.tsv.gz,/opt/vep/.vep/Plugins/data/grch38/gnomad.genomes.r3.0.indel_grch38.tsv.gz <span class="dt">\</span></span>
<span id="cb23-14"><a href="variant-annotation.html#cb23-14" aria-hidden="true" tabindex="-1"></a>--plugin PrimateAI,/opt/vep/.vep/Plugins/data/grch38/PrimateAI_scores_v0.2_GRCh38_sorted.tsv.bgz <span class="dt">\</span></span>
<span id="cb23-15"><a href="variant-annotation.html#cb23-15" aria-hidden="true" tabindex="-1"></a>--plugin SpliceAI,snv=/opt/vep/.vep/Plugins/data/grch38/spliceai_scores.raw.snv.hg38.vcf.gz,indel=/opt/vep/.vep/Plugins/data/grch38/spliceai_scores.raw.indel.hg38.vcf.gz <span class="dt">\</span></span>
<span id="cb23-16"><a href="variant-annotation.html#cb23-16" aria-hidden="true" tabindex="-1"></a>--plugin EVE,file=/opt/vep/.vep/Plugins/data/grch38/eve_merged.vcf.gz <span class="dt">\</span></span>
<span id="cb23-17"><a href="variant-annotation.html#cb23-17" aria-hidden="true" tabindex="-1"></a>--plugin Mastermind,/opt/vep/.vep/Plugins/data/grch38/mastermind_cited_variants_reference-2023.04.02-grch38.vcf.gz <span class="dt">\</span></span>
<span id="cb23-18"><a href="variant-annotation.html#cb23-18" aria-hidden="true" tabindex="-1"></a>--plugin dbNSFP,/opt/vep/.vep/Plugins/data/grch38/dbNSFP4.3a_grch38.gz,MutationTaster_score,MutationTaster_pred <span class="dt">\</span></span>
<span id="cb23-19"><a href="variant-annotation.html#cb23-19" aria-hidden="true" tabindex="-1"></a>--plugin REVEL,/opt/vep/.vep/Plugins/data/grch38/new_tabbed_revel_grch38.tsv.gz <span class="dt">\</span></span>
<span id="cb23-20"><a href="variant-annotation.html#cb23-20" aria-hidden="true" tabindex="-1"></a>--custom /opt/vep/.vep/Plugins/data/grch38/clinvar.vcf.gz,ClinVar,vcf,exact,0,CLNSIG,CLNREVSTAT,CLNDN <span class="dt">\</span></span>
<span id="cb23-21"><a href="variant-annotation.html#cb23-21" aria-hidden="true" tabindex="-1"></a>--cache <span class="dt">\</span></span>
<span id="cb23-22"><a href="variant-annotation.html#cb23-22" aria-hidden="true" tabindex="-1"></a>--merged <span class="dt">\</span></span>
<span id="cb23-23"><a href="variant-annotation.html#cb23-23" aria-hidden="true" tabindex="-1"></a>--offline <span class="dt">\</span></span>
<span id="cb23-24"><a href="variant-annotation.html#cb23-24" aria-hidden="true" tabindex="-1"></a>--use_given_ref <span class="dt">\</span></span>
<span id="cb23-25"><a href="variant-annotation.html#cb23-25" aria-hidden="true" tabindex="-1"></a>--force_overwrite <span class="dt">\</span></span>
<span id="cb23-26"><a href="variant-annotation.html#cb23-26" aria-hidden="true" tabindex="-1"></a>--assembly GRCh38 <span class="dt">\</span></span>
<span id="cb23-27"><a href="variant-annotation.html#cb23-27" aria-hidden="true" tabindex="-1"></a>--fork 2 <span class="dt">\</span></span>
<span id="cb23-28"><a href="variant-annotation.html#cb23-28" aria-hidden="true" tabindex="-1"></a>--buffer_size 1000 <span class="dt">\</span></span>
<span id="cb23-29"><a href="variant-annotation.html#cb23-29" aria-hidden="true" tabindex="-1"></a>--tab</span></code></pre></div>
<div id="filtration" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> Filtration<a href="variant-annotation.html#filtration" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We perform a first round of filtering with VEP to reduce the size of the VCF to be loaded into R. In particular, we select variants with high or moderate impact and a gnomAD exome allele frequency smaller than 0.05.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb24-1"><a href="variant-annotation.html#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="ex">apptainer</span> exec <span class="at">-B</span> vep_data:/opt/vep/.vep vep.sif /opt/vep/src/ensembl-vep/filter_vep <span class="dt">\</span></span>
<span id="cb24-2"><a href="variant-annotation.html#cb24-2" aria-hidden="true" tabindex="-1"></a>-i /opt/vep/.vep/ensembl_annotated.txt <span class="dt">\</span></span>
<span id="cb24-3"><a href="variant-annotation.html#cb24-3" aria-hidden="true" tabindex="-1"></a>-out /opt/vep/.vep/ensembl_ann_vep_filt.txt <span class="dt">\</span></span>
<span id="cb24-4"><a href="variant-annotation.html#cb24-4" aria-hidden="true" tabindex="-1"></a>--filter <span class="st">&quot;(IMPACT is HIGH or IMPACT is MODERATE) and (gnomADe_AF &lt;= 0.05 or not gnomADe_AF)&quot;</span> <span class="dt">\</span></span>
<span id="cb24-5"><a href="variant-annotation.html#cb24-5" aria-hidden="true" tabindex="-1"></a>--force_overwrite <span class="dt">\</span></span>
<span id="cb24-6"><a href="variant-annotation.html#cb24-6" aria-hidden="true" tabindex="-1"></a>--format tab </span></code></pre></div>
<p>We load the phased variants into R and select autosomal variants that have passed all variant calling filters.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="variant-annotation.html#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb25-2"><a href="variant-annotation.html#cb25-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(VariantAnnotation)</span>
<span id="cb25-3"><a href="variant-annotation.html#cb25-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Set sample names</span></span>
<span id="cb25-4"><a href="variant-annotation.html#cb25-4" aria-hidden="true" tabindex="-1"></a>mother <span class="ot">=</span> <span class="st">&quot;HG004&quot;</span></span>
<span id="cb25-5"><a href="variant-annotation.html#cb25-5" aria-hidden="true" tabindex="-1"></a>father <span class="ot">=</span> <span class="st">&quot;HG003&quot;</span></span>
<span id="cb25-6"><a href="variant-annotation.html#cb25-6" aria-hidden="true" tabindex="-1"></a>proband <span class="ot">=</span> <span class="st">&quot;HG002&quot;</span></span>
<span id="cb25-7"><a href="variant-annotation.html#cb25-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Load WhatsHap phased variants</span></span>
<span id="cb25-8"><a href="variant-annotation.html#cb25-8" aria-hidden="true" tabindex="-1"></a>wh <span class="ot">&lt;-</span> <span class="fu">readVcf</span>(<span class="st">&quot;data/AJ_phased_whatshap_refined.vcf&quot;</span>, <span class="st">&quot;hg38&quot;</span>)</span>
<span id="cb25-9"><a href="variant-annotation.html#cb25-9" aria-hidden="true" tabindex="-1"></a>wh <span class="ot">&lt;-</span> <span class="fu">expand</span>(wh)</span>
<span id="cb25-10"><a href="variant-annotation.html#cb25-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Variant info</span></span>
<span id="cb25-11"><a href="variant-annotation.html#cb25-11" aria-hidden="true" tabindex="-1"></a>pos <span class="ot">&lt;-</span> <span class="fu">rowRanges</span>(wh) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>()</span>
<span id="cb25-12"><a href="variant-annotation.html#cb25-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Genotype</span></span>
<span id="cb25-13"><a href="variant-annotation.html#cb25-13" aria-hidden="true" tabindex="-1"></a>wh_geno <span class="ot">&lt;-</span> <span class="fu">geno</span>(wh)<span class="sc">$</span>GT <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb25-14"><a href="variant-annotation.html#cb25-14" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;mother&quot;</span><span class="ot">=</span> <span class="fu">sym</span>(mother), <span class="st">&quot;father&quot;</span><span class="ot">=</span> <span class="fu">sym</span>(father), <span class="st">&quot;proband&quot;</span><span class="ot">=</span> <span class="fu">sym</span>(proband))</span>
<span id="cb25-15"><a href="variant-annotation.html#cb25-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Phasing block</span></span>
<span id="cb25-16"><a href="variant-annotation.html#cb25-16" aria-hidden="true" tabindex="-1"></a>wh_ps <span class="ot">&lt;-</span> <span class="fu">geno</span>(wh)<span class="sc">$</span>PS <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb25-17"><a href="variant-annotation.html#cb25-17" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;PS_mother&quot;</span> <span class="ot">=</span> <span class="fu">sym</span>(mother), <span class="st">&quot;PS_father&quot;</span><span class="ot">=</span> <span class="fu">sym</span>(father), <span class="st">&quot;PS_proband&quot;</span><span class="ot">=</span> <span class="fu">sym</span>(proband))</span>
<span id="cb25-18"><a href="variant-annotation.html#cb25-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Genotype quality</span></span>
<span id="cb25-19"><a href="variant-annotation.html#cb25-19" aria-hidden="true" tabindex="-1"></a>wh_gq <span class="ot">&lt;-</span> <span class="fu">geno</span>(wh)<span class="sc">$</span>GQ <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb25-20"><a href="variant-annotation.html#cb25-20" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;GQ_mother&quot;</span> <span class="ot">=</span> <span class="fu">sym</span>(mother), <span class="st">&quot;GQ_father&quot;</span><span class="ot">=</span> <span class="fu">sym</span>(father), <span class="st">&quot;GQ_proband&quot;</span><span class="ot">=</span> <span class="fu">sym</span>(proband))</span>
<span id="cb25-21"><a href="variant-annotation.html#cb25-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Allelic depth</span></span>
<span id="cb25-22"><a href="variant-annotation.html#cb25-22" aria-hidden="true" tabindex="-1"></a>wh_ad <span class="ot">&lt;-</span> <span class="fu">geno</span>(wh)<span class="sc">$</span>AD <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb25-23"><a href="variant-annotation.html#cb25-23" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;AD_ref_mother&quot;</span> <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">sym</span>(mother),<span class="st">&quot;.1&quot;</span>), <span class="st">&quot;AD_ref_father&quot;</span><span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">sym</span>(father),<span class="st">&quot;.1&quot;</span>), <span class="st">&quot;AD_ref_proband&quot;</span><span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">sym</span>(proband),<span class="st">&quot;.1&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb25-24"><a href="variant-annotation.html#cb25-24" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;AD_alt_mother&quot;</span> <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">sym</span>(mother),<span class="st">&quot;.2&quot;</span>), <span class="st">&quot;AD_alt_father&quot;</span><span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">sym</span>(father),<span class="st">&quot;.2&quot;</span>), <span class="st">&quot;AD_alt_proband&quot;</span><span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">sym</span>(proband),<span class="st">&quot;.2&quot;</span>))</span>
<span id="cb25-25"><a href="variant-annotation.html#cb25-25" aria-hidden="true" tabindex="-1"></a><span class="co"># Join variant info, genotype, phasing block, genotype quality and allelic depth</span></span>
<span id="cb25-26"><a href="variant-annotation.html#cb25-26" aria-hidden="true" tabindex="-1"></a>wh_geno0 <span class="ot">&lt;-</span> <span class="fu">bind_cols</span>(pos,wh_geno,wh_gq,wh_ps,wh_ad)</span>
<span id="cb25-27"><a href="variant-annotation.html#cb25-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify alternate alleles with number</span></span>
<span id="cb25-28"><a href="variant-annotation.html#cb25-28" aria-hidden="true" tabindex="-1"></a>wh_geno01 <span class="ot">&lt;-</span> wh_geno0 <span class="sc">%&gt;%</span> </span>
<span id="cb25-29"><a href="variant-annotation.html#cb25-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(seqnames,start,end,width,REF) <span class="sc">%&gt;%</span> </span>
<span id="cb25-30"><a href="variant-annotation.html#cb25-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">counter =</span> <span class="fu">row_number</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb25-31"><a href="variant-annotation.html#cb25-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb25-32"><a href="variant-annotation.html#cb25-32" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(FILTER<span class="sc">==</span><span class="st">&quot;PASS&quot;</span>)</span>
<span id="cb25-33"><a href="variant-annotation.html#cb25-33" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert variant formatting to VEP format. Add 1 to indel start site.</span></span>
<span id="cb25-34"><a href="variant-annotation.html#cb25-34" aria-hidden="true" tabindex="-1"></a>wh_geno01 <span class="ot">&lt;-</span> wh_geno01 <span class="sc">%&gt;%</span> </span>
<span id="cb25-35"><a href="variant-annotation.html#cb25-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Location =</span> <span class="fu">ifelse</span>(</span>
<span id="cb25-36"><a href="variant-annotation.html#cb25-36" aria-hidden="true" tabindex="-1"></a>    width<span class="sc">==</span><span class="dv">1</span>, <span class="fu">ifelse</span>(</span>
<span id="cb25-37"><a href="variant-annotation.html#cb25-37" aria-hidden="true" tabindex="-1"></a>      <span class="fu">nchar</span>(ALT)<span class="sc">==</span><span class="dv">1</span>, <span class="fu">paste</span>(seqnames,start,<span class="at">sep=</span><span class="st">&quot;:&quot;</span>), <span class="fu">paste0</span>(seqnames,<span class="st">&quot;:&quot;</span>,start,<span class="st">&quot;-&quot;</span>,start<span class="sc">+</span><span class="dv">1</span>)), <span class="fu">ifelse</span>(</span>
<span id="cb25-38"><a href="variant-annotation.html#cb25-38" aria-hidden="true" tabindex="-1"></a>        start<span class="sc">+</span><span class="dv">1</span><span class="sc">==</span>end, <span class="fu">paste0</span>(seqnames,<span class="st">&quot;:&quot;</span>,start<span class="sc">+</span><span class="dv">1</span>), <span class="fu">paste0</span>(seqnames,<span class="st">&quot;:&quot;</span>,start<span class="sc">+</span><span class="dv">1</span>,<span class="st">&quot;-&quot;</span>,end)))) <span class="sc">%&gt;%</span> </span>
<span id="cb25-39"><a href="variant-annotation.html#cb25-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Allele =</span> <span class="fu">ifelse</span>(<span class="fu">substr</span>(REF,<span class="dv">1</span>,<span class="dv">1</span>)<span class="sc">==</span><span class="fu">substr</span>(ALT,<span class="dv">1</span>,<span class="dv">1</span>), <span class="fu">substr</span>(ALT,<span class="dv">2</span>,<span class="fu">nchar</span>(ALT)), ALT))</span>
<span id="cb25-40"><a href="variant-annotation.html#cb25-40" aria-hidden="true" tabindex="-1"></a>wh_geno01<span class="sc">$</span>Allele[wh_geno01<span class="sc">$</span>Allele<span class="sc">==</span><span class="st">&quot;&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;-&quot;</span></span>
<span id="cb25-41"><a href="variant-annotation.html#cb25-41" aria-hidden="true" tabindex="-1"></a><span class="co"># Overlapping deletions</span></span>
<span id="cb25-42"><a href="variant-annotation.html#cb25-42" aria-hidden="true" tabindex="-1"></a>wh_geno_overlap <span class="ot">&lt;-</span> wh_geno01 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(ALT<span class="sc">==</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb25-43"><a href="variant-annotation.html#cb25-43" aria-hidden="true" tabindex="-1"></a>wh_geno1 <span class="ot">&lt;-</span> wh_geno01 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(ALT<span class="sc">!=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb25-44"><a href="variant-annotation.html#cb25-44" aria-hidden="true" tabindex="-1"></a><span class="co"># Select columns</span></span>
<span id="cb25-45"><a href="variant-annotation.html#cb25-45" aria-hidden="true" tabindex="-1"></a>wh_geno1 <span class="ot">&lt;-</span> wh_geno1[<span class="fu">c</span>(<span class="dv">27</span>,<span class="dv">28</span>,<span class="dv">11</span><span class="sc">:</span><span class="dv">25</span>,<span class="dv">26</span>)]</span>
<span id="cb25-46"><a href="variant-annotation.html#cb25-46" aria-hidden="true" tabindex="-1"></a><span class="co"># Select X chromosome</span></span>
<span id="cb25-47"><a href="variant-annotation.html#cb25-47" aria-hidden="true" tabindex="-1"></a>wh_geno_X <span class="ot">&lt;-</span> wh_geno1 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">str_detect</span>(Location,<span class="st">&quot;chrX&quot;</span>))</span>
<span id="cb25-48"><a href="variant-annotation.html#cb25-48" aria-hidden="true" tabindex="-1"></a><span class="co"># Select autosomal variants</span></span>
<span id="cb25-49"><a href="variant-annotation.html#cb25-49" aria-hidden="true" tabindex="-1"></a>wh_geno_auto <span class="ot">&lt;-</span> wh_geno1 <span class="sc">%&gt;%</span></span>
<span id="cb25-50"><a href="variant-annotation.html#cb25-50" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(Location,<span class="st">&quot;chrX&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb25-51"><a href="variant-annotation.html#cb25-51" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(Location,<span class="st">&quot;chrY&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb25-52"><a href="variant-annotation.html#cb25-52" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(Location,<span class="st">&quot;M&quot;</span>))</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["Location"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Allele"],"name":[2],"type":["chr"],"align":["left"]},{"label":["proband"],"name":[3],"type":["chr"],"align":["left"]},{"label":["father"],"name":[4],"type":["chr"],"align":["left"]},{"label":["mother"],"name":[5],"type":["chr"],"align":["left"]},{"label":["GQ_proband"],"name":[6],"type":["int"],"align":["right"]},{"label":["GQ_father"],"name":[7],"type":["int"],"align":["right"]},{"label":["GQ_mother"],"name":[8],"type":["int"],"align":["right"]},{"label":["PS_proband"],"name":[9],"type":["int"],"align":["right"]},{"label":["PS_father"],"name":[10],"type":["int"],"align":["right"]},{"label":["PS_mother"],"name":[11],"type":["int"],"align":["right"]},{"label":["AD_ref_proband"],"name":[12],"type":["int"],"align":["right"]},{"label":["AD_ref_father"],"name":[13],"type":["int"],"align":["right"]},{"label":["AD_ref_mother"],"name":[14],"type":["int"],"align":["right"]},{"label":["AD_alt_proband"],"name":[15],"type":["int"],"align":["right"]},{"label":["AD_alt_father"],"name":[16],"type":["int"],"align":["right"]},{"label":["AD_alt_mother"],"name":[17],"type":["int"],"align":["right"]},{"label":["counter"],"name":[18],"type":["int"],"align":["right"]}],"data":[{"1":"chr1:826893","2":"A","3":"1/1","4":"1/1","5":"0|1","6":"99","7":"99","8":"99","9":"NA","10":"NA","11":"65797","12":"0","13":"0","14":"130","15":"279","16":"232","17":"124","18":"1"},{"1":"chr1:827092","2":"T","3":"0/0","4":"0/0","5":"1|0","6":"68","7":"65","8":"99","9":"NA","10":"NA","11":"65797","12":"40","13":"37","14":"33","15":"0","16":"0","17":"6","18":"1"},{"1":"chr1:827209","2":"C","3":"1/1","4":"1/1","5":"0|1","6":"77","7":"99","8":"87","9":"NA","10":"NA","11":"65797","12":"1","13":"0","14":"4","15":"21","16":"24","17":"13","18":"1"},{"1":"chr1:827212","2":"G","3":"1/1","4":"1/1","5":"0|1","6":"99","7":"99","8":"87","9":"NA","10":"NA","11":"65797","12":"0","13":"0","14":"4","15":"19","16":"22","17":"13","18":"1"},{"1":"chr1:827221","2":"C","3":"1/1","4":"1/1","5":"0|1","6":"90","7":"90","8":"90","9":"NA","10":"NA","11":"65797","12":"0","13":"0","14":"4","15":"16","16":"16","17":"12","18":"1"},{"1":"chr1:827252","2":"A","3":"1/1","4":"1/1","5":"0|1","6":"78","7":"78","8":"23","9":"NA","10":"NA","11":"65797","12":"0","13":"0","14":"3","15":"12","16":"12","17":"5","18":"1"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>Manually remove the hashtag before the line of column titles in ensembl_ann_vep_filt.txt. We load the VEP annotated variants into R and select variants with an allele frequency &lt;0.01 that fall in RefSeq transcripts.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="variant-annotation.html#cb26-1" aria-hidden="true" tabindex="-1"></a>result <span class="ot">&lt;-</span> <span class="fu">read_table</span>(<span class="st">&quot;data/AJ_ensembl_ann_vep_filt_09_01_23.txt&quot;</span>, <span class="at">comment =</span> <span class="st">&quot;#&quot;</span>, <span class="at">na=</span><span class="st">&quot;-&quot;</span>, <span class="at">col_types =</span> <span class="fu">list</span>(</span>
<span id="cb26-2"><a href="variant-annotation.html#cb26-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">Uploaded_variation=</span> <span class="fu">col_factor</span>(), <span class="at">Gene=</span><span class="fu">col_factor</span>(), <span class="at">Feature=</span><span class="fu">col_factor</span>(), <span class="at">MANE_PLUS_CLINICAL =</span> <span class="fu">col_character</span>(), </span>
<span id="cb26-3"><a href="variant-annotation.html#cb26-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">MOTIF_NAME =</span> <span class="fu">col_character</span>(), <span class="at">HIGH_INF_POS =</span> <span class="fu">col_character</span>(), <span class="at">TRANSCRIPTION_FACTORS =</span> <span class="fu">col_character</span>(), </span>
<span id="cb26-4"><a href="variant-annotation.html#cb26-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">PUBMED=</span><span class="fu">col_character</span>(), <span class="at">MOTIF_POS =</span> <span class="fu">col_character</span>()))</span>
<span id="cb26-5"><a href="variant-annotation.html#cb26-5" aria-hidden="true" tabindex="-1"></a>result<span class="sc">$</span>Allele[<span class="fu">is.na</span>(result<span class="sc">$</span>Allele)] <span class="ot">&lt;-</span> <span class="st">&quot;-&quot;</span></span>
<span id="cb26-6"><a href="variant-annotation.html#cb26-6" aria-hidden="true" tabindex="-1"></a>result1 <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span> as_tibble <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(wh_geno1, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;Location&quot;</span>,<span class="st">&quot;Allele&quot;</span>))</span>
<span id="cb26-7"><a href="variant-annotation.html#cb26-7" aria-hidden="true" tabindex="-1"></a>refseq0 <span class="ot">&lt;-</span> result1 <span class="sc">%&gt;%</span> </span>
<span id="cb26-8"><a href="variant-annotation.html#cb26-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(SOURCE <span class="sc">==</span> <span class="st">&quot;RefSeq&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb26-9"><a href="variant-annotation.html#cb26-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(gnomADe_AF<span class="sc">&lt;=</span><span class="fl">0.01</span><span class="sc">|</span><span class="fu">is.na</span>(gnomADe_AF)) <span class="sc">%&gt;%</span> </span>
<span id="cb26-10"><a href="variant-annotation.html#cb26-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>(<span class="fu">is.na</span>(gnomADe_AF) <span class="sc">&amp;</span> gnomADg_AF <span class="sc">&gt;</span> <span class="fl">0.01</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb26-11"><a href="variant-annotation.html#cb26-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>(<span class="fu">str_detect</span>(Feature,<span class="st">&quot;XM&quot;</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb26-12"><a href="variant-annotation.html#cb26-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select_if</span>(<span class="sc">~</span><span class="fu">any</span>(<span class="sc">!</span><span class="fu">is.na</span>(.)))</span></code></pre></div>
<p>We add gene-level information from OMIM, Human Phenotype Ontology, ClinGen, GenCC, Orphanet.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="variant-annotation.html#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ontologyIndex)</span>
<span id="cb27-2"><a href="variant-annotation.html#cb27-2" aria-hidden="true" tabindex="-1"></a><span class="co"># OMIM</span></span>
<span id="cb27-3"><a href="variant-annotation.html#cb27-3" aria-hidden="true" tabindex="-1"></a>omim <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">&quot;data/genemap2_edit.txt&quot;</span>, <span class="at">name_repair=</span><span class="st">&quot;universal&quot;</span>, <span class="at">col_types=</span><span class="fu">list</span>(<span class="at">Entrez.Gene.ID =</span> <span class="fu">col_factor</span>())) <span class="sc">%&gt;%</span> </span>
<span id="cb27-4"><a href="variant-annotation.html#cb27-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">OMIM_Phenotypes =</span> Phenotypes) <span class="sc">%&gt;%</span> </span>
<span id="cb27-5"><a href="variant-annotation.html#cb27-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(Entrez.Gene.ID,OMIM_Phenotypes)</span>
<span id="cb27-6"><a href="variant-annotation.html#cb27-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Human Phenotype Ontology</span></span>
<span id="cb27-7"><a href="variant-annotation.html#cb27-7" aria-hidden="true" tabindex="-1"></a>hpo_data <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">&quot;data/genes_to_phenotype.txt&quot;</span>, <span class="at">col_types=</span><span class="fu">list</span>(<span class="at">entrez_gene_id =</span> <span class="fu">col_factor</span>()))</span>
<span id="cb27-8"><a href="variant-annotation.html#cb27-8" aria-hidden="true" tabindex="-1"></a>hpo <span class="ot">&lt;-</span> <span class="fu">get_ontology</span>(<span class="at">file =</span> <span class="st">&#39;data/hp_obo.txt&#39;</span>, <span class="at">propagate_relationships =</span> <span class="st">&quot;is_a&quot;</span>, <span class="at">extract_tags =</span> <span class="st">&quot;minimal&quot;</span>)</span>
<span id="cb27-9"><a href="variant-annotation.html#cb27-9" aria-hidden="true" tabindex="-1"></a>hpo_kidney <span class="ot">&lt;-</span> <span class="fu">get_descendants</span>(hpo, <span class="at">roots=</span><span class="st">&quot;HP:0000077&quot;</span>) <span class="co">#Abnormality of the kidney</span></span>
<span id="cb27-10"><a href="variant-annotation.html#cb27-10" aria-hidden="true" tabindex="-1"></a>gen_hpo_kidney <span class="ot">&lt;-</span> hpo_data <span class="sc">%&gt;%</span> </span>
<span id="cb27-11"><a href="variant-annotation.html#cb27-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(hpo_term_id <span class="sc">%in%</span> hpo_kidney) <span class="sc">%&gt;%</span> </span>
<span id="cb27-12"><a href="variant-annotation.html#cb27-12" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(frequency_hpo) <span class="sc">%&gt;%</span> </span>
<span id="cb27-13"><a href="variant-annotation.html#cb27-13" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">frequency =</span> <span class="fu">case_when</span>(</span>
<span id="cb27-14"><a href="variant-annotation.html#cb27-14" aria-hidden="true" tabindex="-1"></a>    frequency_hpo <span class="sc">==</span> <span class="st">&quot;HP:0040284&quot;</span> <span class="sc">~</span> <span class="st">&quot; (very rare)&quot;</span>, frequency_hpo <span class="sc">==</span> <span class="st">&quot;HP:0040283&quot;</span> <span class="sc">~</span> <span class="st">&quot; (occasional)&quot;</span>, </span>
<span id="cb27-15"><a href="variant-annotation.html#cb27-15" aria-hidden="true" tabindex="-1"></a>    frequency_hpo <span class="sc">==</span> <span class="st">&quot;HP:0040282&quot;</span> <span class="sc">~</span> <span class="st">&quot; (frequent)&quot;</span>, frequency_hpo <span class="sc">==</span> <span class="st">&quot;HP:0040281&quot;</span> <span class="sc">~</span> <span class="st">&quot; (very frequent)&quot;</span>, </span>
<span id="cb27-16"><a href="variant-annotation.html#cb27-16" aria-hidden="true" tabindex="-1"></a>    frequency_hpo <span class="sc">==</span> <span class="st">&quot;HP:0040280&quot;</span> <span class="sc">~</span> <span class="st">&quot; (obligate)&quot;</span>, <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">&quot;&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb27-17"><a href="variant-annotation.html#cb27-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(<span class="st">&quot;renal_hpo_term&quot;</span>, <span class="fu">c</span>(hpo_term_name, frequency), <span class="at">remove =</span> <span class="cn">FALSE</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb27-18"><a href="variant-annotation.html#cb27-18" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(entrez_gene_id) <span class="sc">%&gt;%</span> </span>
<span id="cb27-19"><a href="variant-annotation.html#cb27-19" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">renal_hpo_term =</span> <span class="fu">toString</span>(<span class="fu">unique</span>(renal_hpo_term)))</span>
<span id="cb27-20"><a href="variant-annotation.html#cb27-20" aria-hidden="true" tabindex="-1"></a>gen_hpo_extra <span class="ot">&lt;-</span> hpo_data <span class="sc">%&gt;%</span> </span>
<span id="cb27-21"><a href="variant-annotation.html#cb27-21" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(frequency_hpo <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;HP:0040281&quot;</span>, <span class="st">&quot;HP:0040280&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb27-22"><a href="variant-annotation.html#cb27-22" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(frequency_hpo) <span class="sc">%&gt;%</span></span>
<span id="cb27-23"><a href="variant-annotation.html#cb27-23" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>(hpo_term_id <span class="sc">%in%</span> hpo_kidney)) <span class="sc">%&gt;%</span> </span>
<span id="cb27-24"><a href="variant-annotation.html#cb27-24" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(entrez_gene_id) <span class="sc">%&gt;%</span> </span>
<span id="cb27-25"><a href="variant-annotation.html#cb27-25" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">extrarenal_hpo_term =</span> <span class="fu">toString</span>(<span class="fu">unique</span>(hpo_term_name)))</span>
<span id="cb27-26"><a href="variant-annotation.html#cb27-26" aria-hidden="true" tabindex="-1"></a><span class="co"># ClinGen </span></span>
<span id="cb27-27"><a href="variant-annotation.html#cb27-27" aria-hidden="true" tabindex="-1"></a>clingen <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/Clingen-Curation.csv&quot;</span>, <span class="at">skip =</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb27-28"><a href="variant-annotation.html#cb27-28" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">assertion=</span><span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">s*</span><span class="sc">\\</span><span class="st">([^</span><span class="sc">\\</span><span class="st">)]+</span><span class="sc">\\</span><span class="st">)&quot;</span>,<span class="st">&quot;&quot;</span>, gene_disease_validity_assertion_classifications)) <span class="sc">%&gt;%</span></span>
<span id="cb27-29"><a href="variant-annotation.html#cb27-29" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">assertion=</span><span class="fu">paste</span>(<span class="st">&quot;(&quot;</span>,assertion,<span class="st">&quot;)&quot;</span>,<span class="at">sep=</span><span class="st">&quot;&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb27-30"><a href="variant-annotation.html#cb27-30" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">assertion=</span><span class="fu">if_else</span>(assertion<span class="sc">==</span><span class="st">&quot;(NA)&quot;</span>,<span class="cn">NA</span>,assertion)) <span class="sc">%&gt;%</span></span>
<span id="cb27-31"><a href="variant-annotation.html#cb27-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(<span class="st">&quot;disease&quot;</span>, <span class="fu">c</span>(disease_label, assertion), <span class="at">remove =</span> <span class="cn">FALSE</span>, <span class="at">na.rm=</span><span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">&quot; &quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb27-32"><a href="variant-annotation.html#cb27-32" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(hgnc_id) <span class="sc">%&gt;%</span> </span>
<span id="cb27-33"><a href="variant-annotation.html#cb27-33" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">clingen_disease_label =</span> <span class="fu">paste</span>(disease,<span class="at">collapse=</span><span class="st">&quot;; &quot;</span>))</span>
<span id="cb27-34"><a href="variant-annotation.html#cb27-34" aria-hidden="true" tabindex="-1"></a><span class="co"># GenCC including Orphanet data</span></span>
<span id="cb27-35"><a href="variant-annotation.html#cb27-35" aria-hidden="true" tabindex="-1"></a>gencc <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">&quot;data/gencc-submissions.tsv&quot;</span>)</span>
<span id="cb27-36"><a href="variant-annotation.html#cb27-36" aria-hidden="true" tabindex="-1"></a>gencc_def <span class="ot">&lt;-</span> gencc <span class="sc">%&gt;%</span> </span>
<span id="cb27-37"><a href="variant-annotation.html#cb27-37" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(submitter_title <span class="sc">!=</span> <span class="st">&quot;Orphanet&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb27-38"><a href="variant-annotation.html#cb27-38" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(classification_title <span class="sc">==</span> <span class="st">&quot;Definitive&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb27-39"><a href="variant-annotation.html#cb27-39" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(gene_symbol, gene_curie, disease_title) <span class="sc">%&gt;%</span> </span>
<span id="cb27-40"><a href="variant-annotation.html#cb27-40" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(</span>
<span id="cb27-41"><a href="variant-annotation.html#cb27-41" aria-hidden="true" tabindex="-1"></a>    <span class="at">submitter_title =</span> <span class="fu">toString</span>(<span class="fu">unique</span>(submitter_title)), <span class="at">moi_title =</span> <span class="fu">toString</span>(<span class="fu">unique</span>(moi_title)), </span>
<span id="cb27-42"><a href="variant-annotation.html#cb27-42" aria-hidden="true" tabindex="-1"></a>    <span class="at">disease_curie =</span> <span class="fu">toString</span>(<span class="fu">unique</span>(disease_curie)), <span class="at">disease_original_curie =</span> <span class="fu">toString</span>(<span class="fu">unique</span>(disease_original_curie)), </span>
<span id="cb27-43"><a href="variant-annotation.html#cb27-43" aria-hidden="true" tabindex="-1"></a>    <span class="at">classification_title =</span> <span class="fu">toString</span>(classification_title), <span class="at">submitted_as_pmids =</span> <span class="fu">toString</span>(<span class="fu">unique</span>(submitted_as_pmids))) <span class="sc">%&gt;%</span></span>
<span id="cb27-44"><a href="variant-annotation.html#cb27-44" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">submitter =</span> <span class="fu">paste</span>(<span class="st">&quot;(&quot;</span>, submitter_title, <span class="st">&quot;)&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb27-45"><a href="variant-annotation.html#cb27-45" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">unite</span>(<span class="st">&quot;gencc_disease_definitive&quot;</span>, <span class="fu">c</span>(disease_title, submitter), <span class="at">remove =</span> <span class="cn">FALSE</span>, <span class="at">sep =</span> <span class="st">&quot; &quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb27-46"><a href="variant-annotation.html#cb27-46" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(gene_curie, gencc_disease_definitive) <span class="sc">%&gt;%</span> </span>
<span id="cb27-47"><a href="variant-annotation.html#cb27-47" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(gene_curie) <span class="sc">%&gt;%</span> </span>
<span id="cb27-48"><a href="variant-annotation.html#cb27-48" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">gencc_disease_definitive =</span> <span class="fu">paste</span>(<span class="fu">unique</span>(gencc_disease_definitive), <span class="at">collapse=</span><span class="st">&quot;; &quot;</span>))</span>
<span id="cb27-49"><a href="variant-annotation.html#cb27-49" aria-hidden="true" tabindex="-1"></a>gencc_strong <span class="ot">&lt;-</span> gencc <span class="sc">%&gt;%</span> </span>
<span id="cb27-50"><a href="variant-annotation.html#cb27-50" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(submitter_title <span class="sc">!=</span> <span class="st">&quot;Orphanet&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb27-51"><a href="variant-annotation.html#cb27-51" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(classification_title <span class="sc">==</span> <span class="st">&quot;Strong&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb27-52"><a href="variant-annotation.html#cb27-52" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(gene_symbol, gene_curie, disease_title) <span class="sc">%&gt;%</span> </span>
<span id="cb27-53"><a href="variant-annotation.html#cb27-53" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(</span>
<span id="cb27-54"><a href="variant-annotation.html#cb27-54" aria-hidden="true" tabindex="-1"></a>    <span class="at">submitter_title =</span> <span class="fu">toString</span>(<span class="fu">unique</span>(submitter_title)), <span class="at">moi_title =</span> <span class="fu">toString</span>(<span class="fu">unique</span>(moi_title)), </span>
<span id="cb27-55"><a href="variant-annotation.html#cb27-55" aria-hidden="true" tabindex="-1"></a>    <span class="at">disease_curie =</span> <span class="fu">toString</span>(<span class="fu">unique</span>(disease_curie)), <span class="at">disease_original_curie =</span> <span class="fu">toString</span>(<span class="fu">unique</span>(disease_original_curie)), </span>
<span id="cb27-56"><a href="variant-annotation.html#cb27-56" aria-hidden="true" tabindex="-1"></a>    <span class="at">classification_title =</span> <span class="fu">toString</span>(classification_title), <span class="at">submitted_as_pmids =</span> <span class="fu">toString</span>(<span class="fu">unique</span>(submitted_as_pmids))) <span class="sc">%&gt;%</span></span>
<span id="cb27-57"><a href="variant-annotation.html#cb27-57" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">submitter =</span> <span class="fu">paste</span>(<span class="st">&quot;(&quot;</span>, submitter_title, <span class="st">&quot;)&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb27-58"><a href="variant-annotation.html#cb27-58" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">unite</span>(<span class="st">&quot;gencc_disease_strong&quot;</span>, <span class="fu">c</span>(disease_title, submitter), <span class="at">remove =</span> <span class="cn">FALSE</span>, <span class="at">sep =</span> <span class="st">&quot; &quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb27-59"><a href="variant-annotation.html#cb27-59" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(gene_curie, gencc_disease_strong) <span class="sc">%&gt;%</span> </span>
<span id="cb27-60"><a href="variant-annotation.html#cb27-60" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(gene_curie) <span class="sc">%&gt;%</span> </span>
<span id="cb27-61"><a href="variant-annotation.html#cb27-61" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">gencc_disease_strong =</span> <span class="fu">paste</span>(<span class="fu">unique</span>(gencc_disease_strong), <span class="at">collapse=</span><span class="st">&quot;; &quot;</span>))</span>
<span id="cb27-62"><a href="variant-annotation.html#cb27-62" aria-hidden="true" tabindex="-1"></a>orphanet <span class="ot">&lt;-</span> gencc <span class="sc">%&gt;%</span> </span>
<span id="cb27-63"><a href="variant-annotation.html#cb27-63" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(submitter_title <span class="sc">==</span> <span class="st">&quot;Orphanet&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb27-64"><a href="variant-annotation.html#cb27-64" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(gene_curie) <span class="sc">%&gt;%</span> </span>
<span id="cb27-65"><a href="variant-annotation.html#cb27-65" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">orphanet_disease =</span> <span class="fu">toString</span>(disease_title), <span class="at">orphanet_moi =</span> <span class="fu">toString</span>(<span class="fu">unique</span>(moi_title)))</span>
<span id="cb27-66"><a href="variant-annotation.html#cb27-66" aria-hidden="true" tabindex="-1"></a><span class="co"># Join</span></span>
<span id="cb27-67"><a href="variant-annotation.html#cb27-67" aria-hidden="true" tabindex="-1"></a>refseq01 <span class="ot">&lt;-</span> refseq0 <span class="sc">%&gt;%</span> </span>
<span id="cb27-68"><a href="variant-annotation.html#cb27-68" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(omim, <span class="at">by=</span><span class="fu">join_by</span>(Gene<span class="sc">==</span>Entrez.Gene.ID)) <span class="sc">%&gt;%</span></span>
<span id="cb27-69"><a href="variant-annotation.html#cb27-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(gen_hpo_kidney, <span class="at">by=</span><span class="fu">join_by</span>(Gene<span class="sc">==</span>entrez_gene_id)) <span class="sc">%&gt;%</span></span>
<span id="cb27-70"><a href="variant-annotation.html#cb27-70" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(gen_hpo_extra, <span class="at">by=</span><span class="fu">join_by</span>(Gene<span class="sc">==</span>entrez_gene_id)) <span class="sc">%&gt;%</span></span>
<span id="cb27-71"><a href="variant-annotation.html#cb27-71" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(clingen, <span class="at">by=</span><span class="fu">join_by</span>(HGNC_ID<span class="sc">==</span>hgnc_id)) <span class="sc">%&gt;%</span></span>
<span id="cb27-72"><a href="variant-annotation.html#cb27-72" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(gencc_def, <span class="at">by=</span><span class="fu">join_by</span>(HGNC_ID<span class="sc">==</span>gene_curie)) <span class="sc">%&gt;%</span></span>
<span id="cb27-73"><a href="variant-annotation.html#cb27-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(gencc_strong, <span class="at">by=</span><span class="fu">join_by</span>(HGNC_ID<span class="sc">==</span>gene_curie)) <span class="sc">%&gt;%</span></span>
<span id="cb27-74"><a href="variant-annotation.html#cb27-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(orphanet, <span class="at">by=</span><span class="fu">join_by</span>(HGNC_ID<span class="sc">==</span>gene_curie)) <span class="sc">%&gt;%</span></span>
<span id="cb27-75"><a href="variant-annotation.html#cb27-75" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(SYMBOL, Uploaded_variation, Location, Feature, HGVSc,   HGVSp, Consequence, ZYG, EXON,  INTRON, gnomADe_AF, gnomADg_AF, MAX_AF, MAX_AF_POPS,    ClinVar,    ClinVar_CLNSIG, OMIM_Phenotypes, renal_hpo_term, extrarenal_hpo_term,   REVEL,  SIFT,   PolyPhen,   CADD_PHRED, PrimateAI,  SpliceAI_pred,  EVE_CLASS,  EVE_SCORE,  MutationTaster_pred,    MutationTaster_score,   orphanet_disease,   orphanet_moi,   Mastermind_MMID3,   Existing_variation, PUBMED, mother, father, proband, Allele,    counter, AD_ref_mother, AD_ref_father, AD_ref_proband,  AD_alt_mother, AD_alt_father, AD_alt_proband, GQ_mother,    GQ_father,  GQ_proband, PS_mother,  PS_father,  PS_proband, CANONICAL,  MANE_SELECT,    MANE_PLUS_CLINICAL, IMPACT, Gene, HGNC_ID, Amino_acids, Codons, AF, AFR_AF, AMR_AF, EAS_AF, EUR_AF, SAS_AF, gnomADe_AFR_AF, gnomADe_AMR_AF, gnomADe_ASJ_AF, gnomADe_EAS_AF, gnomADe_FIN_AF, gnomADe_NFE_AF, gnomADe_OTH_AF, gnomADe_SAS_AF, gnomADg_AFR_AF, gnomADg_AMI_AF, gnomADg_AMR_AF, gnomADg_ASJ_AF, gnomADg_EAS_AF, gnomADg_FIN_AF, gnomADg_MID_AF, gnomADg_NFE_AF, gnomADg_OTH_AF, gnomADg_SAS_AF)</span>
<span id="cb27-76"><a href="variant-annotation.html#cb27-76" aria-hidden="true" tabindex="-1"></a><span class="co"># X chromosome</span></span>
<span id="cb27-77"><a href="variant-annotation.html#cb27-77" aria-hidden="true" tabindex="-1"></a>refseq_X <span class="ot">&lt;-</span> refseq01 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">str_detect</span>(Uploaded_variation,<span class="st">&quot;X&quot;</span>))</span>
<span id="cb27-78"><a href="variant-annotation.html#cb27-78" aria-hidden="true" tabindex="-1"></a><span class="co"># Autosomal chromosomes</span></span>
<span id="cb27-79"><a href="variant-annotation.html#cb27-79" aria-hidden="true" tabindex="-1"></a>refseq <span class="ot">&lt;-</span> refseq01 <span class="sc">%&gt;%</span></span>
<span id="cb27-80"><a href="variant-annotation.html#cb27-80" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(Uploaded_variation,<span class="st">&quot;X&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb27-81"><a href="variant-annotation.html#cb27-81" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(Uploaded_variation,<span class="st">&quot;Y&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb27-82"><a href="variant-annotation.html#cb27-82" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(Uploaded_variation,<span class="st">&quot;M&quot;</span>))</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["SYMBOL"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Uploaded_variation"],"name":[2],"type":["fct"],"align":["left"]},{"label":["Location"],"name":[3],"type":["chr"],"align":["left"]},{"label":["Feature"],"name":[4],"type":["fct"],"align":["left"]},{"label":["HGVSc"],"name":[5],"type":["chr"],"align":["left"]},{"label":["HGVSp"],"name":[6],"type":["chr"],"align":["left"]},{"label":["Consequence"],"name":[7],"type":["chr"],"align":["left"]},{"label":["ZYG"],"name":[8],"type":["chr"],"align":["left"]},{"label":["EXON"],"name":[9],"type":["chr"],"align":["left"]},{"label":["INTRON"],"name":[10],"type":["chr"],"align":["left"]},{"label":["gnomADe_AF"],"name":[11],"type":["dbl"],"align":["right"]},{"label":["gnomADg_AF"],"name":[12],"type":["dbl"],"align":["right"]},{"label":["MAX_AF"],"name":[13],"type":["dbl"],"align":["right"]},{"label":["MAX_AF_POPS"],"name":[14],"type":["chr"],"align":["left"]},{"label":["ClinVar"],"name":[15],"type":["dbl"],"align":["right"]},{"label":["ClinVar_CLNSIG"],"name":[16],"type":["chr"],"align":["left"]},{"label":["OMIM_Phenotypes"],"name":[17],"type":["chr"],"align":["left"]},{"label":["renal_hpo_term"],"name":[18],"type":["chr"],"align":["left"]},{"label":["extrarenal_hpo_term"],"name":[19],"type":["chr"],"align":["left"]},{"label":["REVEL"],"name":[20],"type":["dbl"],"align":["right"]},{"label":["SIFT"],"name":[21],"type":["chr"],"align":["left"]},{"label":["PolyPhen"],"name":[22],"type":["chr"],"align":["left"]},{"label":["CADD_PHRED"],"name":[23],"type":["dbl"],"align":["right"]},{"label":["PrimateAI"],"name":[24],"type":["dbl"],"align":["right"]},{"label":["SpliceAI_pred"],"name":[25],"type":["chr"],"align":["left"]},{"label":["EVE_CLASS"],"name":[26],"type":["chr"],"align":["left"]},{"label":["EVE_SCORE"],"name":[27],"type":["dbl"],"align":["right"]},{"label":["MutationTaster_pred"],"name":[28],"type":["chr"],"align":["left"]},{"label":["MutationTaster_score"],"name":[29],"type":["chr"],"align":["left"]},{"label":["orphanet_disease"],"name":[30],"type":["chr"],"align":["left"]},{"label":["orphanet_moi"],"name":[31],"type":["chr"],"align":["left"]},{"label":["Mastermind_MMID3"],"name":[32],"type":["chr"],"align":["left"]},{"label":["Existing_variation"],"name":[33],"type":["chr"],"align":["left"]},{"label":["PUBMED"],"name":[34],"type":["chr"],"align":["left"]},{"label":["mother"],"name":[35],"type":["chr"],"align":["left"]},{"label":["father"],"name":[36],"type":["chr"],"align":["left"]},{"label":["proband"],"name":[37],"type":["chr"],"align":["left"]},{"label":["Allele"],"name":[38],"type":["chr"],"align":["left"]},{"label":["counter"],"name":[39],"type":["int"],"align":["right"]},{"label":["AD_ref_mother"],"name":[40],"type":["int"],"align":["right"]},{"label":["AD_ref_father"],"name":[41],"type":["int"],"align":["right"]},{"label":["AD_ref_proband"],"name":[42],"type":["int"],"align":["right"]},{"label":["AD_alt_mother"],"name":[43],"type":["int"],"align":["right"]},{"label":["AD_alt_father"],"name":[44],"type":["int"],"align":["right"]},{"label":["AD_alt_proband"],"name":[45],"type":["int"],"align":["right"]},{"label":["GQ_mother"],"name":[46],"type":["int"],"align":["right"]},{"label":["GQ_father"],"name":[47],"type":["int"],"align":["right"]},{"label":["GQ_proband"],"name":[48],"type":["int"],"align":["right"]},{"label":["PS_mother"],"name":[49],"type":["int"],"align":["right"]},{"label":["PS_father"],"name":[50],"type":["int"],"align":["right"]},{"label":["PS_proband"],"name":[51],"type":["int"],"align":["right"]},{"label":["CANONICAL"],"name":[52],"type":["chr"],"align":["left"]},{"label":["MANE_SELECT"],"name":[53],"type":["chr"],"align":["left"]},{"label":["MANE_PLUS_CLINICAL"],"name":[54],"type":["chr"],"align":["left"]},{"label":["IMPACT"],"name":[55],"type":["chr"],"align":["left"]},{"label":["Gene"],"name":[56],"type":["fct"],"align":["left"]},{"label":["HGNC_ID"],"name":[57],"type":["chr"],"align":["left"]},{"label":["Amino_acids"],"name":[58],"type":["chr"],"align":["left"]},{"label":["Codons"],"name":[59],"type":["chr"],"align":["left"]},{"label":["AF"],"name":[60],"type":["dbl"],"align":["right"]},{"label":["AFR_AF"],"name":[61],"type":["dbl"],"align":["right"]},{"label":["AMR_AF"],"name":[62],"type":["dbl"],"align":["right"]},{"label":["EAS_AF"],"name":[63],"type":["dbl"],"align":["right"]},{"label":["EUR_AF"],"name":[64],"type":["dbl"],"align":["right"]},{"label":["SAS_AF"],"name":[65],"type":["dbl"],"align":["right"]},{"label":["gnomADe_AFR_AF"],"name":[66],"type":["dbl"],"align":["right"]},{"label":["gnomADe_AMR_AF"],"name":[67],"type":["dbl"],"align":["right"]},{"label":["gnomADe_ASJ_AF"],"name":[68],"type":["dbl"],"align":["right"]},{"label":["gnomADe_EAS_AF"],"name":[69],"type":["dbl"],"align":["right"]},{"label":["gnomADe_FIN_AF"],"name":[70],"type":["dbl"],"align":["right"]},{"label":["gnomADe_NFE_AF"],"name":[71],"type":["dbl"],"align":["right"]},{"label":["gnomADe_OTH_AF"],"name":[72],"type":["dbl"],"align":["right"]},{"label":["gnomADe_SAS_AF"],"name":[73],"type":["dbl"],"align":["right"]},{"label":["gnomADg_AFR_AF"],"name":[74],"type":["dbl"],"align":["right"]},{"label":["gnomADg_AMI_AF"],"name":[75],"type":["dbl"],"align":["right"]},{"label":["gnomADg_AMR_AF"],"name":[76],"type":["dbl"],"align":["right"]},{"label":["gnomADg_ASJ_AF"],"name":[77],"type":["dbl"],"align":["right"]},{"label":["gnomADg_EAS_AF"],"name":[78],"type":["dbl"],"align":["right"]},{"label":["gnomADg_FIN_AF"],"name":[79],"type":["dbl"],"align":["right"]},{"label":["gnomADg_MID_AF"],"name":[80],"type":["dbl"],"align":["right"]},{"label":["gnomADg_NFE_AF"],"name":[81],"type":["dbl"],"align":["right"]},{"label":["gnomADg_OTH_AF"],"name":[82],"type":["dbl"],"align":["right"]},{"label":["gnomADg_SAS_AF"],"name":[83],"type":["dbl"],"align":["right"]}],"data":[{"1":"RNF223","2":"chr1_1072326_G/A","3":"chr1:1072326","4":"NM_001205252.2","5":"NM_001205252.2:c.241C>T","6":"NP_001192181.1:p.Arg81Trp","7":"missense_variant","8":"HET","9":"2/2","10":"NA","11":"1.255e-03","12":"1.058e-03","13":"0.018730","14":"gnomADg_ASJ","15":"NA","16":"NA","17":"NA","18":"NA","19":"NA","20":"0.298","21":"deleterious(0)","22":"probably_damaging(1)","23":"25.7","24":"0.8687824","25":"RNF223|0.00|0.00|0.00|0.00|-42|42|-36|-33","26":"NA","27":"NA","28":"D","29":"0.514264","30":"NA","31":"NA","32":"RNF223:R81W","33":"rs576663974","34":"NA","35":"0/0","36":"0|1","37":"1|0","38":"A","39":"1","40":"40","41":"77","42":"106","43":"0","44":"68","45":"82","46":"90","47":"99","48":"99","49":"NA","50":"65797","51":"65797","52":"YES","53":"ENST00000453464.3","54":"NA","55":"MODERATE","56":"401934","57":"HGNC:40020","58":"R/W","59":"Cgg/Tgg","60":"2e-04","61":"0","62":"0.0014","63":"0","64":"0","65":"0","66":"0.00000","67":"1.162e-03","68":"0.0177300","69":"3.086e-04","70":"0.0008711","71":"0.0008430","72":"0.001089","73":"0.0003011","74":"0.0002654","75":"0.0000","76":"0.001963","77":"0.018730","78":"0.0003861","79":"0.0005648","80":"0.0000","81":"0.000632","82":"0.0009569","83":"0.0004137"},{"1":"CDK11A","2":"chr1_1708856_TCCTCCTCC/-","3":"chr1:1708856-1708864","4":"NM_001313896.2","5":"NM_001313896.2:c.942_950del","6":"NP_001300825.1:p.Glu322_Glu324del","7":"inframe_deletion","8":"HET","9":"9/20","10":"NA","11":"5.767e-04","12":"3.144e-01","13":"0.463600","14":"gnomADg_ASJ","15":"NA","16":"NA","17":"NA","18":"NA","19":"NA","20":"NA","21":"NA","22":"NA","23":"16.2","24":"NA","25":"NA","26":"NA","27":"NA","28":"NA","29":"NA","30":"NA","31":"NA","32":"NA","33":"rs199942750","34":"NA","35":"0|1","36":"0/0","37":"0|1","38":"-","39":"1","40":"6","41":"39","42":"32","43":"34","44":"0","45":"15","46":"99","47":"68","48":"99","49":"65797","50":"NA","51":"65797","52":"NA","53":"NA","54":"NA","55":"MODERATE","56":"728642","57":"HGNC:1730","58":"EEEE/E","59":"gaGGAGGAGGAa/gaa","60":"NA","61":"NA","62":"NA","63":"NA","64":"NA","65":"NA","66":"0.00102","67":"1.800e-03","68":"0.0000000","69":"0.000e+00","70":"0.0000000","71":"0.0003032","72":"0.000000","73":"0.0003441","74":"0.1058000","75":"0.3029","76":"0.342100","77":"0.463600","78":"0.1708000","79":"0.4138000","80":"0.3871","81":"0.435600","82":"0.3298000","83":"0.2548000"},{"1":"CDK11A","2":"chr1_1708856_TCCTCCTCC/-","3":"chr1:1708856-1708864","4":"NM_001313982.2","5":"NM_001313982.2:c.930_938del","6":"NP_001300911.1:p.Glu318_Glu320del","7":"inframe_deletion","8":"HET","9":"9/20","10":"NA","11":"5.767e-04","12":"3.144e-01","13":"0.463600","14":"gnomADg_ASJ","15":"NA","16":"NA","17":"NA","18":"NA","19":"NA","20":"NA","21":"NA","22":"NA","23":"16.2","24":"NA","25":"NA","26":"NA","27":"NA","28":"NA","29":"NA","30":"NA","31":"NA","32":"NA","33":"rs199942750","34":"NA","35":"0|1","36":"0/0","37":"0|1","38":"-","39":"1","40":"6","41":"39","42":"32","43":"34","44":"0","45":"15","46":"99","47":"68","48":"99","49":"65797","50":"NA","51":"65797","52":"NA","53":"NA","54":"NA","55":"MODERATE","56":"728642","57":"HGNC:1730","58":"EEEE/E","59":"gaGGAGGAGGAa/gaa","60":"NA","61":"NA","62":"NA","63":"NA","64":"NA","65":"NA","66":"0.00102","67":"1.800e-03","68":"0.0000000","69":"0.000e+00","70":"0.0000000","71":"0.0003032","72":"0.000000","73":"0.0003441","74":"0.1058000","75":"0.3029","76":"0.342100","77":"0.463600","78":"0.1708000","79":"0.4138000","80":"0.3871","81":"0.435600","82":"0.3298000","83":"0.2548000"},{"1":"CDK11A","2":"chr1_1708856_TCCTCCTCC/-","3":"chr1:1708856-1708864","4":"NM_024011.4","5":"NM_024011.4:c.933_941del","6":"NP_076916.2:p.Glu319_Glu321del","7":"inframe_deletion","8":"HET","9":"9/20","10":"NA","11":"5.767e-04","12":"3.144e-01","13":"0.463600","14":"gnomADg_ASJ","15":"NA","16":"NA","17":"NA","18":"NA","19":"NA","20":"NA","21":"NA","22":"NA","23":"16.2","24":"NA","25":"NA","26":"NA","27":"NA","28":"NA","29":"NA","30":"NA","31":"NA","32":"NA","33":"rs199942750","34":"NA","35":"0|1","36":"0/0","37":"0|1","38":"-","39":"1","40":"6","41":"39","42":"32","43":"34","44":"0","45":"15","46":"99","47":"68","48":"99","49":"65797","50":"NA","51":"65797","52":"YES","53":"ENST00000404249.8","54":"NA","55":"MODERATE","56":"728642","57":"HGNC:1730","58":"EEEE/E","59":"gaGGAGGAGGAa/gaa","60":"NA","61":"NA","62":"NA","63":"NA","64":"NA","65":"NA","66":"0.00102","67":"1.800e-03","68":"0.0000000","69":"0.000e+00","70":"0.0000000","71":"0.0003032","72":"0.000000","73":"0.0003441","74":"0.1058000","75":"0.3029","76":"0.342100","77":"0.463600","78":"0.1708000","79":"0.4138000","80":"0.3871","81":"0.435600","82":"0.3298000","83":"0.2548000"},{"1":"CDK11A","2":"chr1_1708856_TCCTCCTCC/-","3":"chr1:1708856-1708864","4":"NM_033529.4","5":"NM_033529.4:c.903_911del","6":"NP_277071.2:p.Glu309_Glu311del","7":"inframe_deletion","8":"HET","9":"9/20","10":"NA","11":"5.767e-04","12":"3.144e-01","13":"0.463600","14":"gnomADg_ASJ","15":"NA","16":"NA","17":"NA","18":"NA","19":"NA","20":"NA","21":"NA","22":"NA","23":"16.2","24":"NA","25":"NA","26":"NA","27":"NA","28":"NA","29":"NA","30":"NA","31":"NA","32":"NA","33":"rs199942750","34":"NA","35":"0|1","36":"0/0","37":"0|1","38":"-","39":"1","40":"6","41":"39","42":"32","43":"34","44":"0","45":"15","46":"99","47":"68","48":"99","49":"65797","50":"NA","51":"65797","52":"NA","53":"NA","54":"NA","55":"MODERATE","56":"728642","57":"HGNC:1730","58":"EEEE/E","59":"gaGGAGGAGGAa/gaa","60":"NA","61":"NA","62":"NA","63":"NA","64":"NA","65":"NA","66":"0.00102","67":"1.800e-03","68":"0.0000000","69":"0.000e+00","70":"0.0000000","71":"0.0003032","72":"0.000000","73":"0.0003441","74":"0.1058000","75":"0.3029","76":"0.342100","77":"0.463600","78":"0.1708000","79":"0.4138000","80":"0.3871","81":"0.435600","82":"0.3298000","83":"0.2548000"},{"1":"NPHP4","2":"chr1_5864452_C/A","3":"chr1:5864452","4":"NM_001291593.2","5":"NM_001291593.2:c.2343G>T","6":"NP_001278522.1:p.Arg781Ser","7":"missense_variant","8":"HET","9":"25/27","10":"NA","11":"4.767e-05","12":"2.628e-05","13":"0.001152","14":"gnomADg_ASJ","15":"969225","16":"Uncertain_significance","17":"Senior-Loken syndrome 4, 606996 (3), Autosomal recessive; Nephronophthisis 4, 606966 (3), Autosomal recessive","18":"Stage 5 chronic kidney disease (very frequent), Nephronophthisis (frequent), Tubulointerstitial fibrosis, Renal tubular atrophy, Polyuria, Stage 5 chronic kidney disease, Renal corticomedullary cysts, Nephronophthisis","19":"Short stature, Hypertension, Global developmental delay, Abnormality of retinal pigmentation, Retinal dystrophy","20":"0.664","21":"deleterious(0)","22":"benign(0.383)","23":"24.2","24":"0.4835634","25":"NPHP4|0.04|0.00|0.04|0.00|-9|20|6|-39","26":"Pathogenic","27":"0.8008494","28":"D","29":"0.99534","30":"Senior-Loken syndrome, nephronophthisis 1","31":"Autosomal recessive","32":"NA","33":"rs367814493","34":"NA","35":"0/0","36":"1|0","37":"1|0","38":"A","39":"1","40":"33","41":"78","42":"109","43":"0","44":"69","45":"69","46":"99","47":"99","48":"99","49":"NA","50":"65797","51":"65797","52":"NA","53":"NA","54":"NA","55":"MODERATE","56":"261734","57":"HGNC:19104","58":"R/S","59":"agG/agT","60":"NA","61":"NA","62":"NA","63":"NA","64":"NA","65":"NA","66":"0.00000","67":"6.087e-05","68":"0.0008323","69":"5.999e-05","70":"0.0000000","71":"0.0000000","72":"0.000000","73":"0.0000000","74":"0.0000000","75":"0.0000","76":"0.000000","77":"0.001152","78":"0.0000000","79":"0.0000000","80":"0.0000","81":"0.000000","82":"0.0000000","83":"0.0000000"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="de-novo-variants" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">5.2</span> <em>De Novo</em> Variants<a href="variant-annotation.html#de-novo-variants" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We select <em>de novo</em> variants in the proband.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="variant-annotation.html#cb28-1" aria-hidden="true" tabindex="-1"></a>geno_novo_all <span class="ot">&lt;-</span> wh_geno_auto <span class="sc">%&gt;%</span> </span>
<span id="cb28-2"><a href="variant-annotation.html#cb28-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span> ( ( ( <span class="fu">str_detect</span>(mother, <span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)) <span class="sc">|</span> <span class="fu">str_detect</span>(mother, <span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>) ) <span class="sc">&amp;</span> </span>
<span id="cb28-3"><a href="variant-annotation.html#cb28-3" aria-hidden="true" tabindex="-1"></a>                 (<span class="fu">str_detect</span>(father, <span class="fu">substr</span>(proband,<span class="dv">3</span>,<span class="dv">3</span>)) <span class="sc">|</span> <span class="fu">str_detect</span>(father, <span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>) ) ) <span class="sc">|</span> </span>
<span id="cb28-4"><a href="variant-annotation.html#cb28-4" aria-hidden="true" tabindex="-1"></a>               ( ( <span class="fu">str_detect</span>(father, <span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)) <span class="sc">|</span> <span class="fu">str_detect</span>(father, <span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>) ) <span class="sc">&amp;</span> </span>
<span id="cb28-5"><a href="variant-annotation.html#cb28-5" aria-hidden="true" tabindex="-1"></a>                   (<span class="fu">str_detect</span>(mother, <span class="fu">substr</span>(proband,<span class="dv">3</span>,<span class="dv">3</span>)) <span class="sc">|</span> <span class="fu">str_detect</span>(mother, <span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>) ) ) ) ) <span class="sc">%&gt;%</span></span>
<span id="cb28-6"><a href="variant-annotation.html#cb28-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>(<span class="fu">str_detect</span>(mother,<span class="fu">as.character</span>(counter)) <span class="sc">|</span> <span class="fu">str_detect</span>(father,<span class="fu">as.character</span>(counter))) <span class="sc">|</span></span>
<span id="cb28-7"><a href="variant-annotation.html#cb28-7" aria-hidden="true" tabindex="-1"></a>                  (<span class="fu">str_detect</span>(mother, <span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)) <span class="sc">&amp;</span> <span class="fu">str_detect</span>(mother, <span class="fu">substr</span>(proband,<span class="dv">3</span>,<span class="dv">3</span>))) <span class="sc">|</span></span>
<span id="cb28-8"><a href="variant-annotation.html#cb28-8" aria-hidden="true" tabindex="-1"></a>                  (<span class="fu">str_detect</span>(father, <span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)) <span class="sc">&amp;</span> <span class="fu">str_detect</span>(father, <span class="fu">substr</span>(proband,<span class="dv">3</span>,<span class="dv">3</span>)))</span>
<span id="cb28-9"><a href="variant-annotation.html#cb28-9" aria-hidden="true" tabindex="-1"></a>                  )</span></code></pre></div>
<p>We discard <em>de novo</em> mutations back to the reference allele (e.g.Â 1/1, 1/1 â&gt; 0/1) since they are unlikely to be pathogenic.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="variant-annotation.html#cb29-1" aria-hidden="true" tabindex="-1"></a>geno_novo_not_ref <span class="ot">&lt;-</span> wh_geno_auto <span class="sc">%&gt;%</span> </span>
<span id="cb29-2"><a href="variant-annotation.html#cb29-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span> ( ( ( <span class="fu">str_detect</span>(mother, <span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)) <span class="sc">|</span> <span class="fu">str_detect</span>(mother, <span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>) <span class="sc">|</span> <span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)<span class="sc">==</span><span class="dv">0</span> ) <span class="sc">&amp;</span> </span>
<span id="cb29-3"><a href="variant-annotation.html#cb29-3" aria-hidden="true" tabindex="-1"></a>                 (<span class="fu">str_detect</span>(father, <span class="fu">substr</span>(proband,<span class="dv">3</span>,<span class="dv">3</span>)) <span class="sc">|</span> <span class="fu">str_detect</span>(father, <span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>) <span class="sc">|</span> <span class="fu">substr</span>(proband,<span class="dv">3</span>,<span class="dv">3</span>)<span class="sc">==</span><span class="dv">0</span> ) ) <span class="sc">|</span> </span>
<span id="cb29-4"><a href="variant-annotation.html#cb29-4" aria-hidden="true" tabindex="-1"></a>               ( ( <span class="fu">str_detect</span>(father, <span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)) <span class="sc">|</span> <span class="fu">str_detect</span>(father, <span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>) <span class="sc">|</span> <span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)<span class="sc">==</span><span class="dv">0</span> ) <span class="sc">&amp;</span> </span>
<span id="cb29-5"><a href="variant-annotation.html#cb29-5" aria-hidden="true" tabindex="-1"></a>                   (<span class="fu">str_detect</span>(mother, <span class="fu">substr</span>(proband,<span class="dv">3</span>,<span class="dv">3</span>)) <span class="sc">|</span> <span class="fu">str_detect</span>(mother, <span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>) <span class="sc">|</span> <span class="fu">substr</span>(proband,<span class="dv">3</span>,<span class="dv">3</span>)<span class="sc">==</span><span class="dv">0</span> ) ) ) ) <span class="sc">%&gt;%</span></span>
<span id="cb29-6"><a href="variant-annotation.html#cb29-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>(<span class="fu">str_detect</span>(mother,<span class="fu">as.character</span>(counter)) <span class="sc">|</span> <span class="fu">str_detect</span>(father,<span class="fu">as.character</span>(counter))) <span class="sc">|</span></span>
<span id="cb29-7"><a href="variant-annotation.html#cb29-7" aria-hidden="true" tabindex="-1"></a>                  (<span class="fu">str_detect</span>(mother, <span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)) <span class="sc">&amp;</span> <span class="fu">str_detect</span>(mother, <span class="fu">substr</span>(proband,<span class="dv">3</span>,<span class="dv">3</span>))) <span class="sc">|</span></span>
<span id="cb29-8"><a href="variant-annotation.html#cb29-8" aria-hidden="true" tabindex="-1"></a>                  (<span class="fu">str_detect</span>(father, <span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)) <span class="sc">&amp;</span> <span class="fu">str_detect</span>(father, <span class="fu">substr</span>(proband,<span class="dv">3</span>,<span class="dv">3</span>)))</span>
<span id="cb29-9"><a href="variant-annotation.html#cb29-9" aria-hidden="true" tabindex="-1"></a>                  )</span></code></pre></div>
<p>We increase the cutoff for genotype quality to ensure that the the <em>de novo</em> variants are not a result of incorrect variant calling. Set GQ&gt;=30, i.e, the probability that the call is incorrect is =&lt;0.1%.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="variant-annotation.html#cb30-1" aria-hidden="true" tabindex="-1"></a>geno_novo_not_ref_high_qual <span class="ot">&lt;-</span> geno_novo_not_ref <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(GQ_mother<span class="sc">&gt;=</span><span class="dv">30</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(GQ_father<span class="sc">&gt;=</span><span class="dv">30</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(GQ_proband<span class="sc">&gt;=</span><span class="dv">30</span>)</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["Location"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Allele"],"name":[2],"type":["chr"],"align":["left"]},{"label":["proband"],"name":[3],"type":["chr"],"align":["left"]},{"label":["father"],"name":[4],"type":["chr"],"align":["left"]},{"label":["mother"],"name":[5],"type":["chr"],"align":["left"]},{"label":["GQ_proband"],"name":[6],"type":["int"],"align":["right"]},{"label":["GQ_father"],"name":[7],"type":["int"],"align":["right"]},{"label":["GQ_mother"],"name":[8],"type":["int"],"align":["right"]},{"label":["PS_proband"],"name":[9],"type":["int"],"align":["right"]},{"label":["PS_father"],"name":[10],"type":["int"],"align":["right"]},{"label":["PS_mother"],"name":[11],"type":["int"],"align":["right"]},{"label":["AD_ref_proband"],"name":[12],"type":["int"],"align":["right"]},{"label":["AD_ref_father"],"name":[13],"type":["int"],"align":["right"]},{"label":["AD_ref_mother"],"name":[14],"type":["int"],"align":["right"]},{"label":["AD_alt_proband"],"name":[15],"type":["int"],"align":["right"]},{"label":["AD_alt_father"],"name":[16],"type":["int"],"align":["right"]},{"label":["AD_alt_mother"],"name":[17],"type":["int"],"align":["right"]},{"label":["counter"],"name":[18],"type":["int"],"align":["right"]}],"data":[{"1":"chr1:15425826-15425829","2":"-","3":"0/1","4":"0/0","5":"0/0","6":"99","7":"38","8":"38","9":"NA","10":"NA","11":"NA","12":"106","13":"38","14":"33","15":"70","16":"0","17":"0","18":"1"},{"1":"chr1:64663551-64663552","2":"-","3":"2/2","4":"0/2","5":"0/1","6":"52","7":"83","8":"46","9":"NA","10":"NA","11":"NA","12":"0","13":"6","14":"4","15":"0","16":"1","17":"2","18":"1"},{"1":"chr1:64663551-64663552","2":"A","3":"2/2","4":"0/2","5":"0/1","6":"52","7":"83","8":"46","9":"NA","10":"NA","11":"NA","12":"0","13":"6","14":"4","15":"12","16":"6","17":"0","18":"2"},{"1":"chr1:118081059-118081064","2":"ACAC","3":"0/3","4":"0/2","5":"0/1","6":"99","7":"99","8":"99","9":"NA","10":"NA","11":"NA","12":"38","13":"15","14":"19","15":"7","16":"0","17":"0","18":"3"},{"1":"chr1:149390802-149390803","2":"GGCGGCGGCGGCGGC","3":"2/2","4":"2/3","5":"1/1","6":"56","7":"99","8":"48","9":"NA","10":"NA","11":"NA","12":"0","13":"0","14":"0","15":"0","16":"0","17":"16","18":"1"},{"1":"chr1:149390802-149390803","2":"GGCGGCGGCGGCGGCGGCGGC","3":"2/2","4":"2/3","5":"1/1","6":"56","7":"99","8":"48","9":"NA","10":"NA","11":"NA","12":"0","13":"0","14":"0","15":"19","16":"5","17":"0","18":"2"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>We select <em>de novo</em> variants found in the filtered VEP annotation. For each variant, we chose a single gene transcript that contains the variant, prioritizing canonical and curated transcripts.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="variant-annotation.html#cb31-1" aria-hidden="true" tabindex="-1"></a>prob_novo <span class="ot">&lt;-</span> refseq <span class="sc">%&gt;%</span> <span class="fu">inner_join</span>(geno_novo_not_ref)</span>
<span id="cb31-2"><a href="variant-annotation.html#cb31-2" aria-hidden="true" tabindex="-1"></a>prob_novo <span class="ot">&lt;-</span> prob_novo <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(Uploaded_variation) <span class="sc">%&gt;%</span> </span>
<span id="cb31-3"><a href="variant-annotation.html#cb31-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(CANONICAL), <span class="fu">as.character</span>(Feature),<span class="at">.by_group =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb31-4"><a href="variant-annotation.html#cb31-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Feature <span class="sc">==</span> Feature[<span class="dv">1</span>])</span>
<span id="cb31-5"><a href="variant-annotation.html#cb31-5" aria-hidden="true" tabindex="-1"></a>prob_novo_high_qual <span class="ot">&lt;-</span> refseq <span class="sc">%&gt;%</span> <span class="fu">inner_join</span>(geno_novo_not_ref_high_qual)</span>
<span id="cb31-6"><a href="variant-annotation.html#cb31-6" aria-hidden="true" tabindex="-1"></a>prob_novo_high_qual <span class="ot">&lt;-</span> prob_novo_high_qual <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(Uploaded_variation) <span class="sc">%&gt;%</span> </span>
<span id="cb31-7"><a href="variant-annotation.html#cb31-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(CANONICAL), <span class="fu">as.character</span>(Feature),<span class="at">.by_group =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb31-8"><a href="variant-annotation.html#cb31-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Feature <span class="sc">==</span> Feature[<span class="dv">1</span>])</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["SYMBOL"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Uploaded_variation"],"name":[2],"type":["fct"],"align":["left"]},{"label":["Location"],"name":[3],"type":["chr"],"align":["left"]},{"label":["Feature"],"name":[4],"type":["fct"],"align":["left"]},{"label":["HGVSc"],"name":[5],"type":["chr"],"align":["left"]},{"label":["HGVSp"],"name":[6],"type":["chr"],"align":["left"]},{"label":["Consequence"],"name":[7],"type":["chr"],"align":["left"]},{"label":["ZYG"],"name":[8],"type":["chr"],"align":["left"]},{"label":["EXON"],"name":[9],"type":["chr"],"align":["left"]},{"label":["INTRON"],"name":[10],"type":["chr"],"align":["left"]},{"label":["gnomADe_AF"],"name":[11],"type":["dbl"],"align":["right"]},{"label":["gnomADg_AF"],"name":[12],"type":["dbl"],"align":["right"]},{"label":["MAX_AF"],"name":[13],"type":["dbl"],"align":["right"]},{"label":["MAX_AF_POPS"],"name":[14],"type":["chr"],"align":["left"]},{"label":["ClinVar"],"name":[15],"type":["dbl"],"align":["right"]},{"label":["ClinVar_CLNSIG"],"name":[16],"type":["chr"],"align":["left"]},{"label":["OMIM_Phenotypes"],"name":[17],"type":["chr"],"align":["left"]},{"label":["renal_hpo_term"],"name":[18],"type":["chr"],"align":["left"]},{"label":["extrarenal_hpo_term"],"name":[19],"type":["chr"],"align":["left"]},{"label":["REVEL"],"name":[20],"type":["dbl"],"align":["right"]},{"label":["SIFT"],"name":[21],"type":["chr"],"align":["left"]},{"label":["PolyPhen"],"name":[22],"type":["chr"],"align":["left"]},{"label":["CADD_PHRED"],"name":[23],"type":["dbl"],"align":["right"]},{"label":["PrimateAI"],"name":[24],"type":["dbl"],"align":["right"]},{"label":["SpliceAI_pred"],"name":[25],"type":["chr"],"align":["left"]},{"label":["EVE_CLASS"],"name":[26],"type":["chr"],"align":["left"]},{"label":["EVE_SCORE"],"name":[27],"type":["dbl"],"align":["right"]},{"label":["MutationTaster_pred"],"name":[28],"type":["chr"],"align":["left"]},{"label":["MutationTaster_score"],"name":[29],"type":["chr"],"align":["left"]},{"label":["orphanet_disease"],"name":[30],"type":["chr"],"align":["left"]},{"label":["orphanet_moi"],"name":[31],"type":["chr"],"align":["left"]},{"label":["Mastermind_MMID3"],"name":[32],"type":["chr"],"align":["left"]},{"label":["Existing_variation"],"name":[33],"type":["chr"],"align":["left"]},{"label":["PUBMED"],"name":[34],"type":["chr"],"align":["left"]},{"label":["mother"],"name":[35],"type":["chr"],"align":["left"]},{"label":["father"],"name":[36],"type":["chr"],"align":["left"]},{"label":["proband"],"name":[37],"type":["chr"],"align":["left"]},{"label":["Allele"],"name":[38],"type":["chr"],"align":["left"]},{"label":["counter"],"name":[39],"type":["int"],"align":["right"]},{"label":["AD_ref_mother"],"name":[40],"type":["int"],"align":["right"]},{"label":["AD_ref_father"],"name":[41],"type":["int"],"align":["right"]},{"label":["AD_ref_proband"],"name":[42],"type":["int"],"align":["right"]},{"label":["AD_alt_mother"],"name":[43],"type":["int"],"align":["right"]},{"label":["AD_alt_father"],"name":[44],"type":["int"],"align":["right"]},{"label":["AD_alt_proband"],"name":[45],"type":["int"],"align":["right"]},{"label":["GQ_mother"],"name":[46],"type":["int"],"align":["right"]},{"label":["GQ_father"],"name":[47],"type":["int"],"align":["right"]},{"label":["GQ_proband"],"name":[48],"type":["int"],"align":["right"]},{"label":["PS_mother"],"name":[49],"type":["int"],"align":["right"]},{"label":["PS_father"],"name":[50],"type":["int"],"align":["right"]},{"label":["PS_proband"],"name":[51],"type":["int"],"align":["right"]},{"label":["CANONICAL"],"name":[52],"type":["chr"],"align":["left"]},{"label":["MANE_SELECT"],"name":[53],"type":["chr"],"align":["left"]},{"label":["MANE_PLUS_CLINICAL"],"name":[54],"type":["chr"],"align":["left"]},{"label":["IMPACT"],"name":[55],"type":["chr"],"align":["left"]},{"label":["Gene"],"name":[56],"type":["fct"],"align":["left"]},{"label":["HGNC_ID"],"name":[57],"type":["chr"],"align":["left"]},{"label":["Amino_acids"],"name":[58],"type":["chr"],"align":["left"]},{"label":["Codons"],"name":[59],"type":["chr"],"align":["left"]},{"label":["AF"],"name":[60],"type":["dbl"],"align":["right"]},{"label":["AFR_AF"],"name":[61],"type":["dbl"],"align":["right"]},{"label":["AMR_AF"],"name":[62],"type":["dbl"],"align":["right"]},{"label":["EAS_AF"],"name":[63],"type":["dbl"],"align":["right"]},{"label":["EUR_AF"],"name":[64],"type":["dbl"],"align":["right"]},{"label":["SAS_AF"],"name":[65],"type":["dbl"],"align":["right"]},{"label":["gnomADe_AFR_AF"],"name":[66],"type":["dbl"],"align":["right"]},{"label":["gnomADe_AMR_AF"],"name":[67],"type":["dbl"],"align":["right"]},{"label":["gnomADe_ASJ_AF"],"name":[68],"type":["dbl"],"align":["right"]},{"label":["gnomADe_EAS_AF"],"name":[69],"type":["dbl"],"align":["right"]},{"label":["gnomADe_FIN_AF"],"name":[70],"type":["dbl"],"align":["right"]},{"label":["gnomADe_NFE_AF"],"name":[71],"type":["dbl"],"align":["right"]},{"label":["gnomADe_OTH_AF"],"name":[72],"type":["dbl"],"align":["right"]},{"label":["gnomADe_SAS_AF"],"name":[73],"type":["dbl"],"align":["right"]},{"label":["gnomADg_AFR_AF"],"name":[74],"type":["dbl"],"align":["right"]},{"label":["gnomADg_AMI_AF"],"name":[75],"type":["dbl"],"align":["right"]},{"label":["gnomADg_AMR_AF"],"name":[76],"type":["dbl"],"align":["right"]},{"label":["gnomADg_ASJ_AF"],"name":[77],"type":["dbl"],"align":["right"]},{"label":["gnomADg_EAS_AF"],"name":[78],"type":["dbl"],"align":["right"]},{"label":["gnomADg_FIN_AF"],"name":[79],"type":["dbl"],"align":["right"]},{"label":["gnomADg_MID_AF"],"name":[80],"type":["dbl"],"align":["right"]},{"label":["gnomADg_NFE_AF"],"name":[81],"type":["dbl"],"align":["right"]},{"label":["gnomADg_OTH_AF"],"name":[82],"type":["dbl"],"align":["right"]},{"label":["gnomADg_SAS_AF"],"name":[83],"type":["dbl"],"align":["right"]}],"data":[{"1":"PCDHGB1","2":"chr5_141351402_A/G","3":"chr5:141351402","4":"NM_018922.3","5":"NM_018922.3:c.1142A>G","6":"NP_061745.1:p.Tyr381Cys","7":"missense_variant","8":"HET","9":"1/4","10":"NA","11":"4.108e-06","12":"NA","13":"9.104e-06","14":"gnomADe_NFE","15":"NA","16":"NA","17":"NA","18":"NA","19":"NA","20":"0.084","21":"tolerated_low_confidence(0.14)","22":"possibly_damaging(0.846)","23":"22.800","24":"NA","25":"PCDHGB1|0.00|0.00|0.00|0.00|15|30|-1|-44","26":"NA","27":"NA","28":"N,N,N,N","29":"1,1,1,1","30":"NA","31":"NA","32":"NA","33":"rs1326706527","34":"NA","35":"0/0","36":"0/0","37":"0/1","38":"G","39":"1","40":"33","41":"35","42":"226","43":"0","44":"0","45":"196","46":"39","47":"39","48":"99","49":"NA","50":"NA","51":"NA","52":"YES","53":"ENST00000523390.2","54":"NA","55":"MODERATE","56":"56104","57":"HGNC:8708","58":"Y/C","59":"tAt/tGt","60":"NA","61":"NA","62":"NA","63":"NA","64":"NA","65":"NA","66":"0.000e+00","67":"0.0000000","68":"0.000e+00","69":"0.0000000","70":"0.000e+00","71":"9.104e-06","72":"0.0000000","73":"0.000e+00","74":"NA","75":"NA","76":"NA","77":"NA","78":"NA","79":"NA","80":"NA","81":"NA","82":"NA","83":"NA"},{"1":"HLA-A","2":"chr6_29942825_G/T","3":"chr6:29942825","4":"NM_002116.8","5":"NM_002116.8:c.142G>T","6":"NP_002107.3:p.Ala48Ser","7":"missense_variant","8":"HET","9":"2/8","10":"NA","11":"1.143e-03","12":"3.590e-04","13":"3.568e-03","14":"gnomADe_FIN","15":"NA","16":"NA","17":"{Hypersensitivity syndrome, carbamazepine-induced, susceptibility to}, 608579 (3)","18":"NA","19":"Ocular hypertension, Abnormal chorioretinal morphology, Visual loss, Cystoid macular edema, Vitritis, Photoreceptor layer loss on macular OCT","20":"0.051","21":"tolerated_low_confidence(0.66)","22":"benign(0.011)","23":"3.179","24":"0.4336286","25":"HLA-A|0.04|0.00|0.00|0.00|27|-12|7|25","26":"NA","27":"NA","28":"N,N,N,N","29":"1,1,1,1","30":"NA","31":"NA","32":"HLA-A:A48S","33":"rs41552219,COSV65136702,COSV65138500","34":"NA","35":"0/0","36":"0/0","37":"0/1","38":"T","39":"1","40":"129","41":"83","42":"99","43":"0","44":"0","45":"9","46":"42","47":"42","48":"38","49":"NA","50":"NA","51":"29942652","52":"YES","53":"ENST00000376809.10","54":"NA","55":"MODERATE","56":"3105","57":"HGNC:4931","58":"A/S","59":"Gcc/Tcc","60":"NA","61":"NA","62":"NA","63":"NA","64":"NA","65":"NA","66":"2.644e-04","67":"0.0002447","68":"6.446e-04","69":"0.0009364","70":"3.568e-03","71":"1.562e-03","72":"0.0005267","73":"3.388e-05","74":"0.0001243","75":"0.00000","76":"0.0004886","77":"0.0003797","78":"0.0000000","79":"9.501e-04","80":"0.000000","81":"4.246e-04","82":"0.0006784","83":"0.0002607"},{"1":"MUC21","2":"chr6_30986915_CACCCTC/-","3":"chr6:30986915-30986921","4":"NM_001010909.5","5":"NM_001010909.5:c.742_748del","6":"NP_001010909.2:p.Pro248ValfsTer193","7":"frameshift_variant","8":"HET","9":"2/3","10":"NA","11":"3.588e-05","12":"4.005e-04","13":"5.865e-04","14":"gnomADg_ASJ","15":"NA","16":"NA","17":"NA","18":"NA","19":"NA","20":"NA","21":"NA","22":"NA","23":"22.600","24":"NA","25":"NA","26":"NA","27":"NA","28":"NA","29":"NA","30":"NA","31":"NA","32":"NA","33":"rs777714068","34":"NA","35":"0/0","36":"0/0","37":"0/1","38":"-","39":"1","40":"374","41":"224","42":"281","43":"0","44":"0","45":"38","46":"44","47":"44","48":"99","49":"NA","50":"NA","51":"NA","52":"YES","53":"ENST00000376296.3","54":"NA","55":"HIGH","56":"394263","57":"HGNC:21661","58":"TPS/X","59":"aCACCCTCc/ac","60":"NA","61":"NA","62":"NA","63":"NA","64":"NA","65":"NA","66":"6.172e-05","67":"0.0000000","68":"9.944e-05","69":"0.0000000","70":"4.631e-05","71":"4.410e-05","72":"0.0000000","73":"3.269e-05","74":"0.0003290","75":"0.00000","76":"0.0003385","77":"0.0005865","78":"0.0002045","79":"4.980e-04","80":"0.000000","81":"4.789e-04","82":"0.0004897","83":"0.0000000"},{"1":"HLA-DRB5","2":"chr6_32521978_-/CC","3":"chr6:32521977-32521978","4":"NM_002125.4","5":"NM_002125.4:c.297_298insGG","6":"NP_002116.2:p.Arg100GlyfsTer30","7":"frameshift_variant","8":"HOM","9":"2/6","10":"NA","11":"3.775e-03","12":"4.286e-02","13":"1.434e-01","14":"gnomADg_EAS","15":"NA","16":"NA","17":"NA","18":"NA","19":"NA","20":"NA","21":"NA","22":"NA","23":"6.965","24":"NA","25":"NA","26":"NA","27":"NA","28":"NA","29":"NA","30":"NA","31":"NA","32":"HLA-DRB5:R100fs","33":"rs201925120","34":"NA","35":"1/1","36":"0/0","37":"1/1","38":"CC","39":"1","40":"0","41":"50","42":"0","43":"181","44":"0","45":"221","46":"99","47":"30","48":"99","49":"32521972","50":"NA","51":"32521972","52":"YES","53":"ENST00000374975.4","54":"NA","55":"HIGH","56":"3127","57":"HGNC:4953","58":"-/X","59":"-/GG","60":"NA","61":"NA","62":"NA","63":"NA","64":"NA","65":"NA","66":"1.431e-02","67":"0.0050720","68":"1.667e-03","69":"0.0107800","70":"1.243e-03","71":"9.442e-04","72":"0.0025490","73":"4.937e-03","74":"0.0608700","75":"0.01042","76":"0.0400000","77":"0.0314900","78":"0.1434000","79":"5.286e-02","80":"0.049550","81":"2.266e-02","82":"0.0533800","83":"0.0540100"},{"1":"HLA-DRB5","2":"chr6_32530131_T/G","3":"chr6:32530131","4":"NM_002125.4","5":"NM_002125.4:c.94A>C","6":"NP_002116.2:p.Thr32Pro","7":"missense_variant","8":"HET","9":"1/6","10":"NA","11":"4.338e-06","12":"3.073e-03","13":"1.137e-02","14":"gnomADg_SAS","15":"NA","16":"NA","17":"NA","18":"NA","19":"NA","20":"0.068","21":"tolerated_low_confidence(0.19)","22":"probably_damaging(0.988)","23":"16.810","24":"0.3314275","25":"HLA-DRB5|0.00|0.00|0.00|0.00|-6|-2|-19|-6","26":"NA","27":"NA","28":"N","29":"1","30":"NA","31":"NA","32":"NA","33":"rs796754225","34":"NA","35":"0/0","36":"0/0","37":"0/1","38":"G","39":"1","40":"149","41":"168","42":"163","43":"0","44":"0","45":"28","46":"41","47":"41","48":"99","49":"NA","50":"NA","51":"32530090","52":"YES","53":"ENST00000374975.4","54":"NA","55":"MODERATE","56":"3127","57":"HGNC:4953","58":"T/P","59":"Acc/Ccc","60":"NA","61":"NA","62":"NA","63":"NA","64":"NA","65":"NA","66":"0.000e+00","67":"0.0000000","68":"0.000e+00","69":"0.0000000","70":"4.675e-05","71":"0.000e+00","72":"0.0000000","73":"0.000e+00","74":"0.0046620","75":"0.00000","76":"0.0033310","77":"0.0038240","78":"0.0045890","79":"1.429e-03","80":"0.003333","81":"1.573e-03","82":"0.0059000","83":"0.0113700"},{"1":"RP1L1","2":"chr8_10610091_A/C","3":"chr8:10610091","4":"NM_178857.6","5":"NM_178857.6:c.4007T>G","6":"NP_849188.4:p.Val1336Gly","7":"missense_variant","8":"HET","9":"4/4","10":"NA","11":"0.000e+00","12":"6.157e-05","13":"6.000e-04","14":"gnomADg_EAS","15":"NA","16":"NA","17":"Occult macular dystrophy, 613587 (3), Autosomal dominant; Retinitis pigmentosa 88, 618826 (3), Autosomal recessive","18":"NA","19":"Macular dystrophy, Photophobia, Progressive night blindness, Wide nasal bridge, Abnormality of retinal pigmentation, Hypogonadism, Optic atrophy, Abnormal testis morphology, Atypical scarring of skin, Intellectual disability, Sensorineural hearing impairment, Abnormal retinal vascular morphology, Hypoplasia of penis, Abnormal electroretinogram, Nystagmus, Anteverted nares, Blindness, Conductive hearing impairment","20":"0.021","21":"tolerated(0.26)","22":"unknown(0)","23":"2.276","24":"0.2726049","25":"RP1L1|0.00|0.00|0.00|0.00|-30|-21|50|2","26":"NA","27":"NA","28":"N","29":"1","30":"retinitis pigmentosa, occult macular dystrophy","31":"Autosomal dominant","32":"NA","33":"rs1238834181","34":"NA","35":"0/0","36":"0/0","37":"0/1","38":"C","39":"1","40":"274","41":"173","42":"606","43":"0","44":"0","45":"113","46":"56","47":"56","48":"99","49":"NA","50":"NA","51":"NA","52":"YES","53":"ENST00000382483.4","54":"NA","55":"MODERATE","56":"94137","57":"HGNC:15946","58":"V/G","59":"gTg/gGg","60":"NA","61":"NA","62":"NA","63":"NA","64":"NA","65":"NA","66":"0.000e+00","67":"0.0000000","68":"0.000e+00","69":"0.0000000","70":"0.000e+00","71":"0.000e+00","72":"0.0000000","73":"0.000e+00","74":"0.0000000","75":"0.00000","76":"0.0000000","77":"0.0000000","78":"0.0006000","79":"9.994e-05","80":"0.000000","81":"7.459e-05","82":"0.0000000","83":"0.0000000"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="homozygous-variants" class="section level2 hasAnchor" number="5.3">
<h2><span class="header-section-number">5.3</span> Homozygous Variants<a href="variant-annotation.html#homozygous-variants" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We select the homozygous variants in the proband.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="variant-annotation.html#cb32-1" aria-hidden="true" tabindex="-1"></a>geno_hom <span class="ot">&lt;-</span> wh_geno_auto <span class="sc">%&gt;%</span> </span>
<span id="cb32-2"><a href="variant-annotation.html#cb32-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)<span class="sc">==</span><span class="fu">substr</span>(proband,<span class="dv">3</span>,<span class="dv">3</span>) <span class="sc">&amp;</span> <span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>) <span class="sc">!=</span> <span class="dv">0</span> <span class="sc">&amp;</span> <span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>) <span class="sc">!=</span> <span class="st">&quot;.&quot;</span>)</span></code></pre></div>
<p>We discard the homozygous variants for which one of the parents is also homozygous (e.g.Â 1/1, 0/1 -&gt; 1/1) since these variants are unlikely to be pathogenic.</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="variant-annotation.html#cb33-1" aria-hidden="true" tabindex="-1"></a>geno_hom2 <span class="ot">&lt;-</span> geno_hom <span class="sc">%&gt;%</span></span>
<span id="cb33-2"><a href="variant-annotation.html#cb33-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span> ( (<span class="fu">substr</span>(mother,<span class="dv">1</span>,<span class="dv">1</span>)<span class="sc">==</span><span class="fu">substr</span>(mother,<span class="dv">3</span>,<span class="dv">3</span>) <span class="sc">&amp;</span> <span class="fu">substr</span>(mother,<span class="dv">1</span>,<span class="dv">1</span>)<span class="sc">==</span><span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)) <span class="sc">|</span></span>
<span id="cb33-3"><a href="variant-annotation.html#cb33-3" aria-hidden="true" tabindex="-1"></a>                      (<span class="fu">substr</span>(father,<span class="dv">1</span>,<span class="dv">1</span>)<span class="sc">==</span><span class="fu">substr</span>(father,<span class="dv">3</span>,<span class="dv">3</span>) <span class="sc">&amp;</span> <span class="fu">substr</span>(father,<span class="dv">1</span>,<span class="dv">1</span>)<span class="sc">==</span><span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)) )</span>
<span id="cb33-4"><a href="variant-annotation.html#cb33-4" aria-hidden="true" tabindex="-1"></a>                  )</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["Location"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Allele"],"name":[2],"type":["chr"],"align":["left"]},{"label":["proband"],"name":[3],"type":["chr"],"align":["left"]},{"label":["father"],"name":[4],"type":["chr"],"align":["left"]},{"label":["mother"],"name":[5],"type":["chr"],"align":["left"]},{"label":["GQ_proband"],"name":[6],"type":["int"],"align":["right"]},{"label":["GQ_father"],"name":[7],"type":["int"],"align":["right"]},{"label":["GQ_mother"],"name":[8],"type":["int"],"align":["right"]},{"label":["PS_proband"],"name":[9],"type":["int"],"align":["right"]},{"label":["PS_father"],"name":[10],"type":["int"],"align":["right"]},{"label":["PS_mother"],"name":[11],"type":["int"],"align":["right"]},{"label":["AD_ref_proband"],"name":[12],"type":["int"],"align":["right"]},{"label":["AD_ref_father"],"name":[13],"type":["int"],"align":["right"]},{"label":["AD_ref_mother"],"name":[14],"type":["int"],"align":["right"]},{"label":["AD_alt_proband"],"name":[15],"type":["int"],"align":["right"]},{"label":["AD_alt_father"],"name":[16],"type":["int"],"align":["right"]},{"label":["AD_alt_mother"],"name":[17],"type":["int"],"align":["right"]},{"label":["counter"],"name":[18],"type":["int"],"align":["right"]}],"data":[{"1":"chr1:1629847-1629853","2":"-","3":"1/1","4":"0|1","5":"1|0","6":"18","7":"25","8":"9","9":"NA","10":"65797","11":"65797","12":"0","13":"5","14":"1","15":"2","16":"0","17":"4","18":"1"},{"1":"chr1:2371909","2":"-","3":"1/1","4":"1|0","5":"1|0","6":"3","7":"75","8":"24","9":"NA","10":"65797","11":"65797","12":"1","13":"7","14":"8","15":"2","16":"3","17":"1","18":"1"},{"1":"chr1:3385157","2":"T","3":"1/1","4":"1|0","5":"1|0","6":"99","7":"99","8":"99","9":"NA","10":"65797","11":"65797","12":"2","13":"47","14":"53","15":"113","16":"36","17":"46","18":"1"},{"1":"chr1:3396350","2":"G","3":"1/1","4":"1|0","5":"1|0","6":"99","7":"99","8":"99","9":"NA","10":"65797","11":"65797","12":"0","13":"138","14":"161","15":"295","16":"156","17":"139","18":"1"},{"1":"chr1:3815401","2":"A","3":"1/1","4":"1|0","5":"1|0","6":"99","7":"99","8":"99","9":"NA","10":"65797","11":"65797","12":"0","13":"108","14":"127","15":"270","16":"94","17":"124","18":"1"},{"1":"chr1:3825693","2":"T","3":"1/1","4":"1|0","5":"1|0","6":"99","7":"99","8":"99","9":"NA","10":"65797","11":"65797","12":"0","13":"13","14":"11","15":"34","16":"13","17":"15","18":"1"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>We select the homozygous variants found in the filtered VEP annotation.</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="variant-annotation.html#cb34-1" aria-hidden="true" tabindex="-1"></a>prob_hom <span class="ot">&lt;-</span> refseq <span class="sc">%&gt;%</span> <span class="fu">inner_join</span>(geno_hom2)</span>
<span id="cb34-2"><a href="variant-annotation.html#cb34-2" aria-hidden="true" tabindex="-1"></a>prob_hom <span class="ot">&lt;-</span> prob_hom <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(Uploaded_variation) <span class="sc">%&gt;%</span> </span>
<span id="cb34-3"><a href="variant-annotation.html#cb34-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(CANONICAL), <span class="fu">as.character</span>(Feature),<span class="at">.by_group =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb34-4"><a href="variant-annotation.html#cb34-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Feature <span class="sc">==</span> Feature[<span class="dv">1</span>])</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["SYMBOL"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Uploaded_variation"],"name":[2],"type":["fct"],"align":["left"]},{"label":["Location"],"name":[3],"type":["chr"],"align":["left"]},{"label":["Feature"],"name":[4],"type":["fct"],"align":["left"]},{"label":["HGVSc"],"name":[5],"type":["chr"],"align":["left"]},{"label":["HGVSp"],"name":[6],"type":["chr"],"align":["left"]},{"label":["Consequence"],"name":[7],"type":["chr"],"align":["left"]},{"label":["ZYG"],"name":[8],"type":["chr"],"align":["left"]},{"label":["EXON"],"name":[9],"type":["chr"],"align":["left"]},{"label":["INTRON"],"name":[10],"type":["chr"],"align":["left"]},{"label":["gnomADe_AF"],"name":[11],"type":["dbl"],"align":["right"]},{"label":["gnomADg_AF"],"name":[12],"type":["dbl"],"align":["right"]},{"label":["MAX_AF"],"name":[13],"type":["dbl"],"align":["right"]},{"label":["MAX_AF_POPS"],"name":[14],"type":["chr"],"align":["left"]},{"label":["ClinVar"],"name":[15],"type":["dbl"],"align":["right"]},{"label":["ClinVar_CLNSIG"],"name":[16],"type":["chr"],"align":["left"]},{"label":["OMIM_Phenotypes"],"name":[17],"type":["chr"],"align":["left"]},{"label":["renal_hpo_term"],"name":[18],"type":["chr"],"align":["left"]},{"label":["extrarenal_hpo_term"],"name":[19],"type":["chr"],"align":["left"]},{"label":["REVEL"],"name":[20],"type":["dbl"],"align":["right"]},{"label":["SIFT"],"name":[21],"type":["chr"],"align":["left"]},{"label":["PolyPhen"],"name":[22],"type":["chr"],"align":["left"]},{"label":["CADD_PHRED"],"name":[23],"type":["dbl"],"align":["right"]},{"label":["PrimateAI"],"name":[24],"type":["dbl"],"align":["right"]},{"label":["SpliceAI_pred"],"name":[25],"type":["chr"],"align":["left"]},{"label":["EVE_CLASS"],"name":[26],"type":["chr"],"align":["left"]},{"label":["EVE_SCORE"],"name":[27],"type":["dbl"],"align":["right"]},{"label":["MutationTaster_pred"],"name":[28],"type":["chr"],"align":["left"]},{"label":["MutationTaster_score"],"name":[29],"type":["chr"],"align":["left"]},{"label":["orphanet_disease"],"name":[30],"type":["chr"],"align":["left"]},{"label":["orphanet_moi"],"name":[31],"type":["chr"],"align":["left"]},{"label":["Mastermind_MMID3"],"name":[32],"type":["chr"],"align":["left"]},{"label":["Existing_variation"],"name":[33],"type":["chr"],"align":["left"]},{"label":["PUBMED"],"name":[34],"type":["chr"],"align":["left"]},{"label":["mother"],"name":[35],"type":["chr"],"align":["left"]},{"label":["father"],"name":[36],"type":["chr"],"align":["left"]},{"label":["proband"],"name":[37],"type":["chr"],"align":["left"]},{"label":["Allele"],"name":[38],"type":["chr"],"align":["left"]},{"label":["counter"],"name":[39],"type":["int"],"align":["right"]},{"label":["AD_ref_mother"],"name":[40],"type":["int"],"align":["right"]},{"label":["AD_ref_father"],"name":[41],"type":["int"],"align":["right"]},{"label":["AD_ref_proband"],"name":[42],"type":["int"],"align":["right"]},{"label":["AD_alt_mother"],"name":[43],"type":["int"],"align":["right"]},{"label":["AD_alt_father"],"name":[44],"type":["int"],"align":["right"]},{"label":["AD_alt_proband"],"name":[45],"type":["int"],"align":["right"]},{"label":["GQ_mother"],"name":[46],"type":["int"],"align":["right"]},{"label":["GQ_father"],"name":[47],"type":["int"],"align":["right"]},{"label":["GQ_proband"],"name":[48],"type":["int"],"align":["right"]},{"label":["PS_mother"],"name":[49],"type":["int"],"align":["right"]},{"label":["PS_father"],"name":[50],"type":["int"],"align":["right"]},{"label":["PS_proband"],"name":[51],"type":["int"],"align":["right"]},{"label":["CANONICAL"],"name":[52],"type":["chr"],"align":["left"]},{"label":["MANE_SELECT"],"name":[53],"type":["chr"],"align":["left"]},{"label":["MANE_PLUS_CLINICAL"],"name":[54],"type":["chr"],"align":["left"]},{"label":["IMPACT"],"name":[55],"type":["chr"],"align":["left"]},{"label":["Gene"],"name":[56],"type":["fct"],"align":["left"]},{"label":["HGNC_ID"],"name":[57],"type":["chr"],"align":["left"]},{"label":["Amino_acids"],"name":[58],"type":["chr"],"align":["left"]},{"label":["Codons"],"name":[59],"type":["chr"],"align":["left"]},{"label":["AF"],"name":[60],"type":["dbl"],"align":["right"]},{"label":["AFR_AF"],"name":[61],"type":["dbl"],"align":["right"]},{"label":["AMR_AF"],"name":[62],"type":["dbl"],"align":["right"]},{"label":["EAS_AF"],"name":[63],"type":["dbl"],"align":["right"]},{"label":["EUR_AF"],"name":[64],"type":["dbl"],"align":["right"]},{"label":["SAS_AF"],"name":[65],"type":["dbl"],"align":["right"]},{"label":["gnomADe_AFR_AF"],"name":[66],"type":["dbl"],"align":["right"]},{"label":["gnomADe_AMR_AF"],"name":[67],"type":["dbl"],"align":["right"]},{"label":["gnomADe_ASJ_AF"],"name":[68],"type":["dbl"],"align":["right"]},{"label":["gnomADe_EAS_AF"],"name":[69],"type":["dbl"],"align":["right"]},{"label":["gnomADe_FIN_AF"],"name":[70],"type":["dbl"],"align":["right"]},{"label":["gnomADe_NFE_AF"],"name":[71],"type":["dbl"],"align":["right"]},{"label":["gnomADe_OTH_AF"],"name":[72],"type":["dbl"],"align":["right"]},{"label":["gnomADe_SAS_AF"],"name":[73],"type":["dbl"],"align":["right"]},{"label":["gnomADg_AFR_AF"],"name":[74],"type":["dbl"],"align":["right"]},{"label":["gnomADg_AMI_AF"],"name":[75],"type":["dbl"],"align":["right"]},{"label":["gnomADg_AMR_AF"],"name":[76],"type":["dbl"],"align":["right"]},{"label":["gnomADg_ASJ_AF"],"name":[77],"type":["dbl"],"align":["right"]},{"label":["gnomADg_EAS_AF"],"name":[78],"type":["dbl"],"align":["right"]},{"label":["gnomADg_FIN_AF"],"name":[79],"type":["dbl"],"align":["right"]},{"label":["gnomADg_MID_AF"],"name":[80],"type":["dbl"],"align":["right"]},{"label":["gnomADg_NFE_AF"],"name":[81],"type":["dbl"],"align":["right"]},{"label":["gnomADg_OTH_AF"],"name":[82],"type":["dbl"],"align":["right"]},{"label":["gnomADg_SAS_AF"],"name":[83],"type":["dbl"],"align":["right"]}],"data":[{"1":"TBC1D2","2":"chr9_98199422_C/T","3":"chr9:98199422","4":"NM_001267571.2","5":"NM_001267571.2:c.2746G>A","6":"NP_001254500.1:p.Glu916Lys","7":"missense_variant","8":"HOM","9":"13/13","10":"NA","11":"0.0075510","12":"0.005973","13":"0.02104","14":"gnomADg_ASJ","15":"770388","16":"Benign","17":"NA","18":"NA","19":"NA","20":"0.028","21":"deleterious(0.04)","22":"NA","23":"13.460","24":"0.3835646","25":"TBC1D2|0.00|0.00|0.00|0.00|28|41|29|46","26":"NA","27":"NA","28":"D,N,N,N","29":"0.999997,0.999999,0.999998,1","30":"NA","31":"NA","32":"TBC1D2:E456K,TBC1D2:E905K,TBC1D2:E916K","33":"rs34769888","34":"NA","35":"0|1","36":"0|1","37":"1/1","38":"T","39":"1","40":"17","41":"18","42":"0","43":"30","44":"20","45":"67","46":"99","47":"99","48":"99","49":"154795","50":"154795","51":"NA","52":"YES","53":"ENST00000465784.7","54":"NA","55":"MODERATE","56":"55357","57":"HGNC:18026","58":"E/K","59":"Gag/Aag","60":"0.004","61":"0","62":"0.0043","63":"0","64":"0.006","65":"0.0112","66":"0.0020390","67":"0.0041070","68":"0.0175300","69":"1.089e-04","70":"0.0075790","71":"0.0085020","72":"0.0075090","73":"1.202e-02","74":"0.001786","75":"0.002193","76":"0.003925","77":"0.02104","78":"0.0003857","79":"0.005651","80":"0.003165","81":"0.00813","82":"0.004776","83":"0.01531"},{"1":"TAS2R43","2":"chr12_11091469_-/TT","3":"chr12:11091468-11091469","4":"NM_176884.2","5":"NM_176884.2:c.761_762insAA","6":"NP_795365.2:p.Ser254ArgfsTer27","7":"frameshift_variant","8":"HOM","9":"1/1","10":"NA","11":"0.0005273","12":"0.329200","13":"0.48440","14":"gnomADg_AMI","15":"NA","16":"NA","17":"NA","18":"NA","19":"NA","20":"NA","21":"NA","22":"NA","23":"3.365","24":"NA","25":"NA","26":"NA","27":"NA","28":"NA","29":"NA","30":"NA","31":"NA","32":"PRH1-TAS2R14:5UTR,PRH1-TAS2R14:5UTRint,PRH1-TAS2R14:5UTRinta,PRH1:5UTR,PRH1:5UTRint,PRH1:5UTRinta,TAS2R43:S254fs","33":"rs760672236,COSV67859539","34":"NA","35":"1|0","36":"0|1","37":"1/1","38":"TT","39":"1","40":"387","41":"330","42":"0","43":"69","44":"61","45":"70","46":"99","47":"99","48":"99","49":"125960","50":"125960","51":"NA","52":"YES","53":"ENST00000531678.1","54":"NA","55":"HIGH","56":"259289","57":"HGNC:18875","58":"S/RX","59":"agt/agAAt","60":"NA","61":"NA","62":"NA","63":"NA","64":"NA","65":"NA","66":"0.0003650","67":"0.0004966","68":"0.0000000","69":"0.000e+00","70":"0.0008549","71":"0.0008820","72":"0.0000000","73":"0.000e+00","74":"0.133100","75":"0.484400","76":"0.368700","77":"0.40010","78":"0.4458000","79":"0.460100","80":"0.408600","81":"0.42050","82":"0.341500","83":"0.35950"},{"1":"TAS2R43","2":"chr12_11091472_CC/-","3":"chr12:11091472-11091473","4":"NM_176884.2","5":"NM_176884.2:c.757_758del","6":"NP_795365.2:p.Gly253LysfsTer15","7":"frameshift_variant","8":"HOM","9":"1/1","10":"NA","11":"0.0002696","12":"0.362100","13":"0.52070","14":"gnomADg_AMI","15":"NA","16":"NA","17":"NA","18":"NA","19":"NA","20":"NA","21":"NA","22":"NA","23":"14.430","24":"NA","25":"TAS2R43|0.04|0.10|0.00|0.00|-38|13|27|-50","26":"NA","27":"NA","28":"NA","29":"NA","30":"NA","31":"NA","32":"PRH1-TAS2R14:5UTR,PRH1-TAS2R14:5UTRint,PRH1-TAS2R14:5UTRinta,PRH1:5UTR,PRH1:5UTRint,PRH1:5UTRinta,TAS2R43:G253fs","33":"rs765322779,COSV67852826","34":"NA","35":"1|0","36":"0|1","37":"1/1","38":"-","39":"1","40":"394","41":"334","42":"0","43":"72","44":"64","45":"73","46":"99","47":"99","48":"99","49":"125960","50":"125960","51":"NA","52":"YES","53":"ENST00000531678.1","54":"NA","55":"HIGH","56":"259289","57":"HGNC:18875","58":"G/X","59":"GGa/a","60":"NA","61":"NA","62":"NA","63":"NA","64":"NA","65":"NA","66":"0.0001462","67":"0.0002491","68":"0.0000000","69":"0.000e+00","70":"0.0004284","71":"0.0004608","72":"0.0000000","73":"0.000e+00","74":"0.159800","75":"0.520700","76":"0.400700","77":"0.41560","78":"0.4632000","79":"0.493800","80":"0.409100","81":"0.45230","82":"0.360300","83":"0.37600"},{"1":"TAS2R43","2":"chr12_11091492_C/T","3":"chr12:11091492","4":"NM_176884.2","5":"NM_176884.2:c.738G>A","6":"NP_795365.2:p.Met246Ile","7":"missense_variant","8":"HOM","9":"1/1","10":"NA","11":"0.0006609","12":"0.436000","13":"0.59040","14":"gnomADg_AMI","15":"NA","16":"NA","17":"NA","18":"NA","19":"NA","20":"0.015","21":"tolerated(1)","22":"benign(0)","23":"0.001","24":"0.3738058","25":"TAS2R43|0.01|0.01|0.01|0.00|-8|14|48|0","26":"NA","27":"NA","28":"P","29":"1","30":"NA","31":"NA","32":"PRH1-TAS2R14:5UTR,PRH1-TAS2R14:5UTRint,PRH1-TAS2R14:5UTRinta,PRH1:5UTR,PRH1:5UTRint,PRH1:5UTRinta,TAS2R43:M246I","33":"rs199768488,COSV67852484","34":"NA","35":"1|0","36":"0|1","37":"1/1","38":"T","39":"1","40":"336","41":"275","42":"0","43":"88","44":"84","45":"95","46":"99","47":"99","48":"99","49":"125960","50":"125960","51":"NA","52":"YES","53":"ENST00000531678.1","54":"NA","55":"MODERATE","56":"259289","57":"HGNC:18875","58":"M/I","59":"atG/atA","60":"NA","61":"NA","62":"NA","63":"NA","64":"NA","65":"NA","66":"0.0003587","67":"0.0004266","68":"0.0000000","69":"0.000e+00","70":"0.0014960","71":"0.0011050","72":"0.0002041","73":"0.000e+00","74":"0.227800","75":"0.590400","76":"0.464900","77":"0.47900","78":"0.4878000","79":"0.551500","80":"0.481800","81":"0.52600","82":"0.425100","83":"0.44150"},{"1":"TAS2R43","2":"chr12_11091497_T/C","3":"chr12:11091497","4":"NM_176884.2","5":"NM_176884.2:c.733A>G","6":"NP_795365.2:p.Ile245Val","7":"missense_variant","8":"HOM","9":"1/1","10":"NA","11":"0.0006957","12":"0.442500","13":"0.59800","14":"gnomADg_AMI","15":"NA","16":"NA","17":"NA","18":"NA","19":"NA","20":"0.014","21":"tolerated(0.39)","22":"benign(0)","23":"0.001","24":"0.2216672","25":"TAS2R43|0.00|0.01|0.02|0.00|-5|9|5|-5","26":"NA","27":"NA","28":"P","29":"1","30":"NA","31":"NA","32":"PRH1-TAS2R14:5UTR,PRH1-TAS2R14:5UTRint,PRH1-TAS2R14:5UTRinta,PRH1:5UTR,PRH1:5UTRint,PRH1:5UTRinta,TAS2R43:I245V","33":"rs752123624,COSV67852487","34":"NA","35":"1|0","36":"0|1","37":"1/1","38":"C","39":"1","40":"327","41":"263","42":"0","43":"88","44":"87","45":"103","46":"99","47":"99","48":"99","49":"125960","50":"125960","51":"NA","52":"YES","53":"ENST00000531678.1","54":"NA","55":"MODERATE","56":"259289","57":"HGNC:18875","58":"I/V","59":"Ata/Gta","60":"NA","61":"NA","62":"NA","63":"NA","64":"NA","65":"NA","66":"0.0007149","67":"0.0003913","68":"0.0000000","69":"5.694e-05","70":"0.0014970","71":"0.0011290","72":"0.0002039","73":"0.000e+00","74":"0.234100","75":"0.598000","76":"0.468700","77":"0.48460","78":"0.4895000","79":"0.558500","80":"0.481800","81":"0.53290","82":"0.433300","83":"0.44300"},{"1":"TAS2R43","2":"chr12_11091503_G/C","3":"chr12:11091503","4":"NM_176884.2","5":"NM_176884.2:c.727C>G","6":"NP_795365.2:p.Leu243Val","7":"missense_variant","8":"HOM","9":"1/1","10":"NA","11":"0.0038350","12":"0.449000","13":"0.60800","14":"gnomADg_AMI","15":"NA","16":"NA","17":"NA","18":"NA","19":"NA","20":"0.024","21":"tolerated(0.24)","22":"benign(0.003)","23":"0.002","24":"0.2723038","25":"TAS2R43|0.02|0.04|0.01|0.01|3|-19|34|37","26":"NA","27":"NA","28":"P","29":"1","30":"NA","31":"NA","32":"PRH1-TAS2R14:5UTR,PRH1-TAS2R14:5UTRint,PRH1-TAS2R14:5UTRinta,PRH1:5UTR,PRH1:5UTRint,PRH1:5UTRinta,TAS2R43:L243V","33":"rs199943442,COSV67852490,COSV67855574","34":"NA","35":"1|0","36":"0|1","37":"1/1","38":"C","39":"1","40":"300","41":"247","42":"0","43":"89","44":"86","45":"105","46":"99","47":"99","48":"99","49":"125960","50":"125960","51":"NA","52":"YES","53":"ENST00000531678.1","54":"NA","55":"MODERATE","56":"259289","57":"HGNC:18875","58":"L/V","59":"Ctg/Gtg","60":"NA","61":"NA","62":"NA","63":"NA","64":"NA","65":"NA","66":"0.0028580","67":"0.0025330","68":"0.0003336","69":"5.741e-05","70":"0.0060930","71":"0.0066370","72":"0.0014560","73":"7.064e-05","74":"0.240200","75":"0.608000","76":"0.475100","77":"0.48650","78":"0.4892000","79":"0.564900","80":"0.481800","81":"0.53970","82":"0.436500","83":"0.44790"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="compound-heterozygous-variants" class="section level2 hasAnchor" number="5.4">
<h2><span class="header-section-number">5.4</span> Compound Heterozygous Variants<a href="variant-annotation.html#compound-heterozygous-variants" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We select the unphased compound heterozygous variants.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="variant-annotation.html#cb35-1" aria-hidden="true" tabindex="-1"></a>prob_2plus_het <span class="ot">&lt;-</span> refseq <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)<span class="sc">!=</span><span class="fu">substr</span>(proband,<span class="dv">3</span>,<span class="dv">3</span>)) </span>
<span id="cb35-2"><a href="variant-annotation.html#cb35-2" aria-hidden="true" tabindex="-1"></a>prob_2plus_het <span class="ot">&lt;-</span> prob_2plus_het <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(Feature) <span class="sc">%&gt;%</span> </span>
<span id="cb35-3"><a href="variant-annotation.html#cb35-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">length</span>(<span class="fu">unique</span>(PS_proband))<span class="sc">&gt;=</span><span class="dv">2</span> <span class="sc">|</span> <span class="fu">length</span>(<span class="fu">is.na</span>(PS_proband)[<span class="fu">is.na</span>(PS_proband)<span class="sc">==</span><span class="cn">TRUE</span>])<span class="sc">&gt;=</span><span class="dv">2</span> ) <span class="sc">%&gt;%</span> </span>
<span id="cb35-4"><a href="variant-annotation.html#cb35-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb35-5"><a href="variant-annotation.html#cb35-5" aria-hidden="true" tabindex="-1"></a>prob_2plus_het2 <span class="ot">&lt;-</span> prob_2plus_het <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(Gene) <span class="sc">%&gt;%</span> </span>
<span id="cb35-6"><a href="variant-annotation.html#cb35-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(CANONICAL), <span class="fu">as.character</span>(Feature),<span class="at">.by_group =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb35-7"><a href="variant-annotation.html#cb35-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Feature <span class="sc">==</span> Feature[<span class="dv">1</span>])</span></code></pre></div>
<p>We select the phased compound heterozygous variants.</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="variant-annotation.html#cb36-1" aria-hidden="true" tabindex="-1"></a>prob_comp_het <span class="ot">&lt;-</span> refseq <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(Feature, PS_proband) <span class="sc">%&gt;%</span> </span>
<span id="cb36-2"><a href="variant-annotation.html#cb36-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">any</span>(<span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)<span class="sc">&gt;=</span><span class="dv">1</span> <span class="sc">&amp;</span> <span class="fu">substr</span>(proband,<span class="dv">2</span>,<span class="dv">2</span>)<span class="sc">==</span><span class="st">&quot;|&quot;</span>) <span class="sc">&amp;</span> <span class="fu">any</span>(<span class="fu">substr</span>(proband,<span class="dv">3</span>,<span class="dv">3</span>)<span class="sc">&gt;=</span><span class="dv">1</span> <span class="sc">&amp;</span> <span class="fu">substr</span>(proband,<span class="dv">2</span>,<span class="dv">2</span>)<span class="sc">==</span><span class="st">&quot;|&quot;</span>))</span>
<span id="cb36-3"><a href="variant-annotation.html#cb36-3" aria-hidden="true" tabindex="-1"></a>prob_comp_het2 <span class="ot">&lt;-</span> prob_comp_het <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(Gene) <span class="sc">%&gt;%</span> </span>
<span id="cb36-4"><a href="variant-annotation.html#cb36-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(CANONICAL), <span class="fu">as.character</span>(Feature),<span class="at">.by_group =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb36-5"><a href="variant-annotation.html#cb36-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Feature <span class="sc">==</span> Feature[<span class="dv">1</span>])</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["SYMBOL"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Uploaded_variation"],"name":[2],"type":["fct"],"align":["left"]},{"label":["Location"],"name":[3],"type":["chr"],"align":["left"]},{"label":["Feature"],"name":[4],"type":["fct"],"align":["left"]},{"label":["HGVSc"],"name":[5],"type":["chr"],"align":["left"]},{"label":["HGVSp"],"name":[6],"type":["chr"],"align":["left"]},{"label":["Consequence"],"name":[7],"type":["chr"],"align":["left"]},{"label":["ZYG"],"name":[8],"type":["chr"],"align":["left"]},{"label":["EXON"],"name":[9],"type":["chr"],"align":["left"]},{"label":["INTRON"],"name":[10],"type":["chr"],"align":["left"]},{"label":["gnomADe_AF"],"name":[11],"type":["dbl"],"align":["right"]},{"label":["gnomADg_AF"],"name":[12],"type":["dbl"],"align":["right"]},{"label":["MAX_AF"],"name":[13],"type":["dbl"],"align":["right"]},{"label":["MAX_AF_POPS"],"name":[14],"type":["chr"],"align":["left"]},{"label":["ClinVar"],"name":[15],"type":["dbl"],"align":["right"]},{"label":["ClinVar_CLNSIG"],"name":[16],"type":["chr"],"align":["left"]},{"label":["OMIM_Phenotypes"],"name":[17],"type":["chr"],"align":["left"]},{"label":["renal_hpo_term"],"name":[18],"type":["chr"],"align":["left"]},{"label":["extrarenal_hpo_term"],"name":[19],"type":["chr"],"align":["left"]},{"label":["REVEL"],"name":[20],"type":["dbl"],"align":["right"]},{"label":["SIFT"],"name":[21],"type":["chr"],"align":["left"]},{"label":["PolyPhen"],"name":[22],"type":["chr"],"align":["left"]},{"label":["CADD_PHRED"],"name":[23],"type":["dbl"],"align":["right"]},{"label":["PrimateAI"],"name":[24],"type":["dbl"],"align":["right"]},{"label":["SpliceAI_pred"],"name":[25],"type":["chr"],"align":["left"]},{"label":["EVE_CLASS"],"name":[26],"type":["chr"],"align":["left"]},{"label":["EVE_SCORE"],"name":[27],"type":["dbl"],"align":["right"]},{"label":["MutationTaster_pred"],"name":[28],"type":["chr"],"align":["left"]},{"label":["MutationTaster_score"],"name":[29],"type":["chr"],"align":["left"]},{"label":["orphanet_disease"],"name":[30],"type":["chr"],"align":["left"]},{"label":["orphanet_moi"],"name":[31],"type":["chr"],"align":["left"]},{"label":["Mastermind_MMID3"],"name":[32],"type":["chr"],"align":["left"]},{"label":["Existing_variation"],"name":[33],"type":["chr"],"align":["left"]},{"label":["PUBMED"],"name":[34],"type":["chr"],"align":["left"]},{"label":["mother"],"name":[35],"type":["chr"],"align":["left"]},{"label":["father"],"name":[36],"type":["chr"],"align":["left"]},{"label":["proband"],"name":[37],"type":["chr"],"align":["left"]},{"label":["Allele"],"name":[38],"type":["chr"],"align":["left"]},{"label":["counter"],"name":[39],"type":["int"],"align":["right"]},{"label":["AD_ref_mother"],"name":[40],"type":["int"],"align":["right"]},{"label":["AD_ref_father"],"name":[41],"type":["int"],"align":["right"]},{"label":["AD_ref_proband"],"name":[42],"type":["int"],"align":["right"]},{"label":["AD_alt_mother"],"name":[43],"type":["int"],"align":["right"]},{"label":["AD_alt_father"],"name":[44],"type":["int"],"align":["right"]},{"label":["AD_alt_proband"],"name":[45],"type":["int"],"align":["right"]},{"label":["GQ_mother"],"name":[46],"type":["int"],"align":["right"]},{"label":["GQ_father"],"name":[47],"type":["int"],"align":["right"]},{"label":["GQ_proband"],"name":[48],"type":["int"],"align":["right"]},{"label":["PS_mother"],"name":[49],"type":["int"],"align":["right"]},{"label":["PS_father"],"name":[50],"type":["int"],"align":["right"]},{"label":["PS_proband"],"name":[51],"type":["int"],"align":["right"]},{"label":["CANONICAL"],"name":[52],"type":["chr"],"align":["left"]},{"label":["MANE_SELECT"],"name":[53],"type":["chr"],"align":["left"]},{"label":["MANE_PLUS_CLINICAL"],"name":[54],"type":["chr"],"align":["left"]},{"label":["IMPACT"],"name":[55],"type":["chr"],"align":["left"]},{"label":["Gene"],"name":[56],"type":["fct"],"align":["left"]},{"label":["HGNC_ID"],"name":[57],"type":["chr"],"align":["left"]},{"label":["Amino_acids"],"name":[58],"type":["chr"],"align":["left"]},{"label":["Codons"],"name":[59],"type":["chr"],"align":["left"]},{"label":["AF"],"name":[60],"type":["dbl"],"align":["right"]},{"label":["AFR_AF"],"name":[61],"type":["dbl"],"align":["right"]},{"label":["AMR_AF"],"name":[62],"type":["dbl"],"align":["right"]},{"label":["EAS_AF"],"name":[63],"type":["dbl"],"align":["right"]},{"label":["EUR_AF"],"name":[64],"type":["dbl"],"align":["right"]},{"label":["SAS_AF"],"name":[65],"type":["dbl"],"align":["right"]},{"label":["gnomADe_AFR_AF"],"name":[66],"type":["dbl"],"align":["right"]},{"label":["gnomADe_AMR_AF"],"name":[67],"type":["dbl"],"align":["right"]},{"label":["gnomADe_ASJ_AF"],"name":[68],"type":["dbl"],"align":["right"]},{"label":["gnomADe_EAS_AF"],"name":[69],"type":["dbl"],"align":["right"]},{"label":["gnomADe_FIN_AF"],"name":[70],"type":["dbl"],"align":["right"]},{"label":["gnomADe_NFE_AF"],"name":[71],"type":["dbl"],"align":["right"]},{"label":["gnomADe_OTH_AF"],"name":[72],"type":["dbl"],"align":["right"]},{"label":["gnomADe_SAS_AF"],"name":[73],"type":["dbl"],"align":["right"]},{"label":["gnomADg_AFR_AF"],"name":[74],"type":["dbl"],"align":["right"]},{"label":["gnomADg_AMI_AF"],"name":[75],"type":["dbl"],"align":["right"]},{"label":["gnomADg_AMR_AF"],"name":[76],"type":["dbl"],"align":["right"]},{"label":["gnomADg_ASJ_AF"],"name":[77],"type":["dbl"],"align":["right"]},{"label":["gnomADg_EAS_AF"],"name":[78],"type":["dbl"],"align":["right"]},{"label":["gnomADg_FIN_AF"],"name":[79],"type":["dbl"],"align":["right"]},{"label":["gnomADg_MID_AF"],"name":[80],"type":["dbl"],"align":["right"]},{"label":["gnomADg_NFE_AF"],"name":[81],"type":["dbl"],"align":["right"]},{"label":["gnomADg_OTH_AF"],"name":[82],"type":["dbl"],"align":["right"]},{"label":["gnomADg_SAS_AF"],"name":[83],"type":["dbl"],"align":["right"]}],"data":[{"1":"ATOH8","2":"chr2_85754660_-/CGCACCGCCCGCGCGCCC","3":"chr2:85754659-85754660","4":"NM_032827.7","5":"NM_032827.7:c.474_491dup","6":"NP_116216.2:p.Pro160_Pro165dup","7":"inframe_insertion","8":"HET","9":"1/3","10":"NA","11":"0.0001603","12":"0.0018700","13":"0.021730","14":"gnomADg_ASJ","15":"NA","16":"NA","17":"NA","18":"NA","19":"NA","20":"NA","21":"NA","22":"NA","23":"21.300","24":"NA","25":"NA","26":"NA","27":"NA","28":"NA","29":"NA","30":"NA","31":"NA","32":"NA","33":"rs759894659","34":"NA","35":"0/0","36":"0|1","37":"1|0","38":"CGCACCGCCCGCGCGCCC","39":"1","40":"35","41":"35","42":"46","43":"0","44":"34","45":"44","46":"87","47":"99","48":"99","49":"NA","50":"41404","51":"41404","52":"YES","53":"ENST00000306279.4","54":"NA","55":"MODERATE","56":"84913","57":"HGNC:24126","58":"C/CAPPARP","59":"tgc/tgCGCACCGCCCGCGCGCCCc","60":"NA","61":"NA","62":"NA","63":"NA","64":"NA","65":"NA","66":"0.0000000","67":"0.0000000","68":"0.001314","69":"0.000e+00","70":"0.0006551","71":"0.0001230","72":"0.000000","73":"0.0000000","74":"0.0004176","75":"0.000000","76":"0.001451","77":"0.021730","78":"0.0002027","79":"0.002016","80":"0.000000","81":"0.0019950","82":"0.003880","83":"0.0004264"},{"1":"ATOH8","2":"chr2_85754718_T/G","3":"chr2:85754718","4":"NM_032827.7","5":"NM_032827.7:c.529T>G","6":"NP_116216.2:p.Ser177Ala","7":"missense_variant","8":"HET","9":"1/3","10":"NA","11":"0.0093180","12":"0.0084310","13":"0.020740","14":"gnomADg_ASJ","15":"NA","16":"NA","17":"NA","18":"NA","19":"NA","20":"0.190","21":"tolerated_low_confidence(0.27)","22":"benign(0.007)","23":"3.078","24":"0.7492155","25":"ATOH8|0.00|0.00|0.00|0.00|-11|-29|-1|5","26":"NA","27":"NA","28":"N","29":"1","30":"NA","31":"NA","32":"ATOH8:S177A","33":"rs188300460,COSV100092949","34":"NA","35":"0|1","36":"0/0","37":"0|1","38":"G","39":"1","40":"90","41":"43","42":"90","43":"110","44":"0","45":"99","46":"99","47":"80","48":"99","49":"41404","50":"NA","51":"41404","52":"YES","53":"ENST00000306279.4","54":"NA","55":"MODERATE","56":"84913","57":"HGNC:24126","58":"S/A","59":"Tcc/Gcc","60":"0.0048","61":"0.0000","62":"0.0072","63":"0","64":"0.0129","65":"0.0061","66":"0.0015150","67":"0.0052200","68":"0.019130","69":"0.000e+00","70":"0.0074570","71":"0.0134000","72":"0.010460","73":"0.0066240","74":"0.0019880","75":"0.016450","76":"0.008075","77":"0.020740","78":"0.0000000","79":"0.008898","80":"0.012820","81":"0.0123200","82":"0.009615","83":"0.0067090"},{"1":"GPR35","2":"chr2_240630025_G/A","3":"chr2:240630025","4":"NM_005301.5","5":"NM_005301.5:c.73G>A","6":"NP_005292.2:p.Ala25Thr","7":"missense_variant","8":"HET","9":"2/2","10":"NA","11":"0.0097040","12":"0.0093890","13":"0.019900","14":"EUR","15":"777143","16":"Benign","17":"NA","18":"Renal insufficiency (occasional)","19":"Autoimmunity","20":"0.024","21":"tolerated(0.54)","22":"benign(0)","23":"0.013","24":"0.2661452","25":"GPR35|0.00|0.00|0.00|0.00|28|44|-1|-40","26":"NA","27":"NA","28":"N,N,N,N,N","29":"1,1,1,1,1","30":"NA","31":"NA","32":"GPR35:A25T,GPR35:A56T","33":"rs35146537","34":"23549274","35":"1|0","36":"0/0","37":"0|1","38":"A","39":"1","40":"108","41":"67","42":"143","43":"119","44":"0","45":"126","46":"99","47":"98","48":"99","49":"41404","50":"NA","51":"41404","52":"YES","53":"ENST00000407714.2","54":"NA","55":"MODERATE","56":"2859","57":"HGNC:4492","58":"A/T","59":"Gcc/Acc","60":"0.0048","61":"0.0008","62":"0.0014","63":"0","64":"0.0199","65":"0.0020","66":"0.0020980","67":"0.0033280","68":"0.004583","69":"5.439e-05","70":"0.0164900","71":"0.0145400","72":"0.011600","73":"0.0053900","74":"0.0025340","75":"0.001096","76":"0.004578","77":"0.002017","78":"0.0000000","79":"0.019200","80":"0.000000","81":"0.0148300","82":"0.007184","83":"0.0037250"},{"1":"GPR35","2":"chr2_240630178_G/A","3":"chr2:240630178","4":"NM_005301.5","5":"NM_005301.5:c.226G>A","6":"NP_005292.2:p.Val76Met","7":"missense_variant","8":"HET","9":"2/2","10":"NA","11":"0.0056370","12":"0.0175700","13":"0.081700","14":"AFR","15":"791899","16":"Benign","17":"NA","18":"Renal insufficiency (occasional)","19":"Autoimmunity","20":"0.140","21":"tolerated(0.96)","22":"benign(0.048)","23":"0.089","24":"0.2208744","25":"GPR35|0.00|0.00|0.00|0.00|28|45|-17|-1","26":"NA","27":"NA","28":"N,N,N,N,N","29":"1,1,1,1,1","30":"NA","31":"NA","32":"GPR35:V107M,GPR35:V76M","33":"rs13387859","34":"33073184","35":"0/0","36":"1|0","37":"1|0","38":"A","39":"1","40":"145","41":"109","42":"101","43":"0","44":"100","45":"135","46":"96","47":"99","48":"99","49":"NA","50":"41404","51":"41404","52":"YES","53":"ENST00000407714.2","54":"NA","55":"MODERATE","56":"2859","57":"HGNC:4492","58":"V/M","59":"Gtg/Atg","60":"0.0230","61":"0.0817","62":"0.0101","63":"0","64":"0.0000","65":"0.0000","66":"0.0638900","67":"0.0030770","68":"0.017210","69":"0.000e+00","70":"0.0000000","71":"0.0002431","72":"0.002834","73":"0.0000000","74":"0.0598600","75":"0.000000","76":"0.005496","77":"0.016140","78":"0.0000000","79":"0.000000","80":"0.000000","81":"0.0003529","82":"0.013860","83":"0.0000000"},{"1":"FLNB","2":"chr3_58131983_T/C","3":"chr3:58131983","4":"NM_001164317.2","5":"NM_001164317.2:c.4447T>C","6":"NP_001157789.1:p.Phe1483Leu","7":"missense_variant","8":"HET","9":"26/47","10":"NA","11":"0.0006696","12":"0.0007027","13":"0.006329","14":"gnomADg_MID","15":"809499","16":"Uncertain_significance","17":"Larsen syndrome, 150250 (3), Autosomal dominant; Atelosteogenesis, type I, 108720 (3), Autosomal dominant; Atelosteogenesis, type III, 108721 (3), Autosomal dominant; Spondylocarpotarsal synostosis syndrome, 272460 (3), Autosomal recessive; Boomerang dysplasia, 112310 (3), Autosomal dominant","18":"Multiple renal cysts (occasional), Renal cyst","19":"Severe short-limb dwarfism, Aplasia/Hypoplasia of the fibula, Micromelia, Decreased response to growth hormone stimulation test, Abnormality of tibia morphology, Narrow chest, Poorly ossified vertebrae, Flat face, Joint hyperflexibility, Hypertelorism, Broad thumb, Brachydactyly, Prominent forehead, Malar flattening, Depressed nasal bridge, Short nail, Large joint dislocations, Broad distal phalanx of finger, Short distal phalanx of finger","20":"0.173","21":"deleterious_low_confidence(0.03)","22":"NA","23":"20.300","24":"0.5470113","25":"FLNB|0.00|0.00|0.03|0.00|-36|-25|36|-5","26":"NA","27":"NA","28":"D,D,D,D,D,D,D,N","29":"1,1,1,1,1,1,1,0.999908","30":"atelosteogenesis type I, spondylocarpotarsal synostosis syndrome, Larsen syndrome, atelosteogenesis type III","31":"Autosomal dominant, Autosomal recessive","32":"NA","33":"rs142631042","34":"NA","35":"0/0","36":"0|1","37":"1|0","38":"C","39":"1","40":"47","41":"199","42":"265","43":"0","44":"198","45":"258","46":"99","47":"99","48":"99","49":"NA","50":"197630","51":"197630","52":"NA","53":"NA","54":"NA","55":"MODERATE","56":"2317","57":"HGNC:3755","58":"F/L","59":"Ttt/Ctt","60":"0.0010","61":"0.0000","62":"0.0043","63":"0","64":"0.0020","65":"0.0000","66":"0.0001317","67":"0.0005292","68":"0.002615","69":"0.000e+00","70":"0.0000000","71":"0.0007712","72":"0.001420","73":"0.0003521","74":"0.0001206","75":"0.001096","76":"0.001963","77":"0.003746","78":"0.0000000","79":"0.000000","80":"0.006329","81":"0.0006908","82":"0.003820","83":"0.0002068"},{"1":"FLNB","2":"chr3_58159621_T/C","3":"chr3:58159621","4":"NM_001164317.2","5":"NM_001164317.2:c.7049T>C","6":"NP_001157789.1:p.Ile2350Thr","7":"missense_variant","8":"HET","9":"43/47","10":"NA","11":"0.0074340","12":"0.0072130","13":"0.044770","14":"gnomADe_ASJ","15":"258120","16":"Benign/Likely_benign","17":"Larsen syndrome, 150250 (3), Autosomal dominant; Atelosteogenesis, type I, 108720 (3), Autosomal dominant; Atelosteogenesis, type III, 108721 (3), Autosomal dominant; Spondylocarpotarsal synostosis syndrome, 272460 (3), Autosomal recessive; Boomerang dysplasia, 112310 (3), Autosomal dominant","18":"Multiple renal cysts (occasional), Renal cyst","19":"Severe short-limb dwarfism, Aplasia/Hypoplasia of the fibula, Micromelia, Decreased response to growth hormone stimulation test, Abnormality of tibia morphology, Narrow chest, Poorly ossified vertebrae, Flat face, Joint hyperflexibility, Hypertelorism, Broad thumb, Brachydactyly, Prominent forehead, Malar flattening, Depressed nasal bridge, Short nail, Large joint dislocations, Broad distal phalanx of finger, Short distal phalanx of finger","20":"0.747","21":"deleterious_low_confidence(0)","22":"NA","23":"28.000","24":"0.8164495","25":"FLNB|0.01|0.00|0.00|0.00|-23|-24|-7|-24","26":"NA","27":"NA","28":"D,D,D,D,D,D,D,D","29":"0.99999,0.99999,0.99999,0.999998,0.999998,0.999998,0.99999,1","30":"atelosteogenesis type I, spondylocarpotarsal synostosis syndrome, Larsen syndrome, atelosteogenesis type III","31":"Autosomal dominant, Autosomal recessive","32":"FLNB:I2295T,FLNB:I2308T,FLNB:I2319T,FLNB:I2350T","33":"rs116826041,COSV55888965","34":"25741868","35":"0|1","36":"0/0","37":"0|1","38":"C","39":"1","40":"83","41":"39","42":"76","43":"54","44":"0","45":"62","46":"99","47":"79","48":"99","49":"197630","50":"NA","51":"197630","52":"NA","53":"NA","54":"NA","55":"MODERATE","56":"2317","57":"HGNC:3755","58":"I/T","59":"aTt/aCt","60":"0.0024","61":"0.0000","62":"0.0058","63":"0","64":"0.0080","65":"0.0000","66":"0.0018510","67":"0.0078080","68":"0.044770","69":"0.000e+00","70":"0.0015750","71":"0.0086840","72":"0.013220","73":"0.0004899","74":"0.0022220","75":"0.009868","76":"0.009687","77":"0.043520","78":"0.0000000","79":"0.002828","80":"0.003165","81":"0.0094390","82":"0.011020","83":"0.0002070"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="x-chromosome" class="section level2 hasAnchor" number="5.5">
<h2><span class="header-section-number">5.5</span> X Chromosome<a href="variant-annotation.html#x-chromosome" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We remove X chromosome calls that are âheterozygousâ for male individuals, proband and father.</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="variant-annotation.html#cb37-1" aria-hidden="true" tabindex="-1"></a>wh_geno_X_filt <span class="ot">&lt;-</span> wh_geno_X <span class="sc">%&gt;%</span> </span>
<span id="cb37-2"><a href="variant-annotation.html#cb37-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(( <span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)<span class="sc">==</span><span class="fu">substr</span>(proband,<span class="dv">3</span>,<span class="dv">3</span>) <span class="sc">&amp;</span> <span class="fu">substr</span>(father,<span class="dv">1</span>,<span class="dv">1</span>)<span class="sc">==</span><span class="fu">substr</span>(father,<span class="dv">3</span>,<span class="dv">3</span>)))</span>
<span id="cb37-3"><a href="variant-annotation.html#cb37-3" aria-hidden="true" tabindex="-1"></a>prob_X <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(refseq_X, wh_geno_X) <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(Gene) <span class="sc">%&gt;%</span> </span>
<span id="cb37-4"><a href="variant-annotation.html#cb37-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(CANONICAL), <span class="fu">as.character</span>(Feature),<span class="at">.by_group =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb37-5"><a href="variant-annotation.html#cb37-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Feature <span class="sc">==</span> Feature[<span class="dv">1</span>])</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["SYMBOL"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Uploaded_variation"],"name":[2],"type":["fct"],"align":["left"]},{"label":["Location"],"name":[3],"type":["chr"],"align":["left"]},{"label":["Feature"],"name":[4],"type":["fct"],"align":["left"]},{"label":["HGVSc"],"name":[5],"type":["chr"],"align":["left"]},{"label":["HGVSp"],"name":[6],"type":["chr"],"align":["left"]},{"label":["Consequence"],"name":[7],"type":["chr"],"align":["left"]},{"label":["ZYG"],"name":[8],"type":["chr"],"align":["left"]},{"label":["EXON"],"name":[9],"type":["chr"],"align":["left"]},{"label":["INTRON"],"name":[10],"type":["chr"],"align":["left"]},{"label":["gnomADe_AF"],"name":[11],"type":["dbl"],"align":["right"]},{"label":["gnomADg_AF"],"name":[12],"type":["dbl"],"align":["right"]},{"label":["MAX_AF"],"name":[13],"type":["dbl"],"align":["right"]},{"label":["MAX_AF_POPS"],"name":[14],"type":["chr"],"align":["left"]},{"label":["ClinVar"],"name":[15],"type":["dbl"],"align":["right"]},{"label":["ClinVar_CLNSIG"],"name":[16],"type":["chr"],"align":["left"]},{"label":["OMIM_Phenotypes"],"name":[17],"type":["chr"],"align":["left"]},{"label":["renal_hpo_term"],"name":[18],"type":["chr"],"align":["left"]},{"label":["extrarenal_hpo_term"],"name":[19],"type":["chr"],"align":["left"]},{"label":["REVEL"],"name":[20],"type":["dbl"],"align":["right"]},{"label":["SIFT"],"name":[21],"type":["chr"],"align":["left"]},{"label":["PolyPhen"],"name":[22],"type":["chr"],"align":["left"]},{"label":["CADD_PHRED"],"name":[23],"type":["dbl"],"align":["right"]},{"label":["PrimateAI"],"name":[24],"type":["dbl"],"align":["right"]},{"label":["SpliceAI_pred"],"name":[25],"type":["chr"],"align":["left"]},{"label":["EVE_CLASS"],"name":[26],"type":["chr"],"align":["left"]},{"label":["EVE_SCORE"],"name":[27],"type":["dbl"],"align":["right"]},{"label":["MutationTaster_pred"],"name":[28],"type":["chr"],"align":["left"]},{"label":["MutationTaster_score"],"name":[29],"type":["chr"],"align":["left"]},{"label":["orphanet_disease"],"name":[30],"type":["chr"],"align":["left"]},{"label":["orphanet_moi"],"name":[31],"type":["chr"],"align":["left"]},{"label":["Mastermind_MMID3"],"name":[32],"type":["chr"],"align":["left"]},{"label":["Existing_variation"],"name":[33],"type":["chr"],"align":["left"]},{"label":["PUBMED"],"name":[34],"type":["chr"],"align":["left"]},{"label":["mother"],"name":[35],"type":["chr"],"align":["left"]},{"label":["father"],"name":[36],"type":["chr"],"align":["left"]},{"label":["proband"],"name":[37],"type":["chr"],"align":["left"]},{"label":["Allele"],"name":[38],"type":["chr"],"align":["left"]},{"label":["counter"],"name":[39],"type":["int"],"align":["right"]},{"label":["AD_ref_mother"],"name":[40],"type":["int"],"align":["right"]},{"label":["AD_ref_father"],"name":[41],"type":["int"],"align":["right"]},{"label":["AD_ref_proband"],"name":[42],"type":["int"],"align":["right"]},{"label":["AD_alt_mother"],"name":[43],"type":["int"],"align":["right"]},{"label":["AD_alt_father"],"name":[44],"type":["int"],"align":["right"]},{"label":["AD_alt_proband"],"name":[45],"type":["int"],"align":["right"]},{"label":["GQ_mother"],"name":[46],"type":["int"],"align":["right"]},{"label":["GQ_father"],"name":[47],"type":["int"],"align":["right"]},{"label":["GQ_proband"],"name":[48],"type":["int"],"align":["right"]},{"label":["PS_mother"],"name":[49],"type":["int"],"align":["right"]},{"label":["PS_father"],"name":[50],"type":["int"],"align":["right"]},{"label":["PS_proband"],"name":[51],"type":["int"],"align":["right"]},{"label":["CANONICAL"],"name":[52],"type":["chr"],"align":["left"]},{"label":["MANE_SELECT"],"name":[53],"type":["chr"],"align":["left"]},{"label":["MANE_PLUS_CLINICAL"],"name":[54],"type":["chr"],"align":["left"]},{"label":["IMPACT"],"name":[55],"type":["chr"],"align":["left"]},{"label":["Gene"],"name":[56],"type":["fct"],"align":["left"]},{"label":["HGNC_ID"],"name":[57],"type":["chr"],"align":["left"]},{"label":["Amino_acids"],"name":[58],"type":["chr"],"align":["left"]},{"label":["Codons"],"name":[59],"type":["chr"],"align":["left"]},{"label":["AF"],"name":[60],"type":["dbl"],"align":["right"]},{"label":["AFR_AF"],"name":[61],"type":["dbl"],"align":["right"]},{"label":["AMR_AF"],"name":[62],"type":["dbl"],"align":["right"]},{"label":["EAS_AF"],"name":[63],"type":["dbl"],"align":["right"]},{"label":["EUR_AF"],"name":[64],"type":["dbl"],"align":["right"]},{"label":["SAS_AF"],"name":[65],"type":["dbl"],"align":["right"]},{"label":["gnomADe_AFR_AF"],"name":[66],"type":["dbl"],"align":["right"]},{"label":["gnomADe_AMR_AF"],"name":[67],"type":["dbl"],"align":["right"]},{"label":["gnomADe_ASJ_AF"],"name":[68],"type":["dbl"],"align":["right"]},{"label":["gnomADe_EAS_AF"],"name":[69],"type":["dbl"],"align":["right"]},{"label":["gnomADe_FIN_AF"],"name":[70],"type":["dbl"],"align":["right"]},{"label":["gnomADe_NFE_AF"],"name":[71],"type":["dbl"],"align":["right"]},{"label":["gnomADe_OTH_AF"],"name":[72],"type":["dbl"],"align":["right"]},{"label":["gnomADe_SAS_AF"],"name":[73],"type":["dbl"],"align":["right"]},{"label":["gnomADg_AFR_AF"],"name":[74],"type":["dbl"],"align":["right"]},{"label":["gnomADg_AMI_AF"],"name":[75],"type":["dbl"],"align":["right"]},{"label":["gnomADg_AMR_AF"],"name":[76],"type":["dbl"],"align":["right"]},{"label":["gnomADg_ASJ_AF"],"name":[77],"type":["dbl"],"align":["right"]},{"label":["gnomADg_EAS_AF"],"name":[78],"type":["dbl"],"align":["right"]},{"label":["gnomADg_FIN_AF"],"name":[79],"type":["dbl"],"align":["right"]},{"label":["gnomADg_MID_AF"],"name":[80],"type":["dbl"],"align":["right"]},{"label":["gnomADg_NFE_AF"],"name":[81],"type":["dbl"],"align":["right"]},{"label":["gnomADg_OTH_AF"],"name":[82],"type":["dbl"],"align":["right"]},{"label":["gnomADg_SAS_AF"],"name":[83],"type":["dbl"],"align":["right"]}],"data":[{"1":"ASMTL","2":"chrX_1418092_T/G","3":"chrX:1418092","4":"NM_004192.4","5":"NM_004192.4:c.1403A>C","6":"NP_004183.2:p.Glu468Ala","7":"missense_variant","8":"HET","9":"11/13","10":"NA","11":"0.0018330","12":"1.236e-03","13":"0.020150","14":"gnomADe_ASJ","15":"NA","16":"NA","17":"NA","18":"NA","19":"NA","20":"0.082","21":"tolerated(0.1)","22":"benign(0.043)","23":"7.627","24":"0.5237793","25":"ASMTL|0.00|0.00|0.00|0.00|-29|24|-8|-28","26":"NA","27":"NA","28":"N,N,N,N","29":"0.981743,0.981743,0.981743,0.981743","30":"NA","31":"NA","32":"NA","33":"rs200070508","34":"NA","35":"0|1","36":"0/0","37":"0|1","38":"G","39":"1","40":"56","41":"34","42":"58","43":"38","44":"0","45":"43","46":"99","47":"89","48":"99","49":"284193","50":"NA","51":"284193","52":"YES","53":"ENST00000381317.9","54":"NA","55":"MODERATE","56":"8623","57":"HGNC:751","58":"E/A","59":"gAg/gCg","60":"NA","61":"NA","62":"NA","63":"NA","64":"NA","65":"NA","66":"0.000194","67":"0.0011760","68":"0.020150","69":"0.0000000","70":"1.165e-03","71":"1.254e-03","72":"0.003864","73":"7.277e-04","74":"2.897e-04","75":"0.000000","76":"0.0007860","77":"0.019300","78":"0.000000","79":"0.001320","80":"0.006329","81":"1.029e-03","82":"0.0033490","83":"0.0008282"},{"1":"ASMTL","2":"chrX_1418092_T/G","3":"chrX:1418092","4":"NM_004192.4","5":"NM_004192.4:c.1403A>C","6":"NP_004183.2:p.Glu468Ala","7":"missense_variant","8":"HET","9":"11/13","10":"NA","11":"0.0018330","12":"1.236e-03","13":"0.020150","14":"gnomADe_ASJ","15":"NA","16":"NA","17":"NA","18":"NA","19":"NA","20":"0.082","21":"tolerated(0.1)","22":"benign(0.043)","23":"7.627","24":"0.5237793","25":"ASMTL|0.00|0.00|0.00|0.00|-29|24|-8|-28","26":"NA","27":"NA","28":"N,N,N,N","29":"0.981743,0.981743,0.981743,0.981743","30":"NA","31":"NA","32":"NA","33":"rs200070508","34":"NA","35":"0|1","36":"0/0","37":"0|1","38":"G","39":"1","40":"56","41":"34","42":"58","43":"38","44":"0","45":"43","46":"99","47":"89","48":"99","49":"284193","50":"NA","51":"284193","52":"YES","53":"ENST00000381317.9","54":"NA","55":"MODERATE","56":"8623","57":"HGNC:751","58":"E/A","59":"gAg/gCg","60":"NA","61":"NA","62":"NA","63":"NA","64":"NA","65":"NA","66":"0.000194","67":"0.0011760","68":"0.020150","69":"0.0000000","70":"1.165e-03","71":"1.254e-03","72":"0.003864","73":"7.277e-04","74":"2.897e-04","75":"0.000000","76":"0.0007860","77":"0.019300","78":"0.000000","79":"0.001320","80":"0.006329","81":"1.029e-03","82":"0.0033490","83":"0.0008282"},{"1":"WWC3","2":"chrX_10138908_G/A","3":"chrX:10138908","4":"NM_015691.5","5":"NM_015691.5:c.3428G>A","6":"NP_056506.3:p.Arg1143His","7":"missense_variant","8":"HOM","9":"22/24","10":"NA","11":"0.0010170","12":"6.770e-04","13":"0.008296","14":"gnomADg_ASJ","15":"NA","16":"NA","17":"NA","18":"NA","19":"NA","20":"0.250","21":"deleterious(0.03)","22":"probably_damaging(0.959)","23":"23.800","24":"0.5156125","25":"WWC3|0.00|0.00|0.00|0.00|2|28|28|-1","26":"NA","27":"NA","28":"D,D","29":"0.999928,0.999928","30":"NA","31":"NA","32":"NA","33":"rs138248234","34":"NA","35":"0/1","36":"0/0","37":"1/1","38":"A","39":"1","40":"69","41":"37","42":"0","43":"72","44":"0","45":"69","46":"99","47":"29","48":"99","49":"NA","50":"NA","51":"NA","52":"YES","53":"NA","54":"NA","55":"MODERATE","56":"55841","57":"HGNC:29237","58":"R/H","59":"cGt/cAt","60":"0.0003","61":"0","62":"0.0000","63":"0","64":"0.0013","65":"0","66":"0.000338","67":"0.0009246","68":"0.006501","69":"0.0000000","70":"0.000e+00","71":"1.176e-03","72":"0.001875","73":"5.392e-05","74":"1.295e-04","75":"0.000000","76":"0.0002803","77":"0.008296","78":"0.000000","79":"0.000000","80":"0.004202","81":"8.465e-04","82":"0.0006623","83":"0.0000000"},{"1":"SCML2","2":"chrX_18324912_G/A","3":"chrX:18324912","4":"NM_006089.3","5":"NM_006089.3:c.157C>T","6":"NP_006080.1:p.Arg53Cys","7":"missense_variant","8":"HOM","9":"4/15","10":"NA","11":"0.0001453","12":"8.082e-05","13":"0.002460","14":"gnomADe_ASJ","15":"NA","16":"NA","17":"NA","18":"NA","19":"NA","20":"0.329","21":"deleterious(0)","22":"probably_damaging(0.999)","23":"24.500","24":"0.4405150","25":"SCML2|0.00|0.00|0.00|0.04|48|-5|0|-5","26":"NA","27":"NA","28":"D","29":"0.999992","30":"NA","31":"NA","32":"NA","33":"rs757183756,COSV99319403","34":"NA","35":"0/1","36":"0/0","37":"1/1","38":"A","39":"1","40":"185","41":"35","42":"0","43":"217","44":"0","45":"219","46":"99","47":"39","48":"99","49":"NA","50":"NA","51":"NA","52":"YES","53":"ENST00000251900.9","54":"NA","55":"MODERATE","56":"10389","57":"HGNC:10581","58":"R/C","59":"Cgt/Tgt","60":"NA","61":"NA","62":"NA","63":"NA","64":"NA","65":"NA","66":"0.000000","67":"0.0000000","68":"0.002460","69":"0.0000000","70":"0.000e+00","71":"6.202e-05","72":"0.000000","73":"1.683e-04","74":"3.262e-05","75":"0.000000","76":"0.0000000","77":"0.002269","78":"0.000000","79":"0.000000","80":"0.000000","81":"1.882e-05","82":"0.0006684","83":"0.0000000"},{"1":"SH3KBP1","2":"chrX_19545924_G/C","3":"chrX:19545924","4":"NM_031892.3","5":"NM_031892.3:c.1621C>G","6":"NP_114098.1:p.Gln541Glu","7":"missense_variant,splice_region_variant","8":"HOM","9":"15/18","10":"NA","11":"0.0034200","12":"3.433e-03","13":"0.020310","14":"gnomADe_ASJ","15":"782176","16":"Benign","17":"?Immunodeficiency 61, 300310 (3), X-linked recessive","18":"NA","19":"NA","20":"0.142","21":"deleterious(0.04)","22":"benign(0.303)","23":"21.100","24":"0.6069393","25":"SH3KBP1|0.00|0.00|0.00|0.01|46|-2|7|-2","26":"NA","27":"NA","28":"D,D,D,D","29":"0.999255,0.999901,0.999255,0.999551","30":"NA","31":"NA","32":"SH3KBP1:Q303E,SH3KBP1:Q303srd,SH3KBP1:Q504E,SH3KBP1:Q504srd,SH3KBP1:Q507E,SH3KBP1:Q507srd,SH3KBP1:Q541E,SH3KBP1:Q541srd,SH3KBP1:Q566E,SH3KBP1:Q566srd,SH3KBP1:S260int,SH3KBP1:S260inta,SH3KBP1:S464int,SH3KBP1:S464inta,SH3KBP1:S483int,SH3KBP1:S483inta,SH3KBP1:S498int,SH3KBP1:S498inta,SH3KBP1:S523int,SH3KBP1:S523inta","33":"rs61761898,COSV65648062","34":"NA","35":"0/1","36":"0/0","37":"1/1","38":"C","39":"1","40":"102","41":"34","42":"0","43":"86","44":"0","45":"96","46":"99","47":"21","48":"99","49":"NA","50":"NA","51":"NA","52":"YES","53":"ENST00000397821.8","54":"NA","55":"MODERATE","56":"30011","57":"HGNC:13867","58":"Q/E","59":"Caa/Gaa","60":"0.0019","61":"0","62":"0.0057","63":"0","64":"0.0052","65":"0","66":"0.000380","67":"0.0005107","68":"0.020310","69":"0.0000000","70":"2.812e-03","71":"4.666e-03","72":"0.004420","73":"4.726e-04","74":"5.870e-04","75":"0.005848","76":"0.0038070","77":"0.020050","78":"0.000000","79":"0.003807","80":"0.004202","81":"4.390e-03","82":"0.0059720","83":"0.0007579"},{"1":"SUPT20HL1","2":"chrX_24364309_G/-","3":"chrX:24364309","4":"NM_001136234.3","5":"NM_001136234.3:c.1549del","6":"NP_001129706.3:p.Ala517LeufsTer6","7":"frameshift_variant","8":"HOM","9":"1/1","10":"NA","11":"0.0024730","12":"2.014e-02","13":"0.057340","14":"gnomADg_AFR","15":"NA","16":"NA","17":"NA","18":"NA","19":"NA","20":"NA","21":"NA","22":"NA","23":"0.084","24":"NA","25":"NA","26":"NA","27":"NA","28":"NA","29":"NA","30":"NA","31":"NA","32":"NA","33":"rs1569203595","34":"NA","35":"0|1","36":"1|0","37":"1/1","38":"-","39":"1","40":"18","41":"2","42":"0","43":"7","44":"1","45":"5","46":"66","47":"38","48":"49","49":"284193","50":"284193","51":"NA","52":"YES","53":"ENST00000686983.1","54":"NA","55":"HIGH","56":"100130302","57":"HGNC:30773","58":"A/X","59":"Gct/ct","60":"NA","61":"NA","62":"NA","63":"NA","64":"NA","65":"NA","66":"0.027550","67":"0.0009727","68":"0.003671","69":"0.0003732","70":"8.784e-05","71":"7.959e-04","72":"0.003599","73":"4.737e-03","74":"5.734e-02","75":"0.007874","76":"0.0103800","77":"0.038760","78":"0.004156","79":"0.004331","80":"0.023260","81":"1.069e-02","82":"0.0202300","83":"0.0283300"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="disease-genes" class="section level2 hasAnchor" number="5.6">
<h2><span class="header-section-number">5.6</span> Disease Genes<a href="variant-annotation.html#disease-genes" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We select one entry for each variant in the refseq table, prioritizing canonical and curated transcripts.</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="variant-annotation.html#cb38-1" aria-hidden="true" tabindex="-1"></a>refseq2 <span class="ot">&lt;-</span> refseq01 <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(Uploaded_variation) <span class="sc">%&gt;%</span> </span>
<span id="cb38-2"><a href="variant-annotation.html#cb38-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(CANONICAL), <span class="fu">as.character</span>(Feature),<span class="at">.by_group =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb38-3"><a href="variant-annotation.html#cb38-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Feature <span class="sc">==</span> Feature[<span class="dv">1</span>])</span></code></pre></div>
<p>We select variants that fall in green genes from the UK PanelApp renal broad superpanel.</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="variant-annotation.html#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="co"># UK Panel App</span></span>
<span id="cb39-2"><a href="variant-annotation.html#cb39-2" aria-hidden="true" tabindex="-1"></a>panel <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">&quot;data/renal_superpanel_broad.tsv&quot;</span>)</span>
<span id="cb39-3"><a href="variant-annotation.html#cb39-3" aria-hidden="true" tabindex="-1"></a>panel_green <span class="ot">&lt;-</span> panel <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(GEL_Status<span class="sc">==</span><span class="dv">3</span>)</span>
<span id="cb39-4"><a href="variant-annotation.html#cb39-4" aria-hidden="true" tabindex="-1"></a>refseq_green <span class="ot">&lt;-</span> refseq2 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(HGNC_ID <span class="sc">%in%</span> panel_green<span class="sc">$</span>HGNC)</span>
<span id="cb39-5"><a href="variant-annotation.html#cb39-5" aria-hidden="true" tabindex="-1"></a><span class="co">#Australia Panel App</span></span>
<span id="cb39-6"><a href="variant-annotation.html#cb39-6" aria-hidden="true" tabindex="-1"></a>panel_au <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">&quot;data/Kidneyome_SuperPanel.tsv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb39-7"><a href="variant-annotation.html#cb39-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">duplicated</span>(HGNC)) <span class="sc">%&gt;%</span></span>
<span id="cb39-8"><a href="variant-annotation.html#cb39-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">Status=</span><span class="fu">case_when</span>(</span>
<span id="cb39-9"><a href="variant-annotation.html#cb39-9" aria-hidden="true" tabindex="-1"></a>    GEL_Status<span class="sc">==</span><span class="dv">4</span><span class="sc">|</span>GEL_Status<span class="sc">==</span><span class="dv">3</span> <span class="sc">~</span> <span class="st">&quot;Green&quot;</span>, </span>
<span id="cb39-10"><a href="variant-annotation.html#cb39-10" aria-hidden="true" tabindex="-1"></a>    GEL_Status<span class="sc">==</span><span class="dv">2</span> <span class="sc">~</span> <span class="st">&quot;Amber&quot;</span>, </span>
<span id="cb39-11"><a href="variant-annotation.html#cb39-11" aria-hidden="true" tabindex="-1"></a>    GEL_Status<span class="sc">==</span><span class="dv">1</span><span class="sc">|</span>GEL_Status<span class="sc">==</span><span class="dv">0</span> <span class="sc">~</span> <span class="st">&quot;Red&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb39-12"><a href="variant-annotation.html#cb39-12" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(HGNC,Status)</span>
<span id="cb39-13"><a href="variant-annotation.html#cb39-13" aria-hidden="true" tabindex="-1"></a>refseq_au <span class="ot">&lt;-</span> refseq2 <span class="sc">%&gt;%</span> <span class="fu">inner_join</span>(panel_au, <span class="fu">join_by</span>(HGNC_ID<span class="sc">==</span>HGNC))</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["SYMBOL"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Uploaded_variation"],"name":[2],"type":["fct"],"align":["left"]},{"label":["Location"],"name":[3],"type":["chr"],"align":["left"]},{"label":["Feature"],"name":[4],"type":["fct"],"align":["left"]},{"label":["HGVSc"],"name":[5],"type":["chr"],"align":["left"]},{"label":["HGVSp"],"name":[6],"type":["chr"],"align":["left"]},{"label":["Consequence"],"name":[7],"type":["chr"],"align":["left"]},{"label":["ZYG"],"name":[8],"type":["chr"],"align":["left"]},{"label":["EXON"],"name":[9],"type":["chr"],"align":["left"]},{"label":["INTRON"],"name":[10],"type":["chr"],"align":["left"]},{"label":["gnomADe_AF"],"name":[11],"type":["dbl"],"align":["right"]},{"label":["gnomADg_AF"],"name":[12],"type":["dbl"],"align":["right"]},{"label":["MAX_AF"],"name":[13],"type":["dbl"],"align":["right"]},{"label":["MAX_AF_POPS"],"name":[14],"type":["chr"],"align":["left"]},{"label":["ClinVar"],"name":[15],"type":["dbl"],"align":["right"]},{"label":["ClinVar_CLNSIG"],"name":[16],"type":["chr"],"align":["left"]},{"label":["OMIM_Phenotypes"],"name":[17],"type":["chr"],"align":["left"]},{"label":["renal_hpo_term"],"name":[18],"type":["chr"],"align":["left"]},{"label":["extrarenal_hpo_term"],"name":[19],"type":["chr"],"align":["left"]},{"label":["REVEL"],"name":[20],"type":["dbl"],"align":["right"]},{"label":["SIFT"],"name":[21],"type":["chr"],"align":["left"]},{"label":["PolyPhen"],"name":[22],"type":["chr"],"align":["left"]},{"label":["CADD_PHRED"],"name":[23],"type":["dbl"],"align":["right"]},{"label":["PrimateAI"],"name":[24],"type":["dbl"],"align":["right"]},{"label":["SpliceAI_pred"],"name":[25],"type":["chr"],"align":["left"]},{"label":["EVE_CLASS"],"name":[26],"type":["chr"],"align":["left"]},{"label":["EVE_SCORE"],"name":[27],"type":["dbl"],"align":["right"]},{"label":["MutationTaster_pred"],"name":[28],"type":["chr"],"align":["left"]},{"label":["MutationTaster_score"],"name":[29],"type":["chr"],"align":["left"]},{"label":["orphanet_disease"],"name":[30],"type":["chr"],"align":["left"]},{"label":["orphanet_moi"],"name":[31],"type":["chr"],"align":["left"]},{"label":["Mastermind_MMID3"],"name":[32],"type":["chr"],"align":["left"]},{"label":["Existing_variation"],"name":[33],"type":["chr"],"align":["left"]},{"label":["PUBMED"],"name":[34],"type":["chr"],"align":["left"]},{"label":["mother"],"name":[35],"type":["chr"],"align":["left"]},{"label":["father"],"name":[36],"type":["chr"],"align":["left"]},{"label":["proband"],"name":[37],"type":["chr"],"align":["left"]},{"label":["Allele"],"name":[38],"type":["chr"],"align":["left"]},{"label":["counter"],"name":[39],"type":["int"],"align":["right"]},{"label":["AD_ref_mother"],"name":[40],"type":["int"],"align":["right"]},{"label":["AD_ref_father"],"name":[41],"type":["int"],"align":["right"]},{"label":["AD_ref_proband"],"name":[42],"type":["int"],"align":["right"]},{"label":["AD_alt_mother"],"name":[43],"type":["int"],"align":["right"]},{"label":["AD_alt_father"],"name":[44],"type":["int"],"align":["right"]},{"label":["AD_alt_proband"],"name":[45],"type":["int"],"align":["right"]},{"label":["GQ_mother"],"name":[46],"type":["int"],"align":["right"]},{"label":["GQ_father"],"name":[47],"type":["int"],"align":["right"]},{"label":["GQ_proband"],"name":[48],"type":["int"],"align":["right"]},{"label":["PS_mother"],"name":[49],"type":["int"],"align":["right"]},{"label":["PS_father"],"name":[50],"type":["int"],"align":["right"]},{"label":["PS_proband"],"name":[51],"type":["int"],"align":["right"]},{"label":["CANONICAL"],"name":[52],"type":["chr"],"align":["left"]},{"label":["MANE_SELECT"],"name":[53],"type":["chr"],"align":["left"]},{"label":["MANE_PLUS_CLINICAL"],"name":[54],"type":["chr"],"align":["left"]},{"label":["IMPACT"],"name":[55],"type":["chr"],"align":["left"]},{"label":["Gene"],"name":[56],"type":["fct"],"align":["left"]},{"label":["HGNC_ID"],"name":[57],"type":["chr"],"align":["left"]},{"label":["Amino_acids"],"name":[58],"type":["chr"],"align":["left"]},{"label":["Codons"],"name":[59],"type":["chr"],"align":["left"]},{"label":["AF"],"name":[60],"type":["dbl"],"align":["right"]},{"label":["AFR_AF"],"name":[61],"type":["dbl"],"align":["right"]},{"label":["AMR_AF"],"name":[62],"type":["dbl"],"align":["right"]},{"label":["EAS_AF"],"name":[63],"type":["dbl"],"align":["right"]},{"label":["EUR_AF"],"name":[64],"type":["dbl"],"align":["right"]},{"label":["SAS_AF"],"name":[65],"type":["dbl"],"align":["right"]},{"label":["gnomADe_AFR_AF"],"name":[66],"type":["dbl"],"align":["right"]},{"label":["gnomADe_AMR_AF"],"name":[67],"type":["dbl"],"align":["right"]},{"label":["gnomADe_ASJ_AF"],"name":[68],"type":["dbl"],"align":["right"]},{"label":["gnomADe_EAS_AF"],"name":[69],"type":["dbl"],"align":["right"]},{"label":["gnomADe_FIN_AF"],"name":[70],"type":["dbl"],"align":["right"]},{"label":["gnomADe_NFE_AF"],"name":[71],"type":["dbl"],"align":["right"]},{"label":["gnomADe_OTH_AF"],"name":[72],"type":["dbl"],"align":["right"]},{"label":["gnomADe_SAS_AF"],"name":[73],"type":["dbl"],"align":["right"]},{"label":["gnomADg_AFR_AF"],"name":[74],"type":["dbl"],"align":["right"]},{"label":["gnomADg_AMI_AF"],"name":[75],"type":["dbl"],"align":["right"]},{"label":["gnomADg_AMR_AF"],"name":[76],"type":["dbl"],"align":["right"]},{"label":["gnomADg_ASJ_AF"],"name":[77],"type":["dbl"],"align":["right"]},{"label":["gnomADg_EAS_AF"],"name":[78],"type":["dbl"],"align":["right"]},{"label":["gnomADg_FIN_AF"],"name":[79],"type":["dbl"],"align":["right"]},{"label":["gnomADg_MID_AF"],"name":[80],"type":["dbl"],"align":["right"]},{"label":["gnomADg_NFE_AF"],"name":[81],"type":["dbl"],"align":["right"]},{"label":["gnomADg_OTH_AF"],"name":[82],"type":["dbl"],"align":["right"]},{"label":["gnomADg_SAS_AF"],"name":[83],"type":["dbl"],"align":["right"]},{"label":["Status"],"name":[84],"type":["chr"],"align":["left"]}],"data":[{"1":"NPHP4","2":"chr1_5864452_C/A","3":"chr1:5864452","4":"NM_015102.5","5":"NM_015102.5:c.3882G>T","6":"NP_055917.1:p.Arg1294Ser","7":"missense_variant","8":"HET","9":"28/30","10":"NA","11":"4.767e-05","12":"2.628e-05","13":"0.0011520","14":"gnomADg_ASJ","15":"969225","16":"Uncertain_significance","17":"Senior-Loken syndrome 4, 606996 (3), Autosomal recessive; Nephronophthisis 4, 606966 (3), Autosomal recessive","18":"Stage 5 chronic kidney disease (very frequent), Nephronophthisis (frequent), Tubulointerstitial fibrosis, Renal tubular atrophy, Polyuria, Stage 5 chronic kidney disease, Renal corticomedullary cysts, Nephronophthisis","19":"Short stature, Hypertension, Global developmental delay, Abnormality of retinal pigmentation, Retinal dystrophy","20":"0.664","21":"deleterious(0)","22":"possibly_damaging(0.817)","23":"24.200","24":"0.4835634","25":"NPHP4|0.04|0.00|0.04|0.00|-9|20|6|-39","26":"Pathogenic","27":"0.80084944","28":"D","29":"0.99534","30":"Senior-Loken syndrome, nephronophthisis 1","31":"Autosomal recessive","32":"NA","33":"rs367814493","34":"NA","35":"0/0","36":"1|0","37":"1|0","38":"A","39":"1","40":"33","41":"78","42":"109","43":"0","44":"69","45":"69","46":"99","47":"99","48":"99","49":"NA","50":"65797","51":"65797","52":"YES","53":"ENST00000378156.9","54":"NA","55":"MODERATE","56":"261734","57":"HGNC:19104","58":"R/S","59":"agG/agT","60":"NA","61":"NA","62":"NA","63":"NA","64":"NA","65":"NA","66":"0.000e+00","67":"6.087e-05","68":"0.0008323","69":"5.999e-05","70":"0.0000000","71":"0.0000000","72":"0.0000000","73":"0.000e+00","74":"0.000e+00","75":"0","76":"0.0000000","77":"0.001152","78":"0.0000000","79":"0.0000000","80":"0.000000","81":"0.0000000","82":"0.0000000","83":"0.0000000","84":"Green"},{"1":"AGXT","2":"chr2_240877573_G/A","3":"chr2:240877573","4":"NM_000030.3","5":"NM_000030.3:c.883G>A","6":"NP_000021.1:p.Ala295Thr","7":"missense_variant","8":"HET","9":"9/11","10":"NA","11":"6.672e-03","12":"2.286e-02","13":"0.0908000","14":"AFR","15":"204056","16":"Conflicting_interpretations_of_pathogenicity","17":"Hyperoxaluria, primary, type 1, 259900 (3), Autosomal recessive","18":"Nephrolithiasis (very frequent), Nephrocalcinosis (very frequent), Decreased glomerular filtration rate (frequent), Hematuria (frequent), Stage 5 chronic kidney disease (occasional), Nephrocalcinosis, Hematuria, Renal insufficiency, Calcium oxalate nephrolithiasis","19":"Anemia, Abnormality of circulating enzyme level, Metabolic acidosis, Hyperoxaluria, Calcinosis","20":"0.220","21":"tolerated(0.25)","22":"benign(0.001)","23":"0.744","24":"0.3572417","25":"AGXT|0.00|0.00|0.00|0.00|-22|-27|4|-36","26":"Benign","27":"0.11768815","28":"P","29":"1","30":"primary hyperoxaluria type 1","31":"Autosomal recessive","32":"AGXT:A295T","33":"rs13408961,CM1313658,COSV56754979","34":"25741868,29590070","35":"0/0","36":"1|0","37":"1|0","38":"A","39":"1","40":"46","41":"70","42":"85","43":"0","44":"89","45":"94","46":"77","47":"99","48":"99","49":"NA","50":"41404","51":"41404","52":"YES","53":"ENST00000307503.4","54":"NA","55":"MODERATE","56":"189","57":"HGNC:341","58":"A/T","59":"Gcg/Acg","60":"0.0246","61":"0.0908","62":"0.0014","63":"0.0010","64":"0.0010","65":"0.000","66":"8.379e-02","67":"3.402e-03","68":"0.0202800","69":"2.670e-04","70":"0.0000000","71":"0.0006341","72":"0.0043620","73":"8.783e-05","74":"7.748e-02","75":"0","76":"0.0071930","77":"0.020160","78":"0.0009623","79":"0.0000000","80":"0.006329","81":"0.0005733","82":"0.0191200","83":"0.0004137","84":"Green"},{"1":"SCARB2","2":"chr4_76179654_T/C","3":"chr4:76179654","4":"NM_005506.4","5":"NM_005506.4:c.475A>G","6":"NP_005497.1:p.Met159Val","7":"missense_variant","8":"HET","9":"4/12","10":"NA","11":"8.607e-03","12":"8.370e-03","13":"0.0253200","14":"gnomADg_MID","15":"206702","16":"Benign","17":"Epilepsy, progressive myoclonic 4, with or without renal failure, 254900 (3), Autosomal recessive","18":"Hematuria (occasional), Nephrotic syndrome, Focal segmental glomerulosclerosis, Nephropathy, Renal insufficiency, Glomerulopathy","19":"Splenomegaly, Osteopenia, Anorexia, Hypersplenism, Osteolysis, Thrombocytopenia, Hepatomegaly, Bone pain, Increased bone mineral density, Delayed puberty, Delayed skeletal maturation, Avascular necrosis, Limb ataxia, EEG with polyspike wave complexes, Morning myoclonic jerks","20":"0.126","21":"tolerated(0.17)","22":"benign(0.001)","23":"9.225","24":"0.2870885","25":"SCARB2|0.01|0.00|0.00|0.00|16|-43|41|-1","26":"Benign","27":"0.09772531","28":"N,N","29":"1,0.999936","30":"Unverricht-Lundborg syndrome, action myoclonus-renal failure syndrome","31":"Autosomal recessive","32":"SCARB2:M159V,SCARB2:R92int,SCARB2:R92inta","33":"rs143655258","34":"25741868,25088547","35":"1|0","36":"0/0","37":"0|1","38":"C","39":"1","40":"93","41":"37","42":"108","43":"115","44":"0","45":"101","46":"99","47":"89","48":"99","49":"60220","50":"NA","51":"60220","52":"YES","53":"ENST00000264896.8","54":"NA","55":"MODERATE","56":"950","57":"HGNC:1665","58":"M/V","59":"Atg/Gtg","60":"0.0038","61":"0.0008","62":"0.0014","63":"0.0000","64":"0.0159","65":"0.001","66":"2.153e-03","67":"7.777e-03","68":"0.0129000","69":"1.631e-04","70":"0.0099030","71":"0.0124000","72":"0.0127200","73":"8.166e-04","74":"2.099e-03","75":"0","76":"0.0090940","77":"0.013260","78":"0.0001925","79":"0.0093170","80":"0.025320","81":"0.0126200","82":"0.0138600","83":"0.0012430","84":"Green"},{"1":"FAT1","2":"chr4_186618051_C/T","3":"chr4:186618051","4":"NM_005245.4","5":"NM_005245.4:c.8535G>A","6":"NP_005236.2:p.Met2845Ile","7":"missense_variant","8":"HET","9":"10/27","10":"NA","11":"3.186e-03","12":"2.602e-03","13":"0.0328300","14":"gnomADg_ASJ","15":"781849","16":"Likely_benign","17":"NA","18":"NA","19":"NA","20":"0.043","21":"tolerated_low_confidence(0.36)","22":"benign(0.003)","23":"15.000","24":"0.3124041","25":"FAT1|0.00|0.00|0.00|0.00|-16|19|-6|6","26":"NA","27":"NA","28":"N","29":"0.990862","30":"NA","31":"NA","32":"FAT1:M2845I","33":"rs138948513,COSV101499785","34":"NA","35":"0/0","36":"0|1","37":"1|0","38":"T","39":"1","40":"117","41":"183","42":"224","43":"0","44":"180","45":"181","46":"99","47":"99","48":"99","49":"NA","50":"60220","51":"60220","52":"YES","53":"ENST00000441802.7","54":"NA","55":"MODERATE","56":"2195","57":"HGNC:3595","58":"M/I","59":"atG/atA","60":"0.0006","61":"0.0000","62":"0.0014","63":"0.0000","64":"0.0020","65":"0.000","66":"3.875e-04","67":"1.941e-03","68":"0.0313100","69":"0.000e+00","70":"0.0003249","71":"0.0029380","72":"0.0056220","73":"1.078e-03","74":"6.515e-04","75":"0","76":"0.0031410","77":"0.032830","78":"0.0000000","79":"0.0002824","80":"0.003165","81":"0.0028360","82":"0.0033430","83":"0.0006211","84":"Green"},{"1":"BBS9","2":"chr7_33340944_G/A","3":"chr7:33340944","4":"NM_198428.3","5":"NM_198428.3:c.1246G>A","6":"NP_940820.1:p.Val416Met","7":"missense_variant","8":"HET","9":"11/23","10":"NA","11":"6.070e-03","12":"3.894e-03","13":"0.0175500","14":"gnomADe_AMR","15":"96647","16":"Benign/Likely_benign","17":"Bardet-Biedl syndrome 9, 615986 (3), Autosomal recessive","18":"Multicystic kidney dysplasia (very frequent), Nephrotic syndrome (occasional), Renal insufficiency (very rare)","19":"Pigmentary retinopathy, Intellectual disability, Postaxial hand polydactyly, Abnormal electroretinogram, Obesity","20":"0.155","21":"tolerated(0.35)","22":"benign(0.007)","23":"0.375","24":"0.2559906","25":"BBS9|0.00|0.18|0.01|0.15|2|-47|2|29","26":"Benign","27":"0.29790176","28":"N,N,N,N,N","29":"1,1,1,1,1","30":"Bardet-Biedl syndrome","31":"Autosomal recessive","32":"BBS9:V294M,BBS9:V325M,BBS9:V371M,BBS9:V416M","33":"rs61764067,COSV54169251","34":"25741868,23757202","35":"0|1","36":"0/0","37":"0|1","38":"A","39":"1","40":"114","41":"33","42":"121","43":"104","44":"0","45":"97","46":"99","47":"80","48":"99","49":"228951","50":"NA","51":"228951","52":"YES","53":"ENST00000242067.11","54":"NA","55":"MODERATE","56":"27241","57":"HGNC:30000","58":"V/M","59":"Gtg/Atg","60":"0.0046","61":"0.0000","62":"0.0173","63":"0.0089","64":"0.0010","65":"0.001","66":"4.922e-04","67":"1.755e-02","68":"0.0109100","69":"1.316e-02","70":"0.0061930","71":"0.0029470","72":"0.0076650","73":"1.405e-03","74":"6.763e-04","75":"0","76":"0.0095680","77":"0.009798","78":"0.0154300","79":"0.0074610","80":"0.003165","81":"0.0031030","82":"0.0028790","83":"0.0014500","84":"Green"},{"1":"EGFR","2":"chr7_55173087_G/A","3":"chr7:55173087","4":"NM_005228.5","5":"NM_005228.5:c.2024G>A","6":"NP_005219.2:p.Arg675Gln","7":"missense_variant","8":"HET","9":"17/28","10":"NA","11":"2.132e-04","12":"2.431e-04","13":"0.0009569","14":"gnomADg_OTH","15":"134025","16":"Uncertain_significance","17":"?Inflammatory skin and bowel disease, neonatal, 2, 616069 (3), Autosomal recessive; Nonsmall cell lung cancer, response to tyrosine kinase inhibitor in, 211980 (3), Autosomal dominant, Somatic mutation; Adenocarcinoma of lung, response to tyrosine kinase inhibitor in, 211980 (3), Autosomal dominant, Somatic mutation; {Nonsmall cell lung cancer, susceptibility to}, 211980 (3), Autosomal dominant, Somatic mutation","18":"NA","19":"NA","20":"0.590","21":"deleterious_low_confidence(0)","22":"probably_damaging(0.978)","23":"26.600","24":"0.7896019","25":"EGFR|0.00|0.00|0.00|0.00|2|5|33|37","26":"Uncertain","27":"0.45869527","28":"D,D,D,D","29":"0.992197,0.992197,0.998267,1","30":"neonatal inflammatory skin and bowel disease","31":"Autosomal recessive","32":"EGFR:R408Q,EGFR:R622Q,EGFR:R630Q,EGFR:R675Q","33":"rs150423237,COSV51862598","34":"24728327,28166811,33326033","35":"0|1","36":"0/0","37":"0|1","38":"A","39":"1","40":"58","41":"37","42":"64","43":"53","44":"0","45":"45","46":"99","47":"93","48":"99","49":"228951","50":"NA","51":"228951","52":"YES","53":"ENST00000275493.7","54":"NA","55":"MODERATE","56":"1956","57":"HGNC:3236","58":"R/Q","59":"cGg/cAg","60":"NA","61":"NA","62":"NA","63":"NA","64":"NA","65":"NA","66":"6.155e-05","67":"3.759e-04","68":"0.0001986","69":"5.438e-05","70":"0.0000000","71":"0.0002499","72":"0.0004899","73":"1.633e-04","74":"7.238e-05","75":"0","76":"0.0003273","77":"0.000000","78":"0.0000000","79":"0.0000000","80":"0.000000","81":"0.0003968","82":"0.0009569","83":"0.0000000","84":"Red"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>We combine annotations into a single table.</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="variant-annotation.html#cb40-1" aria-hidden="true" tabindex="-1"></a>refseq3 <span class="ot">&lt;-</span> refseq01 <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(Gene) <span class="sc">%&gt;%</span> </span>
<span id="cb40-2"><a href="variant-annotation.html#cb40-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(CANONICAL), <span class="fu">as.character</span>(Feature),<span class="at">.by_group =</span> T) <span class="sc">%&gt;%</span> <span class="fu">ungroup</span>()</span>
<span id="cb40-3"><a href="variant-annotation.html#cb40-3" aria-hidden="true" tabindex="-1"></a>refseq4 <span class="ot">&lt;-</span> refseq3 <span class="sc">%&gt;%</span> </span>
<span id="cb40-4"><a href="variant-annotation.html#cb40-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(<span class="fu">add_column</span>(geno_novo_not_ref,<span class="at">de_novo=</span><span class="st">&quot;de_novo&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb40-5"><a href="variant-annotation.html#cb40-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(<span class="fu">add_column</span>(geno_novo_not_ref_high_qual,<span class="at">de_novo_gq=</span><span class="st">&quot;de_novo_gq&gt;=30&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb40-6"><a href="variant-annotation.html#cb40-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(<span class="fu">add_column</span>(geno_hom2, <span class="at">homozygote=</span><span class="st">&quot;homozygote&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb40-7"><a href="variant-annotation.html#cb40-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(<span class="fu">add_column</span>(prob_2plus_het, <span class="at">unphased_compound_heterozygote=</span><span class="st">&quot;unphased_compound_heterozygote&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb40-8"><a href="variant-annotation.html#cb40-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(<span class="fu">add_column</span>(prob_comp_het, <span class="at">phased_compound_heterozygote=</span><span class="st">&quot;phased_compound_heterozygote&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb40-9"><a href="variant-annotation.html#cb40-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">zygosity=</span><span class="fu">coalesce</span>(de_novo_gq,de_novo,homozygote,phased_compound_heterozygote,unphased_compound_heterozygote)) <span class="sc">%&gt;%</span></span>
<span id="cb40-10"><a href="variant-annotation.html#cb40-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(de_novo_gq,de_novo,homozygote,phased_compound_heterozygote,unphased_compound_heterozygote)) <span class="sc">%&gt;%</span></span>
<span id="cb40-11"><a href="variant-annotation.html#cb40-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">relocate</span>(zygosity,<span class="at">.after=</span>counter)</span>
<span id="cb40-12"><a href="variant-annotation.html#cb40-12" aria-hidden="true" tabindex="-1"></a>refseq_sum <span class="ot">&lt;-</span> refseq4 <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(Feature) <span class="sc">%&gt;%</span> </span>
<span id="cb40-13"><a href="variant-annotation.html#cb40-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">z=</span><span class="fu">paste</span>(<span class="fu">c</span>(Uploaded_variation,Location,Allele,Consequence,Amino_acids,Codons), <span class="at">collapse =</span> <span class="st">&quot; &quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb40-14"><a href="variant-annotation.html#cb40-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(z,<span class="at">.keep_all =</span> T)</span>
<span id="cb40-15"><a href="variant-annotation.html#cb40-15" aria-hidden="true" tabindex="-1"></a>refseq5 <span class="ot">&lt;-</span> refseq4 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(Feature <span class="sc">%in%</span> refseq_sum<span class="sc">$</span>Feature)</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="statistical-genetics.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="summary.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
