<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>5 Variant Annotation | Whole Exome Sequencing Platform</title>
  <meta name="description" content="This is a pipeline for whole exome sequencing analysis." />
  <meta name="generator" content="bookdown 0.35 and GitBook 2.6.7" />

  <meta property="og:title" content="5 Variant Annotation | Whole Exome Sequencing Platform" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a pipeline for whole exome sequencing analysis." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="5 Variant Annotation | Whole Exome Sequencing Platform" />
  
  <meta name="twitter:description" content="This is a pipeline for whole exome sequencing analysis." />
  

<meta name="author" content="Marc Henein and Zachary Sentell" />


<meta name="date" content="2023-08-18" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="statistical-genetics.html"/>
<link rel="next" href="summary.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Whole Exome Sequencing Platform</a></li>
<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Overview</a></li>
<li class="chapter" data-level="2" data-path="align.html"><a href="align.html"><i class="fa fa-check"></i><b>2</b> Alignment</a>
<ul>
<li class="chapter" data-level="2.1" data-path="align.html"><a href="align.html#pre-processing"><i class="fa fa-check"></i><b>2.1</b> Pre-processing</a></li>
<li class="chapter" data-level="2.2" data-path="align.html"><a href="align.html#mark-duplicates"><i class="fa fa-check"></i><b>2.2</b> Mark Duplicates</a></li>
<li class="chapter" data-level="2.3" data-path="align.html"><a href="align.html#base-quality-score-recalibration"><i class="fa fa-check"></i><b>2.3</b> Base Quality Score Recalibration</a></li>
<li class="chapter" data-level="2.4" data-path="align.html"><a href="align.html#quality-control"><i class="fa fa-check"></i><b>2.4</b> Quality Control</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="calling.html"><a href="calling.html"><i class="fa fa-check"></i><b>3</b> Variant Calling</a>
<ul>
<li class="chapter" data-level="3.1" data-path="calling.html"><a href="calling.html#snvindel-calling"><i class="fa fa-check"></i><b>3.1</b> SNV/Indel Calling</a></li>
<li class="chapter" data-level="3.2" data-path="calling.html"><a href="calling.html#variant-filtration"><i class="fa fa-check"></i><b>3.2</b> Variant Filtration</a></li>
<li class="chapter" data-level="3.3" data-path="calling.html"><a href="calling.html#genotype-posteriors"><i class="fa fa-check"></i><b>3.3</b> Genotype Posteriors</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="statistical-genetics.html"><a href="statistical-genetics.html"><i class="fa fa-check"></i><b>4</b> Statistical Genetics</a>
<ul>
<li class="chapter" data-level="4.1" data-path="statistical-genetics.html"><a href="statistical-genetics.html#phasing"><i class="fa fa-check"></i><b>4.1</b> Phasing</a></li>
<li class="chapter" data-level="4.2" data-path="statistical-genetics.html"><a href="statistical-genetics.html#kinship"><i class="fa fa-check"></i><b>4.2</b> Kinship</a></li>
<li class="chapter" data-level="4.3" data-path="statistical-genetics.html"><a href="statistical-genetics.html#runs-of-homozygosity"><i class="fa fa-check"></i><b>4.3</b> Runs of Homozygosity</a></li>
<li class="chapter" data-level="4.4" data-path="statistical-genetics.html"><a href="statistical-genetics.html#linkage-analysis"><i class="fa fa-check"></i><b>4.4</b> Linkage Analysis</a></li>
<li class="chapter" data-level="4.5" data-path="statistical-genetics.html"><a href="statistical-genetics.html#population-substructure"><i class="fa fa-check"></i><b>4.5</b> Population Substructure</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="variant-annotation.html"><a href="variant-annotation.html"><i class="fa fa-check"></i><b>5</b> Variant Annotation</a>
<ul>
<li class="chapter" data-level="5.1" data-path="variant-annotation.html"><a href="variant-annotation.html#filtration"><i class="fa fa-check"></i><b>5.1</b> Filtration</a></li>
<li class="chapter" data-level="5.2" data-path="variant-annotation.html"><a href="variant-annotation.html#de-novo-variants"><i class="fa fa-check"></i><b>5.2</b> <em>De Novo</em> Variants</a></li>
<li class="chapter" data-level="5.3" data-path="variant-annotation.html"><a href="variant-annotation.html#homozygous-variants"><i class="fa fa-check"></i><b>5.3</b> Homozygous Variants</a></li>
<li class="chapter" data-level="5.4" data-path="variant-annotation.html"><a href="variant-annotation.html#compound-heterozygous-variants"><i class="fa fa-check"></i><b>5.4</b> Compound Heterozygous Variants</a></li>
<li class="chapter" data-level="5.5" data-path="variant-annotation.html"><a href="variant-annotation.html#x-chromosome"><i class="fa fa-check"></i><b>5.5</b> X Chromosome</a></li>
<li class="chapter" data-level="5.6" data-path="variant-annotation.html"><a href="variant-annotation.html#disease-genes"><i class="fa fa-check"></i><b>5.6</b> Disease Genes</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="summary.html"><a href="summary.html"><i class="fa fa-check"></i><b>6</b> Summary</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Whole Exome Sequencing Platform</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="variant-annotation" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">5</span> Variant Annotation<a href="variant-annotation.html#variant-annotation" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Annotate the variants with VEP for molecular consequence, allele frequency and a set of pathogenicity prediction tools.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb19-1"><a href="variant-annotation.html#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="ex">apptainer</span> exec <span class="at">-B</span> vep_data:/opt/vep/.vep vep.sif /opt/vep/src/ensembl-vep/vep <span class="dt">\</span></span>
<span id="cb19-2"><a href="variant-annotation.html#cb19-2" aria-hidden="true" tabindex="-1"></a>--dir_cache /opt/vep/.vep/ <span class="dt">\</span></span>
<span id="cb19-3"><a href="variant-annotation.html#cb19-3" aria-hidden="true" tabindex="-1"></a>--dir_plugins /opt/vep/.vep/Plugins/ <span class="dt">\</span></span>
<span id="cb19-4"><a href="variant-annotation.html#cb19-4" aria-hidden="true" tabindex="-1"></a>-i /opt/vep/.vep/output_vqsr_snps_hard_indels.vcf.gz <span class="dt">\</span></span>
<span id="cb19-5"><a href="variant-annotation.html#cb19-5" aria-hidden="true" tabindex="-1"></a>-o /opt/vep/.vep/ensembl_annotated.txt <span class="dt">\</span></span>
<span id="cb19-6"><a href="variant-annotation.html#cb19-6" aria-hidden="true" tabindex="-1"></a>--individual EC27_proband <span class="dt">\</span></span>
<span id="cb19-7"><a href="variant-annotation.html#cb19-7" aria-hidden="true" tabindex="-1"></a>--everything <span class="dt">\</span></span>
<span id="cb19-8"><a href="variant-annotation.html#cb19-8" aria-hidden="true" tabindex="-1"></a>--check_existing <span class="dt">\</span></span>
<span id="cb19-9"><a href="variant-annotation.html#cb19-9" aria-hidden="true" tabindex="-1"></a>--xref_refseq <span class="dt">\</span></span>
<span id="cb19-10"><a href="variant-annotation.html#cb19-10" aria-hidden="true" tabindex="-1"></a>--fasta /opt/vep/.vep/Homo_sapiens.GRCh38.fa <span class="dt">\</span></span>
<span id="cb19-11"><a href="variant-annotation.html#cb19-11" aria-hidden="true" tabindex="-1"></a>--hgvs <span class="dt">\</span></span>
<span id="cb19-12"><a href="variant-annotation.html#cb19-12" aria-hidden="true" tabindex="-1"></a>--hgvsg <span class="dt">\</span></span>
<span id="cb19-13"><a href="variant-annotation.html#cb19-13" aria-hidden="true" tabindex="-1"></a>--plugin CADD,/opt/vep/.vep/Plugins/data/grch38/whole_genome_SNVs_grch38.tsv.gz,/opt/vep/.vep/Plugins/data/grch38/gnomad.genomes.r3.0.indel_grch38.tsv.gz <span class="dt">\</span></span>
<span id="cb19-14"><a href="variant-annotation.html#cb19-14" aria-hidden="true" tabindex="-1"></a>--plugin PrimateAI,/opt/vep/.vep/Plugins/data/grch38/PrimateAI_scores_v0.2_GRCh38_sorted.tsv.bgz <span class="dt">\</span></span>
<span id="cb19-15"><a href="variant-annotation.html#cb19-15" aria-hidden="true" tabindex="-1"></a>--plugin SpliceAI,snv=/opt/vep/.vep/Plugins/data/grch38/spliceai_scores.raw.snv.hg38.vcf.gz,indel=/opt/vep/.vep/Plugins/data/grch38/spliceai_scores.raw.indel.hg38.vcf.gz <span class="dt">\</span></span>
<span id="cb19-16"><a href="variant-annotation.html#cb19-16" aria-hidden="true" tabindex="-1"></a>--plugin EVE,file=/opt/vep/.vep/Plugins/data/grch38/eve_merged.vcf.gz <span class="dt">\</span></span>
<span id="cb19-17"><a href="variant-annotation.html#cb19-17" aria-hidden="true" tabindex="-1"></a>--plugin Mastermind,/opt/vep/.vep/Plugins/data/grch38/mastermind_cited_variants_reference-2023.04.02-grch38.vcf.gz <span class="dt">\</span></span>
<span id="cb19-18"><a href="variant-annotation.html#cb19-18" aria-hidden="true" tabindex="-1"></a>--plugin dbNSFP,/opt/vep/.vep/Plugins/data/grch38/dbNSFP4.3a_grch38.gz,MutationTaster_score,MutationTaster_pred <span class="dt">\</span></span>
<span id="cb19-19"><a href="variant-annotation.html#cb19-19" aria-hidden="true" tabindex="-1"></a>--plugin REVEL,/opt/vep/.vep/Plugins/data/grch38/new_tabbed_revel_grch38.tsv.gz <span class="dt">\</span></span>
<span id="cb19-20"><a href="variant-annotation.html#cb19-20" aria-hidden="true" tabindex="-1"></a>--custom /opt/vep/.vep/Plugins/data/grch38/clinvar.vcf.gz,ClinVar,vcf,exact,0,CLNSIG,CLNREVSTAT,CLNDN <span class="dt">\</span></span>
<span id="cb19-21"><a href="variant-annotation.html#cb19-21" aria-hidden="true" tabindex="-1"></a>--cache <span class="dt">\</span></span>
<span id="cb19-22"><a href="variant-annotation.html#cb19-22" aria-hidden="true" tabindex="-1"></a>--merged <span class="dt">\</span></span>
<span id="cb19-23"><a href="variant-annotation.html#cb19-23" aria-hidden="true" tabindex="-1"></a>--offline <span class="dt">\</span></span>
<span id="cb19-24"><a href="variant-annotation.html#cb19-24" aria-hidden="true" tabindex="-1"></a>--use_given_ref <span class="dt">\</span></span>
<span id="cb19-25"><a href="variant-annotation.html#cb19-25" aria-hidden="true" tabindex="-1"></a>--force_overwrite <span class="dt">\</span></span>
<span id="cb19-26"><a href="variant-annotation.html#cb19-26" aria-hidden="true" tabindex="-1"></a>--assembly GRCh38 <span class="dt">\</span></span>
<span id="cb19-27"><a href="variant-annotation.html#cb19-27" aria-hidden="true" tabindex="-1"></a>--fork 2 <span class="dt">\</span></span>
<span id="cb19-28"><a href="variant-annotation.html#cb19-28" aria-hidden="true" tabindex="-1"></a>--buffer_size 1000 <span class="dt">\</span></span>
<span id="cb19-29"><a href="variant-annotation.html#cb19-29" aria-hidden="true" tabindex="-1"></a>--tab</span></code></pre></div>
<div id="filtration" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> Filtration<a href="variant-annotation.html#filtration" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Perform a first round of filtering with VEP to reduce the size of the VCF to be loaded into R. In particular, select variants with high or moderate impact and an allele frequency smaller than 0.05.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb20-1"><a href="variant-annotation.html#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="ex">apptainer</span> exec <span class="at">-B</span> vep_data:/opt/vep/.vep vep.sif /opt/vep/src/ensembl-vep/filter_vep <span class="dt">\</span></span>
<span id="cb20-2"><a href="variant-annotation.html#cb20-2" aria-hidden="true" tabindex="-1"></a>-i /opt/vep/.vep/ensembl_annotated.txt <span class="dt">\</span></span>
<span id="cb20-3"><a href="variant-annotation.html#cb20-3" aria-hidden="true" tabindex="-1"></a>-out /opt/vep/.vep/ensembl_ann_vep_filt.txt <span class="dt">\</span></span>
<span id="cb20-4"><a href="variant-annotation.html#cb20-4" aria-hidden="true" tabindex="-1"></a>--filter <span class="st">&quot;(IMPACT is HIGH or IMPACT is MODERATE) and (gnomADe_AF &lt;= 0.05 or not gnomADe_AF)&quot;</span> <span class="dt">\</span></span>
<span id="cb20-5"><a href="variant-annotation.html#cb20-5" aria-hidden="true" tabindex="-1"></a>--force_overwrite <span class="dt">\</span></span>
<span id="cb20-6"><a href="variant-annotation.html#cb20-6" aria-hidden="true" tabindex="-1"></a>--format tab </span></code></pre></div>
<p>Load the WhatsHap phased variants into R and select autosomal variants.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="variant-annotation.html#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb21-2"><a href="variant-annotation.html#cb21-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(VariantAnnotation)</span>
<span id="cb21-3"><a href="variant-annotation.html#cb21-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-4"><a href="variant-annotation.html#cb21-4" aria-hidden="true" tabindex="-1"></a>mother <span class="ot">=</span> <span class="st">&quot;EC25&quot;</span></span>
<span id="cb21-5"><a href="variant-annotation.html#cb21-5" aria-hidden="true" tabindex="-1"></a>father <span class="ot">=</span> <span class="st">&quot;EC26&quot;</span></span>
<span id="cb21-6"><a href="variant-annotation.html#cb21-6" aria-hidden="true" tabindex="-1"></a>proband <span class="ot">=</span> <span class="st">&quot;EC27&quot;</span></span>
<span id="cb21-7"><a href="variant-annotation.html#cb21-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-8"><a href="variant-annotation.html#cb21-8" aria-hidden="true" tabindex="-1"></a>wh <span class="ot">&lt;-</span> <span class="fu">readVcf</span>(<span class="st">&quot;phased_whatshap_refined.vcf&quot;</span>, <span class="st">&quot;hg38&quot;</span>)</span>
<span id="cb21-9"><a href="variant-annotation.html#cb21-9" aria-hidden="true" tabindex="-1"></a>wh <span class="ot">&lt;-</span> <span class="fu">expand</span>(wh)</span>
<span id="cb21-10"><a href="variant-annotation.html#cb21-10" aria-hidden="true" tabindex="-1"></a>pos <span class="ot">&lt;-</span> <span class="fu">rowRanges</span>(wh)</span>
<span id="cb21-11"><a href="variant-annotation.html#cb21-11" aria-hidden="true" tabindex="-1"></a>pos2 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(pos) <span class="sc">%&gt;%</span> as_tibble </span>
<span id="cb21-12"><a href="variant-annotation.html#cb21-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-13"><a href="variant-annotation.html#cb21-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Genotype</span></span>
<span id="cb21-14"><a href="variant-annotation.html#cb21-14" aria-hidden="true" tabindex="-1"></a>wh_geno <span class="ot">&lt;-</span> <span class="fu">geno</span>(wh)<span class="sc">$</span>GT <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb21-15"><a href="variant-annotation.html#cb21-15" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;mother&quot;</span><span class="ot">=</span> <span class="fu">sym</span>(mother), <span class="st">&quot;father&quot;</span><span class="ot">=</span> <span class="fu">sym</span>(father), <span class="st">&quot;proband&quot;</span><span class="ot">=</span> <span class="fu">sym</span>(proband))</span>
<span id="cb21-16"><a href="variant-annotation.html#cb21-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Phasing block</span></span>
<span id="cb21-17"><a href="variant-annotation.html#cb21-17" aria-hidden="true" tabindex="-1"></a>wh_ps <span class="ot">&lt;-</span> <span class="fu">geno</span>(wh)<span class="sc">$</span>PS <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb21-18"><a href="variant-annotation.html#cb21-18" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;PS_mother&quot;</span> <span class="ot">=</span> <span class="fu">sym</span>(mother), <span class="st">&quot;PS_father&quot;</span><span class="ot">=</span> <span class="fu">sym</span>(father), <span class="st">&quot;PS_proband&quot;</span><span class="ot">=</span> <span class="fu">sym</span>(proband))</span>
<span id="cb21-19"><a href="variant-annotation.html#cb21-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Genotype quality</span></span>
<span id="cb21-20"><a href="variant-annotation.html#cb21-20" aria-hidden="true" tabindex="-1"></a>wh_gq <span class="ot">&lt;-</span> <span class="fu">geno</span>(wh)<span class="sc">$</span>GQ <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb21-21"><a href="variant-annotation.html#cb21-21" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;GQ_mother&quot;</span> <span class="ot">=</span> <span class="fu">sym</span>(mother), <span class="st">&quot;GQ_father&quot;</span><span class="ot">=</span> <span class="fu">sym</span>(father), <span class="st">&quot;GQ_proband&quot;</span><span class="ot">=</span> <span class="fu">sym</span>(proband))</span>
<span id="cb21-22"><a href="variant-annotation.html#cb21-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Join genotype, phasing block and genotype quality</span></span>
<span id="cb21-23"><a href="variant-annotation.html#cb21-23" aria-hidden="true" tabindex="-1"></a>wh_geno0 <span class="ot">&lt;-</span> <span class="fu">bind_cols</span>(pos2,wh_geno,wh_gq,wh_ps)</span>
<span id="cb21-24"><a href="variant-annotation.html#cb21-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-25"><a href="variant-annotation.html#cb21-25" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify alternate alleles with number</span></span>
<span id="cb21-26"><a href="variant-annotation.html#cb21-26" aria-hidden="true" tabindex="-1"></a>wh_geno01 <span class="ot">&lt;-</span> wh_geno0 <span class="sc">%&gt;%</span> </span>
<span id="cb21-27"><a href="variant-annotation.html#cb21-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(seqnames,start,end,width,REF) <span class="sc">%&gt;%</span> </span>
<span id="cb21-28"><a href="variant-annotation.html#cb21-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">counter =</span> <span class="fu">row_number</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb21-29"><a href="variant-annotation.html#cb21-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb21-30"><a href="variant-annotation.html#cb21-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-31"><a href="variant-annotation.html#cb21-31" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert variant formatting to VEP format. Add 1 to indel start site.</span></span>
<span id="cb21-32"><a href="variant-annotation.html#cb21-32" aria-hidden="true" tabindex="-1"></a>wh_geno01 <span class="ot">&lt;-</span> wh_geno01 <span class="sc">%&gt;%</span> </span>
<span id="cb21-33"><a href="variant-annotation.html#cb21-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Location =</span> <span class="fu">ifelse</span>(</span>
<span id="cb21-34"><a href="variant-annotation.html#cb21-34" aria-hidden="true" tabindex="-1"></a>    width<span class="sc">==</span><span class="dv">1</span>, <span class="fu">ifelse</span>(</span>
<span id="cb21-35"><a href="variant-annotation.html#cb21-35" aria-hidden="true" tabindex="-1"></a>      <span class="fu">nchar</span>(ALT)<span class="sc">==</span><span class="dv">1</span>, <span class="fu">paste</span>(seqnames,start,<span class="at">sep=</span><span class="st">&quot;:&quot;</span>), <span class="fu">paste0</span>(seqnames,<span class="st">&quot;:&quot;</span>,start,<span class="st">&quot;-&quot;</span>,start<span class="sc">+</span><span class="dv">1</span>)), <span class="fu">ifelse</span>(</span>
<span id="cb21-36"><a href="variant-annotation.html#cb21-36" aria-hidden="true" tabindex="-1"></a>        start<span class="sc">+</span><span class="dv">1</span><span class="sc">==</span>end, <span class="fu">paste0</span>(seqnames,<span class="st">&quot;:&quot;</span>,start<span class="sc">+</span><span class="dv">1</span>), <span class="fu">paste0</span>(seqnames,<span class="st">&quot;:&quot;</span>,start<span class="sc">+</span><span class="dv">1</span>,<span class="st">&quot;-&quot;</span>,end)))) <span class="sc">%&gt;%</span> </span>
<span id="cb21-37"><a href="variant-annotation.html#cb21-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Allele =</span> <span class="fu">ifelse</span>(<span class="fu">substr</span>(REF,<span class="dv">1</span>,<span class="dv">1</span>)<span class="sc">==</span><span class="fu">substr</span>(ALT,<span class="dv">1</span>,<span class="dv">1</span>), <span class="fu">substr</span>(ALT,<span class="dv">2</span>,<span class="fu">nchar</span>(ALT)), ALT))</span>
<span id="cb21-38"><a href="variant-annotation.html#cb21-38" aria-hidden="true" tabindex="-1"></a>wh_geno01<span class="sc">$</span>Allele[wh_geno01<span class="sc">$</span>Allele<span class="sc">==</span><span class="st">&quot;&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;-&quot;</span></span>
<span id="cb21-39"><a href="variant-annotation.html#cb21-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-40"><a href="variant-annotation.html#cb21-40" aria-hidden="true" tabindex="-1"></a><span class="co"># Overlapping deletions</span></span>
<span id="cb21-41"><a href="variant-annotation.html#cb21-41" aria-hidden="true" tabindex="-1"></a>wh_geno_overlap <span class="ot">&lt;-</span> wh_geno01 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(ALT<span class="sc">==</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb21-42"><a href="variant-annotation.html#cb21-42" aria-hidden="true" tabindex="-1"></a>wh_geno1 <span class="ot">&lt;-</span> wh_geno01 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(ALT<span class="sc">!=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb21-43"><a href="variant-annotation.html#cb21-43" aria-hidden="true" tabindex="-1"></a>wh_geno1 <span class="ot">&lt;-</span> wh_geno1[<span class="fu">c</span>(<span class="dv">21</span>,<span class="dv">22</span>,<span class="dv">11</span><span class="sc">:</span><span class="dv">19</span>,<span class="dv">20</span>)]</span>
<span id="cb21-44"><a href="variant-annotation.html#cb21-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-45"><a href="variant-annotation.html#cb21-45" aria-hidden="true" tabindex="-1"></a><span class="co"># Select X chromosome</span></span>
<span id="cb21-46"><a href="variant-annotation.html#cb21-46" aria-hidden="true" tabindex="-1"></a>wh_geno_X <span class="ot">&lt;-</span> wh_geno1 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">str_detect</span>(Location,<span class="st">&quot;chrX&quot;</span>))</span>
<span id="cb21-47"><a href="variant-annotation.html#cb21-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-48"><a href="variant-annotation.html#cb21-48" aria-hidden="true" tabindex="-1"></a><span class="co"># Select autosomal variants</span></span>
<span id="cb21-49"><a href="variant-annotation.html#cb21-49" aria-hidden="true" tabindex="-1"></a>wh_geno_auto <span class="ot">&lt;-</span> wh_geno1 <span class="sc">%&gt;%</span></span>
<span id="cb21-50"><a href="variant-annotation.html#cb21-50" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(Location,<span class="st">&quot;chrX&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb21-51"><a href="variant-annotation.html#cb21-51" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(Location,<span class="st">&quot;chrY&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb21-52"><a href="variant-annotation.html#cb21-52" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(Location,<span class="st">&quot;M&quot;</span>))</span>
<span id="cb21-53"><a href="variant-annotation.html#cb21-53" aria-hidden="true" tabindex="-1"></a>wh_geno_auto</span></code></pre></div>
<p>Manually remove the hashtag before the line of column titles in ensembl_ann_vep_filt.txt. Load the filtered variants into R. Select variants with an allele frequency &lt;0.01 that fall in RefSeq transcripts.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="variant-annotation.html#cb22-1" aria-hidden="true" tabindex="-1"></a>result <span class="ot">&lt;-</span> <span class="fu">read_table</span>(<span class="st">&quot;ensembl_ann_vep_filt_refined_08_07_23.txt&quot;</span>, <span class="at">comment =</span> <span class="st">&quot;#&quot;</span>, <span class="at">na=</span><span class="st">&quot;-&quot;</span>, <span class="at">col_types =</span> <span class="fu">list</span>(</span>
<span id="cb22-2"><a href="variant-annotation.html#cb22-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">Uploaded_variation=</span> <span class="fu">col_factor</span>(), <span class="at">Gene=</span><span class="fu">col_factor</span>(), <span class="at">Feature=</span><span class="fu">col_factor</span>(), <span class="at">MANE_PLUS_CLINICAL =</span> <span class="fu">col_character</span>(), </span>
<span id="cb22-3"><a href="variant-annotation.html#cb22-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">MOTIF_NAME =</span> <span class="fu">col_character</span>(), <span class="at">HIGH_INF_POS =</span> <span class="fu">col_character</span>(), <span class="at">TRANSCRIPTION_FACTORS =</span> <span class="fu">col_character</span>(), </span>
<span id="cb22-4"><a href="variant-annotation.html#cb22-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">PUBMED=</span><span class="fu">col_character</span>(), <span class="at">MOTIF_POS =</span> <span class="fu">col_character</span>()))</span>
<span id="cb22-5"><a href="variant-annotation.html#cb22-5" aria-hidden="true" tabindex="-1"></a>result<span class="sc">$</span>Allele[<span class="fu">is.na</span>(result<span class="sc">$</span>Allele)] <span class="ot">&lt;-</span> <span class="st">&quot;-&quot;</span></span>
<span id="cb22-6"><a href="variant-annotation.html#cb22-6" aria-hidden="true" tabindex="-1"></a>result1 <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span> as_tibble <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(wh_geno1, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;Location&quot;</span>,<span class="st">&quot;Allele&quot;</span>))</span>
<span id="cb22-7"><a href="variant-annotation.html#cb22-7" aria-hidden="true" tabindex="-1"></a>refseq0 <span class="ot">&lt;-</span> result1 <span class="sc">%&gt;%</span> </span>
<span id="cb22-8"><a href="variant-annotation.html#cb22-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(SOURCE <span class="sc">==</span> <span class="st">&quot;RefSeq&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb22-9"><a href="variant-annotation.html#cb22-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(gnomADe_AF<span class="sc">&lt;=</span><span class="fl">0.01</span><span class="sc">|</span><span class="fu">is.na</span>(gnomADe_AF)) <span class="sc">%&gt;%</span> </span>
<span id="cb22-10"><a href="variant-annotation.html#cb22-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>(<span class="fu">is.na</span>(gnomADe_AF) <span class="sc">&amp;</span> gnomADg_AF <span class="sc">&gt;</span> <span class="fl">0.01</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb22-11"><a href="variant-annotation.html#cb22-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>(<span class="fu">str_detect</span>(Feature,<span class="st">&quot;XM&quot;</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb22-12"><a href="variant-annotation.html#cb22-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select_if</span>(<span class="sc">~</span><span class="fu">any</span>(<span class="sc">!</span><span class="fu">is.na</span>(.)))</span></code></pre></div>
<p>Add gene-level information.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="variant-annotation.html#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ontologyIndex)</span>
<span id="cb23-2"><a href="variant-annotation.html#cb23-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-3"><a href="variant-annotation.html#cb23-3" aria-hidden="true" tabindex="-1"></a><span class="co"># OMIM</span></span>
<span id="cb23-4"><a href="variant-annotation.html#cb23-4" aria-hidden="true" tabindex="-1"></a>omim <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">&quot;~/Documents/kidney_panel/omim/genemap2_edit.txt&quot;</span>, <span class="at">name_repair=</span><span class="st">&quot;universal&quot;</span>, <span class="at">col_types=</span><span class="fu">list</span>(<span class="at">Entrez.Gene.ID =</span> <span class="fu">col_factor</span>())) <span class="sc">%&gt;%</span> </span>
<span id="cb23-5"><a href="variant-annotation.html#cb23-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">OMIM_Phenotypes =</span> Phenotypes) <span class="sc">%&gt;%</span> </span>
<span id="cb23-6"><a href="variant-annotation.html#cb23-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(Entrez.Gene.ID,OMIM_Phenotypes)</span>
<span id="cb23-7"><a href="variant-annotation.html#cb23-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-8"><a href="variant-annotation.html#cb23-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Human Phenotype Ontology</span></span>
<span id="cb23-9"><a href="variant-annotation.html#cb23-9" aria-hidden="true" tabindex="-1"></a>hpo_data <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">&quot;~/Documents/kidney_panel/genes_to_phenotype.txt&quot;</span>, <span class="at">col_types=</span><span class="fu">list</span>(<span class="at">entrez_gene_id =</span> <span class="fu">col_factor</span>()))</span>
<span id="cb23-10"><a href="variant-annotation.html#cb23-10" aria-hidden="true" tabindex="-1"></a>hpo <span class="ot">&lt;-</span> <span class="fu">get_ontology</span>(<span class="at">file =</span> <span class="st">&#39;~/Documents/kidney_panel/hp_obo.txt&#39;</span>, <span class="at">propagate_relationships =</span> <span class="st">&quot;is_a&quot;</span>, <span class="at">extract_tags =</span> <span class="st">&quot;minimal&quot;</span>)</span>
<span id="cb23-11"><a href="variant-annotation.html#cb23-11" aria-hidden="true" tabindex="-1"></a>hpo_kidney <span class="ot">&lt;-</span> <span class="fu">get_descendants</span>(hpo, <span class="at">roots=</span><span class="st">&quot;HP:0000077&quot;</span>) <span class="co">#Abnormality of the kidney</span></span>
<span id="cb23-12"><a href="variant-annotation.html#cb23-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-13"><a href="variant-annotation.html#cb23-13" aria-hidden="true" tabindex="-1"></a>gen_hpo_kidney <span class="ot">&lt;-</span> hpo_data <span class="sc">%&gt;%</span> </span>
<span id="cb23-14"><a href="variant-annotation.html#cb23-14" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(hpo_term_id <span class="sc">%in%</span> hpo_kidney) <span class="sc">%&gt;%</span> </span>
<span id="cb23-15"><a href="variant-annotation.html#cb23-15" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(frequency_hpo) <span class="sc">%&gt;%</span> </span>
<span id="cb23-16"><a href="variant-annotation.html#cb23-16" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">frequency =</span> <span class="fu">case_when</span>(</span>
<span id="cb23-17"><a href="variant-annotation.html#cb23-17" aria-hidden="true" tabindex="-1"></a>    frequency_hpo <span class="sc">==</span> <span class="st">&quot;HP:0040284&quot;</span> <span class="sc">~</span> <span class="st">&quot; (very rare)&quot;</span>, frequency_hpo <span class="sc">==</span> <span class="st">&quot;HP:0040283&quot;</span> <span class="sc">~</span> <span class="st">&quot; (occasional)&quot;</span>, </span>
<span id="cb23-18"><a href="variant-annotation.html#cb23-18" aria-hidden="true" tabindex="-1"></a>    frequency_hpo <span class="sc">==</span> <span class="st">&quot;HP:0040282&quot;</span> <span class="sc">~</span> <span class="st">&quot; (frequent)&quot;</span>, frequency_hpo <span class="sc">==</span> <span class="st">&quot;HP:0040281&quot;</span> <span class="sc">~</span> <span class="st">&quot; (very frequent)&quot;</span>, </span>
<span id="cb23-19"><a href="variant-annotation.html#cb23-19" aria-hidden="true" tabindex="-1"></a>    frequency_hpo <span class="sc">==</span> <span class="st">&quot;HP:0040280&quot;</span> <span class="sc">~</span> <span class="st">&quot; (obligate)&quot;</span>, <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">&quot;&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb23-20"><a href="variant-annotation.html#cb23-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(<span class="st">&quot;renal_hpo_term&quot;</span>, <span class="fu">c</span>(hpo_term_name, frequency), <span class="at">remove =</span> <span class="cn">FALSE</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb23-21"><a href="variant-annotation.html#cb23-21" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(entrez_gene_id) <span class="sc">%&gt;%</span> </span>
<span id="cb23-22"><a href="variant-annotation.html#cb23-22" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">renal_hpo_term =</span> <span class="fu">toString</span>(<span class="fu">unique</span>(renal_hpo_term)))</span>
<span id="cb23-23"><a href="variant-annotation.html#cb23-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-24"><a href="variant-annotation.html#cb23-24" aria-hidden="true" tabindex="-1"></a>gen_hpo_extra <span class="ot">&lt;-</span> hpo_data <span class="sc">%&gt;%</span> </span>
<span id="cb23-25"><a href="variant-annotation.html#cb23-25" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(frequency_hpo <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;HP:0040281&quot;</span>, <span class="st">&quot;HP:0040280&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb23-26"><a href="variant-annotation.html#cb23-26" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(frequency_hpo) <span class="sc">%&gt;%</span></span>
<span id="cb23-27"><a href="variant-annotation.html#cb23-27" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>(hpo_term_id <span class="sc">%in%</span> hpo_kidney)) <span class="sc">%&gt;%</span> </span>
<span id="cb23-28"><a href="variant-annotation.html#cb23-28" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(entrez_gene_id) <span class="sc">%&gt;%</span> </span>
<span id="cb23-29"><a href="variant-annotation.html#cb23-29" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">extrarenal_hpo_term =</span> <span class="fu">toString</span>(<span class="fu">unique</span>(hpo_term_name)))</span>
<span id="cb23-30"><a href="variant-annotation.html#cb23-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-31"><a href="variant-annotation.html#cb23-31" aria-hidden="true" tabindex="-1"></a><span class="co"># ClinGen </span></span>
<span id="cb23-32"><a href="variant-annotation.html#cb23-32" aria-hidden="true" tabindex="-1"></a>clingen <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;~/Documents/kidney_panel/Clingen-Curation.csv&quot;</span>, <span class="at">skip =</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb23-33"><a href="variant-annotation.html#cb23-33" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">assertion=</span><span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">s*</span><span class="sc">\\</span><span class="st">([^</span><span class="sc">\\</span><span class="st">)]+</span><span class="sc">\\</span><span class="st">)&quot;</span>,<span class="st">&quot;&quot;</span>, gene_disease_validity_assertion_classifications)) <span class="sc">%&gt;%</span></span>
<span id="cb23-34"><a href="variant-annotation.html#cb23-34" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">assertion=</span><span class="fu">paste</span>(<span class="st">&quot;(&quot;</span>,assertion,<span class="st">&quot;)&quot;</span>,<span class="at">sep=</span><span class="st">&quot;&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb23-35"><a href="variant-annotation.html#cb23-35" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">assertion=</span><span class="fu">if_else</span>(assertion<span class="sc">==</span><span class="st">&quot;(NA)&quot;</span>,<span class="cn">NA</span>,assertion)) <span class="sc">%&gt;%</span></span>
<span id="cb23-36"><a href="variant-annotation.html#cb23-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(<span class="st">&quot;disease&quot;</span>, <span class="fu">c</span>(disease_label, assertion), <span class="at">remove =</span> <span class="cn">FALSE</span>, <span class="at">na.rm=</span><span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">&quot; &quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb23-37"><a href="variant-annotation.html#cb23-37" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(hgnc_id) <span class="sc">%&gt;%</span> </span>
<span id="cb23-38"><a href="variant-annotation.html#cb23-38" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">clingen_disease_label =</span> <span class="fu">paste</span>(disease,<span class="at">collapse=</span><span class="st">&quot;; &quot;</span>))</span>
<span id="cb23-39"><a href="variant-annotation.html#cb23-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-40"><a href="variant-annotation.html#cb23-40" aria-hidden="true" tabindex="-1"></a><span class="co"># GenCC including Orphanet data</span></span>
<span id="cb23-41"><a href="variant-annotation.html#cb23-41" aria-hidden="true" tabindex="-1"></a>gencc <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">&quot;~/Documents/kidney_panel/gencc-submissions.tsv&quot;</span>)</span>
<span id="cb23-42"><a href="variant-annotation.html#cb23-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-43"><a href="variant-annotation.html#cb23-43" aria-hidden="true" tabindex="-1"></a>gencc_def <span class="ot">&lt;-</span> gencc <span class="sc">%&gt;%</span> </span>
<span id="cb23-44"><a href="variant-annotation.html#cb23-44" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(submitter_title <span class="sc">!=</span> <span class="st">&quot;Orphanet&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb23-45"><a href="variant-annotation.html#cb23-45" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(classification_title <span class="sc">==</span> <span class="st">&quot;Definitive&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb23-46"><a href="variant-annotation.html#cb23-46" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(gene_symbol, gene_curie, disease_title) <span class="sc">%&gt;%</span> </span>
<span id="cb23-47"><a href="variant-annotation.html#cb23-47" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(</span>
<span id="cb23-48"><a href="variant-annotation.html#cb23-48" aria-hidden="true" tabindex="-1"></a>    <span class="at">submitter_title =</span> <span class="fu">toString</span>(<span class="fu">unique</span>(submitter_title)), <span class="at">moi_title =</span> <span class="fu">toString</span>(<span class="fu">unique</span>(moi_title)), </span>
<span id="cb23-49"><a href="variant-annotation.html#cb23-49" aria-hidden="true" tabindex="-1"></a>    <span class="at">disease_curie =</span> <span class="fu">toString</span>(<span class="fu">unique</span>(disease_curie)), <span class="at">disease_original_curie =</span> <span class="fu">toString</span>(<span class="fu">unique</span>(disease_original_curie)), </span>
<span id="cb23-50"><a href="variant-annotation.html#cb23-50" aria-hidden="true" tabindex="-1"></a>    <span class="at">classification_title =</span> <span class="fu">toString</span>(classification_title), <span class="at">submitted_as_pmids =</span> <span class="fu">toString</span>(<span class="fu">unique</span>(submitted_as_pmids))) <span class="sc">%&gt;%</span></span>
<span id="cb23-51"><a href="variant-annotation.html#cb23-51" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">submitter =</span> <span class="fu">paste</span>(<span class="st">&quot;(&quot;</span>, submitter_title, <span class="st">&quot;)&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb23-52"><a href="variant-annotation.html#cb23-52" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">unite</span>(<span class="st">&quot;gencc_disease_definitive&quot;</span>, <span class="fu">c</span>(disease_title, submitter), <span class="at">remove =</span> <span class="cn">FALSE</span>, <span class="at">sep =</span> <span class="st">&quot; &quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb23-53"><a href="variant-annotation.html#cb23-53" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(gene_curie, gencc_disease_definitive) <span class="sc">%&gt;%</span> </span>
<span id="cb23-54"><a href="variant-annotation.html#cb23-54" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(gene_curie) <span class="sc">%&gt;%</span> </span>
<span id="cb23-55"><a href="variant-annotation.html#cb23-55" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">gencc_disease_definitive =</span> <span class="fu">paste</span>(<span class="fu">unique</span>(gencc_disease_definitive), <span class="at">collapse=</span><span class="st">&quot;; &quot;</span>))</span>
<span id="cb23-56"><a href="variant-annotation.html#cb23-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-57"><a href="variant-annotation.html#cb23-57" aria-hidden="true" tabindex="-1"></a>gencc_strong <span class="ot">&lt;-</span> gencc <span class="sc">%&gt;%</span> </span>
<span id="cb23-58"><a href="variant-annotation.html#cb23-58" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(submitter_title <span class="sc">!=</span> <span class="st">&quot;Orphanet&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb23-59"><a href="variant-annotation.html#cb23-59" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(classification_title <span class="sc">==</span> <span class="st">&quot;Strong&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb23-60"><a href="variant-annotation.html#cb23-60" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(gene_symbol, gene_curie, disease_title) <span class="sc">%&gt;%</span> </span>
<span id="cb23-61"><a href="variant-annotation.html#cb23-61" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(</span>
<span id="cb23-62"><a href="variant-annotation.html#cb23-62" aria-hidden="true" tabindex="-1"></a>    <span class="at">submitter_title =</span> <span class="fu">toString</span>(<span class="fu">unique</span>(submitter_title)), <span class="at">moi_title =</span> <span class="fu">toString</span>(<span class="fu">unique</span>(moi_title)), </span>
<span id="cb23-63"><a href="variant-annotation.html#cb23-63" aria-hidden="true" tabindex="-1"></a>    <span class="at">disease_curie =</span> <span class="fu">toString</span>(<span class="fu">unique</span>(disease_curie)), <span class="at">disease_original_curie =</span> <span class="fu">toString</span>(<span class="fu">unique</span>(disease_original_curie)), </span>
<span id="cb23-64"><a href="variant-annotation.html#cb23-64" aria-hidden="true" tabindex="-1"></a>    <span class="at">classification_title =</span> <span class="fu">toString</span>(classification_title), <span class="at">submitted_as_pmids =</span> <span class="fu">toString</span>(<span class="fu">unique</span>(submitted_as_pmids))) <span class="sc">%&gt;%</span></span>
<span id="cb23-65"><a href="variant-annotation.html#cb23-65" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">submitter =</span> <span class="fu">paste</span>(<span class="st">&quot;(&quot;</span>, submitter_title, <span class="st">&quot;)&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb23-66"><a href="variant-annotation.html#cb23-66" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">unite</span>(<span class="st">&quot;gencc_disease_strong&quot;</span>, <span class="fu">c</span>(disease_title, submitter), <span class="at">remove =</span> <span class="cn">FALSE</span>, <span class="at">sep =</span> <span class="st">&quot; &quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb23-67"><a href="variant-annotation.html#cb23-67" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(gene_curie, gencc_disease_strong) <span class="sc">%&gt;%</span> </span>
<span id="cb23-68"><a href="variant-annotation.html#cb23-68" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(gene_curie) <span class="sc">%&gt;%</span> </span>
<span id="cb23-69"><a href="variant-annotation.html#cb23-69" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">gencc_disease_strong =</span> <span class="fu">paste</span>(<span class="fu">unique</span>(gencc_disease_strong), <span class="at">collapse=</span><span class="st">&quot;; &quot;</span>))</span>
<span id="cb23-70"><a href="variant-annotation.html#cb23-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-71"><a href="variant-annotation.html#cb23-71" aria-hidden="true" tabindex="-1"></a>orphanet <span class="ot">&lt;-</span> gencc <span class="sc">%&gt;%</span> </span>
<span id="cb23-72"><a href="variant-annotation.html#cb23-72" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(submitter_title <span class="sc">==</span> <span class="st">&quot;Orphanet&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb23-73"><a href="variant-annotation.html#cb23-73" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(gene_curie) <span class="sc">%&gt;%</span> </span>
<span id="cb23-74"><a href="variant-annotation.html#cb23-74" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">orphanet_disease =</span> <span class="fu">toString</span>(disease_title), <span class="at">orphanet_moi =</span> <span class="fu">toString</span>(<span class="fu">unique</span>(moi_title)))</span>
<span id="cb23-75"><a href="variant-annotation.html#cb23-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-76"><a href="variant-annotation.html#cb23-76" aria-hidden="true" tabindex="-1"></a><span class="co"># Join</span></span>
<span id="cb23-77"><a href="variant-annotation.html#cb23-77" aria-hidden="true" tabindex="-1"></a>refseq01 <span class="ot">&lt;-</span> refseq0 <span class="sc">%&gt;%</span> </span>
<span id="cb23-78"><a href="variant-annotation.html#cb23-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(omim, <span class="at">by=</span><span class="fu">join_by</span>(Gene<span class="sc">==</span>Entrez.Gene.ID)) <span class="sc">%&gt;%</span></span>
<span id="cb23-79"><a href="variant-annotation.html#cb23-79" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(gen_hpo_kidney, <span class="at">by=</span><span class="fu">join_by</span>(Gene<span class="sc">==</span>entrez_gene_id)) <span class="sc">%&gt;%</span></span>
<span id="cb23-80"><a href="variant-annotation.html#cb23-80" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(gen_hpo_extra, <span class="at">by=</span><span class="fu">join_by</span>(Gene<span class="sc">==</span>entrez_gene_id)) <span class="sc">%&gt;%</span></span>
<span id="cb23-81"><a href="variant-annotation.html#cb23-81" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(clingen, <span class="at">by=</span><span class="fu">join_by</span>(HGNC_ID<span class="sc">==</span>hgnc_id)) <span class="sc">%&gt;%</span></span>
<span id="cb23-82"><a href="variant-annotation.html#cb23-82" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(gencc_def, <span class="at">by=</span><span class="fu">join_by</span>(HGNC_ID<span class="sc">==</span>gene_curie)) <span class="sc">%&gt;%</span></span>
<span id="cb23-83"><a href="variant-annotation.html#cb23-83" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(gencc_strong, <span class="at">by=</span><span class="fu">join_by</span>(HGNC_ID<span class="sc">==</span>gene_curie)) <span class="sc">%&gt;%</span></span>
<span id="cb23-84"><a href="variant-annotation.html#cb23-84" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(orphanet, <span class="at">by=</span><span class="fu">join_by</span>(HGNC_ID<span class="sc">==</span>gene_curie))</span>
<span id="cb23-85"><a href="variant-annotation.html#cb23-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-86"><a href="variant-annotation.html#cb23-86" aria-hidden="true" tabindex="-1"></a>refseq_X <span class="ot">&lt;-</span> refseq01 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">str_detect</span>(Uploaded_variation,<span class="st">&quot;X&quot;</span>))</span>
<span id="cb23-87"><a href="variant-annotation.html#cb23-87" aria-hidden="true" tabindex="-1"></a>refseq <span class="ot">&lt;-</span> refseq01 <span class="sc">%&gt;%</span></span>
<span id="cb23-88"><a href="variant-annotation.html#cb23-88" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(Uploaded_variation,<span class="st">&quot;X&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb23-89"><a href="variant-annotation.html#cb23-89" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(Uploaded_variation,<span class="st">&quot;Y&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb23-90"><a href="variant-annotation.html#cb23-90" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(Uploaded_variation,<span class="st">&quot;M&quot;</span>))</span>
<span id="cb23-91"><a href="variant-annotation.html#cb23-91" aria-hidden="true" tabindex="-1"></a>refseq</span></code></pre></div>
</div>
<div id="de-novo-variants" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">5.2</span> <em>De Novo</em> Variants<a href="variant-annotation.html#de-novo-variants" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Select <em>de novo</em> variants in the proband.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="variant-annotation.html#cb24-1" aria-hidden="true" tabindex="-1"></a>geno_novo_all <span class="ot">&lt;-</span> wh_geno_auto <span class="sc">%&gt;%</span> </span>
<span id="cb24-2"><a href="variant-annotation.html#cb24-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span> ( ( ( <span class="fu">str_detect</span>(mother, <span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)) <span class="sc">|</span> <span class="fu">str_detect</span>(mother, <span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>) ) <span class="sc">&amp;</span> </span>
<span id="cb24-3"><a href="variant-annotation.html#cb24-3" aria-hidden="true" tabindex="-1"></a>                 (<span class="fu">str_detect</span>(father, <span class="fu">substr</span>(proband,<span class="dv">3</span>,<span class="dv">3</span>)) <span class="sc">|</span> <span class="fu">str_detect</span>(father, <span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>) ) ) <span class="sc">|</span> </span>
<span id="cb24-4"><a href="variant-annotation.html#cb24-4" aria-hidden="true" tabindex="-1"></a>               ( ( <span class="fu">str_detect</span>(father, <span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)) <span class="sc">|</span> <span class="fu">str_detect</span>(father, <span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>) ) <span class="sc">&amp;</span> </span>
<span id="cb24-5"><a href="variant-annotation.html#cb24-5" aria-hidden="true" tabindex="-1"></a>                   (<span class="fu">str_detect</span>(mother, <span class="fu">substr</span>(proband,<span class="dv">3</span>,<span class="dv">3</span>)) <span class="sc">|</span> <span class="fu">str_detect</span>(mother, <span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>) ) ) ) ) <span class="sc">%&gt;%</span></span>
<span id="cb24-6"><a href="variant-annotation.html#cb24-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>(<span class="fu">str_detect</span>(mother,<span class="fu">as.character</span>(counter)) <span class="sc">|</span> <span class="fu">str_detect</span>(father,<span class="fu">as.character</span>(counter))) <span class="sc">|</span></span>
<span id="cb24-7"><a href="variant-annotation.html#cb24-7" aria-hidden="true" tabindex="-1"></a>                  (<span class="fu">str_detect</span>(mother, <span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)) <span class="sc">&amp;</span> <span class="fu">str_detect</span>(mother, <span class="fu">substr</span>(proband,<span class="dv">3</span>,<span class="dv">3</span>))) <span class="sc">|</span></span>
<span id="cb24-8"><a href="variant-annotation.html#cb24-8" aria-hidden="true" tabindex="-1"></a>                  (<span class="fu">str_detect</span>(father, <span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)) <span class="sc">&amp;</span> <span class="fu">str_detect</span>(father, <span class="fu">substr</span>(proband,<span class="dv">3</span>,<span class="dv">3</span>)))</span>
<span id="cb24-9"><a href="variant-annotation.html#cb24-9" aria-hidden="true" tabindex="-1"></a>                  )</span>
<span id="cb24-10"><a href="variant-annotation.html#cb24-10" aria-hidden="true" tabindex="-1"></a>geno_novo_all</span></code></pre></div>
<p>Discard <em>de novo</em> mutations back to the reference allele (e.g. 1/1, 1/1 –&gt; 0/1) since they are unlikely to be pathogenic.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="variant-annotation.html#cb25-1" aria-hidden="true" tabindex="-1"></a>geno_novo_not_ref <span class="ot">&lt;-</span> wh_geno_auto <span class="sc">%&gt;%</span> </span>
<span id="cb25-2"><a href="variant-annotation.html#cb25-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span> ( ( ( <span class="fu">str_detect</span>(mother, <span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)) <span class="sc">|</span> <span class="fu">str_detect</span>(mother, <span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>) <span class="sc">|</span> <span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)<span class="sc">==</span><span class="dv">0</span> ) <span class="sc">&amp;</span> </span>
<span id="cb25-3"><a href="variant-annotation.html#cb25-3" aria-hidden="true" tabindex="-1"></a>                 (<span class="fu">str_detect</span>(father, <span class="fu">substr</span>(proband,<span class="dv">3</span>,<span class="dv">3</span>)) <span class="sc">|</span> <span class="fu">str_detect</span>(father, <span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>) <span class="sc">|</span> <span class="fu">substr</span>(proband,<span class="dv">3</span>,<span class="dv">3</span>)<span class="sc">==</span><span class="dv">0</span> ) ) <span class="sc">|</span> </span>
<span id="cb25-4"><a href="variant-annotation.html#cb25-4" aria-hidden="true" tabindex="-1"></a>               ( ( <span class="fu">str_detect</span>(father, <span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)) <span class="sc">|</span> <span class="fu">str_detect</span>(father, <span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>) <span class="sc">|</span> <span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)<span class="sc">==</span><span class="dv">0</span> ) <span class="sc">&amp;</span> </span>
<span id="cb25-5"><a href="variant-annotation.html#cb25-5" aria-hidden="true" tabindex="-1"></a>                   (<span class="fu">str_detect</span>(mother, <span class="fu">substr</span>(proband,<span class="dv">3</span>,<span class="dv">3</span>)) <span class="sc">|</span> <span class="fu">str_detect</span>(mother, <span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>) <span class="sc">|</span> <span class="fu">substr</span>(proband,<span class="dv">3</span>,<span class="dv">3</span>)<span class="sc">==</span><span class="dv">0</span> ) ) ) ) <span class="sc">%&gt;%</span></span>
<span id="cb25-6"><a href="variant-annotation.html#cb25-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>(<span class="fu">str_detect</span>(mother,<span class="fu">as.character</span>(counter)) <span class="sc">|</span> <span class="fu">str_detect</span>(father,<span class="fu">as.character</span>(counter))) <span class="sc">|</span></span>
<span id="cb25-7"><a href="variant-annotation.html#cb25-7" aria-hidden="true" tabindex="-1"></a>                  (<span class="fu">str_detect</span>(mother, <span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)) <span class="sc">&amp;</span> <span class="fu">str_detect</span>(mother, <span class="fu">substr</span>(proband,<span class="dv">3</span>,<span class="dv">3</span>))) <span class="sc">|</span></span>
<span id="cb25-8"><a href="variant-annotation.html#cb25-8" aria-hidden="true" tabindex="-1"></a>                  (<span class="fu">str_detect</span>(father, <span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)) <span class="sc">&amp;</span> <span class="fu">str_detect</span>(father, <span class="fu">substr</span>(proband,<span class="dv">3</span>,<span class="dv">3</span>)))</span>
<span id="cb25-9"><a href="variant-annotation.html#cb25-9" aria-hidden="true" tabindex="-1"></a>                  )</span>
<span id="cb25-10"><a href="variant-annotation.html#cb25-10" aria-hidden="true" tabindex="-1"></a>geno_novo_not_ref[<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>)]</span></code></pre></div>
<p>Increase the cutoff for genotype quality to ensure that the the <em>de novo</em> variants are not a result of incorrect variant calling. Set GQ&gt;=30, i.e, the probability that the call is incorrect is =&lt;0.1%.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="variant-annotation.html#cb26-1" aria-hidden="true" tabindex="-1"></a>geno_novo_not_ref_high_qual <span class="ot">&lt;-</span> geno_novo_not_ref <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(GQ_mother<span class="sc">&gt;=</span><span class="dv">30</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(GQ_father<span class="sc">&gt;=</span><span class="dv">30</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(GQ_proband<span class="sc">&gt;=</span><span class="dv">30</span>)</span>
<span id="cb26-2"><a href="variant-annotation.html#cb26-2" aria-hidden="true" tabindex="-1"></a>geno_novo_not_ref_high_qual</span></code></pre></div>
<p>Select <em>de novo</em> variants found in the filtered VEP annotation. For each variant, select a single gene transcript that contains the variant.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="variant-annotation.html#cb27-1" aria-hidden="true" tabindex="-1"></a>prob_novo <span class="ot">&lt;-</span> refseq <span class="sc">%&gt;%</span> <span class="fu">inner_join</span>(geno_novo_not_ref)</span>
<span id="cb27-2"><a href="variant-annotation.html#cb27-2" aria-hidden="true" tabindex="-1"></a>prob_novo <span class="ot">&lt;-</span> prob_novo <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(Uploaded_variation) <span class="sc">%&gt;%</span> </span>
<span id="cb27-3"><a href="variant-annotation.html#cb27-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(CANONICAL), <span class="fu">as.character</span>(Feature),<span class="at">.by_group =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb27-4"><a href="variant-annotation.html#cb27-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Feature <span class="sc">==</span> Feature[<span class="dv">1</span>])</span>
<span id="cb27-5"><a href="variant-annotation.html#cb27-5" aria-hidden="true" tabindex="-1"></a>prob_novo_high_qual <span class="ot">&lt;-</span> refseq <span class="sc">%&gt;%</span> <span class="fu">inner_join</span>(geno_novo_not_ref_high_qual)</span>
<span id="cb27-6"><a href="variant-annotation.html#cb27-6" aria-hidden="true" tabindex="-1"></a>prob_novo_high_qual <span class="ot">&lt;-</span> prob_novo_high_qual <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(Uploaded_variation) <span class="sc">%&gt;%</span> </span>
<span id="cb27-7"><a href="variant-annotation.html#cb27-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(CANONICAL), <span class="fu">as.character</span>(Feature),<span class="at">.by_group =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb27-8"><a href="variant-annotation.html#cb27-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Feature <span class="sc">==</span> Feature[<span class="dv">1</span>])</span>
<span id="cb27-9"><a href="variant-annotation.html#cb27-9" aria-hidden="true" tabindex="-1"></a>prob_novo_high_qual</span></code></pre></div>
</div>
<div id="homozygous-variants" class="section level2 hasAnchor" number="5.3">
<h2><span class="header-section-number">5.3</span> Homozygous Variants<a href="variant-annotation.html#homozygous-variants" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Select the homozygous variants in the proband.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="variant-annotation.html#cb28-1" aria-hidden="true" tabindex="-1"></a>geno_hom <span class="ot">&lt;-</span> wh_geno_auto <span class="sc">%&gt;%</span> </span>
<span id="cb28-2"><a href="variant-annotation.html#cb28-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)<span class="sc">==</span><span class="fu">substr</span>(proband,<span class="dv">3</span>,<span class="dv">3</span>) <span class="sc">&amp;</span> <span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>) <span class="sc">!=</span> <span class="dv">0</span> <span class="sc">&amp;</span> <span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>) <span class="sc">!=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb28-3"><a href="variant-annotation.html#cb28-3" aria-hidden="true" tabindex="-1"></a>geno_hom[<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>)]</span></code></pre></div>
<p>Discard the homozygous variants for which one of the parents is also homozygous (e.g. 1/1, 0/1 -&gt; 1/1) since these variants are unlikely to be pathogenic.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="variant-annotation.html#cb29-1" aria-hidden="true" tabindex="-1"></a>geno_hom2 <span class="ot">&lt;-</span> geno_hom <span class="sc">%&gt;%</span></span>
<span id="cb29-2"><a href="variant-annotation.html#cb29-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span> ( (<span class="fu">substr</span>(mother,<span class="dv">1</span>,<span class="dv">1</span>)<span class="sc">==</span><span class="fu">substr</span>(mother,<span class="dv">3</span>,<span class="dv">3</span>) <span class="sc">&amp;</span> <span class="fu">substr</span>(mother,<span class="dv">1</span>,<span class="dv">1</span>)<span class="sc">==</span><span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)) <span class="sc">|</span></span>
<span id="cb29-3"><a href="variant-annotation.html#cb29-3" aria-hidden="true" tabindex="-1"></a>                      (<span class="fu">substr</span>(father,<span class="dv">1</span>,<span class="dv">1</span>)<span class="sc">==</span><span class="fu">substr</span>(father,<span class="dv">3</span>,<span class="dv">3</span>) <span class="sc">&amp;</span> <span class="fu">substr</span>(father,<span class="dv">1</span>,<span class="dv">1</span>)<span class="sc">==</span><span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)) )</span>
<span id="cb29-4"><a href="variant-annotation.html#cb29-4" aria-hidden="true" tabindex="-1"></a>                  )</span>
<span id="cb29-5"><a href="variant-annotation.html#cb29-5" aria-hidden="true" tabindex="-1"></a>geno_hom2[<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>)]</span></code></pre></div>
<p>Select the homozygous variants found in the filtered VEP annotation. For each variant, select a single gene transcript that contains the variant.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="variant-annotation.html#cb30-1" aria-hidden="true" tabindex="-1"></a>prob_hom <span class="ot">&lt;-</span> refseq <span class="sc">%&gt;%</span> <span class="fu">inner_join</span>(geno_hom2)</span>
<span id="cb30-2"><a href="variant-annotation.html#cb30-2" aria-hidden="true" tabindex="-1"></a>prob_hom <span class="ot">&lt;-</span> prob_hom <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(Uploaded_variation) <span class="sc">%&gt;%</span> </span>
<span id="cb30-3"><a href="variant-annotation.html#cb30-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(CANONICAL), <span class="fu">as.character</span>(Feature),<span class="at">.by_group =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb30-4"><a href="variant-annotation.html#cb30-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Feature <span class="sc">==</span> Feature[<span class="dv">1</span>])</span>
<span id="cb30-5"><a href="variant-annotation.html#cb30-5" aria-hidden="true" tabindex="-1"></a>prob_hom</span></code></pre></div>
</div>
<div id="compound-heterozygous-variants" class="section level2 hasAnchor" number="5.4">
<h2><span class="header-section-number">5.4</span> Compound Heterozygous Variants<a href="variant-annotation.html#compound-heterozygous-variants" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Select the unphased compound heterozygous variants.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="variant-annotation.html#cb31-1" aria-hidden="true" tabindex="-1"></a>prob_2plus_het <span class="ot">&lt;-</span> refseq <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)<span class="sc">!=</span><span class="fu">substr</span>(proband,<span class="dv">3</span>,<span class="dv">3</span>)) </span>
<span id="cb31-2"><a href="variant-annotation.html#cb31-2" aria-hidden="true" tabindex="-1"></a>prob_2plus_het <span class="ot">&lt;-</span> prob_2plus_het <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(Feature) <span class="sc">%&gt;%</span> </span>
<span id="cb31-3"><a href="variant-annotation.html#cb31-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">length</span>(<span class="fu">unique</span>(PS_proband))<span class="sc">&gt;=</span><span class="dv">2</span> <span class="sc">|</span> <span class="fu">length</span>(<span class="fu">is.na</span>(PS_proband)[<span class="fu">is.na</span>(PS_proband)<span class="sc">==</span><span class="cn">TRUE</span>])<span class="sc">&gt;=</span><span class="dv">2</span> ) <span class="sc">%&gt;%</span> </span>
<span id="cb31-4"><a href="variant-annotation.html#cb31-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb31-5"><a href="variant-annotation.html#cb31-5" aria-hidden="true" tabindex="-1"></a>prob_2plus_het2 <span class="ot">&lt;-</span> prob_2plus_het <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(Gene) <span class="sc">%&gt;%</span> </span>
<span id="cb31-6"><a href="variant-annotation.html#cb31-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(CANONICAL), <span class="fu">as.character</span>(Feature),<span class="at">.by_group =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb31-7"><a href="variant-annotation.html#cb31-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Feature <span class="sc">==</span> Feature[<span class="dv">1</span>])</span>
<span id="cb31-8"><a href="variant-annotation.html#cb31-8" aria-hidden="true" tabindex="-1"></a>prob_2plus_het2</span></code></pre></div>
<p>Select the phased compound heterozygous variants.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="variant-annotation.html#cb32-1" aria-hidden="true" tabindex="-1"></a>prob_comp_het <span class="ot">&lt;-</span> refseq <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(Feature, PS_proband) <span class="sc">%&gt;%</span> </span>
<span id="cb32-2"><a href="variant-annotation.html#cb32-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">any</span>(<span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)<span class="sc">&gt;=</span><span class="dv">1</span> <span class="sc">&amp;</span> <span class="fu">substr</span>(proband,<span class="dv">2</span>,<span class="dv">2</span>)<span class="sc">==</span><span class="st">&quot;|&quot;</span>) <span class="sc">&amp;</span> <span class="fu">any</span>(<span class="fu">substr</span>(proband,<span class="dv">3</span>,<span class="dv">3</span>)<span class="sc">&gt;=</span><span class="dv">1</span> <span class="sc">&amp;</span> <span class="fu">substr</span>(proband,<span class="dv">2</span>,<span class="dv">2</span>)<span class="sc">==</span><span class="st">&quot;|&quot;</span>)) </span>
<span id="cb32-3"><a href="variant-annotation.html#cb32-3" aria-hidden="true" tabindex="-1"></a>prob_comp_het2 <span class="ot">&lt;-</span> prob_comp_het <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(Gene) <span class="sc">%&gt;%</span> </span>
<span id="cb32-4"><a href="variant-annotation.html#cb32-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(CANONICAL), <span class="fu">as.character</span>(Feature),<span class="at">.by_group =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb32-5"><a href="variant-annotation.html#cb32-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Feature <span class="sc">==</span> Feature[<span class="dv">1</span>])</span>
<span id="cb32-6"><a href="variant-annotation.html#cb32-6" aria-hidden="true" tabindex="-1"></a>prob_comp_het2</span></code></pre></div>
</div>
<div id="x-chromosome" class="section level2 hasAnchor" number="5.5">
<h2><span class="header-section-number">5.5</span> X Chromosome<a href="variant-annotation.html#x-chromosome" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Remove X chromosome calls that are “heterozygous” for male individuals.</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="variant-annotation.html#cb33-1" aria-hidden="true" tabindex="-1"></a>wh_geno_X_filt <span class="ot">&lt;-</span> wh_geno_X <span class="sc">%&gt;%</span> </span>
<span id="cb33-2"><a href="variant-annotation.html#cb33-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(( <span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)<span class="sc">==</span><span class="fu">substr</span>(proband,<span class="dv">3</span>,<span class="dv">3</span>) <span class="sc">&amp;</span> <span class="fu">substr</span>(father,<span class="dv">1</span>,<span class="dv">1</span>)<span class="sc">==</span><span class="fu">substr</span>(father,<span class="dv">3</span>,<span class="dv">3</span>) ))</span>
<span id="cb33-3"><a href="variant-annotation.html#cb33-3" aria-hidden="true" tabindex="-1"></a>prob_X <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(refseq_X, wh_geno_X) <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(Gene) <span class="sc">%&gt;%</span> </span>
<span id="cb33-4"><a href="variant-annotation.html#cb33-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(CANONICAL), <span class="fu">as.character</span>(Feature),<span class="at">.by_group =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb33-5"><a href="variant-annotation.html#cb33-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Feature <span class="sc">==</span> Feature[<span class="dv">1</span>])</span>
<span id="cb33-6"><a href="variant-annotation.html#cb33-6" aria-hidden="true" tabindex="-1"></a>prob_X</span></code></pre></div>
</div>
<div id="disease-genes" class="section level2 hasAnchor" number="5.6">
<h2><span class="header-section-number">5.6</span> Disease Genes<a href="variant-annotation.html#disease-genes" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Select one entry for each variant in refseq table, prioritizing canonical and curated transcripts.</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="variant-annotation.html#cb34-1" aria-hidden="true" tabindex="-1"></a>refseq2 <span class="ot">&lt;-</span> refseq01 <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(Uploaded_variation) <span class="sc">%&gt;%</span> </span>
<span id="cb34-2"><a href="variant-annotation.html#cb34-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(CANONICAL), <span class="fu">as.character</span>(Feature),<span class="at">.by_group =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb34-3"><a href="variant-annotation.html#cb34-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Feature <span class="sc">==</span> Feature[<span class="dv">1</span>])</span>
<span id="cb34-4"><a href="variant-annotation.html#cb34-4" aria-hidden="true" tabindex="-1"></a>refseq2</span></code></pre></div>
<p>Select variants that fall in green genes from the UK PanelApp renal broad superpanel.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="variant-annotation.html#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="co"># UK Panel App</span></span>
<span id="cb35-2"><a href="variant-annotation.html#cb35-2" aria-hidden="true" tabindex="-1"></a>panel <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">&quot;renal_superpanel_broad.tsv&quot;</span>)</span>
<span id="cb35-3"><a href="variant-annotation.html#cb35-3" aria-hidden="true" tabindex="-1"></a>panel_green <span class="ot">&lt;-</span> panel <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(GEL_Status<span class="sc">==</span><span class="dv">3</span>)</span>
<span id="cb35-4"><a href="variant-annotation.html#cb35-4" aria-hidden="true" tabindex="-1"></a>refseq_green <span class="ot">&lt;-</span> refseq2 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(HGNC_ID <span class="sc">%in%</span> panel_green<span class="sc">$</span>HGNC)</span>
<span id="cb35-5"><a href="variant-annotation.html#cb35-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-6"><a href="variant-annotation.html#cb35-6" aria-hidden="true" tabindex="-1"></a><span class="co">#Australia Panel App</span></span>
<span id="cb35-7"><a href="variant-annotation.html#cb35-7" aria-hidden="true" tabindex="-1"></a>panel_au <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">&quot;Kidneyome_SuperPanel.tsv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb35-8"><a href="variant-annotation.html#cb35-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">duplicated</span>(HGNC)) <span class="sc">%&gt;%</span></span>
<span id="cb35-9"><a href="variant-annotation.html#cb35-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">Status=</span><span class="fu">case_when</span>(</span>
<span id="cb35-10"><a href="variant-annotation.html#cb35-10" aria-hidden="true" tabindex="-1"></a>    GEL_Status<span class="sc">==</span><span class="dv">4</span><span class="sc">|</span>GEL_Status<span class="sc">==</span><span class="dv">3</span> <span class="sc">~</span> <span class="st">&quot;Green&quot;</span>, </span>
<span id="cb35-11"><a href="variant-annotation.html#cb35-11" aria-hidden="true" tabindex="-1"></a>    GEL_Status<span class="sc">==</span><span class="dv">2</span> <span class="sc">~</span> <span class="st">&quot;Amber&quot;</span>, </span>
<span id="cb35-12"><a href="variant-annotation.html#cb35-12" aria-hidden="true" tabindex="-1"></a>    GEL_Status<span class="sc">==</span><span class="dv">1</span><span class="sc">|</span>GEL_Status<span class="sc">==</span><span class="dv">0</span> <span class="sc">~</span> <span class="st">&quot;Red&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb35-13"><a href="variant-annotation.html#cb35-13" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(HGNC,Status)</span>
<span id="cb35-14"><a href="variant-annotation.html#cb35-14" aria-hidden="true" tabindex="-1"></a>refseq_au <span class="ot">&lt;-</span> refseq2 <span class="sc">%&gt;%</span> <span class="fu">inner_join</span>(panel_au, <span class="fu">join_by</span>(HGNC_ID<span class="sc">==</span>HGNC))</span>
<span id="cb35-15"><a href="variant-annotation.html#cb35-15" aria-hidden="true" tabindex="-1"></a>refseq_au</span></code></pre></div>
<p>Combine annotations.</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="variant-annotation.html#cb36-1" aria-hidden="true" tabindex="-1"></a>refseq3 <span class="ot">&lt;-</span> refseq01 <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(Gene) <span class="sc">%&gt;%</span> </span>
<span id="cb36-2"><a href="variant-annotation.html#cb36-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(CANONICAL), <span class="fu">as.character</span>(Feature),<span class="at">.by_group =</span> T) <span class="sc">%&gt;%</span> <span class="fu">ungroup</span>()</span>
<span id="cb36-3"><a href="variant-annotation.html#cb36-3" aria-hidden="true" tabindex="-1"></a>refseq4 <span class="ot">&lt;-</span> refseq3 <span class="sc">%&gt;%</span> </span>
<span id="cb36-4"><a href="variant-annotation.html#cb36-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(<span class="fu">add_column</span>(geno_novo_not_ref,<span class="at">de_novo=</span><span class="st">&quot;de_novo&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb36-5"><a href="variant-annotation.html#cb36-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(<span class="fu">add_column</span>(geno_novo_not_ref_high_qual,<span class="at">de_novo_gq=</span><span class="st">&quot;de_novo_gq&gt;=30&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb36-6"><a href="variant-annotation.html#cb36-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(<span class="fu">add_column</span>(geno_hom2, <span class="at">homozygote=</span><span class="st">&quot;homozygote&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb36-7"><a href="variant-annotation.html#cb36-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(<span class="fu">add_column</span>(prob_2plus_het, <span class="at">unphased_compound_heterozygote=</span><span class="st">&quot;unphased_compound_heterozygote&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb36-8"><a href="variant-annotation.html#cb36-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(<span class="fu">add_column</span>(prob_comp_het, <span class="at">phased_compound_heterozygote=</span><span class="st">&quot;phased_compound_heterozygote&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb36-9"><a href="variant-annotation.html#cb36-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">zygosity=</span><span class="fu">coalesce</span>(de_novo_gq,de_novo,homozygote,phased_compound_heterozygote,unphased_compound_heterozygote)) <span class="sc">%&gt;%</span></span>
<span id="cb36-10"><a href="variant-annotation.html#cb36-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(de_novo_gq,de_novo,homozygote,phased_compound_heterozygote,unphased_compound_heterozygote)) <span class="sc">%&gt;%</span></span>
<span id="cb36-11"><a href="variant-annotation.html#cb36-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">relocate</span>(zygosity,<span class="at">.after=</span>counter)</span>
<span id="cb36-12"><a href="variant-annotation.html#cb36-12" aria-hidden="true" tabindex="-1"></a>refseq_sum <span class="ot">&lt;-</span> refseq4 <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(Feature) <span class="sc">%&gt;%</span> </span>
<span id="cb36-13"><a href="variant-annotation.html#cb36-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">z=</span><span class="fu">paste</span>(<span class="fu">c</span>(Uploaded_variation,Location,Allele,Consequence,Amino_acids,Codons), <span class="at">collapse =</span> <span class="st">&quot; &quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-14"><a href="variant-annotation.html#cb36-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(z,<span class="at">.keep_all =</span> T)</span>
<span id="cb36-15"><a href="variant-annotation.html#cb36-15" aria-hidden="true" tabindex="-1"></a>refseq5 <span class="ot">&lt;-</span> refseq4 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(Feature <span class="sc">%in%</span> refseq_sum<span class="sc">$</span>Feature)</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="statistical-genetics.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="summary.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
