<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>5 Variant Annotation | Whole Exome Sequencing Pipeline</title>
  <meta name="description" content="This is a pipeline for whole exome sequencing analysis." />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="5 Variant Annotation | Whole Exome Sequencing Pipeline" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a pipeline for whole exome sequencing analysis." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="5 Variant Annotation | Whole Exome Sequencing Pipeline" />
  
  <meta name="twitter:description" content="This is a pipeline for whole exome sequencing analysis." />
  

<meta name="author" content="Marc Henein and Zachary Sentell" />


<meta name="date" content="2024-05-17" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="statistical-genetics.html"/>
<link rel="next" href="summary.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="libs/pagedtable-1.1/js/pagedtable.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Whole Exome Sequencing Pipeline</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Overview</a></li>
<li class="chapter" data-level="2" data-path="align.html"><a href="align.html"><i class="fa fa-check"></i><b>2</b> Pre-processing</a>
<ul>
<li class="chapter" data-level="2.1" data-path="align.html"><a href="align.html#quality-control-of-raw-reads"><i class="fa fa-check"></i><b>2.1</b> Quality Control of Raw Reads</a></li>
<li class="chapter" data-level="2.2" data-path="align.html"><a href="align.html#alignment"><i class="fa fa-check"></i><b>2.2</b> Alignment</a></li>
<li class="chapter" data-level="2.3" data-path="align.html"><a href="align.html#mark-duplicates"><i class="fa fa-check"></i><b>2.3</b> Mark Duplicates</a></li>
<li class="chapter" data-level="2.4" data-path="align.html"><a href="align.html#base-quality-score-recalibration"><i class="fa fa-check"></i><b>2.4</b> Base Quality Score Recalibration</a></li>
<li class="chapter" data-level="2.5" data-path="align.html"><a href="align.html#quality-control-of-alignment"><i class="fa fa-check"></i><b>2.5</b> Quality Control of Alignment</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="calling.html"><a href="calling.html"><i class="fa fa-check"></i><b>3</b> Variant Calling</a>
<ul>
<li class="chapter" data-level="3.1" data-path="calling.html"><a href="calling.html#snvindel-calling"><i class="fa fa-check"></i><b>3.1</b> SNV/Indel Calling</a></li>
<li class="chapter" data-level="3.2" data-path="calling.html"><a href="calling.html#variant-filtration"><i class="fa fa-check"></i><b>3.2</b> Variant Filtration</a></li>
<li class="chapter" data-level="3.3" data-path="calling.html"><a href="calling.html#genotype-posteriors"><i class="fa fa-check"></i><b>3.3</b> Genotype Posteriors</a></li>
<li class="chapter" data-level="3.4" data-path="calling.html"><a href="calling.html#structural-variants"><i class="fa fa-check"></i><b>3.4</b> Structural Variants</a></li>
<li class="chapter" data-level="3.5" data-path="calling.html"><a href="calling.html#benchmarking"><i class="fa fa-check"></i><b>3.5</b> Benchmarking</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="statistical-genetics.html"><a href="statistical-genetics.html"><i class="fa fa-check"></i><b>4</b> Statistical Genetics</a>
<ul>
<li class="chapter" data-level="4.1" data-path="statistical-genetics.html"><a href="statistical-genetics.html#phasing"><i class="fa fa-check"></i><b>4.1</b> Phasing</a></li>
<li class="chapter" data-level="4.2" data-path="statistical-genetics.html"><a href="statistical-genetics.html#kinship"><i class="fa fa-check"></i><b>4.2</b> Kinship</a></li>
<li class="chapter" data-level="4.3" data-path="statistical-genetics.html"><a href="statistical-genetics.html#runs-of-homozygosity"><i class="fa fa-check"></i><b>4.3</b> Runs of Homozygosity</a></li>
<li class="chapter" data-level="4.4" data-path="statistical-genetics.html"><a href="statistical-genetics.html#linkage-analysis"><i class="fa fa-check"></i><b>4.4</b> Linkage Analysis</a></li>
<li class="chapter" data-level="4.5" data-path="statistical-genetics.html"><a href="statistical-genetics.html#population-substructure"><i class="fa fa-check"></i><b>4.5</b> Population Substructure</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="variant-annotation.html"><a href="variant-annotation.html"><i class="fa fa-check"></i><b>5</b> Variant Annotation</a>
<ul>
<li class="chapter" data-level="5.1" data-path="variant-annotation.html"><a href="variant-annotation.html#variant-effect-predictor"><i class="fa fa-check"></i><b>5.1</b> Variant Effect Predictor</a></li>
<li class="chapter" data-level="5.2" data-path="variant-annotation.html"><a href="variant-annotation.html#filtration"><i class="fa fa-check"></i><b>5.2</b> Filtration</a></li>
<li class="chapter" data-level="5.3" data-path="variant-annotation.html"><a href="variant-annotation.html#de-novo-variants"><i class="fa fa-check"></i><b>5.3</b> <em>De Novo</em> Variants</a></li>
<li class="chapter" data-level="5.4" data-path="variant-annotation.html"><a href="variant-annotation.html#homozygous-variants"><i class="fa fa-check"></i><b>5.4</b> Homozygous Variants</a></li>
<li class="chapter" data-level="5.5" data-path="variant-annotation.html"><a href="variant-annotation.html#compound-heterozygous-variants"><i class="fa fa-check"></i><b>5.5</b> Compound Heterozygous Variants</a></li>
<li class="chapter" data-level="5.6" data-path="variant-annotation.html"><a href="variant-annotation.html#x-chromosome"><i class="fa fa-check"></i><b>5.6</b> X Chromosome</a></li>
<li class="chapter" data-level="5.7" data-path="variant-annotation.html"><a href="variant-annotation.html#disease-genes"><i class="fa fa-check"></i><b>5.7</b> Disease Genes</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="summary.html"><a href="summary.html"><i class="fa fa-check"></i><b>6</b> Summary</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Whole Exome Sequencing Pipeline</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="variant-annotation" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">5</span> Variant Annotation<a href="variant-annotation.html#variant-annotation" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="variant-effect-predictor" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> Variant Effect Predictor<a href="variant-annotation.html#variant-effect-predictor" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We annotate the variants with VEP for molecular consequence, allele frequency, clinical variant databases, i.e. Clinvar, Mastermind, and a set of pathogenicity prediction tools, i.e., CADD, PrimateAI, SpliceAI, EVE, MutationTaster, REVEL.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb24-1"><a href="variant-annotation.html#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="ex">apptainer</span> exec <span class="at">-B</span> vep_data:/opt/vep/.vep vep.sif /opt/vep/src/ensembl-vep/vep <span class="dt">\</span></span>
<span id="cb24-2"><a href="variant-annotation.html#cb24-2" aria-hidden="true" tabindex="-1"></a>--dir_cache /opt/vep/.vep/ <span class="dt">\</span></span>
<span id="cb24-3"><a href="variant-annotation.html#cb24-3" aria-hidden="true" tabindex="-1"></a>--dir_plugins /opt/vep/.vep/Plugins/ <span class="dt">\</span></span>
<span id="cb24-4"><a href="variant-annotation.html#cb24-4" aria-hidden="true" tabindex="-1"></a>-i /opt/vep/.vep/output_vqsr_snps_hard_indels.vcf.gz <span class="dt">\</span></span>
<span id="cb24-5"><a href="variant-annotation.html#cb24-5" aria-hidden="true" tabindex="-1"></a>-o /opt/vep/.vep/ensembl_annotated.txt <span class="dt">\</span></span>
<span id="cb24-6"><a href="variant-annotation.html#cb24-6" aria-hidden="true" tabindex="-1"></a>--individual EC27_proband <span class="dt">\</span></span>
<span id="cb24-7"><a href="variant-annotation.html#cb24-7" aria-hidden="true" tabindex="-1"></a>--everything <span class="dt">\</span></span>
<span id="cb24-8"><a href="variant-annotation.html#cb24-8" aria-hidden="true" tabindex="-1"></a>--check_existing <span class="dt">\</span></span>
<span id="cb24-9"><a href="variant-annotation.html#cb24-9" aria-hidden="true" tabindex="-1"></a>--xref_refseq <span class="dt">\</span></span>
<span id="cb24-10"><a href="variant-annotation.html#cb24-10" aria-hidden="true" tabindex="-1"></a>--fasta /opt/vep/.vep/Homo_sapiens.GRCh38.fa <span class="dt">\</span></span>
<span id="cb24-11"><a href="variant-annotation.html#cb24-11" aria-hidden="true" tabindex="-1"></a>--hgvs <span class="dt">\</span></span>
<span id="cb24-12"><a href="variant-annotation.html#cb24-12" aria-hidden="true" tabindex="-1"></a>--hgvsg <span class="dt">\</span></span>
<span id="cb24-13"><a href="variant-annotation.html#cb24-13" aria-hidden="true" tabindex="-1"></a>--plugin CADD,/opt/vep/.vep/Plugins/data/grch38/whole_genome_SNVs_grch38.tsv.gz,/opt/vep/.vep/Plugins/data/grch38/gnomad.genomes.r3.0.indel_grch38.tsv.gz <span class="dt">\</span></span>
<span id="cb24-14"><a href="variant-annotation.html#cb24-14" aria-hidden="true" tabindex="-1"></a>--plugin PrimateAI,/opt/vep/.vep/Plugins/data/grch38/PrimateAI_scores_v0.2_GRCh38_sorted.tsv.bgz <span class="dt">\</span></span>
<span id="cb24-15"><a href="variant-annotation.html#cb24-15" aria-hidden="true" tabindex="-1"></a>--plugin SpliceAI,snv=/opt/vep/.vep/Plugins/data/grch38/spliceai_scores.raw.snv.hg38.vcf.gz,indel=/opt/vep/.vep/Plugins/data/grch38/spliceai_scores.raw.indel.hg38.vcf.gz <span class="dt">\</span></span>
<span id="cb24-16"><a href="variant-annotation.html#cb24-16" aria-hidden="true" tabindex="-1"></a>--plugin EVE,file=/opt/vep/.vep/Plugins/data/grch38/eve_merged.vcf.gz <span class="dt">\</span></span>
<span id="cb24-17"><a href="variant-annotation.html#cb24-17" aria-hidden="true" tabindex="-1"></a>--plugin Mastermind,/opt/vep/.vep/Plugins/data/grch38/mastermind_cited_variants_reference-2023.04.02-grch38.vcf.gz <span class="dt">\</span></span>
<span id="cb24-18"><a href="variant-annotation.html#cb24-18" aria-hidden="true" tabindex="-1"></a>--plugin dbNSFP,/opt/vep/.vep/Plugins/data/grch38/dbNSFP4.3a_grch38.gz,MutationTaster_score,MutationTaster_pred <span class="dt">\</span></span>
<span id="cb24-19"><a href="variant-annotation.html#cb24-19" aria-hidden="true" tabindex="-1"></a>--plugin REVEL,/opt/vep/.vep/Plugins/data/grch38/new_tabbed_revel_grch38.tsv.gz <span class="dt">\</span></span>
<span id="cb24-20"><a href="variant-annotation.html#cb24-20" aria-hidden="true" tabindex="-1"></a>--custom /opt/vep/.vep/Plugins/data/grch38/clinvar.vcf.gz,ClinVar,vcf,exact,0,CLNSIG,CLNREVSTAT,CLNDN <span class="dt">\</span></span>
<span id="cb24-21"><a href="variant-annotation.html#cb24-21" aria-hidden="true" tabindex="-1"></a>--cache <span class="dt">\</span></span>
<span id="cb24-22"><a href="variant-annotation.html#cb24-22" aria-hidden="true" tabindex="-1"></a>--merged <span class="dt">\</span></span>
<span id="cb24-23"><a href="variant-annotation.html#cb24-23" aria-hidden="true" tabindex="-1"></a>--offline <span class="dt">\</span></span>
<span id="cb24-24"><a href="variant-annotation.html#cb24-24" aria-hidden="true" tabindex="-1"></a>--use_given_ref <span class="dt">\</span></span>
<span id="cb24-25"><a href="variant-annotation.html#cb24-25" aria-hidden="true" tabindex="-1"></a>--force_overwrite <span class="dt">\</span></span>
<span id="cb24-26"><a href="variant-annotation.html#cb24-26" aria-hidden="true" tabindex="-1"></a>--assembly GRCh38 <span class="dt">\</span></span>
<span id="cb24-27"><a href="variant-annotation.html#cb24-27" aria-hidden="true" tabindex="-1"></a>--fork 2 <span class="dt">\</span></span>
<span id="cb24-28"><a href="variant-annotation.html#cb24-28" aria-hidden="true" tabindex="-1"></a>--buffer_size 1000 <span class="dt">\</span></span>
<span id="cb24-29"><a href="variant-annotation.html#cb24-29" aria-hidden="true" tabindex="-1"></a>--tab</span></code></pre></div>
</div>
<div id="filtration" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">5.2</span> Filtration<a href="variant-annotation.html#filtration" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We perform a first round of filtering with VEP to reduce the size of the VCF to be loaded into R. In particular, we select variants with high or moderate impact and a gnomAD exome allele frequency smaller than 0.05.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb25-1"><a href="variant-annotation.html#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="ex">apptainer</span> exec <span class="at">-B</span> vep_data:/opt/vep/.vep vep.sif /opt/vep/src/ensembl-vep/filter_vep <span class="dt">\</span></span>
<span id="cb25-2"><a href="variant-annotation.html#cb25-2" aria-hidden="true" tabindex="-1"></a>-i /opt/vep/.vep/ensembl_annotated.txt <span class="dt">\</span></span>
<span id="cb25-3"><a href="variant-annotation.html#cb25-3" aria-hidden="true" tabindex="-1"></a>-out /opt/vep/.vep/ensembl_ann_vep_filt.txt <span class="dt">\</span></span>
<span id="cb25-4"><a href="variant-annotation.html#cb25-4" aria-hidden="true" tabindex="-1"></a>--filter <span class="st">&quot;(IMPACT is HIGH or IMPACT is MODERATE) and (gnomADe_AF &lt;= 0.05 or not gnomADe_AF)&quot;</span> <span class="dt">\</span></span>
<span id="cb25-5"><a href="variant-annotation.html#cb25-5" aria-hidden="true" tabindex="-1"></a>--force_overwrite <span class="dt">\</span></span>
<span id="cb25-6"><a href="variant-annotation.html#cb25-6" aria-hidden="true" tabindex="-1"></a>--format tab </span></code></pre></div>
<p>We load the phased variants into R and select autosomal variants that have passed all variant calling filters.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="variant-annotation.html#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb26-2"><a href="variant-annotation.html#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(VariantAnnotation)</span>
<span id="cb26-3"><a href="variant-annotation.html#cb26-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Set sample names</span></span>
<span id="cb26-4"><a href="variant-annotation.html#cb26-4" aria-hidden="true" tabindex="-1"></a>mother <span class="ot">=</span> <span class="st">&quot;HG004&quot;</span></span>
<span id="cb26-5"><a href="variant-annotation.html#cb26-5" aria-hidden="true" tabindex="-1"></a>father <span class="ot">=</span> <span class="st">&quot;HG003&quot;</span></span>
<span id="cb26-6"><a href="variant-annotation.html#cb26-6" aria-hidden="true" tabindex="-1"></a>proband <span class="ot">=</span> <span class="st">&quot;HG002&quot;</span></span>
<span id="cb26-7"><a href="variant-annotation.html#cb26-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Load WhatsHap phased variants</span></span>
<span id="cb26-8"><a href="variant-annotation.html#cb26-8" aria-hidden="true" tabindex="-1"></a>wh <span class="ot">&lt;-</span> <span class="fu">readVcf</span>(<span class="st">&quot;data/AJ_phased_whatshap_refined.vcf&quot;</span>, <span class="st">&quot;hg38&quot;</span>)</span>
<span id="cb26-9"><a href="variant-annotation.html#cb26-9" aria-hidden="true" tabindex="-1"></a>wh <span class="ot">&lt;-</span> <span class="fu">expand</span>(wh)</span>
<span id="cb26-10"><a href="variant-annotation.html#cb26-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Variant info</span></span>
<span id="cb26-11"><a href="variant-annotation.html#cb26-11" aria-hidden="true" tabindex="-1"></a>pos <span class="ot">&lt;-</span> <span class="fu">rowRanges</span>(wh) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>()</span>
<span id="cb26-12"><a href="variant-annotation.html#cb26-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Genotype</span></span>
<span id="cb26-13"><a href="variant-annotation.html#cb26-13" aria-hidden="true" tabindex="-1"></a>wh_geno <span class="ot">&lt;-</span> <span class="fu">geno</span>(wh)<span class="sc">$</span>GT <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb26-14"><a href="variant-annotation.html#cb26-14" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;mother&quot;</span><span class="ot">=</span> <span class="fu">sym</span>(mother), <span class="st">&quot;father&quot;</span><span class="ot">=</span> <span class="fu">sym</span>(father), <span class="st">&quot;proband&quot;</span><span class="ot">=</span> <span class="fu">sym</span>(proband))</span>
<span id="cb26-15"><a href="variant-annotation.html#cb26-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Phasing block</span></span>
<span id="cb26-16"><a href="variant-annotation.html#cb26-16" aria-hidden="true" tabindex="-1"></a>wh_ps <span class="ot">&lt;-</span> <span class="fu">geno</span>(wh)<span class="sc">$</span>PS <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb26-17"><a href="variant-annotation.html#cb26-17" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;PS_mother&quot;</span> <span class="ot">=</span> <span class="fu">sym</span>(mother), <span class="st">&quot;PS_father&quot;</span><span class="ot">=</span> <span class="fu">sym</span>(father), <span class="st">&quot;PS_proband&quot;</span><span class="ot">=</span> <span class="fu">sym</span>(proband))</span>
<span id="cb26-18"><a href="variant-annotation.html#cb26-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Genotype quality</span></span>
<span id="cb26-19"><a href="variant-annotation.html#cb26-19" aria-hidden="true" tabindex="-1"></a>wh_gq <span class="ot">&lt;-</span> <span class="fu">geno</span>(wh)<span class="sc">$</span>GQ <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb26-20"><a href="variant-annotation.html#cb26-20" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;GQ_mother&quot;</span> <span class="ot">=</span> <span class="fu">sym</span>(mother), <span class="st">&quot;GQ_father&quot;</span><span class="ot">=</span> <span class="fu">sym</span>(father), <span class="st">&quot;GQ_proband&quot;</span><span class="ot">=</span> <span class="fu">sym</span>(proband))</span>
<span id="cb26-21"><a href="variant-annotation.html#cb26-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Allelic depth</span></span>
<span id="cb26-22"><a href="variant-annotation.html#cb26-22" aria-hidden="true" tabindex="-1"></a>wh_ad <span class="ot">&lt;-</span> <span class="fu">geno</span>(wh)<span class="sc">$</span>AD <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb26-23"><a href="variant-annotation.html#cb26-23" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;AD_ref_mother&quot;</span> <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">sym</span>(mother),<span class="st">&quot;.1&quot;</span>), <span class="st">&quot;AD_ref_father&quot;</span><span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">sym</span>(father),<span class="st">&quot;.1&quot;</span>), <span class="st">&quot;AD_ref_proband&quot;</span><span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">sym</span>(proband),<span class="st">&quot;.1&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb26-24"><a href="variant-annotation.html#cb26-24" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;AD_alt_mother&quot;</span> <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">sym</span>(mother),<span class="st">&quot;.2&quot;</span>), <span class="st">&quot;AD_alt_father&quot;</span><span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">sym</span>(father),<span class="st">&quot;.2&quot;</span>), <span class="st">&quot;AD_alt_proband&quot;</span><span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">sym</span>(proband),<span class="st">&quot;.2&quot;</span>))</span>
<span id="cb26-25"><a href="variant-annotation.html#cb26-25" aria-hidden="true" tabindex="-1"></a><span class="co"># Join variant info, genotype, phasing block, genotype quality and allelic depth</span></span>
<span id="cb26-26"><a href="variant-annotation.html#cb26-26" aria-hidden="true" tabindex="-1"></a>wh_geno0 <span class="ot">&lt;-</span> <span class="fu">bind_cols</span>(pos,wh_geno,wh_gq,wh_ps,wh_ad)</span>
<span id="cb26-27"><a href="variant-annotation.html#cb26-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify alternate alleles with number</span></span>
<span id="cb26-28"><a href="variant-annotation.html#cb26-28" aria-hidden="true" tabindex="-1"></a>wh_geno01 <span class="ot">&lt;-</span> wh_geno0 <span class="sc">%&gt;%</span> </span>
<span id="cb26-29"><a href="variant-annotation.html#cb26-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(seqnames,start,end,width,REF) <span class="sc">%&gt;%</span> </span>
<span id="cb26-30"><a href="variant-annotation.html#cb26-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">counter =</span> <span class="fu">row_number</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb26-31"><a href="variant-annotation.html#cb26-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb26-32"><a href="variant-annotation.html#cb26-32" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(FILTER<span class="sc">==</span><span class="st">&quot;PASS&quot;</span>)</span>
<span id="cb26-33"><a href="variant-annotation.html#cb26-33" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert variant to VEP format. Add 1 to indel start site.</span></span>
<span id="cb26-34"><a href="variant-annotation.html#cb26-34" aria-hidden="true" tabindex="-1"></a>wh_geno01 <span class="ot">&lt;-</span> wh_geno01 <span class="sc">%&gt;%</span> </span>
<span id="cb26-35"><a href="variant-annotation.html#cb26-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Location =</span> <span class="fu">ifelse</span>(</span>
<span id="cb26-36"><a href="variant-annotation.html#cb26-36" aria-hidden="true" tabindex="-1"></a>    width<span class="sc">==</span><span class="dv">1</span>, <span class="fu">ifelse</span>(</span>
<span id="cb26-37"><a href="variant-annotation.html#cb26-37" aria-hidden="true" tabindex="-1"></a>      <span class="fu">nchar</span>(ALT)<span class="sc">==</span><span class="dv">1</span>, <span class="fu">paste</span>(seqnames,start,<span class="at">sep=</span><span class="st">&quot;:&quot;</span>), <span class="fu">paste0</span>(seqnames,<span class="st">&quot;:&quot;</span>,start,<span class="st">&quot;-&quot;</span>,start<span class="sc">+</span><span class="dv">1</span>)), <span class="fu">ifelse</span>(</span>
<span id="cb26-38"><a href="variant-annotation.html#cb26-38" aria-hidden="true" tabindex="-1"></a>        start<span class="sc">+</span><span class="dv">1</span><span class="sc">==</span>end, <span class="fu">paste0</span>(seqnames,<span class="st">&quot;:&quot;</span>,start<span class="sc">+</span><span class="dv">1</span>), <span class="fu">paste0</span>(seqnames,<span class="st">&quot;:&quot;</span>,start<span class="sc">+</span><span class="dv">1</span>,<span class="st">&quot;-&quot;</span>,end)))) <span class="sc">%&gt;%</span> </span>
<span id="cb26-39"><a href="variant-annotation.html#cb26-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Allele =</span> <span class="fu">ifelse</span>(<span class="fu">substr</span>(REF,<span class="dv">1</span>,<span class="dv">1</span>)<span class="sc">==</span><span class="fu">substr</span>(ALT,<span class="dv">1</span>,<span class="dv">1</span>), <span class="fu">substr</span>(ALT,<span class="dv">2</span>,<span class="fu">nchar</span>(ALT)), ALT))</span>
<span id="cb26-40"><a href="variant-annotation.html#cb26-40" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert variant to gnomAD format.</span></span>
<span id="cb26-41"><a href="variant-annotation.html#cb26-41" aria-hidden="true" tabindex="-1"></a>trim <span class="ot">&lt;-</span> <span class="cf">function</span>(a1,a2) {</span>
<span id="cb26-42"><a href="variant-annotation.html#cb26-42" aria-hidden="true" tabindex="-1"></a>  b1 <span class="ot">&lt;-</span> <span class="fu">str_sub</span>(a1,<span class="dv">2</span>,<span class="sc">-</span><span class="dv">1</span>)</span>
<span id="cb26-43"><a href="variant-annotation.html#cb26-43" aria-hidden="true" tabindex="-1"></a>  b2 <span class="ot">&lt;-</span> <span class="fu">str_sub</span>(a2,<span class="dv">2</span>,<span class="sc">-</span><span class="dv">1</span>)</span>
<span id="cb26-44"><a href="variant-annotation.html#cb26-44" aria-hidden="true" tabindex="-1"></a>  <span class="cf">repeat</span> {</span>
<span id="cb26-45"><a href="variant-annotation.html#cb26-45" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">nchar</span>(b1)<span class="sc">==</span><span class="dv">0</span> <span class="sc">|</span> <span class="fu">nchar</span>(b2)<span class="sc">==</span><span class="dv">0</span>) <span class="cf">break</span></span>
<span id="cb26-46"><a href="variant-annotation.html#cb26-46" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">str_sub</span>(b1,<span class="sc">-</span><span class="dv">1</span>,<span class="sc">-</span><span class="dv">1</span>)<span class="sc">!=</span><span class="fu">str_sub</span>(b2,<span class="sc">-</span><span class="dv">1</span>,<span class="sc">-</span><span class="dv">1</span>)) <span class="cf">break</span></span>
<span id="cb26-47"><a href="variant-annotation.html#cb26-47" aria-hidden="true" tabindex="-1"></a>    b1 <span class="ot">&lt;-</span> <span class="fu">str_sub</span>(b1,<span class="dv">1</span>,<span class="sc">-</span><span class="dv">2</span>)</span>
<span id="cb26-48"><a href="variant-annotation.html#cb26-48" aria-hidden="true" tabindex="-1"></a>    b2 <span class="ot">&lt;-</span> <span class="fu">str_sub</span>(b2,<span class="dv">1</span>,<span class="sc">-</span><span class="dv">2</span>)</span>
<span id="cb26-49"><a href="variant-annotation.html#cb26-49" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb26-50"><a href="variant-annotation.html#cb26-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">paste</span>(<span class="fu">str_sub</span>(a1,<span class="dv">1</span>,<span class="dv">1</span>),b1,<span class="st">&quot;-&quot;</span>,<span class="fu">str_sub</span>(a2,<span class="dv">1</span>,<span class="dv">1</span>),b2,<span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb26-51"><a href="variant-annotation.html#cb26-51" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb26-52"><a href="variant-annotation.html#cb26-52" aria-hidden="true" tabindex="-1"></a>wh_geno01 <span class="ot">&lt;-</span> wh_geno01 <span class="sc">%&gt;%</span> rowwise <span class="sc">%&gt;%</span> </span>
<span id="cb26-53"><a href="variant-annotation.html#cb26-53" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">gnomADid =</span> <span class="fu">paste</span>(<span class="fu">substr</span>(seqnames,<span class="dv">4</span>,<span class="dv">5</span>),start,<span class="fu">trim</span>(REF,ALT),<span class="at">sep=</span><span class="st">&quot;-&quot;</span>))</span>
<span id="cb26-54"><a href="variant-annotation.html#cb26-54" aria-hidden="true" tabindex="-1"></a>wh_geno01<span class="sc">$</span>Allele[wh_geno01<span class="sc">$</span>Allele<span class="sc">==</span><span class="st">&quot;&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;-&quot;</span></span>
<span id="cb26-55"><a href="variant-annotation.html#cb26-55" aria-hidden="true" tabindex="-1"></a><span class="co"># Overlapping deletions</span></span>
<span id="cb26-56"><a href="variant-annotation.html#cb26-56" aria-hidden="true" tabindex="-1"></a>wh_geno_overlap <span class="ot">&lt;-</span> wh_geno01 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(ALT<span class="sc">==</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb26-57"><a href="variant-annotation.html#cb26-57" aria-hidden="true" tabindex="-1"></a>wh_geno1 <span class="ot">&lt;-</span> wh_geno01 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(ALT<span class="sc">!=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb26-58"><a href="variant-annotation.html#cb26-58" aria-hidden="true" tabindex="-1"></a><span class="co"># Select columns</span></span>
<span id="cb26-59"><a href="variant-annotation.html#cb26-59" aria-hidden="true" tabindex="-1"></a>wh_geno1 <span class="ot">&lt;-</span> wh_geno1[<span class="fu">c</span>(<span class="dv">27</span>,<span class="dv">28</span>,<span class="dv">11</span><span class="sc">:</span><span class="dv">25</span>,<span class="dv">26</span>,<span class="dv">29</span>)]</span>
<span id="cb26-60"><a href="variant-annotation.html#cb26-60" aria-hidden="true" tabindex="-1"></a><span class="co"># Select X chromosome</span></span>
<span id="cb26-61"><a href="variant-annotation.html#cb26-61" aria-hidden="true" tabindex="-1"></a>wh_geno_X <span class="ot">&lt;-</span> wh_geno1 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">str_detect</span>(Location,<span class="st">&quot;chrX&quot;</span>))</span>
<span id="cb26-62"><a href="variant-annotation.html#cb26-62" aria-hidden="true" tabindex="-1"></a><span class="co"># Select autosomal variants</span></span>
<span id="cb26-63"><a href="variant-annotation.html#cb26-63" aria-hidden="true" tabindex="-1"></a>wh_geno_auto <span class="ot">&lt;-</span> wh_geno1 <span class="sc">%&gt;%</span></span>
<span id="cb26-64"><a href="variant-annotation.html#cb26-64" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(Location,<span class="st">&quot;chrX&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb26-65"><a href="variant-annotation.html#cb26-65" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(Location,<span class="st">&quot;chrY&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb26-66"><a href="variant-annotation.html#cb26-66" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(Location,<span class="st">&quot;M&quot;</span>))</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["Location"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Allele"],"name":[2],"type":["chr"],"align":["left"]},{"label":["proband"],"name":[3],"type":["chr"],"align":["left"]},{"label":["father"],"name":[4],"type":["chr"],"align":["left"]},{"label":["mother"],"name":[5],"type":["chr"],"align":["left"]},{"label":["GQ_proband"],"name":[6],"type":["int"],"align":["right"]},{"label":["GQ_father"],"name":[7],"type":["int"],"align":["right"]},{"label":["GQ_mother"],"name":[8],"type":["int"],"align":["right"]},{"label":["PS_proband"],"name":[9],"type":["int"],"align":["right"]},{"label":["PS_father"],"name":[10],"type":["int"],"align":["right"]},{"label":["PS_mother"],"name":[11],"type":["int"],"align":["right"]},{"label":["AD_ref_proband"],"name":[12],"type":["int"],"align":["right"]},{"label":["AD_ref_father"],"name":[13],"type":["int"],"align":["right"]},{"label":["AD_ref_mother"],"name":[14],"type":["int"],"align":["right"]},{"label":["AD_alt_proband"],"name":[15],"type":["int"],"align":["right"]},{"label":["AD_alt_father"],"name":[16],"type":["int"],"align":["right"]},{"label":["AD_alt_mother"],"name":[17],"type":["int"],"align":["right"]},{"label":["counter"],"name":[18],"type":["int"],"align":["right"]},{"label":["gnomADid"],"name":[19],"type":["chr"],"align":["left"]}],"data":[{"1":"chr1:826893","2":"A","3":"1/1","4":"1/1","5":"0|1","6":"99","7":"99","8":"99","9":"NA","10":"NA","11":"65797","12":"0","13":"0","14":"130","15":"279","16":"232","17":"124","18":"1","19":"1-826893-G-A"},{"1":"chr1:827092","2":"T","3":"0/0","4":"0/0","5":"1|0","6":"68","7":"65","8":"99","9":"NA","10":"NA","11":"65797","12":"40","13":"37","14":"33","15":"0","16":"0","17":"6","18":"1","19":"1-827092-C-T"},{"1":"chr1:827209","2":"C","3":"1/1","4":"1/1","5":"0|1","6":"77","7":"99","8":"87","9":"NA","10":"NA","11":"65797","12":"1","13":"0","14":"4","15":"21","16":"24","17":"13","18":"1","19":"1-827209-G-C"},{"1":"chr1:827212","2":"G","3":"1/1","4":"1/1","5":"0|1","6":"99","7":"99","8":"87","9":"NA","10":"NA","11":"65797","12":"0","13":"0","14":"4","15":"19","16":"22","17":"13","18":"1","19":"1-827212-C-G"},{"1":"chr1:827221","2":"C","3":"1/1","4":"1/1","5":"0|1","6":"90","7":"90","8":"90","9":"NA","10":"NA","11":"65797","12":"0","13":"0","14":"4","15":"16","16":"16","17":"12","18":"1","19":"1-827221-T-C"},{"1":"chr1:827252","2":"A","3":"1/1","4":"1/1","5":"0|1","6":"78","7":"78","8":"23","9":"NA","10":"NA","11":"65797","12":"0","13":"0","14":"3","15":"12","16":"12","17":"5","18":"1","19":"1-827252-T-A"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>We load the VEP annotated variants into R and select variants with an allele frequency &lt;0.01 that fall in RefSeq transcripts.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="variant-annotation.html#cb27-1" aria-hidden="true" tabindex="-1"></a>result <span class="ot">&lt;-</span> <span class="fu">read_table</span>(<span class="st">&quot;data/AJ_ensembl_ann_vep_filt_09_01_23.txt&quot;</span>, <span class="at">comment =</span> <span class="st">&quot;##&quot;</span>, <span class="at">na=</span><span class="st">&quot;-&quot;</span>, <span class="at">col_types =</span> <span class="fu">list</span>(</span>
<span id="cb27-2"><a href="variant-annotation.html#cb27-2" aria-hidden="true" tabindex="-1"></a>  <span class="st">`</span><span class="at">#Uploaded_variation</span><span class="st">`</span><span class="ot">=</span><span class="fu">col_factor</span>(), <span class="at">Gene=</span><span class="fu">col_factor</span>(), <span class="at">Feature=</span><span class="fu">col_factor</span>(), <span class="at">MANE_PLUS_CLINICAL =</span> <span class="fu">col_character</span>(), </span>
<span id="cb27-3"><a href="variant-annotation.html#cb27-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">MOTIF_NAME =</span> <span class="fu">col_character</span>(), <span class="at">HIGH_INF_POS =</span> <span class="fu">col_character</span>(), <span class="at">TRANSCRIPTION_FACTORS =</span> <span class="fu">col_character</span>(), </span>
<span id="cb27-4"><a href="variant-annotation.html#cb27-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">PUBMED=</span><span class="fu">col_character</span>(), <span class="at">MOTIF_POS =</span> <span class="fu">col_character</span>())) <span class="sc">%&gt;%</span></span>
<span id="cb27-5"><a href="variant-annotation.html#cb27-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">Uploaded_variation=</span><span class="st">`</span><span class="at">#Uploaded_variation</span><span class="st">`</span>)</span>
<span id="cb27-6"><a href="variant-annotation.html#cb27-6" aria-hidden="true" tabindex="-1"></a>result<span class="sc">$</span>Allele[<span class="fu">is.na</span>(result<span class="sc">$</span>Allele)] <span class="ot">&lt;-</span> <span class="st">&quot;-&quot;</span></span>
<span id="cb27-7"><a href="variant-annotation.html#cb27-7" aria-hidden="true" tabindex="-1"></a>result1 <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span> as_tibble <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(wh_geno1, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;Location&quot;</span>,<span class="st">&quot;Allele&quot;</span>))</span>
<span id="cb27-8"><a href="variant-annotation.html#cb27-8" aria-hidden="true" tabindex="-1"></a>refseq0 <span class="ot">&lt;-</span> result1 <span class="sc">%&gt;%</span> </span>
<span id="cb27-9"><a href="variant-annotation.html#cb27-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(SOURCE <span class="sc">==</span> <span class="st">&quot;RefSeq&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb27-10"><a href="variant-annotation.html#cb27-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(gnomADe_AF<span class="sc">&lt;=</span><span class="fl">0.01</span><span class="sc">|</span><span class="fu">is.na</span>(gnomADe_AF)) <span class="sc">%&gt;%</span> </span>
<span id="cb27-11"><a href="variant-annotation.html#cb27-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(gnomADg_AF<span class="sc">&lt;=</span><span class="fl">0.01</span><span class="sc">|</span><span class="fu">is.na</span>(gnomADg_AF)) <span class="sc">%&gt;%</span></span>
<span id="cb27-12"><a href="variant-annotation.html#cb27-12" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>(<span class="fu">str_detect</span>(Feature,<span class="st">&quot;XM&quot;</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb27-13"><a href="variant-annotation.html#cb27-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select_if</span>(<span class="sc">~</span><span class="fu">any</span>(<span class="sc">!</span><span class="fu">is.na</span>(.)))</span></code></pre></div>
<p>We add gene-level information from OMIM, Human Phenotype Ontology (HPO), ClinGen, GenCC, Orphanet. We select HPO terms for each gene that are associated with a particular phenotype, in this case, kidney disease.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="variant-annotation.html#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ontologyIndex)</span>
<span id="cb28-2"><a href="variant-annotation.html#cb28-2" aria-hidden="true" tabindex="-1"></a><span class="co"># OMIM</span></span>
<span id="cb28-3"><a href="variant-annotation.html#cb28-3" aria-hidden="true" tabindex="-1"></a>omim <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">&quot;data/genemap2_edit.txt&quot;</span>, <span class="at">name_repair=</span><span class="st">&quot;universal&quot;</span>, <span class="at">col_types=</span><span class="fu">list</span>(<span class="at">Entrez.Gene.ID =</span> <span class="fu">col_factor</span>())) <span class="sc">%&gt;%</span> </span>
<span id="cb28-4"><a href="variant-annotation.html#cb28-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">OMIM_Phenotypes =</span> Phenotypes) <span class="sc">%&gt;%</span> </span>
<span id="cb28-5"><a href="variant-annotation.html#cb28-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(Entrez.Gene.ID,OMIM_Phenotypes)</span>
<span id="cb28-6"><a href="variant-annotation.html#cb28-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Human Phenotype Ontology</span></span>
<span id="cb28-7"><a href="variant-annotation.html#cb28-7" aria-hidden="true" tabindex="-1"></a>hpo_data <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">&quot;data/genes_to_phenotype.txt&quot;</span>, <span class="at">col_types=</span><span class="fu">list</span>(<span class="at">entrez_gene_id =</span> <span class="fu">col_factor</span>()))</span>
<span id="cb28-8"><a href="variant-annotation.html#cb28-8" aria-hidden="true" tabindex="-1"></a>hpo <span class="ot">&lt;-</span> <span class="fu">get_ontology</span>(<span class="at">file =</span> <span class="st">&#39;data/hp_obo.txt&#39;</span>, <span class="at">propagate_relationships =</span> <span class="st">&quot;is_a&quot;</span>, <span class="at">extract_tags =</span> <span class="st">&quot;minimal&quot;</span>)</span>
<span id="cb28-9"><a href="variant-annotation.html#cb28-9" aria-hidden="true" tabindex="-1"></a>hpo_kidney <span class="ot">&lt;-</span> <span class="fu">get_descendants</span>(hpo, <span class="at">roots=</span><span class="st">&quot;HP:0000077&quot;</span>) <span class="co">#Abnormality of the kidney</span></span>
<span id="cb28-10"><a href="variant-annotation.html#cb28-10" aria-hidden="true" tabindex="-1"></a>gen_hpo_kidney <span class="ot">&lt;-</span> hpo_data <span class="sc">%&gt;%</span> </span>
<span id="cb28-11"><a href="variant-annotation.html#cb28-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(hpo_term_id <span class="sc">%in%</span> hpo_kidney) <span class="sc">%&gt;%</span> </span>
<span id="cb28-12"><a href="variant-annotation.html#cb28-12" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(frequency_hpo) <span class="sc">%&gt;%</span> </span>
<span id="cb28-13"><a href="variant-annotation.html#cb28-13" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">frequency =</span> <span class="fu">case_when</span>(</span>
<span id="cb28-14"><a href="variant-annotation.html#cb28-14" aria-hidden="true" tabindex="-1"></a>    frequency_hpo <span class="sc">==</span> <span class="st">&quot;HP:0040284&quot;</span> <span class="sc">~</span> <span class="st">&quot; (very rare)&quot;</span>, frequency_hpo <span class="sc">==</span> <span class="st">&quot;HP:0040283&quot;</span> <span class="sc">~</span> <span class="st">&quot; (occasional)&quot;</span>, </span>
<span id="cb28-15"><a href="variant-annotation.html#cb28-15" aria-hidden="true" tabindex="-1"></a>    frequency_hpo <span class="sc">==</span> <span class="st">&quot;HP:0040282&quot;</span> <span class="sc">~</span> <span class="st">&quot; (frequent)&quot;</span>, frequency_hpo <span class="sc">==</span> <span class="st">&quot;HP:0040281&quot;</span> <span class="sc">~</span> <span class="st">&quot; (very frequent)&quot;</span>, </span>
<span id="cb28-16"><a href="variant-annotation.html#cb28-16" aria-hidden="true" tabindex="-1"></a>    frequency_hpo <span class="sc">==</span> <span class="st">&quot;HP:0040280&quot;</span> <span class="sc">~</span> <span class="st">&quot; (obligate)&quot;</span>, <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">&quot;&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb28-17"><a href="variant-annotation.html#cb28-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(<span class="st">&quot;renal_hpo_term&quot;</span>, <span class="fu">c</span>(hpo_term_name, frequency), <span class="at">remove =</span> <span class="cn">FALSE</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb28-18"><a href="variant-annotation.html#cb28-18" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(entrez_gene_id) <span class="sc">%&gt;%</span> </span>
<span id="cb28-19"><a href="variant-annotation.html#cb28-19" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">renal_hpo_term =</span> <span class="fu">toString</span>(<span class="fu">unique</span>(renal_hpo_term)))</span>
<span id="cb28-20"><a href="variant-annotation.html#cb28-20" aria-hidden="true" tabindex="-1"></a>gen_hpo_extra <span class="ot">&lt;-</span> hpo_data <span class="sc">%&gt;%</span> </span>
<span id="cb28-21"><a href="variant-annotation.html#cb28-21" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(frequency_hpo <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;HP:0040281&quot;</span>, <span class="st">&quot;HP:0040280&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb28-22"><a href="variant-annotation.html#cb28-22" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(frequency_hpo) <span class="sc">%&gt;%</span></span>
<span id="cb28-23"><a href="variant-annotation.html#cb28-23" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>(hpo_term_id <span class="sc">%in%</span> hpo_kidney)) <span class="sc">%&gt;%</span> </span>
<span id="cb28-24"><a href="variant-annotation.html#cb28-24" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(entrez_gene_id) <span class="sc">%&gt;%</span> </span>
<span id="cb28-25"><a href="variant-annotation.html#cb28-25" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">extrarenal_hpo_term =</span> <span class="fu">toString</span>(<span class="fu">unique</span>(hpo_term_name)))</span>
<span id="cb28-26"><a href="variant-annotation.html#cb28-26" aria-hidden="true" tabindex="-1"></a><span class="co"># ClinGen </span></span>
<span id="cb28-27"><a href="variant-annotation.html#cb28-27" aria-hidden="true" tabindex="-1"></a>clingen <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/Clingen-Curation.csv&quot;</span>, <span class="at">skip =</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb28-28"><a href="variant-annotation.html#cb28-28" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">assertion=</span><span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">s*</span><span class="sc">\\</span><span class="st">([^</span><span class="sc">\\</span><span class="st">)]+</span><span class="sc">\\</span><span class="st">)&quot;</span>,<span class="st">&quot;&quot;</span>, gene_disease_validity_assertion_classifications)) <span class="sc">%&gt;%</span></span>
<span id="cb28-29"><a href="variant-annotation.html#cb28-29" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">assertion=</span><span class="fu">paste</span>(<span class="st">&quot;(&quot;</span>,assertion,<span class="st">&quot;)&quot;</span>,<span class="at">sep=</span><span class="st">&quot;&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb28-30"><a href="variant-annotation.html#cb28-30" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">assertion=</span><span class="fu">if_else</span>(assertion<span class="sc">==</span><span class="st">&quot;(NA)&quot;</span>,<span class="cn">NA</span>,assertion)) <span class="sc">%&gt;%</span></span>
<span id="cb28-31"><a href="variant-annotation.html#cb28-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(<span class="st">&quot;disease&quot;</span>, <span class="fu">c</span>(disease_label, assertion), <span class="at">remove =</span> <span class="cn">FALSE</span>, <span class="at">na.rm=</span><span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">&quot; &quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb28-32"><a href="variant-annotation.html#cb28-32" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(hgnc_id) <span class="sc">%&gt;%</span> </span>
<span id="cb28-33"><a href="variant-annotation.html#cb28-33" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">clingen_disease_label =</span> <span class="fu">paste</span>(disease,<span class="at">collapse=</span><span class="st">&quot;; &quot;</span>))</span>
<span id="cb28-34"><a href="variant-annotation.html#cb28-34" aria-hidden="true" tabindex="-1"></a><span class="co"># GenCC including Orphanet data</span></span>
<span id="cb28-35"><a href="variant-annotation.html#cb28-35" aria-hidden="true" tabindex="-1"></a>gencc <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">&quot;data/gencc-submissions.tsv&quot;</span>)</span>
<span id="cb28-36"><a href="variant-annotation.html#cb28-36" aria-hidden="true" tabindex="-1"></a>gencc_def <span class="ot">&lt;-</span> gencc <span class="sc">%&gt;%</span> </span>
<span id="cb28-37"><a href="variant-annotation.html#cb28-37" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(submitter_title <span class="sc">!=</span> <span class="st">&quot;Orphanet&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb28-38"><a href="variant-annotation.html#cb28-38" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(classification_title <span class="sc">==</span> <span class="st">&quot;Definitive&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb28-39"><a href="variant-annotation.html#cb28-39" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(gene_symbol, gene_curie, disease_title) <span class="sc">%&gt;%</span> </span>
<span id="cb28-40"><a href="variant-annotation.html#cb28-40" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(</span>
<span id="cb28-41"><a href="variant-annotation.html#cb28-41" aria-hidden="true" tabindex="-1"></a>    <span class="at">submitter_title =</span> <span class="fu">toString</span>(<span class="fu">unique</span>(submitter_title)), <span class="at">moi_title =</span> <span class="fu">toString</span>(<span class="fu">unique</span>(moi_title)), </span>
<span id="cb28-42"><a href="variant-annotation.html#cb28-42" aria-hidden="true" tabindex="-1"></a>    <span class="at">disease_curie =</span> <span class="fu">toString</span>(<span class="fu">unique</span>(disease_curie)), <span class="at">disease_original_curie =</span> <span class="fu">toString</span>(<span class="fu">unique</span>(disease_original_curie)), </span>
<span id="cb28-43"><a href="variant-annotation.html#cb28-43" aria-hidden="true" tabindex="-1"></a>    <span class="at">classification_title =</span> <span class="fu">toString</span>(classification_title), <span class="at">submitted_as_pmids =</span> <span class="fu">toString</span>(<span class="fu">unique</span>(submitted_as_pmids))) <span class="sc">%&gt;%</span></span>
<span id="cb28-44"><a href="variant-annotation.html#cb28-44" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">submitter =</span> <span class="fu">paste</span>(<span class="st">&quot;(&quot;</span>, submitter_title, <span class="st">&quot;)&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb28-45"><a href="variant-annotation.html#cb28-45" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">unite</span>(<span class="st">&quot;gencc_disease_definitive&quot;</span>, <span class="fu">c</span>(disease_title, submitter), <span class="at">remove =</span> <span class="cn">FALSE</span>, <span class="at">sep =</span> <span class="st">&quot; &quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb28-46"><a href="variant-annotation.html#cb28-46" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(gene_curie, gencc_disease_definitive) <span class="sc">%&gt;%</span> </span>
<span id="cb28-47"><a href="variant-annotation.html#cb28-47" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(gene_curie) <span class="sc">%&gt;%</span> </span>
<span id="cb28-48"><a href="variant-annotation.html#cb28-48" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">gencc_disease_definitive =</span> <span class="fu">paste</span>(<span class="fu">unique</span>(gencc_disease_definitive), <span class="at">collapse=</span><span class="st">&quot;; &quot;</span>))</span>
<span id="cb28-49"><a href="variant-annotation.html#cb28-49" aria-hidden="true" tabindex="-1"></a>gencc_strong <span class="ot">&lt;-</span> gencc <span class="sc">%&gt;%</span> </span>
<span id="cb28-50"><a href="variant-annotation.html#cb28-50" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(submitter_title <span class="sc">!=</span> <span class="st">&quot;Orphanet&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb28-51"><a href="variant-annotation.html#cb28-51" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(classification_title <span class="sc">==</span> <span class="st">&quot;Strong&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb28-52"><a href="variant-annotation.html#cb28-52" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(gene_symbol, gene_curie, disease_title) <span class="sc">%&gt;%</span> </span>
<span id="cb28-53"><a href="variant-annotation.html#cb28-53" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(</span>
<span id="cb28-54"><a href="variant-annotation.html#cb28-54" aria-hidden="true" tabindex="-1"></a>    <span class="at">submitter_title =</span> <span class="fu">toString</span>(<span class="fu">unique</span>(submitter_title)), <span class="at">moi_title =</span> <span class="fu">toString</span>(<span class="fu">unique</span>(moi_title)), </span>
<span id="cb28-55"><a href="variant-annotation.html#cb28-55" aria-hidden="true" tabindex="-1"></a>    <span class="at">disease_curie =</span> <span class="fu">toString</span>(<span class="fu">unique</span>(disease_curie)), <span class="at">disease_original_curie =</span> <span class="fu">toString</span>(<span class="fu">unique</span>(disease_original_curie)), </span>
<span id="cb28-56"><a href="variant-annotation.html#cb28-56" aria-hidden="true" tabindex="-1"></a>    <span class="at">classification_title =</span> <span class="fu">toString</span>(classification_title), <span class="at">submitted_as_pmids =</span> <span class="fu">toString</span>(<span class="fu">unique</span>(submitted_as_pmids))) <span class="sc">%&gt;%</span></span>
<span id="cb28-57"><a href="variant-annotation.html#cb28-57" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">submitter =</span> <span class="fu">paste</span>(<span class="st">&quot;(&quot;</span>, submitter_title, <span class="st">&quot;)&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb28-58"><a href="variant-annotation.html#cb28-58" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">unite</span>(<span class="st">&quot;gencc_disease_strong&quot;</span>, <span class="fu">c</span>(disease_title, submitter), <span class="at">remove =</span> <span class="cn">FALSE</span>, <span class="at">sep =</span> <span class="st">&quot; &quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb28-59"><a href="variant-annotation.html#cb28-59" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(gene_curie, gencc_disease_strong) <span class="sc">%&gt;%</span> </span>
<span id="cb28-60"><a href="variant-annotation.html#cb28-60" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(gene_curie) <span class="sc">%&gt;%</span> </span>
<span id="cb28-61"><a href="variant-annotation.html#cb28-61" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">gencc_disease_strong =</span> <span class="fu">paste</span>(<span class="fu">unique</span>(gencc_disease_strong), <span class="at">collapse=</span><span class="st">&quot;; &quot;</span>))</span>
<span id="cb28-62"><a href="variant-annotation.html#cb28-62" aria-hidden="true" tabindex="-1"></a>orphanet <span class="ot">&lt;-</span> gencc <span class="sc">%&gt;%</span> </span>
<span id="cb28-63"><a href="variant-annotation.html#cb28-63" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(submitter_title <span class="sc">==</span> <span class="st">&quot;Orphanet&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb28-64"><a href="variant-annotation.html#cb28-64" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(gene_curie) <span class="sc">%&gt;%</span> </span>
<span id="cb28-65"><a href="variant-annotation.html#cb28-65" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">orphanet_disease =</span> <span class="fu">toString</span>(disease_title), <span class="at">orphanet_moi =</span> <span class="fu">toString</span>(<span class="fu">unique</span>(moi_title)))</span>
<span id="cb28-66"><a href="variant-annotation.html#cb28-66" aria-hidden="true" tabindex="-1"></a><span class="co"># Join</span></span>
<span id="cb28-67"><a href="variant-annotation.html#cb28-67" aria-hidden="true" tabindex="-1"></a>refseq01 <span class="ot">&lt;-</span> refseq0 <span class="sc">%&gt;%</span> </span>
<span id="cb28-68"><a href="variant-annotation.html#cb28-68" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(omim, <span class="at">by=</span><span class="fu">join_by</span>(Gene<span class="sc">==</span>Entrez.Gene.ID)) <span class="sc">%&gt;%</span></span>
<span id="cb28-69"><a href="variant-annotation.html#cb28-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(gen_hpo_kidney, <span class="at">by=</span><span class="fu">join_by</span>(Gene<span class="sc">==</span>entrez_gene_id)) <span class="sc">%&gt;%</span></span>
<span id="cb28-70"><a href="variant-annotation.html#cb28-70" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(gen_hpo_extra, <span class="at">by=</span><span class="fu">join_by</span>(Gene<span class="sc">==</span>entrez_gene_id)) <span class="sc">%&gt;%</span></span>
<span id="cb28-71"><a href="variant-annotation.html#cb28-71" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(clingen, <span class="at">by=</span><span class="fu">join_by</span>(HGNC_ID<span class="sc">==</span>hgnc_id)) <span class="sc">%&gt;%</span></span>
<span id="cb28-72"><a href="variant-annotation.html#cb28-72" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(gencc_def, <span class="at">by=</span><span class="fu">join_by</span>(HGNC_ID<span class="sc">==</span>gene_curie)) <span class="sc">%&gt;%</span></span>
<span id="cb28-73"><a href="variant-annotation.html#cb28-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(gencc_strong, <span class="at">by=</span><span class="fu">join_by</span>(HGNC_ID<span class="sc">==</span>gene_curie)) <span class="sc">%&gt;%</span></span>
<span id="cb28-74"><a href="variant-annotation.html#cb28-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(orphanet, <span class="at">by=</span><span class="fu">join_by</span>(HGNC_ID<span class="sc">==</span>gene_curie)) <span class="sc">%&gt;%</span></span>
<span id="cb28-75"><a href="variant-annotation.html#cb28-75" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(SYMBOL, Uploaded_variation, Location, Feature, HGVSg, HGVSc,    HGVSp, Consequence, ZYG, EXON,  INTRON, gnomADe_AF, gnomADg_AF, MAX_AF, MAX_AF_POPS,    ClinVar,    ClinVar_CLNSIG, OMIM_Phenotypes, renal_hpo_term, extrarenal_hpo_term,   REVEL,  SIFT,   PolyPhen,   CADD_PHRED, PrimateAI,  SpliceAI_pred,  EVE_CLASS,  EVE_SCORE,  MutationTaster_pred,    MutationTaster_score,   orphanet_disease,   orphanet_moi,   Mastermind_MMID3,   Existing_variation, PUBMED, mother, father, proband, Allele, counter, AD_ref_mother, AD_ref_father, AD_ref_proband, AD_alt_mother, AD_alt_father, AD_alt_proband, GQ_mother,    GQ_father,  GQ_proband, PS_mother,  PS_father,  PS_proband, CANONICAL,  MANE_SELECT,    MANE_PLUS_CLINICAL, IMPACT, Gene, HGNC_ID, Amino_acids, Codons, AF, AFR_AF, AMR_AF, EAS_AF, EUR_AF, SAS_AF, gnomADe_AFR_AF, gnomADe_AMR_AF, gnomADe_ASJ_AF, gnomADe_EAS_AF, gnomADe_FIN_AF, gnomADe_NFE_AF, gnomADe_OTH_AF, gnomADe_SAS_AF, gnomADg_AFR_AF, gnomADg_AMI_AF, gnomADg_AMR_AF, gnomADg_ASJ_AF, gnomADg_EAS_AF, gnomADg_FIN_AF, gnomADg_MID_AF, gnomADg_NFE_AF, gnomADg_OTH_AF, gnomADg_SAS_AF, VARIANT_CLASS, gnomADid)</span>
<span id="cb28-76"><a href="variant-annotation.html#cb28-76" aria-hidden="true" tabindex="-1"></a><span class="co"># X chromosome</span></span>
<span id="cb28-77"><a href="variant-annotation.html#cb28-77" aria-hidden="true" tabindex="-1"></a>refseq_X <span class="ot">&lt;-</span> refseq01 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">str_detect</span>(Uploaded_variation,<span class="st">&quot;X&quot;</span>))</span>
<span id="cb28-78"><a href="variant-annotation.html#cb28-78" aria-hidden="true" tabindex="-1"></a><span class="co"># Autosomal chromosomes</span></span>
<span id="cb28-79"><a href="variant-annotation.html#cb28-79" aria-hidden="true" tabindex="-1"></a>refseq <span class="ot">&lt;-</span> refseq01 <span class="sc">%&gt;%</span></span>
<span id="cb28-80"><a href="variant-annotation.html#cb28-80" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(Uploaded_variation,<span class="st">&quot;X&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb28-81"><a href="variant-annotation.html#cb28-81" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(Uploaded_variation,<span class="st">&quot;Y&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb28-82"><a href="variant-annotation.html#cb28-82" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(Uploaded_variation,<span class="st">&quot;M&quot;</span>))</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["SYMBOL"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Uploaded_variation"],"name":[2],"type":["fct"],"align":["left"]},{"label":["Location"],"name":[3],"type":["chr"],"align":["left"]},{"label":["Feature"],"name":[4],"type":["fct"],"align":["left"]},{"label":["HGVSg"],"name":[5],"type":["chr"],"align":["left"]},{"label":["HGVSc"],"name":[6],"type":["chr"],"align":["left"]},{"label":["HGVSp"],"name":[7],"type":["chr"],"align":["left"]},{"label":["Consequence"],"name":[8],"type":["chr"],"align":["left"]},{"label":["ZYG"],"name":[9],"type":["chr"],"align":["left"]},{"label":["EXON"],"name":[10],"type":["chr"],"align":["left"]},{"label":["INTRON"],"name":[11],"type":["chr"],"align":["left"]},{"label":["gnomADe_AF"],"name":[12],"type":["dbl"],"align":["right"]},{"label":["gnomADg_AF"],"name":[13],"type":["dbl"],"align":["right"]},{"label":["MAX_AF"],"name":[14],"type":["dbl"],"align":["right"]},{"label":["MAX_AF_POPS"],"name":[15],"type":["chr"],"align":["left"]},{"label":["ClinVar"],"name":[16],"type":["dbl"],"align":["right"]},{"label":["ClinVar_CLNSIG"],"name":[17],"type":["chr"],"align":["left"]},{"label":["OMIM_Phenotypes"],"name":[18],"type":["chr"],"align":["left"]},{"label":["renal_hpo_term"],"name":[19],"type":["chr"],"align":["left"]},{"label":["extrarenal_hpo_term"],"name":[20],"type":["chr"],"align":["left"]},{"label":["REVEL"],"name":[21],"type":["dbl"],"align":["right"]},{"label":["SIFT"],"name":[22],"type":["chr"],"align":["left"]},{"label":["PolyPhen"],"name":[23],"type":["chr"],"align":["left"]},{"label":["CADD_PHRED"],"name":[24],"type":["dbl"],"align":["right"]},{"label":["PrimateAI"],"name":[25],"type":["dbl"],"align":["right"]},{"label":["SpliceAI_pred"],"name":[26],"type":["chr"],"align":["left"]},{"label":["EVE_CLASS"],"name":[27],"type":["chr"],"align":["left"]},{"label":["EVE_SCORE"],"name":[28],"type":["dbl"],"align":["right"]},{"label":["MutationTaster_pred"],"name":[29],"type":["chr"],"align":["left"]},{"label":["MutationTaster_score"],"name":[30],"type":["chr"],"align":["left"]},{"label":["orphanet_disease"],"name":[31],"type":["chr"],"align":["left"]},{"label":["orphanet_moi"],"name":[32],"type":["chr"],"align":["left"]},{"label":["Mastermind_MMID3"],"name":[33],"type":["chr"],"align":["left"]},{"label":["Existing_variation"],"name":[34],"type":["chr"],"align":["left"]},{"label":["PUBMED"],"name":[35],"type":["chr"],"align":["left"]},{"label":["mother"],"name":[36],"type":["chr"],"align":["left"]},{"label":["father"],"name":[37],"type":["chr"],"align":["left"]},{"label":["proband"],"name":[38],"type":["chr"],"align":["left"]},{"label":["Allele"],"name":[39],"type":["chr"],"align":["left"]},{"label":["counter"],"name":[40],"type":["int"],"align":["right"]},{"label":["AD_ref_mother"],"name":[41],"type":["int"],"align":["right"]},{"label":["AD_ref_father"],"name":[42],"type":["int"],"align":["right"]},{"label":["AD_ref_proband"],"name":[43],"type":["int"],"align":["right"]},{"label":["AD_alt_mother"],"name":[44],"type":["int"],"align":["right"]},{"label":["AD_alt_father"],"name":[45],"type":["int"],"align":["right"]},{"label":["AD_alt_proband"],"name":[46],"type":["int"],"align":["right"]},{"label":["GQ_mother"],"name":[47],"type":["int"],"align":["right"]},{"label":["GQ_father"],"name":[48],"type":["int"],"align":["right"]},{"label":["GQ_proband"],"name":[49],"type":["int"],"align":["right"]},{"label":["PS_mother"],"name":[50],"type":["int"],"align":["right"]},{"label":["PS_father"],"name":[51],"type":["int"],"align":["right"]},{"label":["PS_proband"],"name":[52],"type":["int"],"align":["right"]},{"label":["CANONICAL"],"name":[53],"type":["chr"],"align":["left"]},{"label":["MANE_SELECT"],"name":[54],"type":["chr"],"align":["left"]},{"label":["MANE_PLUS_CLINICAL"],"name":[55],"type":["chr"],"align":["left"]},{"label":["IMPACT"],"name":[56],"type":["chr"],"align":["left"]},{"label":["Gene"],"name":[57],"type":["fct"],"align":["left"]},{"label":["HGNC_ID"],"name":[58],"type":["chr"],"align":["left"]},{"label":["Amino_acids"],"name":[59],"type":["chr"],"align":["left"]},{"label":["Codons"],"name":[60],"type":["chr"],"align":["left"]},{"label":["AF"],"name":[61],"type":["dbl"],"align":["right"]},{"label":["AFR_AF"],"name":[62],"type":["dbl"],"align":["right"]},{"label":["AMR_AF"],"name":[63],"type":["dbl"],"align":["right"]},{"label":["EAS_AF"],"name":[64],"type":["dbl"],"align":["right"]},{"label":["EUR_AF"],"name":[65],"type":["dbl"],"align":["right"]},{"label":["SAS_AF"],"name":[66],"type":["dbl"],"align":["right"]},{"label":["gnomADe_AFR_AF"],"name":[67],"type":["dbl"],"align":["right"]},{"label":["gnomADe_AMR_AF"],"name":[68],"type":["dbl"],"align":["right"]},{"label":["gnomADe_ASJ_AF"],"name":[69],"type":["dbl"],"align":["right"]},{"label":["gnomADe_EAS_AF"],"name":[70],"type":["dbl"],"align":["right"]},{"label":["gnomADe_FIN_AF"],"name":[71],"type":["dbl"],"align":["right"]},{"label":["gnomADe_NFE_AF"],"name":[72],"type":["dbl"],"align":["right"]},{"label":["gnomADe_OTH_AF"],"name":[73],"type":["dbl"],"align":["right"]},{"label":["gnomADe_SAS_AF"],"name":[74],"type":["dbl"],"align":["right"]},{"label":["gnomADg_AFR_AF"],"name":[75],"type":["dbl"],"align":["right"]},{"label":["gnomADg_AMI_AF"],"name":[76],"type":["dbl"],"align":["right"]},{"label":["gnomADg_AMR_AF"],"name":[77],"type":["dbl"],"align":["right"]},{"label":["gnomADg_ASJ_AF"],"name":[78],"type":["dbl"],"align":["right"]},{"label":["gnomADg_EAS_AF"],"name":[79],"type":["dbl"],"align":["right"]},{"label":["gnomADg_FIN_AF"],"name":[80],"type":["dbl"],"align":["right"]},{"label":["gnomADg_MID_AF"],"name":[81],"type":["dbl"],"align":["right"]},{"label":["gnomADg_NFE_AF"],"name":[82],"type":["dbl"],"align":["right"]},{"label":["gnomADg_OTH_AF"],"name":[83],"type":["dbl"],"align":["right"]},{"label":["gnomADg_SAS_AF"],"name":[84],"type":["dbl"],"align":["right"]},{"label":["VARIANT_CLASS"],"name":[85],"type":["chr"],"align":["left"]},{"label":["gnomADid"],"name":[86],"type":["chr"],"align":["left"]}],"data":[{"1":"RNF223","2":"chr1_1072326_G/A","3":"chr1:1072326","4":"NM_001205252.2","5":"chr1:g.1072326G>A","6":"NM_001205252.2:c.241C>T","7":"NP_001192181.1:p.Arg81Trp","8":"missense_variant","9":"HET","10":"2/2","11":"NA","12":"1.255e-03","13":"1.058e-03","14":"0.018730","15":"gnomADg_ASJ","16":"NA","17":"NA","18":"NA","19":"NA","20":"NA","21":"0.298","22":"deleterious(0)","23":"probably_damaging(1)","24":"25.70","25":"0.8687824","26":"RNF223|0.00|0.00|0.00|0.00|-42|42|-36|-33","27":"NA","28":"NA","29":"D","30":"0.514264","31":"NA","32":"NA","33":"RNF223:R81W","34":"rs576663974","35":"NA","36":"0/0","37":"0|1","38":"1|0","39":"A","40":"1","41":"40","42":"77","43":"106","44":"0","45":"68","46":"82","47":"90","48":"99","49":"99","50":"NA","51":"65797","52":"65797","53":"YES","54":"ENST00000453464.3","55":"NA","56":"MODERATE","57":"401934","58":"HGNC:40020","59":"R/W","60":"Cgg/Tgg","61":"0.0002","62":"0","63":"0.0014","64":"0.0000","65":"0.000","66":"0.000","67":"0","68":"1.162e-03","69":"0.0177300","70":"3.086e-04","71":"0.0008711","72":"0.000843","73":"0.001089","74":"0.0003011","75":"2.654e-04","76":"0","77":"0.001963","78":"0.018730","79":"0.0003861","80":"0.0005648","81":"0","82":"0.0006320","83":"0.0009569","84":"0.0004137","85":"SNV","86":"1-1072326-G-A"},{"1":"NPHP4","2":"chr1_5864452_C/A","3":"chr1:5864452","4":"NM_001291593.2","5":"chr1:g.5864452C>A","6":"NM_001291593.2:c.2343G>T","7":"NP_001278522.1:p.Arg781Ser","8":"missense_variant","9":"HET","10":"25/27","11":"NA","12":"4.767e-05","13":"2.628e-05","14":"0.001152","15":"gnomADg_ASJ","16":"969225","17":"Uncertain_significance","18":"Senior-Loken syndrome 4, 606996 (3), Autosomal recessive; Nephronophthisis 4, 606966 (3), Autosomal recessive","19":"Stage 5 chronic kidney disease (very frequent), Nephronophthisis (frequent), Tubulointerstitial fibrosis, Renal tubular atrophy, Polyuria, Stage 5 chronic kidney disease, Renal corticomedullary cysts, Nephronophthisis","20":"Short stature, Hypertension, Global developmental delay, Abnormality of retinal pigmentation, Retinal dystrophy","21":"0.664","22":"deleterious(0)","23":"benign(0.383)","24":"24.20","25":"0.4835634","26":"NPHP4|0.04|0.00|0.04|0.00|-9|20|6|-39","27":"Pathogenic","28":"0.80084944","29":"D","30":"0.99534","31":"Senior-Loken syndrome, nephronophthisis 1","32":"Autosomal recessive","33":"NA","34":"rs367814493","35":"NA","36":"0/0","37":"1|0","38":"1|0","39":"A","40":"1","41":"33","42":"78","43":"109","44":"0","45":"69","46":"69","47":"99","48":"99","49":"99","50":"NA","51":"65797","52":"65797","53":"NA","54":"NA","55":"NA","56":"MODERATE","57":"261734","58":"HGNC:19104","59":"R/S","60":"agG/agT","61":"NA","62":"NA","63":"NA","64":"NA","65":"NA","66":"NA","67":"0","68":"6.087e-05","69":"0.0008323","70":"5.999e-05","71":"0.0000000","72":"0.000000","73":"0.000000","74":"0.0000000","75":"0.000e+00","76":"0","77":"0.000000","78":"0.001152","79":"0.0000000","80":"0.0000000","81":"0","82":"0.0000000","83":"0.0000000","84":"0.0000000","85":"SNV","86":"1-5864452-C-A"},{"1":"NPHP4","2":"chr1_5864452_C/A","3":"chr1:5864452","4":"NM_001291594.2","5":"chr1:g.5864452C>A","6":"NM_001291594.2:c.2346G>T","7":"NP_001278523.1:p.Arg782Ser","8":"missense_variant","9":"HET","10":"24/26","11":"NA","12":"4.767e-05","13":"2.628e-05","14":"0.001152","15":"gnomADg_ASJ","16":"969225","17":"Uncertain_significance","18":"Senior-Loken syndrome 4, 606996 (3), Autosomal recessive; Nephronophthisis 4, 606966 (3), Autosomal recessive","19":"Stage 5 chronic kidney disease (very frequent), Nephronophthisis (frequent), Tubulointerstitial fibrosis, Renal tubular atrophy, Polyuria, Stage 5 chronic kidney disease, Renal corticomedullary cysts, Nephronophthisis","20":"Short stature, Hypertension, Global developmental delay, Abnormality of retinal pigmentation, Retinal dystrophy","21":"0.664","22":"deleterious(0)","23":"benign(0.383)","24":"24.20","25":"0.4835634","26":"NPHP4|0.04|0.00|0.04|0.00|-9|20|6|-39","27":"Pathogenic","28":"0.80084944","29":"D","30":"0.99534","31":"Senior-Loken syndrome, nephronophthisis 1","32":"Autosomal recessive","33":"NA","34":"rs367814493","35":"NA","36":"0/0","37":"1|0","38":"1|0","39":"A","40":"1","41":"33","42":"78","43":"109","44":"0","45":"69","46":"69","47":"99","48":"99","49":"99","50":"NA","51":"65797","52":"65797","53":"NA","54":"NA","55":"NA","56":"MODERATE","57":"261734","58":"HGNC:19104","59":"R/S","60":"agG/agT","61":"NA","62":"NA","63":"NA","64":"NA","65":"NA","66":"NA","67":"0","68":"6.087e-05","69":"0.0008323","70":"5.999e-05","71":"0.0000000","72":"0.000000","73":"0.000000","74":"0.0000000","75":"0.000e+00","76":"0","77":"0.000000","78":"0.001152","79":"0.0000000","80":"0.0000000","81":"0","82":"0.0000000","83":"0.0000000","84":"0.0000000","85":"SNV","86":"1-5864452-C-A"},{"1":"NPHP4","2":"chr1_5864452_C/A","3":"chr1:5864452","4":"NM_015102.5","5":"chr1:g.5864452C>A","6":"NM_015102.5:c.3882G>T","7":"NP_055917.1:p.Arg1294Ser","8":"missense_variant","9":"HET","10":"28/30","11":"NA","12":"4.767e-05","13":"2.628e-05","14":"0.001152","15":"gnomADg_ASJ","16":"969225","17":"Uncertain_significance","18":"Senior-Loken syndrome 4, 606996 (3), Autosomal recessive; Nephronophthisis 4, 606966 (3), Autosomal recessive","19":"Stage 5 chronic kidney disease (very frequent), Nephronophthisis (frequent), Tubulointerstitial fibrosis, Renal tubular atrophy, Polyuria, Stage 5 chronic kidney disease, Renal corticomedullary cysts, Nephronophthisis","20":"Short stature, Hypertension, Global developmental delay, Abnormality of retinal pigmentation, Retinal dystrophy","21":"0.664","22":"deleterious(0)","23":"possibly_damaging(0.817)","24":"24.20","25":"0.4835634","26":"NPHP4|0.04|0.00|0.04|0.00|-9|20|6|-39","27":"Pathogenic","28":"0.80084944","29":"D","30":"0.99534","31":"Senior-Loken syndrome, nephronophthisis 1","32":"Autosomal recessive","33":"NA","34":"rs367814493","35":"NA","36":"0/0","37":"1|0","38":"1|0","39":"A","40":"1","41":"33","42":"78","43":"109","44":"0","45":"69","46":"69","47":"99","48":"99","49":"99","50":"NA","51":"65797","52":"65797","53":"YES","54":"ENST00000378156.9","55":"NA","56":"MODERATE","57":"261734","58":"HGNC:19104","59":"R/S","60":"agG/agT","61":"NA","62":"NA","63":"NA","64":"NA","65":"NA","66":"NA","67":"0","68":"6.087e-05","69":"0.0008323","70":"5.999e-05","71":"0.0000000","72":"0.000000","73":"0.000000","74":"0.0000000","75":"0.000e+00","76":"0","77":"0.000000","78":"0.001152","79":"0.0000000","80":"0.0000000","81":"0","82":"0.0000000","83":"0.0000000","84":"0.0000000","85":"SNV","86":"1-5864452-C-A"},{"1":"GPR153","2":"chr1_6249964_C/T","3":"chr1:6249964","4":"NM_207370.4","5":"chr1:g.6249964C>T","6":"NM_207370.4:c.1204G>A","7":"NP_997253.2:p.Val402Met","8":"missense_variant","9":"HET","10":"6/6","11":"NA","12":"2.290e-03","13":"6.823e-03","14":"0.183500","15":"gnomADg_EAS","16":"NA","17":"NA","18":"NA","19":"NA","20":"NA","21":"0.035","22":"tolerated(1)","23":"benign(0.001)","24":"12.17","25":"0.7926179","26":"GPR153|0.01|0.00|0.00|0.00|39|-1|-1|1","27":"Benign","28":"0.06354271","29":"P","30":"0.812616","31":"NA","32":"NA","33":"GPR153:V402M","34":"rs78132938","35":"NA","36":"0/1","37":"0/1","38":"0/1","39":"T","40":"1","41":"5","42":"6","43":"3","44":"5","45":"0","46":"8","47":"99","48":"10","49":"68","50":"NA","51":"NA","52":"NA","53":"YES","54":"ENST00000377893.3","55":"NA","56":"MODERATE","57":"387509","58":"HGNC:23618","59":"V/M","60":"Gtg/Atg","61":"0.0357","62":"0","63":"0.0000","64":"0.1756","65":"0.001","66":"0.001","67":"0","68":"0.000e+00","69":"0.0000000","70":"1.649e-01","71":"0.0000000","72":"0.000000","73":"0.009804","74":"0.0000000","75":"9.651e-05","76":"0","77":"0.001112","78":"0.007209","79":"0.1835000","80":"0.0000000","81":"0","82":"0.0001912","83":"0.0052580","84":"0.0039300","85":"SNV","86":"1-6249964-C-T"},{"1":"H6PD","2":"chr1_9244920_CCCAGGCA/-","3":"chr1:9244920-9244927","4":"NM_001282587.2","5":"chr1:g.9244928_9244935del","6":"NM_001282587.2:c.27_34del","7":"NP_001269516.1:p.Pro10ValfsTer38","8":"splice_acceptor_variant,coding_sequence_variant,intron_variant","9":"HET","10":"2/5","11":"1/4","12":"NA","13":"NA","14":"NA","15":"NA","16":"402919","17":"Benign","18":"Cortisone reductase deficiency 1, 604931 (3), Autosomal recessive","19":"NA","20":"NA","21":"NA","22":"NA","23":"NA","24":"22.90","25":"NA","26":"NA","27":"NA","28":"NA","29":"NA","30":"NA","31":"cortisone reductase deficiency","32":"Autosomal dominant","33":"NA","34":"NA","35":"NA","36":"1|0","37":"0/0","38":"0|1","39":"-","40":"1","41":"30","42":"34","43":"40","44":"33","45":"0","46":"35","47":"99","48":"66","49":"99","50":"65797","51":"NA","52":"65797","53":"NA","54":"NA","55":"NA","56":"HIGH","57":"9563","58":"HGNC:4795","59":"NA","60":"NA","61":"NA","62":"NA","63":"NA","64":"NA","65":"NA","66":"NA","67":"NA","68":"NA","69":"NA","70":"NA","71":"NA","72":"NA","73":"NA","74":"NA","75":"NA","76":"NA","77":"NA","78":"NA","79":"NA","80":"NA","81":"NA","82":"NA","83":"NA","84":"NA","85":"deletion","86":"1-9244919-CCCCAGGCA-C"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="de-novo-variants" class="section level2 hasAnchor" number="5.3">
<h2><span class="header-section-number">5.3</span> <em>De Novo</em> Variants<a href="variant-annotation.html#de-novo-variants" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We select <em>de novo</em> variants in the proband.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="variant-annotation.html#cb29-1" aria-hidden="true" tabindex="-1"></a>geno_novo_all <span class="ot">&lt;-</span> wh_geno_auto <span class="sc">%&gt;%</span> </span>
<span id="cb29-2"><a href="variant-annotation.html#cb29-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span> ( ( ( <span class="fu">str_detect</span>(mother, <span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)) <span class="sc">|</span> <span class="fu">str_detect</span>(mother, <span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>) ) <span class="sc">&amp;</span> </span>
<span id="cb29-3"><a href="variant-annotation.html#cb29-3" aria-hidden="true" tabindex="-1"></a>                 (<span class="fu">str_detect</span>(father, <span class="fu">substr</span>(proband,<span class="dv">3</span>,<span class="dv">3</span>)) <span class="sc">|</span> <span class="fu">str_detect</span>(father, <span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>) ) ) <span class="sc">|</span> </span>
<span id="cb29-4"><a href="variant-annotation.html#cb29-4" aria-hidden="true" tabindex="-1"></a>               ( ( <span class="fu">str_detect</span>(father, <span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)) <span class="sc">|</span> <span class="fu">str_detect</span>(father, <span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>) ) <span class="sc">&amp;</span> </span>
<span id="cb29-5"><a href="variant-annotation.html#cb29-5" aria-hidden="true" tabindex="-1"></a>                   (<span class="fu">str_detect</span>(mother, <span class="fu">substr</span>(proband,<span class="dv">3</span>,<span class="dv">3</span>)) <span class="sc">|</span> <span class="fu">str_detect</span>(mother, <span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>) ) ) ) ) <span class="sc">%&gt;%</span></span>
<span id="cb29-6"><a href="variant-annotation.html#cb29-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>(<span class="fu">str_detect</span>(mother,<span class="fu">as.character</span>(counter)) <span class="sc">|</span> <span class="fu">str_detect</span>(father,<span class="fu">as.character</span>(counter))) <span class="sc">|</span></span>
<span id="cb29-7"><a href="variant-annotation.html#cb29-7" aria-hidden="true" tabindex="-1"></a>                  (<span class="fu">str_detect</span>(mother, <span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)) <span class="sc">&amp;</span> <span class="fu">str_detect</span>(mother, <span class="fu">substr</span>(proband,<span class="dv">3</span>,<span class="dv">3</span>))) <span class="sc">|</span></span>
<span id="cb29-8"><a href="variant-annotation.html#cb29-8" aria-hidden="true" tabindex="-1"></a>                  (<span class="fu">str_detect</span>(father, <span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)) <span class="sc">&amp;</span> <span class="fu">str_detect</span>(father, <span class="fu">substr</span>(proband,<span class="dv">3</span>,<span class="dv">3</span>)))</span>
<span id="cb29-9"><a href="variant-annotation.html#cb29-9" aria-hidden="true" tabindex="-1"></a>                  )</span></code></pre></div>
<p>We discard <em>de novo</em> mutations back to the reference allele (e.g. 1/1, 1/1 –&gt; 0/1) since they are unlikely to be pathogenic.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="variant-annotation.html#cb30-1" aria-hidden="true" tabindex="-1"></a>geno_novo_not_ref <span class="ot">&lt;-</span> wh_geno_auto <span class="sc">%&gt;%</span> </span>
<span id="cb30-2"><a href="variant-annotation.html#cb30-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span> ( ( ( <span class="fu">str_detect</span>(mother, <span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)) <span class="sc">|</span> <span class="fu">str_detect</span>(mother, <span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>) <span class="sc">|</span> <span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)<span class="sc">==</span><span class="dv">0</span> ) <span class="sc">&amp;</span> </span>
<span id="cb30-3"><a href="variant-annotation.html#cb30-3" aria-hidden="true" tabindex="-1"></a>                 (<span class="fu">str_detect</span>(father, <span class="fu">substr</span>(proband,<span class="dv">3</span>,<span class="dv">3</span>)) <span class="sc">|</span> <span class="fu">str_detect</span>(father, <span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>) <span class="sc">|</span> <span class="fu">substr</span>(proband,<span class="dv">3</span>,<span class="dv">3</span>)<span class="sc">==</span><span class="dv">0</span> ) ) <span class="sc">|</span> </span>
<span id="cb30-4"><a href="variant-annotation.html#cb30-4" aria-hidden="true" tabindex="-1"></a>               ( ( <span class="fu">str_detect</span>(father, <span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)) <span class="sc">|</span> <span class="fu">str_detect</span>(father, <span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>) <span class="sc">|</span> <span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)<span class="sc">==</span><span class="dv">0</span> ) <span class="sc">&amp;</span> </span>
<span id="cb30-5"><a href="variant-annotation.html#cb30-5" aria-hidden="true" tabindex="-1"></a>                   (<span class="fu">str_detect</span>(mother, <span class="fu">substr</span>(proband,<span class="dv">3</span>,<span class="dv">3</span>)) <span class="sc">|</span> <span class="fu">str_detect</span>(mother, <span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>) <span class="sc">|</span> <span class="fu">substr</span>(proband,<span class="dv">3</span>,<span class="dv">3</span>)<span class="sc">==</span><span class="dv">0</span> ) ) ) ) <span class="sc">%&gt;%</span></span>
<span id="cb30-6"><a href="variant-annotation.html#cb30-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>(<span class="fu">str_detect</span>(mother,<span class="fu">as.character</span>(counter)) <span class="sc">|</span> <span class="fu">str_detect</span>(father,<span class="fu">as.character</span>(counter))) <span class="sc">|</span></span>
<span id="cb30-7"><a href="variant-annotation.html#cb30-7" aria-hidden="true" tabindex="-1"></a>                  (<span class="fu">str_detect</span>(mother, <span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)) <span class="sc">&amp;</span> <span class="fu">str_detect</span>(mother, <span class="fu">substr</span>(proband,<span class="dv">3</span>,<span class="dv">3</span>))) <span class="sc">|</span></span>
<span id="cb30-8"><a href="variant-annotation.html#cb30-8" aria-hidden="true" tabindex="-1"></a>                  (<span class="fu">str_detect</span>(father, <span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)) <span class="sc">&amp;</span> <span class="fu">str_detect</span>(father, <span class="fu">substr</span>(proband,<span class="dv">3</span>,<span class="dv">3</span>)))</span>
<span id="cb30-9"><a href="variant-annotation.html#cb30-9" aria-hidden="true" tabindex="-1"></a>                  )</span></code></pre></div>
<p>We increase the cutoff for genotype quality to ensure that the the <em>de novo</em> variants are not a result of incorrect variant calling. Set GQ&gt;=30, i.e, the probability that the call is incorrect is =&lt;0.1%.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="variant-annotation.html#cb31-1" aria-hidden="true" tabindex="-1"></a>geno_novo_not_ref_high_qual <span class="ot">&lt;-</span> geno_novo_not_ref <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(GQ_mother<span class="sc">&gt;=</span><span class="dv">30</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(GQ_father<span class="sc">&gt;=</span><span class="dv">30</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(GQ_proband<span class="sc">&gt;=</span><span class="dv">30</span>)</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["Location"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Allele"],"name":[2],"type":["chr"],"align":["left"]},{"label":["mother"],"name":[3],"type":["chr"],"align":["left"]},{"label":["father"],"name":[4],"type":["chr"],"align":["left"]},{"label":["proband"],"name":[5],"type":["chr"],"align":["left"]},{"label":["counter"],"name":[6],"type":["int"],"align":["right"]}],"data":[{"1":"chr1:15425826-15425829","2":"-","3":"0/0","4":"0/0","5":"0/1","6":"1"},{"1":"chr1:64663551-64663552","2":"-","3":"0/1","4":"0/2","5":"2/2","6":"1"},{"1":"chr1:64663551-64663552","2":"A","3":"0/1","4":"0/2","5":"2/2","6":"2"},{"1":"chr1:118081059-118081064","2":"ACAC","3":"0/1","4":"0/2","5":"0/3","6":"3"},{"1":"chr1:149390802-149390803","2":"GGCGGCGGCGGCGGC","3":"1/1","4":"2/3","5":"2/2","6":"1"},{"1":"chr1:149390802-149390803","2":"GGCGGCGGCGGCGGCGGCGGC","3":"1/1","4":"2/3","5":"2/2","6":"2"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>We select <em>de novo</em> variants found in the filtered VEP annotation. For each variant, we chose a single gene transcript that contains the variant, prioritizing canonical and curated transcripts.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="variant-annotation.html#cb32-1" aria-hidden="true" tabindex="-1"></a>prob_novo <span class="ot">&lt;-</span> refseq <span class="sc">%&gt;%</span> <span class="fu">inner_join</span>(geno_novo_not_ref)</span>
<span id="cb32-2"><a href="variant-annotation.html#cb32-2" aria-hidden="true" tabindex="-1"></a>prob_novo <span class="ot">&lt;-</span> prob_novo <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(Uploaded_variation) <span class="sc">%&gt;%</span> </span>
<span id="cb32-3"><a href="variant-annotation.html#cb32-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(CANONICAL), <span class="fu">as.character</span>(Feature),<span class="at">.by_group =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb32-4"><a href="variant-annotation.html#cb32-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Feature <span class="sc">==</span> Feature[<span class="dv">1</span>])</span>
<span id="cb32-5"><a href="variant-annotation.html#cb32-5" aria-hidden="true" tabindex="-1"></a>prob_novo_high_qual <span class="ot">&lt;-</span> refseq <span class="sc">%&gt;%</span> <span class="fu">inner_join</span>(geno_novo_not_ref_high_qual)</span>
<span id="cb32-6"><a href="variant-annotation.html#cb32-6" aria-hidden="true" tabindex="-1"></a>prob_novo_high_qual <span class="ot">&lt;-</span> prob_novo_high_qual <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(Uploaded_variation) <span class="sc">%&gt;%</span> </span>
<span id="cb32-7"><a href="variant-annotation.html#cb32-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(CANONICAL), <span class="fu">as.character</span>(Feature),<span class="at">.by_group =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb32-8"><a href="variant-annotation.html#cb32-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Feature <span class="sc">==</span> Feature[<span class="dv">1</span>])</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["SYMBOL"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Uploaded_variation"],"name":[2],"type":["fct"],"align":["left"]},{"label":["HGVSc"],"name":[3],"type":["chr"],"align":["left"]},{"label":["HGVSp"],"name":[4],"type":["chr"],"align":["left"]},{"label":["mother"],"name":[5],"type":["chr"],"align":["left"]},{"label":["father"],"name":[6],"type":["chr"],"align":["left"]},{"label":["proband"],"name":[7],"type":["chr"],"align":["left"]},{"label":["counter"],"name":[8],"type":["int"],"align":["right"]}],"data":[{"1":"DNMT3A","2":"chr2_25234286_C/T","3":"NM_022552.5:c.2732G>A","4":"NP_072046.2:p.Cys911Tyr","5":"0/0","6":"0/0","7":"0/1","8":"1"},{"1":"PRR30","2":"chr2_27137168_A/G","3":"NM_178553.4:c.1162T>C","4":"NP_848648.2:p.Ser388Pro","5":"0/0","6":"0/0","7":"0/1","8":"1"},{"1":"MAB21L4","2":"chr2_240895786_G/T","3":"NM_001085437.3:c.212C>A","4":"NP_001078906.3:p.Ala71Asp","5":"0/0","6":"0/0","7":"0/1","8":"1"},{"1":"STAB1","2":"chr3_52513896_C/G","3":"NM_015136.3:c.3362C>G","4":"NP_055951.2:p.Pro1121Arg","5":"0/0","6":"0/0","7":"0/1","8":"1"},{"1":"PCDHGB1","2":"chr5_141351402_A/G","3":"NM_018922.3:c.1142A>G","4":"NP_061745.1:p.Tyr381Cys","5":"0/0","6":"0/0","7":"0/1","8":"1"},{"1":"HLA-A","2":"chr6_29942825_G/T","3":"NM_002116.8:c.142G>T","4":"NP_002107.3:p.Ala48Ser","5":"0/0","6":"0/0","7":"0/1","8":"1"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="homozygous-variants" class="section level2 hasAnchor" number="5.4">
<h2><span class="header-section-number">5.4</span> Homozygous Variants<a href="variant-annotation.html#homozygous-variants" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We select the homozygous variants in the proband.</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="variant-annotation.html#cb33-1" aria-hidden="true" tabindex="-1"></a>geno_hom <span class="ot">&lt;-</span> wh_geno_auto <span class="sc">%&gt;%</span> </span>
<span id="cb33-2"><a href="variant-annotation.html#cb33-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)<span class="sc">==</span><span class="fu">substr</span>(proband,<span class="dv">3</span>,<span class="dv">3</span>) <span class="sc">&amp;</span> <span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>) <span class="sc">!=</span> <span class="dv">0</span> <span class="sc">&amp;</span> <span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>) <span class="sc">!=</span> <span class="st">&quot;.&quot;</span>)</span></code></pre></div>
<p>We discard the homozygous variants for which one of the parents is also homozygous (e.g. 1/1, 0/1 -&gt; 1/1) since these variants are unlikely to be pathogenic.</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="variant-annotation.html#cb34-1" aria-hidden="true" tabindex="-1"></a>geno_hom2 <span class="ot">&lt;-</span> geno_hom <span class="sc">%&gt;%</span></span>
<span id="cb34-2"><a href="variant-annotation.html#cb34-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span> ( (<span class="fu">substr</span>(mother,<span class="dv">1</span>,<span class="dv">1</span>)<span class="sc">==</span><span class="fu">substr</span>(mother,<span class="dv">3</span>,<span class="dv">3</span>) <span class="sc">&amp;</span> <span class="fu">substr</span>(mother,<span class="dv">1</span>,<span class="dv">1</span>)<span class="sc">==</span><span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)) <span class="sc">|</span></span>
<span id="cb34-3"><a href="variant-annotation.html#cb34-3" aria-hidden="true" tabindex="-1"></a>                      (<span class="fu">substr</span>(father,<span class="dv">1</span>,<span class="dv">1</span>)<span class="sc">==</span><span class="fu">substr</span>(father,<span class="dv">3</span>,<span class="dv">3</span>) <span class="sc">&amp;</span> <span class="fu">substr</span>(father,<span class="dv">1</span>,<span class="dv">1</span>)<span class="sc">==</span><span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)) )</span>
<span id="cb34-4"><a href="variant-annotation.html#cb34-4" aria-hidden="true" tabindex="-1"></a>                  )</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["Location"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Allele"],"name":[2],"type":["chr"],"align":["left"]},{"label":["proband"],"name":[3],"type":["chr"],"align":["left"]},{"label":["father"],"name":[4],"type":["chr"],"align":["left"]},{"label":["mother"],"name":[5],"type":["chr"],"align":["left"]},{"label":["GQ_proband"],"name":[6],"type":["int"],"align":["right"]},{"label":["GQ_father"],"name":[7],"type":["int"],"align":["right"]},{"label":["GQ_mother"],"name":[8],"type":["int"],"align":["right"]},{"label":["PS_proband"],"name":[9],"type":["int"],"align":["right"]},{"label":["PS_father"],"name":[10],"type":["int"],"align":["right"]},{"label":["PS_mother"],"name":[11],"type":["int"],"align":["right"]},{"label":["AD_ref_proband"],"name":[12],"type":["int"],"align":["right"]},{"label":["AD_ref_father"],"name":[13],"type":["int"],"align":["right"]},{"label":["AD_ref_mother"],"name":[14],"type":["int"],"align":["right"]},{"label":["AD_alt_proband"],"name":[15],"type":["int"],"align":["right"]},{"label":["AD_alt_father"],"name":[16],"type":["int"],"align":["right"]},{"label":["AD_alt_mother"],"name":[17],"type":["int"],"align":["right"]},{"label":["counter"],"name":[18],"type":["int"],"align":["right"]},{"label":["gnomADid"],"name":[19],"type":["chr"],"align":["left"]}],"data":[{"1":"chr1:1629847-1629853","2":"-","3":"1/1","4":"0|1","5":"1|0","6":"18","7":"25","8":"9","9":"NA","10":"65797","11":"65797","12":"0","13":"5","14":"1","15":"2","16":"0","17":"4","18":"1","19":"1-1629846-CCCCCCAG-C"},{"1":"chr1:2371909","2":"-","3":"1/1","4":"1|0","5":"1|0","6":"3","7":"75","8":"24","9":"NA","10":"65797","11":"65797","12":"1","13":"7","14":"8","15":"2","16":"3","17":"1","18":"1","19":"1-2371908-CA-C"},{"1":"chr1:3385157","2":"T","3":"1/1","4":"1|0","5":"1|0","6":"99","7":"99","8":"99","9":"NA","10":"65797","11":"65797","12":"2","13":"47","14":"53","15":"113","16":"36","17":"46","18":"1","19":"1-3385157-C-T"},{"1":"chr1:3396350","2":"G","3":"1/1","4":"1|0","5":"1|0","6":"99","7":"99","8":"99","9":"NA","10":"65797","11":"65797","12":"0","13":"138","14":"161","15":"295","16":"156","17":"139","18":"1","19":"1-3396350-T-G"},{"1":"chr1:3815401","2":"A","3":"1/1","4":"1|0","5":"1|0","6":"99","7":"99","8":"99","9":"NA","10":"65797","11":"65797","12":"0","13":"108","14":"127","15":"270","16":"94","17":"124","18":"1","19":"1-3815401-G-A"},{"1":"chr1:3825693","2":"T","3":"1/1","4":"1|0","5":"1|0","6":"99","7":"99","8":"99","9":"NA","10":"65797","11":"65797","12":"0","13":"13","14":"11","15":"34","16":"13","17":"15","18":"1","19":"1-3825693-C-T"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>We select the homozygous variants found in the filtered VEP annotation.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="variant-annotation.html#cb35-1" aria-hidden="true" tabindex="-1"></a>prob_hom <span class="ot">&lt;-</span> refseq <span class="sc">%&gt;%</span> <span class="fu">inner_join</span>(geno_hom2)</span>
<span id="cb35-2"><a href="variant-annotation.html#cb35-2" aria-hidden="true" tabindex="-1"></a>prob_hom <span class="ot">&lt;-</span> prob_hom <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(Uploaded_variation) <span class="sc">%&gt;%</span> </span>
<span id="cb35-3"><a href="variant-annotation.html#cb35-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(CANONICAL), <span class="fu">as.character</span>(Feature),<span class="at">.by_group =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb35-4"><a href="variant-annotation.html#cb35-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Feature <span class="sc">==</span> Feature[<span class="dv">1</span>])</span></code></pre></div>
<p>For each of these variants, we query the gnomAD database for the number of individuals that are homozygous for that variant.</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="variant-annotation.html#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ghql)</span>
<span id="cb36-2"><a href="variant-annotation.html#cb36-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(jsonlite)</span>
<span id="cb36-3"><a href="variant-annotation.html#cb36-3" aria-hidden="true" tabindex="-1"></a>token <span class="ot">&lt;-</span> <span class="fu">Sys.getenv</span>(<span class="st">&quot;GNOMAD_TOKEN&quot;</span>)</span>
<span id="cb36-4"><a href="variant-annotation.html#cb36-4" aria-hidden="true" tabindex="-1"></a>con <span class="ot">&lt;-</span> GraphqlClient<span class="sc">$</span><span class="fu">new</span>(</span>
<span id="cb36-5"><a href="variant-annotation.html#cb36-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">url =</span> <span class="st">&quot;https://gnomad.broadinstitute.org/api&quot;</span>,</span>
<span id="cb36-6"><a href="variant-annotation.html#cb36-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">headers =</span> <span class="fu">list</span>(<span class="at">Authorization =</span> <span class="fu">paste0</span>(<span class="st">&quot;Bearer &quot;</span>, token))</span>
<span id="cb36-7"><a href="variant-annotation.html#cb36-7" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb36-8"><a href="variant-annotation.html#cb36-8" aria-hidden="true" tabindex="-1"></a>qry <span class="ot">&lt;-</span> Query<span class="sc">$</span><span class="fu">new</span>()</span>
<span id="cb36-9"><a href="variant-annotation.html#cb36-9" aria-hidden="true" tabindex="-1"></a>qry<span class="sc">$</span><span class="fu">query</span>(<span class="st">&#39;homozygous&#39;</span>,</span>
<span id="cb36-10"><a href="variant-annotation.html#cb36-10" aria-hidden="true" tabindex="-1"></a><span class="st">&#39;query Variant($var: String!){</span></span>
<span id="cb36-11"><a href="variant-annotation.html#cb36-11" aria-hidden="true" tabindex="-1"></a><span class="st">  variant(variantId: $var, dataset: gnomad_r4) {</span></span>
<span id="cb36-12"><a href="variant-annotation.html#cb36-12" aria-hidden="true" tabindex="-1"></a><span class="st">    exome {</span></span>
<span id="cb36-13"><a href="variant-annotation.html#cb36-13" aria-hidden="true" tabindex="-1"></a><span class="st">      homozygote_count</span></span>
<span id="cb36-14"><a href="variant-annotation.html#cb36-14" aria-hidden="true" tabindex="-1"></a><span class="st">    }</span></span>
<span id="cb36-15"><a href="variant-annotation.html#cb36-15" aria-hidden="true" tabindex="-1"></a><span class="st">    genome {</span></span>
<span id="cb36-16"><a href="variant-annotation.html#cb36-16" aria-hidden="true" tabindex="-1"></a><span class="st">      homozygote_count</span></span>
<span id="cb36-17"><a href="variant-annotation.html#cb36-17" aria-hidden="true" tabindex="-1"></a><span class="st">    }</span></span>
<span id="cb36-18"><a href="variant-annotation.html#cb36-18" aria-hidden="true" tabindex="-1"></a><span class="st">  }</span></span>
<span id="cb36-19"><a href="variant-annotation.html#cb36-19" aria-hidden="true" tabindex="-1"></a><span class="st">}&#39;</span>)</span>
<span id="cb36-20"><a href="variant-annotation.html#cb36-20" aria-hidden="true" tabindex="-1"></a>gnomADid <span class="ot">&lt;-</span> prob_hom<span class="sc">$</span>gnomADid</span>
<span id="cb36-21"><a href="variant-annotation.html#cb36-21" aria-hidden="true" tabindex="-1"></a>s <span class="ot">&lt;-</span> <span class="fu">tibble</span>(<span class="at">gnomADid=</span><span class="cn">NA</span>, <span class="at">gnomADe_hom=</span><span class="cn">NA</span>, <span class="at">gnomADg_hom=</span><span class="cn">NA</span>)</span>
<span id="cb36-22"><a href="variant-annotation.html#cb36-22" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> gnomADid) {</span>
<span id="cb36-23"><a href="variant-annotation.html#cb36-23" aria-hidden="true" tabindex="-1"></a>  variables <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">var =</span> i)</span>
<span id="cb36-24"><a href="variant-annotation.html#cb36-24" aria-hidden="true" tabindex="-1"></a>  tries <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb36-25"><a href="variant-annotation.html#cb36-25" aria-hidden="true" tabindex="-1"></a>  <span class="cf">repeat</span> {</span>
<span id="cb36-26"><a href="variant-annotation.html#cb36-26" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> <span class="fu">try</span>(con<span class="sc">$</span><span class="fu">exec</span>(qry<span class="sc">$</span>queries<span class="sc">$</span>homozygous, variables), <span class="at">silent =</span> <span class="cn">TRUE</span>)</span>
<span id="cb36-27"><a href="variant-annotation.html#cb36-27" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is</span>(x, <span class="st">&#39;try-error&#39;</span>)) {</span>
<span id="cb36-28"><a href="variant-annotation.html#cb36-28" aria-hidden="true" tabindex="-1"></a>      x1 <span class="ot">&lt;-</span> jsonlite<span class="sc">::</span><span class="fu">fromJSON</span>(x)<span class="sc">$</span>data<span class="sc">$</span>variant<span class="sc">$</span>exome<span class="sc">$</span>homozygote_count</span>
<span id="cb36-29"><a href="variant-annotation.html#cb36-29" aria-hidden="true" tabindex="-1"></a>      x2 <span class="ot">&lt;-</span> jsonlite<span class="sc">::</span><span class="fu">fromJSON</span>(x)<span class="sc">$</span>data<span class="sc">$</span>variant<span class="sc">$</span>genome<span class="sc">$</span>homozygote_count</span>
<span id="cb36-30"><a href="variant-annotation.html#cb36-30" aria-hidden="true" tabindex="-1"></a>      <span class="fu">message</span>(<span class="fu">paste</span>(tries,<span class="st">&quot;success&quot;</span>, <span class="at">collapse=</span><span class="st">&quot; &quot;</span>))</span>
<span id="cb36-31"><a href="variant-annotation.html#cb36-31" aria-hidden="true" tabindex="-1"></a>      <span class="cf">break</span></span>
<span id="cb36-32"><a href="variant-annotation.html#cb36-32" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb36-33"><a href="variant-annotation.html#cb36-33" aria-hidden="true" tabindex="-1"></a>    <span class="fu">Sys.sleep</span>(<span class="dv">2</span><span class="sc">*</span>tries)</span>
<span id="cb36-34"><a href="variant-annotation.html#cb36-34" aria-hidden="true" tabindex="-1"></a>    <span class="fu">message</span>(tries)</span>
<span id="cb36-35"><a href="variant-annotation.html#cb36-35" aria-hidden="true" tabindex="-1"></a>    tries <span class="ot">&lt;-</span> tries <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb36-36"><a href="variant-annotation.html#cb36-36" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb36-37"><a href="variant-annotation.html#cb36-37" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">is.null</span>(x1)) x1 <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb36-38"><a href="variant-annotation.html#cb36-38" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">is.null</span>(x2)) x2 <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb36-39"><a href="variant-annotation.html#cb36-39" aria-hidden="true" tabindex="-1"></a>  s <span class="ot">&lt;-</span> s <span class="sc">%&gt;%</span> <span class="fu">add_row</span>(<span class="at">gnomADid=</span>variables<span class="sc">$</span>var,<span class="at">gnomADe_hom=</span>x1, <span class="at">gnomADg_hom=</span>x2)</span>
<span id="cb36-40"><a href="variant-annotation.html#cb36-40" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb36-41"><a href="variant-annotation.html#cb36-41" aria-hidden="true" tabindex="-1"></a>prob_hom <span class="ot">&lt;-</span> <span class="fu">left_join</span>(prob_hom, s) <span class="sc">%&gt;%</span> </span>
<span id="cb36-42"><a href="variant-annotation.html#cb36-42" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">relocate</span>(<span class="fu">c</span>(gnomADe_hom,gnomADg_hom),<span class="at">.after=</span>MAX_AF_POPS)</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["SYMBOL"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Uploaded_variation"],"name":[2],"type":["fct"],"align":["left"]},{"label":["HGVSc"],"name":[3],"type":["chr"],"align":["left"]},{"label":["HGVSp"],"name":[4],"type":["chr"],"align":["left"]},{"label":["gnomADe_hom"],"name":[5],"type":["int"],"align":["right"]},{"label":["gnomADg_hom"],"name":[6],"type":["int"],"align":["right"]},{"label":["proband"],"name":[7],"type":["chr"],"align":["left"]},{"label":["counter"],"name":[8],"type":["int"],"align":["right"]}],"data":[{"1":"TNRC18","2":"chr7_5374171_G/A","3":"NM_001080495.3:c.3113C>T","4":"NP_001073964.2:p.Ser1038Phe","5":"0","6":"NA","7":"1/1","8":"1"},{"1":"TBC1D2","2":"chr9_98199422_C/T","3":"NM_001267571.2:c.2746G>A","4":"NP_001254500.1:p.Glu916Lys","5":"44","6":"5","7":"1/1","8":"1"},{"1":"NPRL3","2":"chr16_138305_GAGGAGGACGGAGCCGGAGGCGGAGGGGGCCT/-","3":"NM_001077350.3:c.-67-2_-38del","4":"NA","5":"37716","6":"9885","7":"1/1","8":"1"},{"1":"FADS6","2":"chr17_74893498_-/GGTTCCATGGGCTCCGTA","3":"NM_178128.6:c.81_98dup","4":"NP_835229.3:p.Thr28_Pro33dup","5":"9027","6":"3569","7":"1/1","8":"1"},{"1":"TOX2","2":"chr20_44051315_A/G","3":"NM_001098797.2:c.421A>G","4":"NP_001092267.1:p.Met141Val","5":"63","6":"6","7":"1/1","8":"1"},{"1":"CTSA","2":"chr20_45891599_CTG/-","3":"NM_000308.4:c.54_56del","4":"NP_000299.3:p.Leu19del","5":"246488","6":"31868","7":"1/1","8":"1"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="compound-heterozygous-variants" class="section level2 hasAnchor" number="5.5">
<h2><span class="header-section-number">5.5</span> Compound Heterozygous Variants<a href="variant-annotation.html#compound-heterozygous-variants" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We select the unphased compound heterozygous variants.</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="variant-annotation.html#cb37-1" aria-hidden="true" tabindex="-1"></a>prob_2plus_het <span class="ot">&lt;-</span> refseq <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)<span class="sc">!=</span><span class="fu">substr</span>(proband,<span class="dv">3</span>,<span class="dv">3</span>)) </span>
<span id="cb37-2"><a href="variant-annotation.html#cb37-2" aria-hidden="true" tabindex="-1"></a>prob_2plus_het <span class="ot">&lt;-</span> prob_2plus_het <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(Feature) <span class="sc">%&gt;%</span> </span>
<span id="cb37-3"><a href="variant-annotation.html#cb37-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">length</span>(<span class="fu">unique</span>(PS_proband))<span class="sc">&gt;=</span><span class="dv">2</span> <span class="sc">|</span> <span class="fu">length</span>(<span class="fu">is.na</span>(PS_proband)[<span class="fu">is.na</span>(PS_proband)<span class="sc">==</span><span class="cn">TRUE</span>])<span class="sc">&gt;=</span><span class="dv">2</span> ) <span class="sc">%&gt;%</span> </span>
<span id="cb37-4"><a href="variant-annotation.html#cb37-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb37-5"><a href="variant-annotation.html#cb37-5" aria-hidden="true" tabindex="-1"></a>prob_2plus_het2 <span class="ot">&lt;-</span> prob_2plus_het <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(Gene) <span class="sc">%&gt;%</span> </span>
<span id="cb37-6"><a href="variant-annotation.html#cb37-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(CANONICAL), <span class="fu">as.character</span>(Feature),<span class="at">.by_group =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb37-7"><a href="variant-annotation.html#cb37-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Feature <span class="sc">==</span> Feature[<span class="dv">1</span>])</span></code></pre></div>
<p>We select the phased compound heterozygous variants.</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="variant-annotation.html#cb38-1" aria-hidden="true" tabindex="-1"></a>prob_comp_het <span class="ot">&lt;-</span> refseq <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(Feature, PS_proband) <span class="sc">%&gt;%</span> </span>
<span id="cb38-2"><a href="variant-annotation.html#cb38-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">any</span>(<span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)<span class="sc">&gt;=</span><span class="dv">1</span> <span class="sc">&amp;</span> <span class="fu">substr</span>(proband,<span class="dv">2</span>,<span class="dv">2</span>)<span class="sc">==</span><span class="st">&quot;|&quot;</span>) <span class="sc">&amp;</span> <span class="fu">any</span>(<span class="fu">substr</span>(proband,<span class="dv">3</span>,<span class="dv">3</span>)<span class="sc">&gt;=</span><span class="dv">1</span> <span class="sc">&amp;</span> <span class="fu">substr</span>(proband,<span class="dv">2</span>,<span class="dv">2</span>)<span class="sc">==</span><span class="st">&quot;|&quot;</span>))</span>
<span id="cb38-3"><a href="variant-annotation.html#cb38-3" aria-hidden="true" tabindex="-1"></a>prob_comp_het2 <span class="ot">&lt;-</span> prob_comp_het <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(Gene) <span class="sc">%&gt;%</span> </span>
<span id="cb38-4"><a href="variant-annotation.html#cb38-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(CANONICAL), <span class="fu">as.character</span>(Feature),<span class="at">.by_group =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb38-5"><a href="variant-annotation.html#cb38-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Feature <span class="sc">==</span> Feature[<span class="dv">1</span>])</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["SYMBOL"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Uploaded_variation"],"name":[2],"type":["fct"],"align":["left"]},{"label":["HGVSc"],"name":[3],"type":["chr"],"align":["left"]},{"label":["HGVSp"],"name":[4],"type":["chr"],"align":["left"]},{"label":["proband"],"name":[5],"type":["chr"],"align":["left"]},{"label":["counter"],"name":[6],"type":["int"],"align":["right"]},{"label":["PS_proband"],"name":[7],"type":["int"],"align":["right"]}],"data":[{"1":"ATOH8","2":"chr2_85754660_-/CGCACCGCCCGCGCGCCC","3":"NM_032827.7:c.474_491dup","4":"NP_116216.2:p.Pro160_Pro165dup","5":"1|0","6":"1","7":"41404"},{"1":"ATOH8","2":"chr2_85754718_T/G","3":"NM_032827.7:c.529T>G","4":"NP_116216.2:p.Ser177Ala","5":"0|1","6":"1","7":"41404"},{"1":"FLNB","2":"chr3_58131983_T/C","3":"NM_001164317.2:c.4447T>C","4":"NP_001157789.1:p.Phe1483Leu","5":"1|0","6":"1","7":"197630"},{"1":"FLNB","2":"chr3_58159621_T/C","3":"NM_001164317.2:c.7049T>C","4":"NP_001157789.1:p.Ile2350Thr","5":"0|1","6":"1","7":"197630"},{"1":"HLA-DRB1","2":"chr6_32580255_T/G","3":"NM_002124.4:c.779A>C","4":"NP_002115.2:p.Gln260Pro","5":"0|1","6":"1","7":"203397"},{"1":"HLA-DRB1","2":"chr6_32580272_T/C","3":"NM_002124.4:c.764-2A>G","4":"NA","5":"1|0","6":"1","7":"203397"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>For two variants within a gene that cannot be phased with trio data, we can estimate phase using EM-based haplotype estimation from the gnomAD database.</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="variant-annotation.html#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Obtain the GRCh37 variant ID to query gnomAD</span></span>
<span id="cb39-2"><a href="variant-annotation.html#cb39-2" aria-hidden="true" tabindex="-1"></a>grch38id <span class="ot">&lt;-</span> prob_2plus_het2<span class="sc">$</span>gnomADid</span>
<span id="cb39-3"><a href="variant-annotation.html#cb39-3" aria-hidden="true" tabindex="-1"></a>qry<span class="sc">$</span><span class="fu">query</span>(<span class="st">&#39;liftover&#39;</span>,</span>
<span id="cb39-4"><a href="variant-annotation.html#cb39-4" aria-hidden="true" tabindex="-1"></a><span class="st">&#39;query liftOver($varID: String!){</span></span>
<span id="cb39-5"><a href="variant-annotation.html#cb39-5" aria-hidden="true" tabindex="-1"></a><span class="st">  liftover(liftover_variant_id: $varID, reference_genome:GRCh38) {</span></span>
<span id="cb39-6"><a href="variant-annotation.html#cb39-6" aria-hidden="true" tabindex="-1"></a><span class="st">    source {</span></span>
<span id="cb39-7"><a href="variant-annotation.html#cb39-7" aria-hidden="true" tabindex="-1"></a><span class="st">      variant_id</span></span>
<span id="cb39-8"><a href="variant-annotation.html#cb39-8" aria-hidden="true" tabindex="-1"></a><span class="st">    }</span></span>
<span id="cb39-9"><a href="variant-annotation.html#cb39-9" aria-hidden="true" tabindex="-1"></a><span class="st">  }</span></span>
<span id="cb39-10"><a href="variant-annotation.html#cb39-10" aria-hidden="true" tabindex="-1"></a><span class="st">}&#39;</span>)</span>
<span id="cb39-11"><a href="variant-annotation.html#cb39-11" aria-hidden="true" tabindex="-1"></a>grch37id <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb39-12"><a href="variant-annotation.html#cb39-12" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> grch38id) {</span>
<span id="cb39-13"><a href="variant-annotation.html#cb39-13" aria-hidden="true" tabindex="-1"></a>  variables <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">varID =</span> i)</span>
<span id="cb39-14"><a href="variant-annotation.html#cb39-14" aria-hidden="true" tabindex="-1"></a>  tries <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb39-15"><a href="variant-annotation.html#cb39-15" aria-hidden="true" tabindex="-1"></a>  <span class="cf">repeat</span> {</span>
<span id="cb39-16"><a href="variant-annotation.html#cb39-16" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> <span class="fu">try</span>(con<span class="sc">$</span><span class="fu">exec</span>(qry<span class="sc">$</span>queries<span class="sc">$</span>liftover, variables), <span class="at">silent =</span> <span class="cn">TRUE</span>)</span>
<span id="cb39-17"><a href="variant-annotation.html#cb39-17" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is</span>(x, <span class="st">&#39;try-error&#39;</span>)) {</span>
<span id="cb39-18"><a href="variant-annotation.html#cb39-18" aria-hidden="true" tabindex="-1"></a>      x <span class="ot">&lt;-</span> jsonlite<span class="sc">::</span><span class="fu">fromJSON</span>(x)<span class="sc">$</span>data<span class="sc">$</span>liftover<span class="sc">$</span>source<span class="sc">$</span>variant_id </span>
<span id="cb39-19"><a href="variant-annotation.html#cb39-19" aria-hidden="true" tabindex="-1"></a>      <span class="fu">message</span>(<span class="fu">paste</span>(tries,<span class="st">&quot;success&quot;</span>, <span class="at">collapse=</span><span class="st">&quot; &quot;</span>))</span>
<span id="cb39-20"><a href="variant-annotation.html#cb39-20" aria-hidden="true" tabindex="-1"></a>      <span class="cf">break</span></span>
<span id="cb39-21"><a href="variant-annotation.html#cb39-21" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb39-22"><a href="variant-annotation.html#cb39-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">Sys.sleep</span>(<span class="dv">2</span><span class="sc">*</span>tries)</span>
<span id="cb39-23"><a href="variant-annotation.html#cb39-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">message</span>(tries)</span>
<span id="cb39-24"><a href="variant-annotation.html#cb39-24" aria-hidden="true" tabindex="-1"></a>    tries <span class="ot">&lt;-</span> tries <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb39-25"><a href="variant-annotation.html#cb39-25" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb39-26"><a href="variant-annotation.html#cb39-26" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">is.null</span>(x)) x <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb39-27"><a href="variant-annotation.html#cb39-27" aria-hidden="true" tabindex="-1"></a>  grch37id <span class="ot">&lt;-</span> <span class="fu">append</span>(grch37id, x)</span>
<span id="cb39-28"><a href="variant-annotation.html#cb39-28" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb39-29"><a href="variant-annotation.html#cb39-29" aria-hidden="true" tabindex="-1"></a>liftover <span class="ot">&lt;-</span> <span class="fu">tibble</span>(<span class="at">gnomADid=</span>grch38id, grch37id)</span>
<span id="cb39-30"><a href="variant-annotation.html#cb39-30" aria-hidden="true" tabindex="-1"></a>prob_2plus_het2 <span class="ot">&lt;-</span> <span class="fu">left_join</span>(prob_2plus_het2, liftover)</span>
<span id="cb39-31"><a href="variant-annotation.html#cb39-31" aria-hidden="true" tabindex="-1"></a><span class="co"># Obtain probability that two variants within a gene are compound heterozygous and phase variants using cutoff of &lt;0.2 for cis and &gt;0.55 for trans</span></span>
<span id="cb39-32"><a href="variant-annotation.html#cb39-32" aria-hidden="true" tabindex="-1"></a>qry<span class="sc">$</span><span class="fu">query</span>(<span class="st">&#39;coocurrence&#39;</span>,</span>
<span id="cb39-33"><a href="variant-annotation.html#cb39-33" aria-hidden="true" tabindex="-1"></a><span class="st">&#39;query Coocurrence($var1: String!, $var2: String!){</span></span>
<span id="cb39-34"><a href="variant-annotation.html#cb39-34" aria-hidden="true" tabindex="-1"></a><span class="st">  variant_cooccurrence(dataset: gnomad_r2_1, variants: [$var1, $var2]) {</span></span>
<span id="cb39-35"><a href="variant-annotation.html#cb39-35" aria-hidden="true" tabindex="-1"></a><span class="st">    p_compound_heterozygous</span></span>
<span id="cb39-36"><a href="variant-annotation.html#cb39-36" aria-hidden="true" tabindex="-1"></a><span class="st">  }</span></span>
<span id="cb39-37"><a href="variant-annotation.html#cb39-37" aria-hidden="true" tabindex="-1"></a><span class="st">}&#39;</span></span>
<span id="cb39-38"><a href="variant-annotation.html#cb39-38" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb39-39"><a href="variant-annotation.html#cb39-39" aria-hidden="true" tabindex="-1"></a>prob_2plus_het_gnom <span class="ot">&lt;-</span> prob_2plus_het2 <span class="sc">%&gt;%</span> </span>
<span id="cb39-40"><a href="variant-annotation.html#cb39-40" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(grch37id)) <span class="sc">%&gt;%</span></span>
<span id="cb39-41"><a href="variant-annotation.html#cb39-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(SYMBOL) <span class="sc">%&gt;%</span></span>
<span id="cb39-42"><a href="variant-annotation.html#cb39-42" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">n</span>()<span class="sc">!=</span><span class="dv">1</span>)</span>
<span id="cb39-43"><a href="variant-annotation.html#cb39-43" aria-hidden="true" tabindex="-1"></a>s <span class="ot">&lt;-</span> <span class="fu">tibble</span>(<span class="at">grch37id=</span><span class="cn">NA</span>, <span class="at">gnomAD_A=</span><span class="cn">NA</span>, <span class="at">gnomAD_B=</span><span class="cn">NA</span>, <span class="at">gnomAD_PS=</span><span class="cn">NA</span>)</span>
<span id="cb39-44"><a href="variant-annotation.html#cb39-44" aria-hidden="true" tabindex="-1"></a>r <span class="ot">&lt;-</span> <span class="fu">tibble</span>(<span class="at">grch37id=</span><span class="cn">NA</span>, <span class="at">gnomAD_cis=</span><span class="cn">NA</span>, <span class="at">gnomAD_trans=</span><span class="cn">NA</span>)</span>
<span id="cb39-45"><a href="variant-annotation.html#cb39-45" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (a <span class="cf">in</span> <span class="fu">unique</span>(prob_2plus_het_gnom<span class="sc">$</span>SYMBOL)) {</span>
<span id="cb39-46"><a href="variant-annotation.html#cb39-46" aria-hidden="true" tabindex="-1"></a>  var <span class="ot">&lt;-</span> prob_2plus_het_gnom[prob_2plus_het_gnom<span class="sc">$</span>SYMBOL<span class="sc">==</span>a,]<span class="sc">$</span>grch37id</span>
<span id="cb39-47"><a href="variant-annotation.html#cb39-47" aria-hidden="true" tabindex="-1"></a>  sub <span class="ot">&lt;-</span> <span class="fu">tibble</span>(<span class="at">grch37id=</span>var, <span class="at">gnomAD_A=</span><span class="cn">NA</span>, <span class="at">gnomAD_B=</span><span class="cn">NA</span>, <span class="at">gnomAD_PS=</span><span class="cn">NA</span>)</span>
<span id="cb39-48"><a href="variant-annotation.html#cb39-48" aria-hidden="true" tabindex="-1"></a>  sub<span class="sc">$</span>gnomAD_A[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb39-49"><a href="variant-annotation.html#cb39-49" aria-hidden="true" tabindex="-1"></a>  sub<span class="sc">$</span>gnomAD_PS[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb39-50"><a href="variant-annotation.html#cb39-50" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>(<span class="fu">length</span>(var)<span class="sc">-</span><span class="dv">1</span>)) {</span>
<span id="cb39-51"><a href="variant-annotation.html#cb39-51" aria-hidden="true" tabindex="-1"></a>    c <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb39-52"><a href="variant-annotation.html#cb39-52" aria-hidden="true" tabindex="-1"></a>    t <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb39-53"><a href="variant-annotation.html#cb39-53" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> (j <span class="cf">in</span> (i<span class="sc">+</span><span class="dv">1</span>)<span class="sc">:</span><span class="fu">length</span>(var)) {</span>
<span id="cb39-54"><a href="variant-annotation.html#cb39-54" aria-hidden="true" tabindex="-1"></a>      variables <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">var1 =</span> var[i], <span class="at">var2 =</span> var[j])</span>
<span id="cb39-55"><a href="variant-annotation.html#cb39-55" aria-hidden="true" tabindex="-1"></a>      tries <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb39-56"><a href="variant-annotation.html#cb39-56" aria-hidden="true" tabindex="-1"></a>      <span class="cf">repeat</span> {</span>
<span id="cb39-57"><a href="variant-annotation.html#cb39-57" aria-hidden="true" tabindex="-1"></a>        y <span class="ot">&lt;-</span> <span class="fu">try</span>(con<span class="sc">$</span><span class="fu">exec</span>(qry<span class="sc">$</span>queries<span class="sc">$</span>coocurrence, variables), <span class="at">silent =</span> <span class="cn">TRUE</span>)</span>
<span id="cb39-58"><a href="variant-annotation.html#cb39-58" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is</span>(y, <span class="st">&#39;try-error&#39;</span>)) {</span>
<span id="cb39-59"><a href="variant-annotation.html#cb39-59" aria-hidden="true" tabindex="-1"></a>          y <span class="ot">&lt;-</span> jsonlite<span class="sc">::</span><span class="fu">fromJSON</span>(y)<span class="sc">$</span>data<span class="sc">$</span>variant_cooccurrence<span class="sc">$</span>p_compound_heterozygous</span>
<span id="cb39-60"><a href="variant-annotation.html#cb39-60" aria-hidden="true" tabindex="-1"></a>          <span class="fu">message</span>(<span class="fu">paste</span>(tries,<span class="st">&quot;success&quot;</span>, <span class="at">collapse=</span><span class="st">&quot; &quot;</span>))</span>
<span id="cb39-61"><a href="variant-annotation.html#cb39-61" aria-hidden="true" tabindex="-1"></a>        <span class="cf">break</span></span>
<span id="cb39-62"><a href="variant-annotation.html#cb39-62" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb39-63"><a href="variant-annotation.html#cb39-63" aria-hidden="true" tabindex="-1"></a>      <span class="fu">Sys.sleep</span>(<span class="dv">2</span><span class="sc">*</span>tries)</span>
<span id="cb39-64"><a href="variant-annotation.html#cb39-64" aria-hidden="true" tabindex="-1"></a>      <span class="fu">message</span>(tries)</span>
<span id="cb39-65"><a href="variant-annotation.html#cb39-65" aria-hidden="true" tabindex="-1"></a>      tries <span class="ot">&lt;-</span> tries <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb39-66"><a href="variant-annotation.html#cb39-66" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb39-67"><a href="variant-annotation.html#cb39-67" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (<span class="fu">is.null</span>(y)) y <span class="ot">&lt;-</span> <span class="st">&quot;NA&quot;</span></span>
<span id="cb39-68"><a href="variant-annotation.html#cb39-68" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (y<span class="sc">&lt;</span><span class="fl">0.2</span>) c <span class="ot">&lt;-</span> <span class="fu">append</span>(c, <span class="fu">paste0</span>(variables<span class="sc">$</span>var2,<span class="st">&quot; (&quot;</span>,<span class="fu">round</span>(y,<span class="at">digits=</span><span class="dv">2</span>),<span class="st">&quot;)&quot;</span>))</span>
<span id="cb39-69"><a href="variant-annotation.html#cb39-69" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (y<span class="sc">&gt;</span><span class="fl">0.55</span> <span class="sc">&amp;</span> y<span class="sc">!=</span><span class="st">&quot;NA&quot;</span>) t <span class="ot">&lt;-</span> <span class="fu">append</span>(t, <span class="fu">paste0</span>(variables<span class="sc">$</span>var2,<span class="st">&quot; (&quot;</span>,<span class="fu">round</span>(y,<span class="at">digits=</span><span class="dv">2</span>),<span class="st">&quot;)&quot;</span>))</span>
<span id="cb39-70"><a href="variant-annotation.html#cb39-70" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.na</span>(sub[[i,<span class="st">&quot;gnomAD_A&quot;</span>]]) <span class="sc">&amp;</span> <span class="fu">is.na</span>(sub[[i,<span class="st">&quot;gnomAD_B&quot;</span>]])) {</span>
<span id="cb39-71"><a href="variant-annotation.html#cb39-71" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (y<span class="sc">&lt;</span><span class="fl">0.2</span>) {</span>
<span id="cb39-72"><a href="variant-annotation.html#cb39-72" aria-hidden="true" tabindex="-1"></a>          sub<span class="sc">$</span>gnomAD_A[j] <span class="ot">&lt;-</span> i</span>
<span id="cb39-73"><a href="variant-annotation.html#cb39-73" aria-hidden="true" tabindex="-1"></a>          sub<span class="sc">$</span>gnomAD_PS[j] <span class="ot">&lt;-</span> sub<span class="sc">$</span>gnomAD_PS[i]</span>
<span id="cb39-74"><a href="variant-annotation.html#cb39-74" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb39-75"><a href="variant-annotation.html#cb39-75" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (y<span class="sc">&gt;</span><span class="fl">0.55</span> <span class="sc">&amp;</span> y<span class="sc">!=</span><span class="st">&quot;NA&quot;</span>) {</span>
<span id="cb39-76"><a href="variant-annotation.html#cb39-76" aria-hidden="true" tabindex="-1"></a>          sub<span class="sc">$</span>gnomAD_B[j] <span class="ot">&lt;-</span> i</span>
<span id="cb39-77"><a href="variant-annotation.html#cb39-77" aria-hidden="true" tabindex="-1"></a>          sub<span class="sc">$</span>gnomAD_PS[j] <span class="ot">&lt;-</span> sub<span class="sc">$</span>gnomAD_PS[i]</span>
<span id="cb39-78"><a href="variant-annotation.html#cb39-78" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb39-79"><a href="variant-annotation.html#cb39-79" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb39-80"><a href="variant-annotation.html#cb39-80" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.na</span>(sub[[i,<span class="st">&quot;gnomAD_B&quot;</span>]]) <span class="sc">&amp;</span> <span class="fu">is.na</span>(sub[[i,<span class="st">&quot;gnomAD_A&quot;</span>]])) {</span>
<span id="cb39-81"><a href="variant-annotation.html#cb39-81" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (y<span class="sc">&lt;</span><span class="fl">0.2</span>) {</span>
<span id="cb39-82"><a href="variant-annotation.html#cb39-82" aria-hidden="true" tabindex="-1"></a>          sub<span class="sc">$</span>gnomAD_B[j] <span class="ot">&lt;-</span> i</span>
<span id="cb39-83"><a href="variant-annotation.html#cb39-83" aria-hidden="true" tabindex="-1"></a>          sub<span class="sc">$</span>gnomAD_PS[j] <span class="ot">&lt;-</span> sub<span class="sc">$</span>gnomAD_PS[i]</span>
<span id="cb39-84"><a href="variant-annotation.html#cb39-84" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb39-85"><a href="variant-annotation.html#cb39-85" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (y<span class="sc">&gt;</span><span class="fl">0.55</span> <span class="sc">&amp;</span> y<span class="sc">!=</span><span class="st">&quot;NA&quot;</span>) {</span>
<span id="cb39-86"><a href="variant-annotation.html#cb39-86" aria-hidden="true" tabindex="-1"></a>          sub<span class="sc">$</span>gnomAD_A[j] <span class="ot">&lt;-</span> i</span>
<span id="cb39-87"><a href="variant-annotation.html#cb39-87" aria-hidden="true" tabindex="-1"></a>          sub<span class="sc">$</span>gnomAD_PS[j] <span class="ot">&lt;-</span> sub<span class="sc">$</span>gnomAD_PS[i]</span>
<span id="cb39-88"><a href="variant-annotation.html#cb39-88" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb39-89"><a href="variant-annotation.html#cb39-89" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb39-90"><a href="variant-annotation.html#cb39-90" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (<span class="fu">is.na</span>(sub[[i,<span class="st">&quot;gnomAD_A&quot;</span>]]) <span class="sc">&amp;</span> <span class="fu">is.na</span>(sub[[i,<span class="st">&quot;gnomAD_B&quot;</span>]])) {</span>
<span id="cb39-91"><a href="variant-annotation.html#cb39-91" aria-hidden="true" tabindex="-1"></a>        sub<span class="sc">$</span>gnomAD_A[i] <span class="ot">&lt;-</span> i</span>
<span id="cb39-92"><a href="variant-annotation.html#cb39-92" aria-hidden="true" tabindex="-1"></a>        sub<span class="sc">$</span>gnomAD_PS[i] <span class="ot">&lt;-</span> i</span>
<span id="cb39-93"><a href="variant-annotation.html#cb39-93" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (y<span class="sc">&lt;</span><span class="fl">0.2</span>) { </span>
<span id="cb39-94"><a href="variant-annotation.html#cb39-94" aria-hidden="true" tabindex="-1"></a>          sub<span class="sc">$</span>gnomAD_A[j] <span class="ot">&lt;-</span> i</span>
<span id="cb39-95"><a href="variant-annotation.html#cb39-95" aria-hidden="true" tabindex="-1"></a>          sub<span class="sc">$</span>gnomAD_PS[j] <span class="ot">&lt;-</span> i</span>
<span id="cb39-96"><a href="variant-annotation.html#cb39-96" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb39-97"><a href="variant-annotation.html#cb39-97" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (y<span class="sc">&gt;</span><span class="fl">0.55</span> <span class="sc">&amp;</span> y<span class="sc">!=</span><span class="st">&quot;NA&quot;</span>) {</span>
<span id="cb39-98"><a href="variant-annotation.html#cb39-98" aria-hidden="true" tabindex="-1"></a>          sub<span class="sc">$</span>gnomAD_B[j] <span class="ot">&lt;-</span> i</span>
<span id="cb39-99"><a href="variant-annotation.html#cb39-99" aria-hidden="true" tabindex="-1"></a>          sub<span class="sc">$</span>gnomAD_PS[j] <span class="ot">&lt;-</span> i</span>
<span id="cb39-100"><a href="variant-annotation.html#cb39-100" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb39-101"><a href="variant-annotation.html#cb39-101" aria-hidden="true" tabindex="-1"></a>      }  </span>
<span id="cb39-102"><a href="variant-annotation.html#cb39-102" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb39-103"><a href="variant-annotation.html#cb39-103" aria-hidden="true" tabindex="-1"></a>    r <span class="ot">&lt;-</span> r <span class="sc">%&gt;%</span> <span class="fu">add_row</span>(<span class="at">grch37id=</span>var[i], <span class="at">gnomAD_cis=</span><span class="fu">str_flatten_comma</span>(c), <span class="at">gnomAD_trans=</span><span class="fu">str_flatten_comma</span>(t))</span>
<span id="cb39-104"><a href="variant-annotation.html#cb39-104" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb39-105"><a href="variant-annotation.html#cb39-105" aria-hidden="true" tabindex="-1"></a>  s <span class="ot">&lt;-</span> <span class="fu">rbind</span>(s,sub)</span>
<span id="cb39-106"><a href="variant-annotation.html#cb39-106" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb39-107"><a href="variant-annotation.html#cb39-107" aria-hidden="true" tabindex="-1"></a>prob_2plus_het2 <span class="ot">&lt;-</span> prob_2plus_het2 <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(r) <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(s) <span class="sc">%&gt;%</span> </span>
<span id="cb39-108"><a href="variant-annotation.html#cb39-108" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">relocate</span>(<span class="fu">c</span>(gnomAD_A,gnomAD_B,gnomAD_PS), <span class="at">.after=</span>PS_proband)</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["SYMBOL"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Uploaded_variation"],"name":[2],"type":["fct"],"align":["left"]},{"label":["HGVSc"],"name":[3],"type":["chr"],"align":["left"]},{"label":["HGVSp"],"name":[4],"type":["chr"],"align":["left"]},{"label":["proband"],"name":[5],"type":["chr"],"align":["left"]},{"label":["PS_proband"],"name":[6],"type":["int"],"align":["right"]},{"label":["gnomAD_A"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["gnomAD_B"],"name":[8],"type":["lgl"],"align":["right"]},{"label":["gnomAD_PS"],"name":[9],"type":["dbl"],"align":["right"]},{"label":["gnomAD_cis"],"name":[10],"type":["chr"],"align":["left"]},{"label":["gnomAD_trans"],"name":[11],"type":["chr"],"align":["left"]}],"data":[{"1":"HLA-DQA1","2":"chr6_32641501_G/C","3":"NM_002122.5:c.274G>C","4":"NP_002113.2:p.Ala92Pro","5":"1|0","6":"203397","7":"1","8":"NA","9":"1","10":"6-32609279-C-T (0), 6-32609299-A-G (0), 6-32609312-A-C (0)","11":""},{"1":"HLA-DQA1","2":"chr6_32641502_C/T","3":"NM_002122.5:c.275C>T","4":"NP_002113.2:p.Ala92Val","5":"1|0","6":"203397","7":"1","8":"NA","9":"1","10":"6-32609299-A-G (0), 6-32609312-A-C (0)","11":""},{"1":"HLA-DQA1","2":"chr6_32641522_A/G","3":"NM_002122.5:c.295A>G","4":"NP_002113.2:p.Met99Val","5":"0/2","6":"32641444","7":"2","8":"NA","9":"1","10":"6-32609312-A-C (0)","11":""},{"1":"HLA-DQA1","2":"chr6_32641535_A/C","3":"NM_002122.5:c.308A>C","4":"NP_002113.2:p.Tyr103Ser","5":"1|0","6":"203397","7":"3","8":"NA","9":"1","10":"NA","11":"NA"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="x-chromosome" class="section level2 hasAnchor" number="5.6">
<h2><span class="header-section-number">5.6</span> X Chromosome<a href="variant-annotation.html#x-chromosome" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We remove X chromosome calls that are “heterozygous” for male individuals, i.e., proband and father. However, some of these calls may be correct if they lie in the pseudoautosomal region.</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="variant-annotation.html#cb40-1" aria-hidden="true" tabindex="-1"></a>wh_geno_X_filt <span class="ot">&lt;-</span> wh_geno_X <span class="sc">%&gt;%</span> </span>
<span id="cb40-2"><a href="variant-annotation.html#cb40-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(( <span class="fu">substr</span>(proband,<span class="dv">1</span>,<span class="dv">1</span>)<span class="sc">==</span><span class="fu">substr</span>(proband,<span class="dv">3</span>,<span class="dv">3</span>) <span class="sc">&amp;</span> <span class="fu">substr</span>(father,<span class="dv">1</span>,<span class="dv">1</span>)<span class="sc">==</span><span class="fu">substr</span>(father,<span class="dv">3</span>,<span class="dv">3</span>)))</span>
<span id="cb40-3"><a href="variant-annotation.html#cb40-3" aria-hidden="true" tabindex="-1"></a>prob_X <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(refseq_X, wh_geno_X_filt) <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(Gene) <span class="sc">%&gt;%</span> </span>
<span id="cb40-4"><a href="variant-annotation.html#cb40-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(CANONICAL), <span class="fu">as.character</span>(Feature),<span class="at">.by_group =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb40-5"><a href="variant-annotation.html#cb40-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Feature <span class="sc">==</span> Feature[<span class="dv">1</span>])</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["SYMBOL"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Uploaded_variation"],"name":[2],"type":["fct"],"align":["left"]},{"label":["HGVSc"],"name":[3],"type":["chr"],"align":["left"]},{"label":["HGVSp"],"name":[4],"type":["chr"],"align":["left"]},{"label":["mother"],"name":[5],"type":["chr"],"align":["left"]},{"label":["father"],"name":[6],"type":["chr"],"align":["left"]},{"label":["proband"],"name":[7],"type":["chr"],"align":["left"]},{"label":["counter"],"name":[8],"type":["int"],"align":["right"]}],"data":[{"1":"WWC3","2":"chrX_10138908_G/A","3":"NM_015691.5:c.3428G>A","4":"NP_056506.3:p.Arg1143His","5":"0/1","6":"0/0","7":"1/1","8":"1"},{"1":"SCML2","2":"chrX_18324912_G/A","3":"NM_006089.3:c.157C>T","4":"NP_006080.1:p.Arg53Cys","5":"0/1","6":"0/0","7":"1/1","8":"1"},{"1":"SH3KBP1","2":"chrX_19545924_G/C","3":"NM_031892.3:c.1621C>G","4":"NP_114098.1:p.Gln541Glu","5":"0/1","6":"0/0","7":"1/1","8":"1"},{"1":"AMER1","2":"chrX_64191414_T/C","3":"NM_152424.4:c.1873A>G","4":"NP_689637.3:p.Thr625Ala","5":"0/1","6":"0/0","7":"1/1","8":"1"},{"1":"RTL9","2":"chrX_110454576_A/G","3":"NM_001385449.1:c.3959A>G","4":"NP_001372378.1:p.Asn1320Ser","5":"0/1","6":"0/0","7":"1/1","8":"1"},{"1":"GRIA3","2":"chrX_123403094_G/A","3":"NM_007325.5:c.1181G>A","4":"NP_015564.5:p.Arg394Gln","5":"0/1","6":"0/0","7":"1/1","8":"1"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="disease-genes" class="section level2 hasAnchor" number="5.7">
<h2><span class="header-section-number">5.7</span> Disease Genes<a href="variant-annotation.html#disease-genes" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We select one entry for each variant in the refseq table, prioritizing canonical and curated transcripts.</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="variant-annotation.html#cb41-1" aria-hidden="true" tabindex="-1"></a>refseq2 <span class="ot">&lt;-</span> refseq01 <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(Uploaded_variation) <span class="sc">%&gt;%</span> </span>
<span id="cb41-2"><a href="variant-annotation.html#cb41-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(CANONICAL), <span class="fu">as.character</span>(Feature),<span class="at">.by_group =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb41-3"><a href="variant-annotation.html#cb41-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Feature <span class="sc">==</span> Feature[<span class="dv">1</span>])</span></code></pre></div>
<p>As an example, we select variants that fall in kidney disease genes, i.e., green genes from the UK PanelApp renal broad superpanel and all genes from the Australia PanelApp kidneyome superpanel.</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="variant-annotation.html#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="co"># UK Panel App</span></span>
<span id="cb42-2"><a href="variant-annotation.html#cb42-2" aria-hidden="true" tabindex="-1"></a>panel <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">&quot;data/renal_superpanel_broad.tsv&quot;</span>)</span>
<span id="cb42-3"><a href="variant-annotation.html#cb42-3" aria-hidden="true" tabindex="-1"></a>panel_green <span class="ot">&lt;-</span> panel <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(GEL_Status<span class="sc">==</span><span class="dv">3</span>)</span>
<span id="cb42-4"><a href="variant-annotation.html#cb42-4" aria-hidden="true" tabindex="-1"></a>refseq_green <span class="ot">&lt;-</span> refseq2 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(HGNC_ID <span class="sc">%in%</span> panel_green<span class="sc">$</span>HGNC)</span>
<span id="cb42-5"><a href="variant-annotation.html#cb42-5" aria-hidden="true" tabindex="-1"></a><span class="co">#Australia Panel App</span></span>
<span id="cb42-6"><a href="variant-annotation.html#cb42-6" aria-hidden="true" tabindex="-1"></a>panel_au <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">&quot;data/Kidneyome_SuperPanel.tsv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb42-7"><a href="variant-annotation.html#cb42-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">duplicated</span>(HGNC)) <span class="sc">%&gt;%</span></span>
<span id="cb42-8"><a href="variant-annotation.html#cb42-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">Status=</span><span class="fu">case_when</span>(</span>
<span id="cb42-9"><a href="variant-annotation.html#cb42-9" aria-hidden="true" tabindex="-1"></a>    GEL_Status<span class="sc">==</span><span class="dv">4</span><span class="sc">|</span>GEL_Status<span class="sc">==</span><span class="dv">3</span> <span class="sc">~</span> <span class="st">&quot;Green&quot;</span>, </span>
<span id="cb42-10"><a href="variant-annotation.html#cb42-10" aria-hidden="true" tabindex="-1"></a>    GEL_Status<span class="sc">==</span><span class="dv">2</span> <span class="sc">~</span> <span class="st">&quot;Amber&quot;</span>, </span>
<span id="cb42-11"><a href="variant-annotation.html#cb42-11" aria-hidden="true" tabindex="-1"></a>    GEL_Status<span class="sc">==</span><span class="dv">1</span><span class="sc">|</span>GEL_Status<span class="sc">==</span><span class="dv">0</span> <span class="sc">~</span> <span class="st">&quot;Red&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb42-12"><a href="variant-annotation.html#cb42-12" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(HGNC,Status)</span>
<span id="cb42-13"><a href="variant-annotation.html#cb42-13" aria-hidden="true" tabindex="-1"></a>refseq_au <span class="ot">&lt;-</span> refseq2 <span class="sc">%&gt;%</span> <span class="fu">inner_join</span>(panel_au, <span class="fu">join_by</span>(HGNC_ID<span class="sc">==</span>HGNC))</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["SYMBOL"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Uploaded_variation"],"name":[2],"type":["fct"],"align":["left"]},{"label":["HGVSg"],"name":[3],"type":["chr"],"align":["left"]},{"label":["HGVSc"],"name":[4],"type":["chr"],"align":["left"]},{"label":["Status"],"name":[5],"type":["chr"],"align":["left"]}],"data":[{"1":"NPHP4","2":"chr1_5864452_C/A","3":"chr1:g.5864452C>A","4":"NM_015102.5:c.3882G>T","5":"Green"},{"1":"SCARB2","2":"chr4_76179654_T/C","3":"chr4:g.76179654T>C","4":"NM_005506.4:c.475A>G","5":"Green"},{"1":"FAT1","2":"chr4_186618051_C/T","3":"chr4:g.186618051C>T","4":"NM_005245.4:c.8535G>A","5":"Green"},{"1":"BBS9","2":"chr7_33340944_G/A","3":"chr7:g.33340944G>A","4":"NM_198428.3:c.1246G>A","5":"Green"},{"1":"EGFR","2":"chr7_55173087_G/A","3":"chr7:g.55173087G>A","4":"NM_005228.5:c.2024G>A","5":"Red"},{"1":"SOX17","2":"chr8_54459525_T/A","3":"chr8:g.54459525T>A","4":"NM_022454.4:c.775T>A","5":"Red"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>We combine annotations into a single table.</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="variant-annotation.html#cb43-1" aria-hidden="true" tabindex="-1"></a>refseq3 <span class="ot">&lt;-</span> refseq01 <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(Gene) <span class="sc">%&gt;%</span> </span>
<span id="cb43-2"><a href="variant-annotation.html#cb43-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(CANONICAL), <span class="fu">as.character</span>(Feature),<span class="at">.by_group =</span> T) <span class="sc">%&gt;%</span> <span class="fu">ungroup</span>()</span>
<span id="cb43-3"><a href="variant-annotation.html#cb43-3" aria-hidden="true" tabindex="-1"></a>refseq4 <span class="ot">&lt;-</span> refseq3 <span class="sc">%&gt;%</span> </span>
<span id="cb43-4"><a href="variant-annotation.html#cb43-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(<span class="fu">add_column</span>(geno_novo_not_ref,<span class="at">de_novo=</span><span class="st">&quot;de_novo&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb43-5"><a href="variant-annotation.html#cb43-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(<span class="fu">add_column</span>(geno_novo_not_ref_high_qual,<span class="at">de_novo_gq=</span><span class="st">&quot;de_novo_gq&gt;=30&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb43-6"><a href="variant-annotation.html#cb43-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(<span class="fu">add_column</span>(geno_hom2, <span class="at">homozygote=</span><span class="st">&quot;homozygote&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb43-7"><a href="variant-annotation.html#cb43-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(<span class="fu">add_column</span>(prob_2plus_het, <span class="at">unphased_compound_heterozygote=</span><span class="st">&quot;unphased_compound_heterozygote&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb43-8"><a href="variant-annotation.html#cb43-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(<span class="fu">add_column</span>(prob_comp_het, <span class="at">phased_compound_heterozygote=</span><span class="st">&quot;phased_compound_heterozygote&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb43-9"><a href="variant-annotation.html#cb43-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">zygosity=</span><span class="fu">coalesce</span>(de_novo_gq,de_novo,homozygote,phased_compound_heterozygote,unphased_compound_heterozygote)) <span class="sc">%&gt;%</span></span>
<span id="cb43-10"><a href="variant-annotation.html#cb43-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(de_novo_gq,de_novo,homozygote,phased_compound_heterozygote,unphased_compound_heterozygote)) <span class="sc">%&gt;%</span></span>
<span id="cb43-11"><a href="variant-annotation.html#cb43-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">relocate</span>(zygosity,<span class="at">.after=</span>counter)</span>
<span id="cb43-12"><a href="variant-annotation.html#cb43-12" aria-hidden="true" tabindex="-1"></a>refseq_sum <span class="ot">&lt;-</span> refseq4 <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(Feature) <span class="sc">%&gt;%</span> </span>
<span id="cb43-13"><a href="variant-annotation.html#cb43-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">z=</span><span class="fu">paste</span>(<span class="fu">c</span>(Uploaded_variation,Location,Allele,Consequence,Amino_acids,Codons), <span class="at">collapse =</span> <span class="st">&quot; &quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb43-14"><a href="variant-annotation.html#cb43-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(z,<span class="at">.keep_all =</span> T)</span>
<span id="cb43-15"><a href="variant-annotation.html#cb43-15" aria-hidden="true" tabindex="-1"></a>refseq5 <span class="ot">&lt;-</span> refseq4 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(Feature <span class="sc">%in%</span> refseq_sum<span class="sc">$</span>Feature)</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="statistical-genetics.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="summary.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
